<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-11-03">

<title>hambiDoubleEvo - WGS genetic parallelism analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/wgs/01_format_variants.html">Population WGS</a></li><li class="breadcrumb-item"><a href="../../R/wgs/02_parallelism.html">2. Parallelism analysis</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">hambiDoubleEvo</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/01_rpkm2tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/02_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Analyze and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Population WGS</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/02_parallelism.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Metagenome sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Variant plotting</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/04_mutation_dynamics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Mutational dynamics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">2</span> Read data</a></li>
  <li><a href="#parallelism-at-nucleotide-level" id="toc-parallelism-at-nucleotide-level" class="nav-link" data-scroll-target="#parallelism-at-nucleotide-level"><span class="header-section-number">3</span> Parallelism at nucleotide level</a>
  <ul class="collapse">
  <li><a href="#nucleotide-parallelism-by-treatment" id="toc-nucleotide-parallelism-by-treatment" class="nav-link" data-scroll-target="#nucleotide-parallelism-by-treatment"><span class="header-section-number">3.1</span> Nucleotide parallelism by treatment</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">3.2</span> Conclusion</a></li>
  </ul></li>
  <li><a href="#parallelism-at-gene-level" id="toc-parallelism-at-gene-level" class="nav-link" data-scroll-target="#parallelism-at-gene-level"><span class="header-section-number">4</span> Parallelism at gene level</a>
  <ul class="collapse">
  <li><a href="#prepare-input-data-for-non-syn-mutations-only" id="toc-prepare-input-data-for-non-syn-mutations-only" class="nav-link" data-scroll-target="#prepare-input-data-for-non-syn-mutations-only"><span class="header-section-number">4.1</span> Prepare input data for non-syn mutations only</a></li>
  <li><a href="#g-score" id="toc-g-score" class="nav-link" data-scroll-target="#g-score"><span class="header-section-number">4.2</span> G score</a>
  <ul class="collapse">
  <li><a href="#simulating-the-number-of-replicates-with-a-parallel-gene-hit" id="toc-simulating-the-number-of-replicates-with-a-parallel-gene-hit" class="nav-link" data-scroll-target="#simulating-the-number-of-replicates-with-a-parallel-gene-hit"><span class="header-section-number">4.2.1</span> Simulating the number of replicates with a parallel gene hit</a></li>
  <li><a href="#visualize-the-survival-curves-for-each-treatment-combination" id="toc-visualize-the-survival-curves-for-each-treatment-combination" class="nav-link" data-scroll-target="#visualize-the-survival-curves-for-each-treatment-combination"><span class="header-section-number">4.2.2</span> Visualize the survival curves for each treatment combination</a></li>
  <li><a href="#inspecting-individual-genes" id="toc-inspecting-individual-genes" class="nav-link" data-scroll-target="#inspecting-individual-genes"><span class="header-section-number">4.2.3</span> Inspecting individual genes</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#convergentdivergent-evolution" id="toc-convergentdivergent-evolution" class="nav-link" data-scroll-target="#convergentdivergent-evolution"><span class="header-section-number">5</span> Convergent/divergent evolution</a>
  <ul class="collapse">
  <li><a href="#all-species" id="toc-all-species" class="nav-link" data-scroll-target="#all-species"><span class="header-section-number">5.1</span> All species</a>
  <ul class="collapse">
  <li><a href="#jensen-shannon-divergence" id="toc-jensen-shannon-divergence" class="nav-link" data-scroll-target="#jensen-shannon-divergence"><span class="header-section-number">5.1.1</span> Jensen Shannon Divergence</a></li>
  <li><a href="#enrichment-of-individual-cog-categories" id="toc-enrichment-of-individual-cog-categories" class="nav-link" data-scroll-target="#enrichment-of-individual-cog-categories"><span class="header-section-number">5.1.2</span> Enrichment of individual COG categories</a></li>
  <li><a href="#cog-distribution-plot" id="toc-cog-distribution-plot" class="nav-link" data-scroll-target="#cog-distribution-plot"><span class="header-section-number">5.1.3</span> COG Distribution Plot</a></li>
  </ul></li>
  <li><a href="#species-present-in-the-metagenomes" id="toc-species-present-in-the-metagenomes" class="nav-link" data-scroll-target="#species-present-in-the-metagenomes"><span class="header-section-number">5.2</span> Species present in the metagenomes</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/wgs/01_format_variants.html">Population WGS</a></li><li class="breadcrumb-item"><a href="../../R/wgs/02_parallelism.html">2. Parallelism analysis</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">WGS genetic parallelism analysis</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 3, 2024</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook reads in the formatted WGS variant frequencies and annotations. It then uses statistical tests to determine which mutations and which functional categories appeared across biological replicates within condition more often than would be expected by chance. It also looks for mutations/functions that arise within treatment categories more than expected by chance.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Libraries and global variables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb1-8"><a href="#cb1-8"></a></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"wgs"</span>, <span class="st">"utils_parallelism.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Set up some directories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"wgs"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"wgs"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"wgs"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-8"><a href="#cb2-8"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read data</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"mutect_parsed_annotated.rds"</span>))</span>
<span id="cb3-2"><a href="#cb3-2"></a>degentab <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb3-3"><a href="#cb3-3"></a>genome_len <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_genome_len.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360050722212-FAQ-for-Mutect2">The mutect2 documentation</a> says that any variant without “PASS” is false positive and should be removed. We will follow this convention here and only focus on PASSing mutations.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># note that 1287 got an extra replicate of replicate "A" sequenced. It was called replicate "B" in the</span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># files and I will combine it with the other replicate "A"</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>tofilter <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="fu">filter</span>(strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span> <span class="sc">&amp;</span> replicate <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"A"</span>,<span class="st">"B"</span>))</span>
<span id="cb4-5"><a href="#cb4-5"></a></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co"># this is to make sure we filter out duplicates by randomly taking one</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co"># from each group BUT to make sure to select duplicates that have a PASS</span></span>
<span id="cb4-8"><a href="#cb4-8"></a>tocombine <span class="ot">&lt;-</span> tofilter <span class="sc">%&gt;%</span> </span>
<span id="cb4-9"><a href="#cb4-9"></a>  <span class="fu">group_by</span>(chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb4-10"><a href="#cb4-10"></a>  <span class="fu">mutate</span>(<span class="at">W =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb4-11"><a href="#cb4-11"></a>  <span class="fu">mutate</span>(<span class="at">W1 =</span> <span class="fu">case_when</span>(W <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> filter <span class="sc">==</span> <span class="st">"PASS"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb4-12"><a href="#cb4-12"></a>                       W <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> filter <span class="sc">!=</span> <span class="st">"PASS"</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb4-13"><a href="#cb4-13"></a>                       W <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-14"><a href="#cb4-14"></a>  <span class="fu">mutate</span>(<span class="at">W2 =</span> <span class="fu">if_else</span>(W <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(W1) <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-15"><a href="#cb4-15"></a>  <span class="fu">mutate</span>(<span class="at">W3 =</span> W2 <span class="sc">+</span> W1) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16"></a>  <span class="fu">relocate</span>(W, W1, W2, W3) <span class="sc">%&gt;%</span> </span>
<span id="cb4-17"><a href="#cb4-17"></a>  <span class="fu">filter</span>(W3 <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-18"><a href="#cb4-18"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-19"><a href="#cb4-19"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>W, <span class="sc">-</span>W1, <span class="sc">-</span>W2, <span class="sc">-</span>W3) <span class="sc">%&gt;%</span> </span>
<span id="cb4-20"><a href="#cb4-20"></a>  <span class="fu">mutate</span>(<span class="at">replicate =</span> <span class="st">"A"</span>)</span>
<span id="cb4-21"><a href="#cb4-21"></a></span>
<span id="cb4-22"><a href="#cb4-22"></a>mgvars_filt <span class="ot">&lt;-</span> <span class="fu">anti_join</span>(mgvars, tofilter) <span class="sc">%&gt;%</span></span>
<span id="cb4-23"><a href="#cb4-23"></a>  <span class="fu">bind_rows</span>(tocombine) <span class="sc">%&gt;%</span> </span>
<span id="cb4-24"><a href="#cb4-24"></a>  <span class="fu">mutate</span>(<span class="at">replicate =</span> <span class="fu">if_else</span>(replicate <span class="sc">==</span> <span class="st">"B"</span>, <span class="st">"C"</span>, replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-25"><a href="#cb4-25"></a>  <span class="co"># we are filtering by requiring a depth of at least 10 reads</span></span>
<span id="cb4-26"><a href="#cb4-26"></a>  <span class="co"># and that the variant passed the mutect2 internal filters</span></span>
<span id="cb4-27"><a href="#cb4-27"></a>  <span class="fu">filter</span>(filter <span class="sc">==</span> <span class="st">"PASS"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-28"><a href="#cb4-28"></a>  <span class="fu">group_by</span>(chrom, pos, ref, alt, replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb4-29"><a href="#cb4-29"></a>  <span class="fu">filter</span>(freq_alt <span class="sc">==</span> <span class="fu">max</span>(freq_alt)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-30"><a href="#cb4-30"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="parallelism-at-nucleotide-level" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Parallelism at nucleotide level</h1>
<p>Looking at all mutations (inside and outside of coding sequences, synonymous and nonsynonymous)</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>mgvars_filt <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="fu">mutate</span>(<span class="at">mi=</span><span class="fu">n_distinct</span>(replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="fu">count</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="fu">group_by</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="fu">mutate</span>(<span class="at">tot =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-8"><a href="#cb5-8"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mi))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["strainID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["mi"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]},{"label":["tot"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"HAMBI_0006","2":"3","3":"39","4":"1384"},{"1":"HAMBI_0097","2":"3","3":"3","4":"1384"},{"1":"HAMBI_0105","2":"3","3":"9","4":"1384"},{"1":"HAMBI_1287","2":"3","3":"3","4":"1384"},{"1":"HAMBI_1292","2":"3","3":"27","4":"1384"},{"1":"HAMBI_1842","2":"3","3":"42","4":"1384"},{"1":"HAMBI_2160","2":"3","3":"792","4":"1384"},{"1":"HAMBI_2494","2":"3","3":"9","4":"1384"},{"1":"HAMBI_2659","2":"3","3":"6","4":"1384"},{"1":"HAMBI_2792","2":"3","3":"418","4":"1384"},{"1":"HAMBI_3031","2":"3","3":"3","4":"1384"},{"1":"HAMBI_3237","2":"3","3":"33","4":"1384"},{"1":"HAMBI_0006","2":"2","3":"2","4":"250"},{"1":"HAMBI_0097","2":"2","3":"2","4":"250"},{"1":"HAMBI_0105","2":"2","3":"2","4":"250"},{"1":"HAMBI_0262","2":"2","3":"8","4":"250"},{"1":"HAMBI_0403","2":"2","3":"2","4":"250"},{"1":"HAMBI_1279","2":"2","3":"2","4":"250"},{"1":"HAMBI_1287","2":"2","3":"4","4":"250"},{"1":"HAMBI_1292","2":"2","3":"4","4":"250"},{"1":"HAMBI_1299","2":"2","3":"6","4":"250"},{"1":"HAMBI_1842","2":"2","3":"8","4":"250"},{"1":"HAMBI_1923","2":"2","3":"14","4":"250"},{"1":"HAMBI_1977","2":"2","3":"4","4":"250"},{"1":"HAMBI_2160","2":"2","3":"22","4":"250"},{"1":"HAMBI_2443","2":"2","3":"36","4":"250"},{"1":"HAMBI_2494","2":"2","3":"34","4":"250"},{"1":"HAMBI_2792","2":"2","3":"76","4":"250"},{"1":"HAMBI_3237","2":"2","3":"24","4":"250"},{"1":"HAMBI_0006","2":"1","3":"945","4":"8969"},{"1":"HAMBI_0097","2":"1","3":"42","4":"8969"},{"1":"HAMBI_0105","2":"1","3":"39","4":"8969"},{"1":"HAMBI_0262","2":"1","3":"30","4":"8969"},{"1":"HAMBI_0403","2":"1","3":"61","4":"8969"},{"1":"HAMBI_1279","2":"1","3":"62","4":"8969"},{"1":"HAMBI_1287","2":"1","3":"90","4":"8969"},{"1":"HAMBI_1292","2":"1","3":"248","4":"8969"},{"1":"HAMBI_1299","2":"1","3":"68","4":"8969"},{"1":"HAMBI_1842","2":"1","3":"42","4":"8969"},{"1":"HAMBI_1923","2":"1","3":"138","4":"8969"},{"1":"HAMBI_1972","2":"1","3":"68","4":"8969"},{"1":"HAMBI_1977","2":"1","3":"66","4":"8969"},{"1":"HAMBI_2159","2":"1","3":"2263","4":"8969"},{"1":"HAMBI_2160","2":"1","3":"79","4":"8969"},{"1":"HAMBI_2164","2":"1","3":"736","4":"8969"},{"1":"HAMBI_2443","2":"1","3":"39","4":"8969"},{"1":"HAMBI_2494","2":"1","3":"42","4":"8969"},{"1":"HAMBI_2659","2":"1","3":"49","4":"8969"},{"1":"HAMBI_2792","2":"1","3":"121","4":"8969"},{"1":"HAMBI_3031","2":"1","3":"1886","4":"8969"},{"1":"HAMBI_3237","2":"1","3":"1855","4":"8969"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>We identified 22 species with detectable mutations (basically everything, except for 1299 for which the sequencing failed). Interestingly, (as shown above) there are many species with a huge amount of parallel mutations across all three replicates. For example there are 1497 sites with the exact same mutation in all three sequenced replicates. Most of these are not fixed mutations (but some of them are), but what I realized when digging deeper is that most of these extremely parallel mutations reside on predicted plasmid sequences, in mobile genetic elements (like transposases), and in predicted prophages. Certainly something interesting is going on in these regions of these species, but I don’t think that short read sequencing is really sufficient to tell us what is going on. Instead short reads seem to give lots of short indels in very small windows of the genome producing a lot of noise. So I think it is best to try and remove these kinds of mutations before proceeding with the analysis.</p>
<p>We used the tools MGEfinder v1.0.3 with database v1.0.2 <span class="citation" data-cites="durrant2020a">(<a href="#ref-durrant2020a" role="doc-biblioref">Durrant et al. 2020</a>)</span> and geNomad v1.7.4 <span class="citation" data-cites="camargo2023">(<a href="#ref-camargo2023" role="doc-biblioref">Camargo et al. 2023</a>)</span> to identify mobile elements, plasmids and integrated viruses (prophages) in the genomes and excluded these regions from subsequent analyses.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>mgefinder <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"MGE_finder_HAMBI_combined.tsv"</span>))</span>
<span id="cb6-2"><a href="#cb6-2"></a>genomad <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"genomad_HAMBI_combined.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="co"># exclude very large plasmids over 1Mbp from the filtering</span></span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="fu">filter</span>((end <span class="sc">-</span> start) <span class="sc">&lt;</span> <span class="dv">1000000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>mgvars_filt_mb <span class="ot">&lt;-</span> mgvars_filt <span class="sc">%&gt;%</span> </span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="fu">left_join</span>(mgefinder, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start.y <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start.y, <span class="sc">-</span>end.y, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6"></a>  <span class="fu">left_join</span>(genomad, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-7"><a href="#cb7-7"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-8"><a href="#cb7-8"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start, <span class="sc">-</span>end, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb7-9"><a href="#cb7-9"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-10"><a href="#cb7-10"></a>  <span class="fu">rename</span>(<span class="at">start =</span> start.x, <span class="at">end =</span> end.x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>mgvars_filt_mb <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  <span class="co"># remove the annotations for saving</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(feature_type<span class="sc">:</span><span class="fu">last_col</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"wgsvars_filt_mb_nonsyn.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>mgvars_filt_mb <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="fu">mutate</span>(<span class="at">mi=</span><span class="fu">n_distinct</span>(replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="fu">count</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="fu">group_by</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7"></a>  <span class="fu">mutate</span>(<span class="at">tot =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mi))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["strainID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["mi"],"name":[2],"type":["int"],"align":["right"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]},{"label":["tot"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"HAMBI_0006","2":"3","3":"39","4":"63"},{"1":"HAMBI_0097","2":"3","3":"3","4":"63"},{"1":"HAMBI_1287","2":"3","3":"3","4":"63"},{"1":"HAMBI_1842","2":"3","3":"3","4":"63"},{"1":"HAMBI_2494","2":"3","3":"9","4":"63"},{"1":"HAMBI_2659","2":"3","3":"6","4":"63"},{"1":"HAMBI_0006","2":"2","3":"2","4":"98"},{"1":"HAMBI_0097","2":"2","3":"2","4":"98"},{"1":"HAMBI_0262","2":"2","3":"8","4":"98"},{"1":"HAMBI_0403","2":"2","3":"2","4":"98"},{"1":"HAMBI_1279","2":"2","3":"2","4":"98"},{"1":"HAMBI_1287","2":"2","3":"4","4":"98"},{"1":"HAMBI_1299","2":"2","3":"6","4":"98"},{"1":"HAMBI_1977","2":"2","3":"4","4":"98"},{"1":"HAMBI_2443","2":"2","3":"36","4":"98"},{"1":"HAMBI_2494","2":"2","3":"32","4":"98"},{"1":"HAMBI_0006","2":"1","3":"945","4":"4738"},{"1":"HAMBI_0097","2":"1","3":"34","4":"4738"},{"1":"HAMBI_0262","2":"1","3":"30","4":"4738"},{"1":"HAMBI_0403","2":"1","3":"59","4":"4738"},{"1":"HAMBI_1279","2":"1","3":"62","4":"4738"},{"1":"HAMBI_1287","2":"1","3":"72","4":"4738"},{"1":"HAMBI_1299","2":"1","3":"68","4":"4738"},{"1":"HAMBI_1842","2":"1","3":"41","4":"4738"},{"1":"HAMBI_1972","2":"1","3":"68","4":"4738"},{"1":"HAMBI_1977","2":"1","3":"66","4":"4738"},{"1":"HAMBI_2159","2":"1","3":"1464","4":"4738"},{"1":"HAMBI_2164","2":"1","3":"437","4":"4738"},{"1":"HAMBI_2443","2":"1","3":"39","4":"4738"},{"1":"HAMBI_2494","2":"1","3":"42","4":"4738"},{"1":"HAMBI_2659","2":"1","3":"49","4":"4738"},{"1":"HAMBI_3237","2":"1","3":"1262","4":"4738"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>OK this seems a lot more reasonable here… Now there are much fewer mutations that are parallel across the replicates.</p>
<p>To put these observations in context, we can compare them to a simple null model in which mutations are uniformly distributed across the sites in the genome. We can then check how many sites with 1, 2, and 3-fold multiplicity that we would expect by chance. We define nucleotide parallelism as the number of mutations occurring at the same site in the genome in independent populations. For each site, we define the multiplicity, <span class="math inline">\(m_{i}\)</span>, as the number of populations (i.e., replicates) with a mutation detected at that site, so that the multiplicity in this experiment can range from one to three.</p>
<p>The expected number of mutations with <span class="math inline">\(m_{i} \geq m\)</span> in a sample of total mutations size <span class="math inline">\(n_{tot}\)</span> is:</p>
<p><span class="math display">\[ S(m) \approx \sum_{n \geq m} \frac{n}{n_{tot}} \cdot L_{tot} \cdot \frac{ \left( \frac{n_{tot}}{L_{tot}} \right) ^n}{n!} e^{-n_{tot}/L_{tot}} \]</span></p>
<p>where <span class="math inline">\(L_{tot}\)</span> is the total number of bases in the genome.</p>
<section id="nucleotide-parallelism-by-treatment" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="nucleotide-parallelism-by-treatment"><span class="header-section-number">3.1</span> Nucleotide parallelism by treatment</h2>
<p>Here we will do the estimation separately for each species <span class="math inline">\(\times\)</span> treatment combination.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>nuc_survival <span class="ot">&lt;-</span> mgvars_filt_mb <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, pos, ref, alt, replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="co"># multiplicity = number of replicate populations each unique mutation is</span></span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="co"># observed in group by the mutation position, with alternative allele to the</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="co"># grouping</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">summarize</span>(<span class="at">m =</span> <span class="fu">n_distinct</span>(replicate), <span class="at">.by =</span> <span class="fu">c</span>(strainID, pos, ref, alt)) <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="co"># now calculate the total number of mutations across all replicates so we need</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="co"># to ungroup by mutation position/alt allele but because we still want to</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>  <span class="co"># determine this value by treatment category we keep the treatment category</span></span>
<span id="cb10-10"><a href="#cb10-10"></a>  <span class="co"># grouping. However, this should be changed if you want to for example average</span></span>
<span id="cb10-11"><a href="#cb10-11"></a>  <span class="co"># over all the treatment conditions on a species basis</span></span>
<span id="cb10-12"><a href="#cb10-12"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span></span>
<span id="cb10-13"><a href="#cb10-13"></a>  <span class="fu">count</span>(m, <span class="at">name =</span> <span class="st">"m_count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-14"><a href="#cb10-14"></a>  <span class="fu">mutate</span>(<span class="at">n =</span> m <span class="sc">*</span> m_count,</span>
<span id="cb10-15"><a href="#cb10-15"></a>         <span class="at">Ntot =</span> <span class="fu">sum</span>(n),</span>
<span id="cb10-16"><a href="#cb10-16"></a>         <span class="at">perc =</span> n <span class="sc">/</span> Ntot <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-17"><a href="#cb10-17"></a>  <span class="fu">left_join</span>(genome_len, <span class="at">by =</span> <span class="fu">join_by</span>(strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb10-18"><a href="#cb10-18"></a>  <span class="fu">arrange</span>(<span class="fu">cur_group_id</span>(), <span class="fu">desc</span>(m)) <span class="sc">%&gt;%</span></span>
<span id="cb10-19"><a href="#cb10-19"></a>  <span class="co">#  dpois() tells the probability mass at a given number of counts. Here we</span></span>
<span id="cb10-20"><a href="#cb10-20"></a>  <span class="co">#  want to get the probability of observing n mutations with multiplicity</span></span>
<span id="cb10-21"><a href="#cb10-21"></a>  <span class="co">#  = mi (i.e. the counts of mi in the observed data). We assume that</span></span>
<span id="cb10-22"><a href="#cb10-22"></a>  <span class="co">#  mutations independently occur on the genome of size Ltot at a rate of</span></span>
<span id="cb10-23"><a href="#cb10-23"></a>  <span class="co">#  lambda = Ntot/Ltot and that generally the events are rare. Thus this</span></span>
<span id="cb10-24"><a href="#cb10-24"></a>  <span class="co">#  situation can be modeled by the Poisson distribution. We can get the</span></span>
<span id="cb10-25"><a href="#cb10-25"></a>  <span class="co">#  binned number of mutations per level of multiplicity m by multiplying</span></span>
<span id="cb10-26"><a href="#cb10-26"></a>  <span class="co">#  the probability by the length of the genome and the binned mutations</span></span>
<span id="cb10-27"><a href="#cb10-27"></a>  <span class="co">#  divided by the total number of mutations.</span></span>
<span id="cb10-28"><a href="#cb10-28"></a>  <span class="fu">mutate</span>(<span class="at">m_count_expected =</span> <span class="fu">cumsum</span>((m_count <span class="sc">/</span> Ntot) <span class="sc">*</span></span>
<span id="cb10-29"><a href="#cb10-29"></a>                                     total_len <span class="sc">*</span></span>
<span id="cb10-30"><a href="#cb10-30"></a>                                     <span class="fu">dpois</span>(m, <span class="at">lambda =</span> Ntot <span class="sc">/</span> total_len))) <span class="sc">%&gt;%</span></span>
<span id="cb10-31"><a href="#cb10-31"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>num_contigs) <span class="sc">%&gt;%</span></span>
<span id="cb10-32"><a href="#cb10-32"></a>  <span class="fu">relocate</span>(m, n, Ntot, perc, m_count, m_count_expected) <span class="sc">%&gt;%</span></span>
<span id="cb10-33"><a href="#cb10-33"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb10-34"><a href="#cb10-34"></a></span>
<span id="cb10-35"><a href="#cb10-35"></a><span class="co"># setup for plotting</span></span>
<span id="cb10-36"><a href="#cb10-36"></a>nuc_survival_plot <span class="ot">&lt;-</span> nuc_survival <span class="sc">%&gt;%</span> </span>
<span id="cb10-37"><a href="#cb10-37"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb10-38"><a href="#cb10-38"></a>  <span class="co"># when there is only one multiplicity observed for a mutation filter such</span></span>
<span id="cb10-39"><a href="#cb10-39"></a>  <span class="co"># that the multiplicty of that mutation must be greater than one.</span></span>
<span id="cb10-40"><a href="#cb10-40"></a>  <span class="co"># Otherwise include all remaining mutations (m &gt; 0)</span></span>
<span id="cb10-41"><a href="#cb10-41"></a>  <span class="fu">filter</span>(<span class="fu">case_when</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> m <span class="sc">&gt;</span> <span class="dv">1</span>,</span>
<span id="cb10-42"><a href="#cb10-42"></a>                       <span class="cn">TRUE</span> <span class="sc">~</span> m <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-43"><a href="#cb10-43"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"m_count"</span>, <span class="st">"m_count_expected"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-44"><a href="#cb10-44"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">paste</span>(strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-45"><a href="#cb10-45"></a>  <span class="co"># and make final plot</span></span>
<span id="cb10-46"><a href="#cb10-46"></a>  <span class="fu">plot_nuc_survival</span>(., <span class="dv">5000</span>, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">1000</span>, <span class="dv">10000</span>), <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_parallelism_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Distribution of nucleotide multiplicity <span class="math inline">\((m_{i})\)</span> - i.e., the number of replicates with with the same genomic mutation including site and alternative allele - for each species with multiplicities <span class="math inline">\(\gt 1\)</span>. The observed data is shown in blue while the null expectation is shown in red. In all cases the observed <span class="math inline">\(m_{i}\)</span> exceeded the null expectation meaning that we must reject the simple null model of uniform mutation distribution in favor of the alternative model where mutations are nonrandom and cluster across replicate populations.
</figcaption>
</figure>
</div>
</section>
<section id="conclusion" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">3.2</span> Conclusion</h2>
<p>Nucleotide parallelism results are presented in Figure <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>. For these species this very simple null model mostly predicts that we should expect fewer than two parallel mutations (same position, same alternative allele) across the three replicate populations. For 12 evolved species the observed data show an excess of nucleotide parallelism relative to this simple null expectation. In particular HAMBI_0006, HAMBI_1287, and HAMBI_2494 all have at least 1 mutation that is identical across all three replicate evolved populations. However, multi-hit sites are only a very small fraction of the total observed mutations across all species (~ 3% across species with multi-hit sites). Thus, we will continue our analysis but looking at the gene level.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>nuc_survival <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">mutate</span>(<span class="at">Ntot_f =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  <span class="fu">summarize</span>(<span class="at">perc =</span> <span class="fu">sum</span>(n)<span class="sc">/</span>Ntot_f<span class="sc">*</span><span class="dv">100</span>, <span class="at">.by =</span> <span class="st">"m"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">mutate</span>(<span class="at">cum_perc =</span> <span class="fu">cumsum</span>(perc))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["m"],"name":[1],"type":["int"],"align":["right"]},{"label":["perc"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["cum_perc"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"1.285977","3":"1.285977"},{"1":"2","2":"2.000408","3":"3.286385"},{"1":"1","2":"96.713615","3":"100.000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
<section id="parallelism-at-gene-level" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Parallelism at gene level</h1>
<p>If selection pressures and mutation rates did not vary between genes, the number of mutations in each gene should be proportional to the target gene size. We assume that gene length is a primary driver of the target size. We then define a multiplicity for each gene according to:</p>
<p><span id="eq-01"><span class="math display">\[m_{i} = n_{i} \cdot \frac{\overline{L}}{L_{i}} \tag{1}\]</span></span></p>
<p>where <span class="math inline">\(n_{i}\)</span> is the number of nonsynonymous mutations (i.e., amino acid changing) in <span class="math inline">\(gene_{i}\)</span> across all replicate populations per species, <span class="math inline">\(L_{i}\)</span> is the total number of nonsynonymous sites (i.e., excluding 4-fold degenerate sites) in <span class="math inline">\(gene_{i}\)</span>, and <span class="math inline">\(\overline{L}\)</span> is the average value of <span class="math inline">\(L_{i}\)</span> across all genes in the genome. This definition ensures that under the null hypothesis, all genes have the same expected multiplicity <span class="math inline">\(\overline{m} = n_{tot}/n_{genes}\)</span>.</p>
<p>We next define a null model for gene multiplicity that assumes mutations are assigned to genes with probability:</p>
<p><span id="eq-02"><span class="math display">\[p_{i} \propto L_{i} r_{i} \tag{2}\]</span></span></p>
<p>for some set of enrichment factors <span class="math inline">\(r_{i}\)</span>. In the alternate model, the maximum likelihood estimator of the enrichment factors <span class="math inline">\(r_{i}\)</span> is the ratio of observed to expected gene multiplicities, <span class="math inline">\(r_{i} = m_{i}/\overline{m}\)</span>. The net increase of the log-likelihood relative to the null model (<span class="math inline">\(r_{i} = 1\)</span>) is then given by:</p>
<p><span id="eq-03"><span class="math display">\[\Delta \ell = \sum_{i} n_{i} \log \left ( \frac{m_{i}}{\overline{m}} \right ) \tag{3}\]</span></span></p>
<p>This likelihood ratio estimator is equivalent to the total G-score introduced in earlier work <span class="citation" data-cites="tenaillon2016">(<a href="#ref-tenaillon2016" role="doc-biblioref">Tenaillon et al. 2016</a>)</span>. Here we assess statistical significance using permutation tests following Shoemaker <em>et al.</em> <span class="citation" data-cites="shoemaker2021">(<a href="#ref-shoemaker2021" role="doc-biblioref">Shoemaker et al. 2021</a>)</span>. For permutations tests (n = 10,000) we randomly resample mutations to the observed <span class="math inline">\(n_{tot}\)</span> set size from each species using the multinomial distribution, where the probability of sampling a non-synonymous mutation at <span class="math inline">\(gene_{i}\)</span> was given by <span class="math inline">\(p_{i} = {L_{i}} / {L_{tot}}\)</span> where <span class="math inline">\(L_{i}\)</span> is the total number of nonsynonymous sites in <span class="math inline">\(gene_{i}\)</span> and <span class="math inline">\(L_{tot}\)</span> is the total number of nonsynonymous sites in the genome.</p>
<p>We next identify specific genes enriched for mutations following the criteria of Good <em>et. al</em> . <span class="citation" data-cites="good2017">(<a href="#ref-good2017" role="doc-biblioref">Good et al. 2017</a>)</span>. We calculate a value for each gene using the equation:</p>
<p><span id="eq-04"><span class="math display">\[P_{i} = \sum_{n \geq n_{i}} \frac{ \left ( \frac{n_{tot} L_{i}}{ \overline{L} N_{genes}} \right )^{n}}{n!} e^{- \frac{n_{tot} L_{i} }{\overline{L} N_{genes}}} \tag{4}\]</span></span></p>
<p>Here we only consider genes with <span class="math inline">\(n_{i} \geq 3\)</span> to exclude low values driven primarily by gene length. Under the null hypothesis, the expected number of genes with <span class="math inline">\(P_{i} \geq p\)</span> can be found using a Poisson survival curve given by:</p>
<p><span id="eq-05"><span class="math display">\[\overline{N}(P) \approx \sum_{i=1}^{N_{genes}} \sum_{n=3}^{\infty} \theta (P - P_{i}(n, L_{i}) \cdot \frac{ \left( \frac{n_{tot} L_{i}}{\overline{L} N_{genes}} \right)^{n}}{n!} e^{-\left( \frac{n_{tot} L_{i}}{\overline{L} N_{genes}} \right)} \tag{5}\]</span></span></p>
<p>We can compare this expected number to the observed number of genes <span class="math inline">\(N(P)\)</span> using a critical value (<span class="math inline">\(P^{\ast}\)</span>) such that</p>
<p><span id="eq-06"><span class="math display">\[\frac{\overline{N}(P^{\ast})}{N(P^{\ast})} \leq \alpha \tag{6}\]</span></span></p>
<p>for a given FDR <span class="math inline">\(\alpha\)</span> value of 0.05. For this value we then define the set of significantly enriched genes as:</p>
<p><span id="eq-07"><span class="math display">\[I = \left\{ i : P_{i} \leq P^{\ast} \left(\alpha \right) \right\} \tag{7}\]</span></span></p>
<section id="prepare-input-data-for-non-syn-mutations-only" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="prepare-input-data-for-non-syn-mutations-only"><span class="header-section-number">4.1</span> Prepare input data for non-syn mutations only</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>mgvars_filt_ns <span class="ot">&lt;-</span> mgvars_filt_mb <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="co"># exclude intragenic, intergenic, and synonymous mutations. Also exclude</span></span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="co"># fusion functions because these are weird and also rare. Excluding the</span></span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="co"># modifier category also ensures that we filter out any tRNAs with mutations</span></span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(effect, <span class="st">"intergenic|intragenic|synonymous|fusion"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(impact, <span class="st">"MODIFIER"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, replicate, chrom<span class="sc">:</span>alt, locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="fu">relocate</span>(strainID, locus_tag)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="g-score" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="g-score"><span class="header-section-number">4.2</span> G score</h2>
<p>Warning! this takes some time</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>muts_by_sp <span class="ot">&lt;-</span> mgvars_filt_ns <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">mutate</span>(<span class="at">groupid =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>  <span class="fu">group_by</span>(groupid, strainID)</span>
<span id="cb13-6"><a href="#cb13-6"></a></span>
<span id="cb13-7"><a href="#cb13-7"></a><span class="co"># doing 10000 permutations</span></span>
<span id="cb13-8"><a href="#cb13-8"></a>result_sp <span class="ot">&lt;-</span> <span class="fu">run_full_gene_parallelism_pipeline</span>(muts_by_sp, degentab, <span class="dv">10000</span>)</span>
<span id="cb13-9"><a href="#cb13-9"></a></span>
<span id="cb13-10"><a href="#cb13-10"></a><span class="co"># Save this for later</span></span>
<span id="cb13-11"><a href="#cb13-11"></a><span class="fu">write_rds</span>(result_sp, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"parallelism_results_sp_nomobile.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co"># Read in previous results</span></span>
<span id="cb14-2"><a href="#cb14-2"></a>result_sp <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"parallelism_results_sp_nomobile.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="simulating-the-number-of-replicates-with-a-parallel-gene-hit" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="simulating-the-number-of-replicates-with-a-parallel-gene-hit"><span class="header-section-number">4.2.1</span> Simulating the number of replicates with a parallel gene hit</h3>
<p>As another sanity check, we took the resamplings from the null distribution and calculated the number of replicates with a parallel gene hit (i.e.&nbsp;the same gene hit in <span class="math inline">\(\geq 2\)</span> experimental replicates) and compared that to the observed distribution. The null distribution reflected the process of sampling a mutation under a multinomial distribution with probability proportional to the total number of nonsynonymous sites <span class="math inline">\(L_{i}\)</span> in <span class="math inline">\(gene_{i}\)</span>. We observed that in 9/16 cases the null distribution produced fewer parallel gene hits than observed in the data from 2 experimental replicates. In all 16 cases the null distribution produced fewer parallel gene hits than observed in the data from 3 experimental replicates. Thus, for some suspicious genes with very high mutational density (see below) we required these genes to be present in all three replicates to include them in our final significant gene lists.</p>
<div id="fig-03a" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03a-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_parallelism_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03a-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Number of genes (y axis) with nonsynonymous mutation hits in <span class="math inline">\(N_{replicates} = \{n : 1 \leq n \leq 3\}\)</span> (x axis). Blue bars depict the distribution observed in the data, while red bars depict the null distribution from resampling mutations to the observed <span class="math inline">\(n_{tot}\)</span> set size from each species using the multinomial distribution, where the probability of sampling a non-synonymous mutation at <span class="math inline">\(gene_{i}\)</span> was dependent only on the number of nonsynonymous sites in the gene.
</figcaption>
</figure>
</div>
<div id="fig-03b" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03b-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_parallelism_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03b-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Same as in <a href="#fig-03a" class="quarto-xref">Figure&nbsp;2</a> but for different treatment combinations.
</figcaption>
</figure>
</div>
</section>
<section id="visualize-the-survival-curves-for-each-treatment-combination" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="visualize-the-survival-curves-for-each-treatment-combination"><span class="header-section-number">4.2.2</span> Visualize the survival curves for each treatment combination</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>survcurv <span class="ot">&lt;-</span> result_sp<span class="sc">$</span>output_df2plot <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(xend)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb15-5"><a href="#cb15-5"></a>    <span class="fu">geom_segment</span>(</span>
<span id="cb15-6"><a href="#cb15-6"></a>      <span class="fu">aes</span>(</span>
<span id="cb15-7"><a href="#cb15-7"></a>        <span class="at">x =</span> x,</span>
<span id="cb15-8"><a href="#cb15-8"></a>        <span class="at">y =</span> y,</span>
<span id="cb15-9"><a href="#cb15-9"></a>        <span class="at">xend =</span> xend,</span>
<span id="cb15-10"><a href="#cb15-10"></a>        <span class="at">yend =</span> yend</span>
<span id="cb15-11"><a href="#cb15-11"></a>      ),</span>
<span id="cb15-12"><a href="#cb15-12"></a>      <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb15-13"><a href="#cb15-13"></a>      <span class="at">linewidth =</span> <span class="fl">0.25</span>, <span class="at">color =</span> <span class="st">"red"</span></span>
<span id="cb15-14"><a href="#cb15-14"></a>    ) <span class="sc">+</span></span>
<span id="cb15-15"><a href="#cb15-15"></a>    <span class="fu">geom_step</span>( <span class="fu">aes</span>(<span class="at">x =</span> obs_p, <span class="at">y =</span> value, <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb15-16"><a href="#cb15-16"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> label, <span class="at">ncol =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb15-17"><a href="#cb15-17"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"$-log_{10}P$"</span>), <span class="at">y =</span> <span class="st">"Number of genes"</span>) <span class="sc">+</span></span>
<span id="cb15-18"><a href="#cb15-18"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"grey50"</span>)) <span class="sc">+</span></span>
<span id="cb15-19"><a href="#cb15-19"></a>    <span class="fu">scale_y_log10</span>(</span>
<span id="cb15-20"><a href="#cb15-20"></a>      <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb15-21"><a href="#cb15-21"></a>      <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">trans_format</span>(<span class="st">"log10"</span>, scales<span class="sc">::</span><span class="fu">math_format</span>(<span class="dv">10</span> <span class="sc">^</span>.x))</span>
<span id="cb15-22"><a href="#cb15-22"></a>    ) <span class="sc">+</span></span>
<span id="cb15-23"><a href="#cb15-23"></a>    <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb15-24"><a href="#cb15-24"></a>    <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">200</span>)) <span class="sc">+</span></span>
<span id="cb15-25"><a href="#cb15-25"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb15-26"><a href="#cb15-26"></a>    <span class="fu">theme</span>(</span>
<span id="cb15-27"><a href="#cb15-27"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-28"><a href="#cb15-28"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb15-29"><a href="#cb15-29"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb15-30"><a href="#cb15-30"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_parallelism_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: The observed <span class="math inline">\((N)\)</span> (blue line) and expected number <span class="math inline">\((\overline{N})\)</span> (grey line) of genes with <span class="math inline">\(P_{i} \leq P\)</span>, as a function of <span class="math inline">\(P\)</span>. The red dashed line denotes the genome-wide significance threshold <span class="math inline">\(P^{*}\)</span> for <span class="math inline">\(\alpha = 0.05\)</span> defined in Equation <a href="#eq-06" class="quarto-xref">Equation&nbsp;6</a>.
</figcaption>
</figure>
</div>
</section>
<section id="inspecting-individual-genes" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="inspecting-individual-genes"><span class="header-section-number">4.2.3</span> Inspecting individual genes</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>result_sp<span class="sc">$</span>output_gscore <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2"></a>  <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["groupid"],"name":[1],"type":["int"],"align":["right"]},{"label":["observed_g_score"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["pvalue"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["strainID"],"name":[4],"type":["chr"],"align":["left"]}],"data":[{"1":"1","2":"2.332718","3":"0.0000","4":"HAMBI_0006"},{"1":"2","2":"6.257592","3":"0.0000","4":"HAMBI_0097"},{"1":"3","2":"4.947171","3":"0.0209","4":"HAMBI_0262"},{"1":"4","2":"4.804133","3":"0.0028","4":"HAMBI_0403"},{"1":"5","2":"5.129346","3":"0.0000","4":"HAMBI_1279"},{"1":"8","2":"5.026324","3":"0.0433","4":"HAMBI_1842"},{"1":"9","2":"4.583821","3":"0.0001","4":"HAMBI_1972"},{"1":"10","2":"5.627219","3":"0.0000","4":"HAMBI_1977"},{"1":"11","2":"2.502411","3":"0.0000","4":"HAMBI_2159"},{"1":"12","2":"3.047936","3":"0.0052","4":"HAMBI_2164"},{"1":"13","2":"5.262213","3":"0.0000","4":"HAMBI_2443"},{"1":"14","2":"4.892053","3":"0.0000","4":"HAMBI_2494"},{"1":"16","2":"2.169519","3":"0.0001","4":"HAMBI_3237"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Also some genes tend to have a large number of consecutive mutations which is probably due to incorrect read mapping and not real mutations so we will try and filter these out. We’ll do this by filtering on genes with a mutational density <span class="math inline">\(d_{g} \gt 0.15\)</span>, where the mutational density in gene <span class="math inline">\(g\)</span> was defined as</p>
<p><span class="math display">\[d_{g} = \frac{|A|_{g}}{b_{g} - a_{g}}\]</span></p>
<p>for all genome positions <span class="math inline">\(x\)</span> on the interval <span class="math inline">\(\{x | a_{g} \leq x \leq b_{g}\}\)</span> where <span class="math inline">\(b_{g}\)</span> is the greatest position and <span class="math inline">\(a_{g}\)</span> is the smallest position of observed mutations in gene <span class="math inline">\(g\)</span> and <span class="math inline">\(|A|_{g}\)</span> is the cardinality of the mutation set.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>high_density_genes <span class="ot">&lt;-</span> mgvars_filt_ns <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2"></a>  <span class="fu">group_by</span>(strainID, locus_tag, replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="fu">mutate</span>(<span class="at">mutwindox =</span> pos <span class="sc">-</span> <span class="fu">lag</span>(pos),</span>
<span id="cb17-4"><a href="#cb17-4"></a>         <span class="at">mut_num =</span> <span class="fu">n</span>(),</span>
<span id="cb17-5"><a href="#cb17-5"></a>         <span class="at">mut_dens =</span> <span class="fu">n</span>()<span class="sc">/</span>(<span class="fu">max</span>(pos) <span class="sc">-</span>  <span class="fu">min</span>(pos))) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-7"><a href="#cb17-7"></a>  <span class="fu">filter</span>(mut_dens <span class="sc">!=</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-8"><a href="#cb17-8"></a>  <span class="fu">filter</span>(mut_dens <span class="sc">&gt;</span> <span class="fl">0.15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-9"><a href="#cb17-9"></a>  <span class="fu">distinct</span>(locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10"></a>  <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>output_gene_table_filt <span class="ot">&lt;-</span> result_sp<span class="sc">$</span>output_gene_table <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2"></a>  <span class="co"># only genes exceeding the critical pstar value from the survival curves above</span></span>
<span id="cb18-3"><a href="#cb18-3"></a>  <span class="fu">filter</span>(neg_log10P <span class="sc">&gt;=</span> pstar) <span class="sc">%&gt;%</span> </span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="co"># only include genes with at least 3 hits in the same gene or found in more than one replicate.</span></span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="fu">filter</span>(observed_hits_n_i <span class="sc">&gt;=</span> <span class="dv">3</span> <span class="sc">|</span> n_replicate <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6"></a>  <span class="co"># filters out the problematic genes above unless the number of replicates detected is 2 or more</span></span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="fu">filter</span>(<span class="fu">case_when</span>(n_replicate <span class="sc">&lt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="sc">!</span>(locus_tag <span class="sc">%in%</span> high_density_genes), </span>
<span id="cb18-8"><a href="#cb18-8"></a>                   <span class="cn">TRUE</span> <span class="sc">~</span> n_replicate <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-9"><a href="#cb18-9"></a>  <span class="fu">left_join</span>(<span class="fu">distinct</span>(<span class="fu">select</span>(mgvars_filt, locus_tag, product, cds_length<span class="sc">:</span>gene)),</span>
<span id="cb18-10"><a href="#cb18-10"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag))</span>
<span id="cb18-11"><a href="#cb18-11"></a></span>
<span id="cb18-12"><a href="#cb18-12"></a>output_gene_table_filt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["groupid"],"name":[1],"type":["int"],"align":["right"]},{"label":["locus_tag"],"name":[2],"type":["chr"],"align":["left"]},{"label":["neg_log10P"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["strainID"],"name":[4],"type":["chr"],"align":["left"]},{"label":["pstar"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["nonsyn_sites"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["observed_hits_n_i"],"name":[7],"type":["int"],"align":["right"]},{"label":["n_replicate"],"name":[8],"type":["int"],"align":["right"]},{"label":["gene_multiplicity_m_i"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["A"],"name":[10],"type":["int"],"align":["right"]},{"label":["C"],"name":[11],"type":["int"],"align":["right"]},{"label":["E"],"name":[12],"type":["int"],"align":["right"]},{"label":["product"],"name":[13],"type":["chr"],"align":["left"]},{"label":["cds_length"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["ns_length"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["f0"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["f2"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["f3"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["f4"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["gene"],"name":[20],"type":["chr"],"align":["left"]}],"data":[{"1":"1","2":"H0006_03238","3":"9.598285","4":"HAMBI_0006","5":"9.598285","6":"575","7":"3","8":"3","9":"4.1417546","10":"1","11":"1","12":"1","13":"HTH-type transcriptional repressor GlaR","14":"696","15":"575","16":"448","17":"114","18":"13","19":"121","20":"glaR_2"},{"1":"1","2":"H0006_02025","3":"45.294967","4":"HAMBI_0006","5":"9.598285","6":"1015","7":"12","8":"3","9":"9.3852567","10":"4","11":"4","12":"4","13":"hypothetical protein","14":"1239","15":"1015","16":"793","17":"203","18":"19","19":"224","20":"NA"},{"1":"2","2":"H0097_02729","3":"15.327500","4":"HAMBI_0097","5":"15.327500","6":"1176","7":"3","8":"3","9":"1.9014216","10":"1","11":"1","12":"1","13":"hypothetical protein","14":"1377","15":"1176","16":"905","17":"247","18":"24","19":"201","20":"NA"},{"1":"3","2":"H0262_00934","3":"7.475263","4":"HAMBI_0262","5":"7.475263","6":"3374","7":"2","8":"2","9":"0.4430084","10":"1","11":"0","12":"1","13":"DNA-directed RNA polymerase subunit beta","14":"4116","15":"3374","16":"2685","17":"610","18":"79","19":"742","20":"rpoB"},{"1":"3","2":"H0262_00732","3":"9.359634","4":"HAMBI_0262","5":"7.475263","6":"1306","7":"2","8":"2","9":"1.1444949","10":"0","11":"1","12":"1","13":"Adaptive-response sensory-kinase SasA","14":"1677","15":"1306","16":"1049","17":"231","18":"26","19":"371","20":"sasA_7"},{"1":"3","2":"H0262_00089","3":"10.311679","4":"HAMBI_0262","5":"7.475263","6":"810","7":"2","8":"2","9":"1.8453214","10":"1","11":"1","12":"0","13":"putative ABC transporter phosphonate/phosphite binding protein PhnD2","14":"1005","15":"810","16":"657","17":"134","18":"19","19":"195","20":"phnD2"},{"1":"4","2":"H0403_00124","3":"15.463626","4":"HAMBI_0403","5":"10.903080","6":"943","7":"3","8":"3","9":"2.5029135","10":"1","11":"1","12":"1","13":"Adaptive-response sensory-kinase SasA","14":"1173","15":"943","16":"743","17":"182","18":"18","19":"230","20":"sasA_2"},{"1":"4","2":"H0403_04847","3":"20.433397","4":"HAMBI_0403","5":"10.903080","6":"1203","7":"4","8":"3","9":"2.6159572","10":"1","11":"1","12":"2","13":"Glycerol kinase","14":"1497","15":"1203","16":"973","17":"210","18":"20","19":"294","20":"glpK"},{"1":"5","2":"H1279_00765","3":"19.904266","4":"HAMBI_1279","5":"19.904266","6":"204","7":"3","8":"2","9":"11.6119159","10":"1","11":"2","12":"0","13":"RNA polymerase sigma factor RpoS","14":"240","15":"204","16":"155","17":"48","18":"1","19":"36","20":"rpoS_1"},{"1":"5","2":"H1279_02096","3":"42.930117","4":"HAMBI_1279","5":"19.904266","6":"704","7":"8","8":"1","9":"8.9728441","10":"0","11":"0","12":"8","13":"Vitamin B12 import system permease protein BtuC","14":"882","15":"704","16":"564","17":"126","18":"14","19":"178","20":"btuC"},{"1":"6","2":"H1287_00316","3":"10.073569","4":"HAMBI_1287","5":"9.349728","6":"726","7":"2","8":"2","9":"2.1958013","10":"1","11":"1","12":"0","13":"Transcriptional regulatory protein OmpR","14":"873","15":"726","16":"565","17":"146","18":"15","19":"147","20":"ompR_1"},{"1":"6","2":"H1287_02172","3":"20.975671","4":"HAMBI_1287","5":"9.349728","6":"923","7":"4","8":"3","9":"3.4542833","10":"1","11":"2","12":"1","13":"hypothetical protein","14":"1104","15":"923","16":"715","17":"184","18":"24","19":"181","20":"NA"},{"1":"8","2":"H1842_00573","3":"14.480959","4":"HAMBI_1842","5":"9.756535","6":"1757","7":"3","8":"3","9":"1.3357785","10":"1","11":"1","12":"1","13":"Polyphosphate kinase","14":"2139","15":"1757","16":"1397","17":"310","18":"50","19":"382","20":"ppk"},{"1":"8","2":"H1842_00504","3":"16.270626","4":"HAMBI_1842","5":"9.756535","6":"966","7":"3","8":"3","9":"2.4295681","10":"1","11":"1","12":"1","13":"hypothetical protein","14":"1212","15":"966","16":"784","17":"155","18":"27","19":"246","20":"NA"},{"1":"9","2":"H1972_02204","3":"9.932559","4":"HAMBI_1972","5":"9.932559","6":"667","7":"2","8":"2","9":"2.3912623","10":"1","11":"0","12":"1","13":"Septum site-determining protein MinD","14":"813","15":"667","16":"518","17":"130","18":"19","19":"146","20":"minD"},{"1":"9","2":"H1972_02723","3":"11.294714","4":"HAMBI_1972","5":"9.932559","6":"337","7":"2","8":"2","9":"4.7328544","10":"0","11":"1","12":"1","13":"hypothetical protein","14":"402","15":"337","16":"265","17":"68","18":"4","19":"65","20":"NA"},{"1":"9","2":"H1972_02205","3":"12.006219","4":"HAMBI_1972","5":"9.932559","6":"236","7":"2","8":"2","9":"6.7583556","10":"1","11":"1","12":"0","13":"Cell division topological specificity factor","14":"270","15":"236","16":"172","17":"57","18":"7","19":"34","20":"minE"},{"1":"10","2":"H1977_00727","3":"19.411635","4":"HAMBI_1977","5":"9.274343","6":"407","7":"3","8":"2","9":"5.9737569","10":"0","11":"2","12":"1","13":"Peptidoglycan-associated lipoprotein","14":"504","15":"407","16":"314","17":"86","18":"7","19":"97","20":"pal_1"},{"1":"10","2":"H1977_04823","3":"31.063477","4":"HAMBI_1977","5":"9.274343","6":"755","7":"5","8":"3","9":"5.3671502","10":"1","11":"3","12":"1","13":"UDP-3-O-acyl-N-acetylglucosamine deacetylase","14":"912","15":"755","16":"598","17":"139","18":"18","19":"157","20":"lpxC"},{"1":"11","2":"H2159_02256","3":"7.307044","4":"HAMBI_2159","5":"7.307044","6":"939","7":"3","8":"1","9":"2.5741666","10":"0","11":"0","12":"3","13":"4-oxalomesaconate tautomerase","14":"1200","15":"939","16":"793","17":"129","18":"17","19":"261","20":"galD"},{"1":"11","2":"H2159_02042","3":"9.697768","4":"HAMBI_2159","5":"7.307044","6":"1156","7":"4","8":"1","9":"2.7879382","10":"0","11":"0","12":"4","13":"Outer membrane protein OprM","14":"1488","15":"1156","16":"982","17":"162","18":"12","19":"332","20":"oprM_4"},{"1":"11","2":"H2159_00397","3":"9.878062","4":"HAMBI_2159","5":"7.307044","6":"1103","7":"4","8":"1","9":"2.9219008","10":"0","11":"0","12":"4","13":"Formyl-CoA:oxalate CoA-transferase","14":"1353","15":"1103","16":"891","17":"190","18":"22","19":"250","20":"frc_1"},{"1":"11","2":"H2159_01932","3":"10.134239","4":"HAMBI_2159","5":"7.307044","6":"1032","7":"4","8":"1","9":"3.1229230","10":"0","11":"0","12":"4","13":"Histidinol-phosphate aminotransferase","14":"1341","15":"1032","16":"860","17":"160","18":"12","19":"309","20":"hisC_3"},{"1":"11","2":"H2159_03087","3":"10.935307","4":"HAMBI_2159","5":"7.307044","6":"839","7":"4","8":"1","9":"3.8413070","10":"4","11":"0","12":"0","13":"Ketol-acid reductoisomerase (NADP(+))","14":"1017","15":"839","16":"676","17":"142","18":"21","19":"178","20":"ilvC"},{"1":"11","2":"H2159_01969","3":"11.563045","4":"HAMBI_2159","5":"7.307044","6":"714","7":"4","8":"1","9":"4.5138047","10":"0","11":"0","12":"4","13":"Glycine cleavage system transcriptional activator","14":"894","15":"714","16":"576","17":"129","18":"9","19":"180","20":"gcvA_3"},{"1":"11","2":"H2159_00326","3":"12.684995","4":"HAMBI_2159","5":"7.307044","6":"536","7":"4","8":"1","9":"6.0127921","10":"4","11":"0","12":"0","13":"hypothetical protein","14":"648","15":"536","16":"416","17":"112","18":"8","19":"112","20":"NA"},{"1":"11","2":"H2159_01966","3":"14.409013","4":"HAMBI_2159","5":"7.307044","6":"346","7":"4","8":"2","9":"9.3146143","10":"2","11":"2","12":"0","13":"hypothetical protein","14":"423","15":"346","16":"273","17":"69","18":"4","19":"77","20":"NA"},{"1":"11","2":"H2159_02708","3":"17.700831","4":"HAMBI_2159","5":"7.307044","6":"908","7":"6","8":"1","9":"5.3241022","10":"6","11":"0","12":"0","13":"S-(hydroxymethyl)glutathione dehydrogenase","14":"1107","15":"908","16":"737","17":"143","18":"28","19":"199","20":"flhA_2"},{"1":"11","2":"H2159_03364","3":"20.944826","4":"HAMBI_2159","5":"7.307044","6":"1605","7":"8","8":"1","9":"4.0160206","10":"8","11":"0","12":"0","13":"Phosphomethylpyrimidine synthase","14":"1932","15":"1605","16":"1294","17":"265","18":"46","19":"327","20":"thiC"},{"1":"11","2":"H2159_01982","3":"21.552718","4":"HAMBI_2159","5":"7.307044","6":"3125","7":"10","8":"1","9":"2.5782852","10":"10","11":"0","12":"0","13":"Bifunctional protein PutA","14":"3942","15":"3125","16":"2580","17":"485","18":"60","19":"817","20":"putA"},{"1":"11","2":"H2159_02466","3":"36.736801","4":"HAMBI_2159","5":"7.307044","6":"1058","7":"12","8":"1","9":"9.1385346","10":"12","11":"0","12":"0","13":"Cell division protein FtsP","14":"1296","15":"1058","16":"878","17":"168","18":"12","19":"238","20":"ftsP_1"},{"1":"11","2":"H2159_00234","3":"59.762651","4":"HAMBI_2159","5":"7.307044","6":"823","7":"24","8":"2","9":"23.4959165","10":"23","11":"0","12":"1","13":"Alcohol dehydrogenase","14":"1026","15":"823","16":"683","17":"118","18":"22","19":"203","20":"NA"},{"1":"11","2":"H2159_00323","3":"59.762651","4":"HAMBI_2159","5":"7.307044","6":"376","7":"18","8":"1","9":"38.5714215","10":"18","11":"0","12":"0","13":"hypothetical protein","14":"474","15":"376","16":"305","17":"61","18":"10","19":"98","20":"NA"},{"1":"11","2":"H2159_00325","3":"59.762651","4":"HAMBI_2159","5":"7.307044","6":"723","7":"26","8":"1","9":"28.9745056","10":"26","11":"0","12":"0","13":"Thioredoxin reductase","14":"894","15":"723","16":"593","17":"108","18":"22","19":"171","20":"trxB_1"},{"1":"11","2":"H2159_01425","3":"59.762651","4":"HAMBI_2159","5":"7.307044","6":"1453","7":"17","8":"2","9":"9.4268000","10":"16","11":"0","12":"1","13":"hypothetical protein","14":"1821","15":"1453","16":"1188","17":"244","18":"21","19":"368","20":"NA"},{"1":"12","2":"H2164_00335","3":"9.762398","4":"HAMBI_2164","5":"8.709893","6":"1243","7":"3","8":"1","9":"1.9090852","10":"3","11":"0","12":"0","13":"Long-chain-fatty-acid--CoA ligase FadD13","14":"1542","15":"1243","16":"1005","17":"216","18":"22","19":"299","20":"NA"},{"1":"12","2":"H2164_03223","3":"11.890227","4":"HAMBI_2164","5":"8.709893","6":"606","7":"3","8":"1","9":"3.9158299","10":"3","11":"0","12":"0","13":"GTP cyclohydrolase 1 type 2","14":"747","15":"606","16":"480","17":"116","18":"10","19":"141","20":"ybgI"},{"1":"13","2":"H2443_02447","3":"9.451882","4":"HAMBI_2443","5":"9.451882","6":"1161","7":"2","8":"2","9":"1.2749904","10":"0","11":"1","12":"1","13":"hypothetical protein","14":"1416","15":"1161","16":"919","17":"212","18":"30","19":"255","20":"NA"},{"1":"13","2":"H2443_03044","3":"10.026868","4":"HAMBI_2443","5":"9.451882","6":"870","7":"2","8":"2","9":"1.7014526","10":"0","11":"1","12":"1","13":"Pseudaminic acid synthase","14":"1077","15":"870","16":"699","17":"150","18":"21","19":"207","20":"pseI"},{"1":"13","2":"H2443_02540","3":"24.250682","4":"HAMBI_2443","5":"9.451882","6":"4888","7":"6","8":"1","9":"0.9085089","10":"0","11":"6","12":"0","13":"hypothetical protein","14":"6297","15":"4888","16":"4134","17":"753","18":"1","19":"1409","20":"NA"},{"1":"13","2":"H2443_02537","3":"34.657359","4":"HAMBI_2443","5":"9.451882","6":"849","7":"6","8":"2","9":"5.2306141","10":"3","11":"3","12":"0","13":"hypothetical protein","14":"1083","15":"849","16":"699","17":"137","18":"13","19":"234","20":"NA"},{"1":"14","2":"H2494_03556","3":"7.020468","4":"HAMBI_2494","5":"7.020468","6":"3454","7":"2","8":"2","9":"0.4597794","10":"1","11":"1","12":"0","13":"DNA-directed RNA polymerase subunit beta'","14":"4242","15":"3454","16":"2777","17":"591","18":"86","19":"788","20":"rpoC"},{"1":"14","2":"H2494_00166","3":"8.376476","4":"HAMBI_2494","5":"7.020468","6":"1741","7":"2","8":"2","9":"0.9121644","10":"1","11":"1","12":"0","13":"Acetone carboxylase beta subunit","14":"2145","15":"1741","16":"1417","17":"291","18":"33","19":"404","20":"acxA_1"},{"1":"14","2":"H2494_04782","3":"9.418470","4":"HAMBI_2494","5":"7.020468","6":"1031","7":"2","8":"2","9":"1.5403280","10":"1","11":"1","12":"0","13":"Phosphoribosylamine--glycine ligase","14":"1278","15":"1031","16":"854","17":"155","18":"22","19":"247","20":"purD"},{"1":"14","2":"H2494_02668","3":"9.729423","4":"HAMBI_2494","5":"7.020468","6":"882","7":"2","8":"2","9":"1.8005422","10":"1","11":"1","12":"0","13":"Flavin-dependent trigonelline monooxygenase%2C oxygenase component","14":"1059","15":"882","16":"695","17":"169","18":"18","19":"177","20":"tgnB_2"},{"1":"14","2":"H2494_03519","3":"9.884432","4":"HAMBI_2494","5":"7.020468","6":"816","7":"2","8":"2","9":"1.9461742","10":"1","11":"1","12":"0","13":"Delta-aminolevulinic acid dehydratase","14":"999","15":"816","16":"667","17":"134","18":"15","19":"183","20":"hemB"},{"1":"14","2":"H2494_00327","3":"9.979494","4":"HAMBI_2494","5":"7.020468","6":"778","7":"2","8":"2","9":"2.0412316","10":"1","11":"1","12":"0","13":"Formate dehydrogenase%2C nitrate-inducible%2C iron-sulfur subunit","14":"936","15":"778","16":"620","17":"145","18":"13","19":"158","20":"fdnH"},{"1":"14","2":"H2494_01194","3":"10.133965","4":"HAMBI_2494","5":"7.020468","6":"720","7":"2","8":"2","9":"2.2056641","10":"1","11":"1","12":"0","13":"HTH-type transcriptional regulator YhaJ","14":"909","15":"720","16":"585","17":"123","18":"12","19":"189","20":"yhaJ_1"},{"1":"14","2":"H2494_01989","3":"10.167480","4":"HAMBI_2494","5":"7.020468","6":"708","7":"2","8":"2","9":"2.2430483","10":"1","11":"1","12":"0","13":"HTH-type transcriptional regulator GltR","14":"915","15":"708","16":"591","17":"109","18":"8","19":"207","20":"gltR"},{"1":"14","2":"H2494_04167","3":"10.178778","4":"HAMBI_2494","5":"7.020468","6":"704","7":"2","8":"2","9":"2.2557929","10":"1","11":"1","12":"0","13":"4-diphosphocytidyl-2-C-methyl-D-erythritol kinase","14":"882","15":"704","16":"575","17":"122","18":"7","19":"178","20":"ispE"},{"1":"14","2":"H2494_04848","3":"11.404065","4":"HAMBI_2494","5":"7.020468","6":"381","7":"2","8":"2","9":"4.1681842","10":"1","11":"1","12":"0","13":"hypothetical protein","14":"468","15":"381","16":"312","17":"58","18":"11","19":"87","20":"NA"},{"1":"14","2":"H2494_00163","3":"13.509799","4":"HAMBI_2494","5":"7.020468","6":"1629","7":"3","8":"3","9":"1.4623188","10":"1","11":"1","12":"1","13":"Acetoin dehydrogenase operon transcriptional activator AcoR","14":"2028","15":"1629","16":"1316","17":"278","18":"35","19":"399","20":"acoR_1"},{"1":"14","2":"H2494_03421","3":"13.832729","4":"HAMBI_2494","5":"7.020468","6":"1462","7":"3","8":"3","9":"1.6293552","10":"1","11":"1","12":"1","13":"hypothetical protein","14":"1881","15":"1462","16":"1228","17":"217","18":"17","19":"419","20":"NA"},{"1":"14","2":"H2494_04708","3":"26.614418","4":"HAMBI_2494","5":"7.020468","6":"1026","7":"5","8":"3","9":"3.8695862","10":"2","11":"2","12":"1","13":"DNA-cytosine methyltransferase","14":"1275","15":"1026","16":"832","17":"177","18":"17","19":"249","20":"dcm"},{"1":"15","2":"H2659_03614","3":"7.802649","4":"HAMBI_2659","5":"7.802649","6":"2386","7":"2","8":"2","9":"0.6529547","10":"1","11":"1","12":"0","13":"Vitamin B12 transporter BtuB","14":"2958","15":"2386","16":"1931","17":"415","18":"40","19":"572","20":"btuB_26"},{"1":"15","2":"H2659_03908","3":"8.049110","4":"HAMBI_2659","5":"7.802649","6":"2107","7":"2","8":"2","9":"0.7394162","10":"1","11":"0","12":"1","13":"Sensor histidine kinase RcsC","14":"2643","15":"2107","16":"1669","17":"413","18":"25","19":"536","20":"rcsC_14"},{"1":"15","2":"H2659_03917","3":"13.164033","4":"HAMBI_2659","5":"7.802649","6":"1877","7":"3","8":"3","9":"1.2450320","10":"1","11":"1","12":"1","13":"Vitamin B12 transporter BtuB","14":"2298","15":"1877","16":"1493","17":"352","18":"32","19":"421","20":"btuB_32"},{"1":"16","2":"H3237_03943","3":"9.915656","4":"HAMBI_3237","5":"8.897199","6":"426","7":"3","8":"1","9":"4.8959073","10":"0","11":"0","12":"3","13":"hypothetical protein","14":"528","15":"426","16":"340","17":"83","18":"3","19":"102","20":"NA"},{"1":"16","2":"H3237_04517","3":"10.758041","4":"HAMBI_3237","5":"8.897199","6":"974","7":"4","8":"1","9":"2.8551081","10":"0","11":"0","12":"4","13":"hypothetical protein","14":"1272","15":"974","16":"818","17":"143","18":"13","19":"298","20":"NA"},{"1":"16","2":"H3237_00478","3":"11.273316","4":"HAMBI_3237","5":"8.897199","6":"853","7":"4","8":"1","9":"3.2601118","10":"0","11":"0","12":"4","13":"Oligosaccharides import ATP-binding protein MsmX","14":"1089","15":"853","16":"692","17":"147","18":"14","19":"236","20":"msmX_2"},{"1":"16","2":"H3237_02296","3":"16.923100","4":"HAMBI_3237","5":"8.897199","6":"563","7":"5","8":"1","9":"6.1742347","10":"0","11":"0","12":"5","13":"High-affinity branched-chain amino acid transport ATP-binding protein LivF","14":"705","15":"563","16":"450","17":"98","18":"15","19":"142","20":"livF_6"},{"1":"16","2":"H3237_03349","3":"17.455500","4":"HAMBI_3237","5":"8.897199","6":"178","7":"4","8":"1","9":"15.6228951","10":"0","11":"0","12":"4","13":"hypothetical protein","14":"207","15":"178","16":"134","17":"41","18":"3","19":"29","20":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="convergentdivergent-evolution" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Convergent/divergent evolution</h1>
<p>So now we have a list of enriched genes for each species. We can see that, for example, some gene functions pop up multiple times (like transcription factors and sensory kinases). Are these different functions popping up across species more that we would expect by chance? We’ll try and test this using a simulation based approach</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># Read in results of eggnogmapper which maps a COG category to every gene it can in each genome</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>eggnograw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_all.emapper.annotations.xz"</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>cog_description <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="sc">~</span>COG_category_single,                                                  <span class="sc">~</span>COG_category_long,</span>
<span id="cb20-3"><a href="#cb20-3"></a>            <span class="st">"J"</span>,               <span class="st">"J - Translation, ribosomal structure and biogenesis"</span>,</span>
<span id="cb20-4"><a href="#cb20-4"></a>            <span class="st">"A"</span>,                               <span class="st">"A - RNA processing and modification"</span>,</span>
<span id="cb20-5"><a href="#cb20-5"></a>            <span class="st">"K"</span>,                                                 <span class="st">"K – Transcription"</span>,</span>
<span id="cb20-6"><a href="#cb20-6"></a>            <span class="st">"L"</span>,                         <span class="st">"L - Replication, recombination and repair"</span>,</span>
<span id="cb20-7"><a href="#cb20-7"></a>            <span class="st">"B"</span>,                              <span class="st">"B - Chromatin structure and dynamics"</span>,</span>
<span id="cb20-8"><a href="#cb20-8"></a>            <span class="st">"D"</span>,    <span class="st">"D - Cell cycle control, cell division, chromosome partitioning"</span>,</span>
<span id="cb20-9"><a href="#cb20-9"></a>            <span class="st">"Y"</span>,                                             <span class="st">"Y - Nuclear structure"</span>,</span>
<span id="cb20-10"><a href="#cb20-10"></a>            <span class="st">"V"</span>,                                            <span class="st">"V - Defense mechanisms"</span>,</span>
<span id="cb20-11"><a href="#cb20-11"></a>            <span class="st">"T"</span>,                                <span class="st">"T - Signal transduction mechanisms"</span>,</span>
<span id="cb20-12"><a href="#cb20-12"></a>            <span class="st">"M"</span>,                        <span class="st">"M - Cell wall/membrane/envelope biogenesis"</span>,</span>
<span id="cb20-13"><a href="#cb20-13"></a>            <span class="st">"N"</span>,                                                 <span class="st">"N - Cell motility"</span>,</span>
<span id="cb20-14"><a href="#cb20-14"></a>            <span class="st">"Z"</span>,                                                  <span class="st">"Z – Cytoskeleton"</span>,</span>
<span id="cb20-15"><a href="#cb20-15"></a>            <span class="st">"W"</span>,                                      <span class="st">"W - Extracellular structures"</span>,</span>
<span id="cb20-16"><a href="#cb20-16"></a>            <span class="st">"U"</span>, <span class="st">"U - Intracellular trafficking, secretion, and vesicular transport"</span>,</span>
<span id="cb20-17"><a href="#cb20-17"></a>            <span class="st">"O"</span>,  <span class="st">"O - Posttranslational modification, protein turnover, chaperones"</span>,</span>
<span id="cb20-18"><a href="#cb20-18"></a>            <span class="st">"X"</span>,                              <span class="st">"X - Mobilome: prophages, transposons"</span>,</span>
<span id="cb20-19"><a href="#cb20-19"></a>            <span class="st">"C"</span>,                              <span class="st">"C - Energy production and conversion"</span>,</span>
<span id="cb20-20"><a href="#cb20-20"></a>            <span class="st">"G"</span>,                         <span class="st">"G - Carbohydrate transport and metabolism"</span>,</span>
<span id="cb20-21"><a href="#cb20-21"></a>            <span class="st">"E"</span>,                           <span class="st">"E - Amino acid transport and metabolism"</span>,</span>
<span id="cb20-22"><a href="#cb20-22"></a>            <span class="st">"F"</span>,                           <span class="st">"F - Nucleotide transport and metabolism"</span>,</span>
<span id="cb20-23"><a href="#cb20-23"></a>            <span class="st">"H"</span>,                             <span class="st">"H - Coenzyme transport and metabolism"</span>,</span>
<span id="cb20-24"><a href="#cb20-24"></a>            <span class="st">"I"</span>,                                <span class="st">"I - Lipid transport and metabolism"</span>,</span>
<span id="cb20-25"><a href="#cb20-25"></a>            <span class="st">"P"</span>,                        <span class="st">"P - Inorganic ion transport and metabolism"</span>,</span>
<span id="cb20-26"><a href="#cb20-26"></a>            <span class="st">"Q"</span>,  <span class="st">"Q - Secondary metabolites biosynthesis, transport and catabolism"</span>,</span>
<span id="cb20-27"><a href="#cb20-27"></a>            <span class="st">"R"</span>,                              <span class="st">"R - General function prediction only"</span>,</span>
<span id="cb20-28"><a href="#cb20-28"></a>            <span class="st">"S"</span>,                                              <span class="st">"S - Function unknown"</span></span>
<span id="cb20-29"><a href="#cb20-29"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="co"># formatting the eggnog dataframe for downstream use</span></span>
<span id="cb21-2"><a href="#cb21-2"></a>enrichedspprefix <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">paste0</span>(<span class="st">"H"</span>, <span class="fu">str_extract</span>(output_gene_table_filt<span class="sc">$</span>locus_tag, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>)))</span>
<span id="cb21-3"><a href="#cb21-3"></a>enrichedsp <span class="ot">&lt;-</span> <span class="fu">unique</span>(output_gene_table_filt<span class="sc">$</span>strainID)</span>
<span id="cb21-4"><a href="#cb21-4"></a><span class="fu">names</span>(enrichedsp) <span class="ot">&lt;-</span> enrichedspprefix</span>
<span id="cb21-5"><a href="#cb21-5"></a></span>
<span id="cb21-6"><a href="#cb21-6"></a><span class="co"># use a seed for reproducibility because we are randomly sampling the split COGs</span></span>
<span id="cb21-7"><a href="#cb21-7"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">6783543</span>, eggnog <span class="ot">&lt;-</span> eggnograw <span class="sc">%&gt;%</span>  </span>
<span id="cb21-8"><a href="#cb21-8"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(query, <span class="fu">paste</span>(enrichedspprefix, <span class="at">collapse=</span><span class="st">"|"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9"></a>  <span class="fu">mutate</span>(<span class="at">COG_category =</span> <span class="fu">if_else</span>(COG_category <span class="sc">==</span> <span class="st">"-"</span>, <span class="cn">NA_character_</span>, COG_category)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-10"><a href="#cb21-10"></a>  <span class="co"># If multiple cog categories, split them into a character vector then randomly sample</span></span>
<span id="cb21-11"><a href="#cb21-11"></a>  <span class="co"># one of them</span></span>
<span id="cb21-12"><a href="#cb21-12"></a>  <span class="fu">mutate</span>(<span class="at">COG_category1 =</span> <span class="fu">strsplit</span>(COG_category, <span class="at">split =</span> <span class="st">""</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-13"><a href="#cb21-13"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-14"><a href="#cb21-14"></a>  <span class="fu">mutate</span>(<span class="at">COG_category_single =</span> <span class="fu">sample</span>(COG_category1, <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-15"><a href="#cb21-15"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-16"><a href="#cb21-16"></a>  <span class="fu">mutate</span>(<span class="at">strmatchvar =</span> <span class="fu">str_extract</span>(query, <span class="st">"^[:alnum:]*"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-17"><a href="#cb21-17"></a>  <span class="fu">mutate</span>(<span class="at">strainID =</span> <span class="fu">str_replace_all</span>(strmatchvar, enrichedsp)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-18"><a href="#cb21-18"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>strmatchvar, <span class="sc">-</span>COG_category1) <span class="sc">%&gt;%</span> </span>
<span id="cb21-19"><a href="#cb21-19"></a>  <span class="fu">rename</span>(<span class="at">locus_tag =</span> query) <span class="sc">%&gt;%</span> </span>
<span id="cb21-20"><a href="#cb21-20"></a>  <span class="fu">left_join</span>(cog_description, <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-21"><a href="#cb21-21"></a>  <span class="fu">relocate</span>(strainID, locus_tag, COG_category_single, COG_category_long))</span>
<span id="cb21-22"><a href="#cb21-22"></a></span>
<span id="cb21-23"><a href="#cb21-23"></a><span class="fu">write_tsv</span>(eggnog, here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_all_eggnog_formatted.tsv.xz"</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="co"># read back the formatted eggnog data</span></span>
<span id="cb22-2"><a href="#cb22-2"></a>eggnog <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_all_eggnog_formatted.tsv.xz"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>save a copy of significantly parallel genes as a table</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="fu">left_join</span>(output_gene_table_filt, eggnog, </span>
<span id="cb23-2"><a href="#cb23-2"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3"></a>  <span class="fu">rename</span>(<span class="at">prokka_annotation =</span> product) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"enriched_parallel_genes.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="all-species" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="all-species"><span class="header-section-number">5.1</span> All species</h2>
<p>Calculate the probability of each gene (ns_sites/total_ns_sites) and link to COG categories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>eggnog_nslen <span class="ot">&lt;-</span> <span class="fu">left_join</span>(eggnog, degentab,</span>
<span id="cb24-2"><a href="#cb24-2"></a>                          <span class="at">by =</span> <span class="fu">join_by</span>(strainID, locus_tag)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, locus_tag, COG_category_single, ns_length) <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb24-6"><a href="#cb24-6"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> ns_length<span class="sc">/</span><span class="fu">sum</span>(ns_length), </span>
<span id="cb24-7"><a href="#cb24-7"></a>         <span class="at">cog_genes =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb24-8"><a href="#cb24-8"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="jensen-shannon-divergence" class="level3" data-number="5.1.1">
<h3 data-number="5.1.1" class="anchored" data-anchor-id="jensen-shannon-divergence"><span class="header-section-number">5.1.1</span> Jensen Shannon Divergence</h3>
<p>This function draws n genes from each species with a probability proportional to gene length, where n = the number of significant parallel genes observed from the data. We repeat this simulation 10 000 times and below we check if 1) the Jensen Shannon divergence between the distribution of the COG categories drawn randomly and the background (i.e., the distribution of COG categories in the whole genome) is greater than the Jensen Shannon divergence between the background and the observed distribution of COG categories in significantly parallel genes. The idea is that if some COG categories are enriched/depleted in the parallel genes, then the JSD divergence between the observed and the background should be larger than the JSD between the random sampled genes (sampling probability based only on gene length) and the background. We calculate an empirical P value by “counting” how many times the simulated JSD exceeds the observed.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>gene_draw <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb25-2"><a href="#cb25-2"></a>  output_gene_table_filt <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3"></a>    <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4"></a>    <span class="fu">summarize</span>(<span class="at">nsiggenes =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5"></a>    <span class="fu">left_join</span>(eggnog_nslen,</span>
<span id="cb25-6"><a href="#cb25-6"></a>              <span class="at">by =</span> <span class="fu">join_by</span>(strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb25-7"><a href="#cb25-7"></a>    <span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span><span class="fu">c</span>(strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb25-8"><a href="#cb25-8"></a>    <span class="fu">mutate</span>(<span class="at">locus_tag =</span> <span class="fu">map</span>(</span>
<span id="cb25-9"><a href="#cb25-9"></a>      data,</span>
<span id="cb25-10"><a href="#cb25-10"></a>      \(x) <span class="fu">sample</span>(</span>
<span id="cb25-11"><a href="#cb25-11"></a>        <span class="at">x =</span> x<span class="sc">$</span>locus_tag,</span>
<span id="cb25-12"><a href="#cb25-12"></a>        <span class="at">size =</span> <span class="fu">unique</span>(x<span class="sc">$</span>nsiggenes),</span>
<span id="cb25-13"><a href="#cb25-13"></a>        <span class="at">prob =</span> x<span class="sc">$</span>p,</span>
<span id="cb25-14"><a href="#cb25-14"></a>        <span class="at">replace =</span> <span class="cn">FALSE</span></span>
<span id="cb25-15"><a href="#cb25-15"></a>      )</span>
<span id="cb25-16"><a href="#cb25-16"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb25-17"><a href="#cb25-17"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>data) <span class="sc">%&gt;%</span></span>
<span id="cb25-18"><a href="#cb25-18"></a>    <span class="fu">unnest</span>(<span class="at">cols =</span> locus_tag) </span>
<span id="cb25-19"><a href="#cb25-19"></a>}</span>
<span id="cb25-20"><a href="#cb25-20"></a></span>
<span id="cb25-21"><a href="#cb25-21"></a><span class="co"># take 10 000 random draws of the genes. Use a seed for reproducibility</span></span>
<span id="cb25-22"><a href="#cb25-22"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">12367</span>,</span>
<span id="cb25-23"><a href="#cb25-23"></a>                 genes_simulated <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10000</span>, <span class="sc">~</span><span class="fu">gene_draw</span>(), <span class="at">.progress=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-24"><a href="#cb25-24"></a>                   <span class="fu">list_rbind</span>(<span class="at">names_to =</span> <span class="st">"id"</span>))</span>
<span id="cb25-25"><a href="#cb25-25"></a></span>
<span id="cb25-26"><a href="#cb25-26"></a><span class="co"># save the output so we don't need to run again</span></span>
<span id="cb25-27"><a href="#cb25-27"></a><span class="fu">write_rds</span>(genes_simulated, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"genes_simulated_nomobile.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Some data formatting</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="co"># read data back</span></span>
<span id="cb26-2"><a href="#cb26-2"></a>genes_simulated <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"genes_simulated_nomobile.rds"</span>))</span>
<span id="cb26-3"><a href="#cb26-3"></a></span>
<span id="cb26-4"><a href="#cb26-4"></a><span class="co"># map the locus_tags to COG categories for the simulated</span></span>
<span id="cb26-5"><a href="#cb26-5"></a><span class="co"># gene draws</span></span>
<span id="cb26-6"><a href="#cb26-6"></a>COGs_distribution_simulated <span class="ot">&lt;-</span> genes_simulated <span class="sc">%&gt;%</span></span>
<span id="cb26-7"><a href="#cb26-7"></a>  <span class="fu">left_join</span>(eggnog_nslen, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, locus_tag)) <span class="sc">%&gt;%</span></span>
<span id="cb26-8"><a href="#cb26-8"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"nsim"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="#cb26-10"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb26-11"><a href="#cb26-11"></a></span>
<span id="cb26-12"><a href="#cb26-12"></a><span class="co"># get the discrete distribution of COGs in the significant parallel genes</span></span>
<span id="cb26-13"><a href="#cb26-13"></a>COGs_distribution_observed <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb26-14"><a href="#cb26-14"></a>                                        <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-15"><a href="#cb26-15"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"nobs"</span>) </span>
<span id="cb26-16"><a href="#cb26-16"></a></span>
<span id="cb26-17"><a href="#cb26-17"></a><span class="co"># get the discrete distribution of COGs across the whole genome of all species</span></span>
<span id="cb26-18"><a href="#cb26-18"></a><span class="co"># this is the null case (i.e. the background that we would expect if there was no</span></span>
<span id="cb26-19"><a href="#cb26-19"></a><span class="co"># enrichment)</span></span>
<span id="cb26-20"><a href="#cb26-20"></a>COGs_distribution_nullexpectation <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb26-21"><a href="#cb26-21"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"nnull"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we calculate the Jensen Shannon divergence between the background and the simulated data to estimate an empirical P value</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="co"># function to calculate the Jensen Shannon divergene of a pair of</span></span>
<span id="cb27-2"><a href="#cb27-2"></a><span class="co"># distributions</span></span>
<span id="cb27-3"><a href="#cb27-3"></a>JSD_pair <span class="ot">&lt;-</span> <span class="cf">function</span>(p, q){</span>
<span id="cb27-4"><a href="#cb27-4"></a>  m <span class="ot">&lt;-</span> <span class="fl">0.5</span> <span class="sc">*</span> (p <span class="sc">+</span> q)</span>
<span id="cb27-5"><a href="#cb27-5"></a>  JS <span class="ot">&lt;-</span>  <span class="fl">0.5</span> <span class="sc">*</span> (<span class="fu">sum</span>(p <span class="sc">*</span> <span class="fu">log</span>(p<span class="sc">/</span>m)) <span class="sc">+</span> <span class="fu">sum</span>(q <span class="sc">*</span> <span class="fu">log</span>(q<span class="sc">/</span>m)))</span>
<span id="cb27-6"><a href="#cb27-6"></a>  <span class="fu">return</span>(JS)</span>
<span id="cb27-7"><a href="#cb27-7"></a>}</span>
<span id="cb27-8"><a href="#cb27-8"></a></span>
<span id="cb27-9"><a href="#cb27-9"></a><span class="co"># Calculate the Jensen Shannon divergence between the COG distribution observed in </span></span>
<span id="cb27-10"><a href="#cb27-10"></a><span class="co"># enriched genes and the COG distribution in the genomic background</span></span>
<span id="cb27-11"><a href="#cb27-11"></a>JSD_observed <span class="ot">&lt;-</span> <span class="fu">left_join</span>(COGs_distribution_observed, COGs_distribution_nullexpectation,</span>
<span id="cb27-12"><a href="#cb27-12"></a>                          <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-13"><a href="#cb27-13"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> nobs<span class="sc">/</span><span class="fu">sum</span>(nobs), </span>
<span id="cb27-14"><a href="#cb27-14"></a>         <span class="at">q =</span> nnull<span class="sc">/</span><span class="fu">sum</span>(nnull)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-15"><a href="#cb27-15"></a>  <span class="fu">summarize</span>(<span class="at">JSD =</span> <span class="fu">JSD_pair</span>(p, q)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-16"><a href="#cb27-16"></a>  <span class="fu">pull</span>()</span>
<span id="cb27-17"><a href="#cb27-17"></a></span>
<span id="cb27-18"><a href="#cb27-18"></a><span class="co"># now calculate JSD between the simulated draws and the backgound distribution</span></span>
<span id="cb27-19"><a href="#cb27-19"></a>JSD_simulated <span class="ot">&lt;-</span> <span class="fu">left_join</span>(COGs_distribution_simulated, </span>
<span id="cb27-20"><a href="#cb27-20"></a>                           COGs_distribution_nullexpectation,</span>
<span id="cb27-21"><a href="#cb27-21"></a>                           <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-22"><a href="#cb27-22"></a>  <span class="co"># drops from calculation categories that aren't in both the observed data</span></span>
<span id="cb27-23"><a href="#cb27-23"></a>  <span class="co"># and the resampled data. Need to do this because JSD can't handle zero values</span></span>
<span id="cb27-24"><a href="#cb27-24"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-25"><a href="#cb27-25"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb27-26"><a href="#cb27-26"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> nsim<span class="sc">/</span><span class="fu">sum</span>(nsim), </span>
<span id="cb27-27"><a href="#cb27-27"></a>         <span class="at">q =</span> nnull<span class="sc">/</span><span class="fu">sum</span>(nnull)) <span class="sc">%&gt;%</span></span>
<span id="cb27-28"><a href="#cb27-28"></a>  <span class="fu">summarize</span>(<span class="at">JSD =</span> <span class="fu">JSD_pair</span>(p, q)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-29"><a href="#cb27-29"></a>  <span class="fu">pull</span>(JSD)</span>
<span id="cb27-30"><a href="#cb27-30"></a></span>
<span id="cb27-31"><a href="#cb27-31"></a><span class="co"># proportion of JSD measures between simulated and the background that are greater</span></span>
<span id="cb27-32"><a href="#cb27-32"></a><span class="co"># than the JSD between the observed and the background</span></span>
<span id="cb27-33"><a href="#cb27-33"></a>global_p <span class="ot">&lt;-</span> <span class="fu">length</span>(JSD_simulated[JSD_simulated <span class="sc">&gt;=</span> JSD_observed])<span class="sc">/</span><span class="fu">length</span>(JSD_simulated)</span>
<span id="cb27-34"><a href="#cb27-34"></a></span>
<span id="cb27-35"><a href="#cb27-35"></a>global_p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.397</code></pre>
</div>
</div>
</section>
<section id="enrichment-of-individual-cog-categories" class="level3" data-number="5.1.2">
<h3 data-number="5.1.2" class="anchored" data-anchor-id="enrichment-of-individual-cog-categories"><span class="header-section-number">5.1.2</span> Enrichment of individual COG categories</h3>
<p>OK so the overall distribution of COG categories in the parallel genes is not significantly different than in the rest of the genome - ~40% of the time the JSD between the COG distribution in a random sample and the background is greater than the observed COG distribution and the background.</p>
<section id="hypergeometric-test" class="level4" data-number="5.1.2.1">
<h4 data-number="5.1.2.1" class="anchored" data-anchor-id="hypergeometric-test"><span class="header-section-number">5.1.2.1</span> Hypergeometric Test</h4>
<p>We can test enrichment in individual COG categories in individual species using the hypergeometric test</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="fu">library</span>(qvalue)</span>
<span id="cb29-2"><a href="#cb29-2"></a></span>
<span id="cb29-3"><a href="#cb29-3"></a>par2test <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb29-4"><a href="#cb29-4"></a>                      <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb29-5"><a href="#cb29-5"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb29-7"><a href="#cb29-7"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"n_cog_par"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-8"><a href="#cb29-8"></a>  <span class="fu">mutate</span>(<span class="at">n_par =</span> <span class="fu">sum</span>(n_cog_par))</span>
<span id="cb29-9"><a href="#cb29-9"></a></span>
<span id="cb29-10"><a href="#cb29-10"></a>back2test <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb29-11"><a href="#cb29-11"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-12"><a href="#cb29-12"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb29-13"><a href="#cb29-13"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"n_cog_background"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-14"><a href="#cb29-14"></a>  <span class="fu">mutate</span>(<span class="at">n_background =</span> <span class="fu">sum</span>(n_cog_background))</span>
<span id="cb29-15"><a href="#cb29-15"></a></span>
<span id="cb29-16"><a href="#cb29-16"></a>phypres <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par2test, back2test, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-17"><a href="#cb29-17"></a>  <span class="fu">mutate</span>(<span class="at">p_enrich =</span> <span class="fu">enricher</span>(n_background, n_par, n_cog_background, n_cog_par, <span class="at">over=</span><span class="cn">TRUE</span>),</span>
<span id="cb29-18"><a href="#cb29-18"></a>         <span class="at">p_deplete =</span> <span class="fu">enricher</span>(n_background, n_par, n_cog_background, n_cog_par, <span class="at">over=</span><span class="cn">FALSE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-19"><a href="#cb29-19"></a>  <span class="fu">mutate</span>(<span class="at">p_enrich_adj =</span> <span class="fu">p.adjust</span>(p_enrich, <span class="at">method =</span> <span class="st">"fdr"</span>),</span>
<span id="cb29-20"><a href="#cb29-20"></a>         <span class="at">p_deplete_adj =</span> <span class="fu">p.adjust</span>(p_deplete, <span class="at">method =</span> <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-21"><a href="#cb29-21"></a>  <span class="fu">arrange</span>(p_enrich_adj)</span>
<span id="cb29-22"><a href="#cb29-22"></a></span>
<span id="cb29-23"><a href="#cb29-23"></a>qenrich <span class="ot">&lt;-</span> <span class="fu">qvalue_truncp</span>(phypres<span class="sc">$</span>p_enrich)</span>
<span id="cb29-24"><a href="#cb29-24"></a>qdeplete <span class="ot">&lt;-</span> <span class="fu">qvalue_truncp</span>(phypres<span class="sc">$</span>p_deplete)</span>
<span id="cb29-25"><a href="#cb29-25"></a></span>
<span id="cb29-26"><a href="#cb29-26"></a>phypres<span class="sc">$</span>q_enrich <span class="ot">&lt;-</span> qenrich<span class="sc">$</span>qvalues</span>
<span id="cb29-27"><a href="#cb29-27"></a>phypres<span class="sc">$</span>q_deplete <span class="ot">&lt;-</span> qdeplete<span class="sc">$</span>qvalues</span>
<span id="cb29-28"><a href="#cb29-28"></a></span>
<span id="cb29-29"><a href="#cb29-29"></a>phypres <span class="sc">%&gt;%</span> </span>
<span id="cb29-30"><a href="#cb29-30"></a>  <span class="fu">relocate</span>(p_enrich, q_enrich, p_enrich_adj) <span class="sc">%&gt;%</span> </span>
<span id="cb29-31"><a href="#cb29-31"></a>  <span class="fu">arrange</span>(p_enrich)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["p_enrich"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["q_enrich"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["p_enrich_adj"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["strainID"],"name":[4],"type":["chr"],"align":["left"]},{"label":["COG_category_single"],"name":[5],"type":["chr"],"align":["left"]},{"label":["n_cog_par"],"name":[6],"type":["int"],"align":["right"]},{"label":["n_par"],"name":[7],"type":["int"],"align":["right"]},{"label":["n_cog_background"],"name":[8],"type":["int"],"align":["right"]},{"label":["n_background"],"name":[9],"type":["int"],"align":["right"]},{"label":["p_deplete"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["p_deplete_adj"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["q_deplete"],"name":[12],"type":["dbl"],"align":["right"]}],"data":[{"1":"0.0004981552","2":"0.006001026","3":"0.0009963104","4":"HAMBI_1972","5":"D","6":"2","7":"3","8":"48","9":"3671","10":"0.9999979","11":"0.9999979","12":"1"},{"1":"0.0022129352","2":"0.013329061","3":"0.0022129352","4":"HAMBI_2443","5":"M","6":"2","7":"2","8":"214","9":"4539","10":"1.0000000","11":"1.0000000","12":"1"},{"1":"0.0037698804","2":"0.015137954","3":"0.0037698804","4":"HAMBI_1977","5":"M","6":"2","7":"2","8":"330","9":"5367","10":"1.0000000","11":"1.0000000","12":"1"},{"1":"0.0103181518","2":"0.031074402","3":"0.0206363037","4":"HAMBI_2659","5":"P","6":"2","7":"3","8":"228","9":"3802","10":"0.9997870","11":"0.9997870","12":"1"},{"1":"0.0296318561","2":"0.070205985","3":"0.2074229929","4":"HAMBI_2159","5":"C","6":"4","7":"15","8":"424","9":"5158","10":"0.9944963","11":"0.9944963","12":"1"},{"1":"0.0349674954","2":"0.070205985","3":"0.2797399632","4":"HAMBI_2494","5":"F","6":"2","7":"13","8":"123","9":"5324","10":"0.9970946","11":"0.9970946","12":"1"},{"1":"0.0497793258","2":"0.085666659","3":"0.0834974894","4":"HAMBI_0403","5":"F","6":"1","7":"2","8":"117","9":"4642","10":"0.9993700","11":"0.9993700","12":"1"},{"1":"0.0663295741","2":"0.086204485","3":"0.1326591481","4":"HAMBI_1287","5":"T","6":"1","7":"2","8":"139","9":"4121","10":"0.9988702","11":"0.9988702","12":"1"},{"1":"0.0702610073","2":"0.086204485","3":"0.1405220146","4":"HAMBI_2164","5":"Q","6":"1","7":"2","8":"194","9":"5424","10":"0.9987271","11":"0.9987271","12":"1"},{"1":"0.0737717759","2":"0.086204485","3":"0.2582012157","4":"HAMBI_2159","5":"O","6":"2","7":"15","8":"156","9":"5158","10":"0.9905510","11":"0.9944963","12":"1"},{"1":"0.0834974894","2":"0.086204485","3":"0.0834974894","4":"HAMBI_0403","5":"T","6":"1","7":"2","8":"198","9":"4642","10":"0.9981894","11":"0.9993700","12":"1"},{"1":"0.0858717361","2":"0.086204485","3":"0.1660984604","4":"HAMBI_1279","5":"H","6":"1","7":"2","8":"179","9":"4078","10":"0.9980836","11":"0.9980836","12":"1"},{"1":"0.1013771997","2":"0.093941567","3":"0.1013771997","4":"HAMBI_0006","5":"K","6":"1","7":"1","8":"530","9":"5228","10":"1.0000000","11":"1.0000000","12":"1"},{"1":"0.1282761429","2":"0.096700245","3":"0.2993110001","4":"HAMBI_2159","5":"H","6":"2","7":"15","8":"216","9":"5158","10":"0.9773049","11":"0.9944963","12":"1"},{"1":"0.1293897450","2":"0.096700245","3":"0.2587794901","4":"HAMBI_1842","5":"P","6":"1","7":"2","8":"291","9":"4348","10":"0.9955351","11":"0.9955351","12":"1"},{"1":"0.1311930677","2":"0.096700245","3":"0.4077117658","4":"HAMBI_2494","5":"K","6":"3","7":"13","8":"528","9":"5324","10":"0.9669328","11":"0.9970946","12":"1"},{"1":"0.1364784118","2":"0.096700245","3":"0.1930183431","4":"HAMBI_0262","5":"T","6":"1","7":"3","8":"137","9":"2871","10":"0.9934291","11":"0.9934291","12":"1"},{"1":"0.1525234282","2":"0.096700245","3":"0.1525234282","4":"HAMBI_2659","5":"T","6":"1","7":"3","8":"204","9":"3802","10":"0.9917078","11":"0.9997870","12":"1"},{"1":"0.1528675055","2":"0.096700245","3":"0.1528675055","4":"HAMBI_1287","5":"K","6":"1","7":"2","8":"328","9":"4121","10":"0.9936828","11":"0.9988702","12":"1"},{"1":"0.1660984604","2":"0.096700245","3":"0.1660984604","4":"HAMBI_1279","5":"K","6":"1","7":"2","8":"354","9":"4078","10":"0.9924839","11":"0.9980836","12":"1"},{"1":"0.1765517664","2":"0.096700245","3":"0.3089655912","4":"HAMBI_2159","5":"K","6":"3","7":"15","8":"506","9":"5158","10":"0.9479062","11":"0.9944963","12":"1"},{"1":"0.1765994694","2":"0.096700245","3":"0.1930183431","4":"HAMBI_0262","5":"P","6":"1","7":"3","8":"180","9":"2871","10":"0.9887543","11":"0.9934291","12":"1"},{"1":"0.1868271780","2":"0.096883139","3":"0.3293719695","4":"HAMBI_3237","5":"S","6":"2","7":"4","8":"1191","9":"5840","10":"0.9713087","11":"0.9797402","12":"1"},{"1":"0.1930183431","2":"0.096883139","3":"0.1930183431","4":"HAMBI_0262","5":"K","6":"1","7":"3","8":"198","9":"2871","10":"0.9864452","11":"0.9934291","12":"1"},{"1":"0.2193308550","2":"0.102556800","3":"0.2193308550","4":"HAMBI_0097","5":"S","6":"1","7":"1","8":"649","9":"2959","10":"1.0000000","11":"1.0000000","12":"1"},{"1":"0.2213486859","2":"0.102556800","3":"0.3293719695","4":"HAMBI_3237","5":"P","6":"1","7":"4","8":"354","9":"5840","10":"0.9797402","11":"0.9797402","12":"1"},{"1":"0.2841412921","2":"0.126774366","3":"0.4077117658","4":"HAMBI_2494","5":"Q","6":"1","7":"13","8":"135","9":"5324","10":"0.9585344","11":"0.9970946","12":"1"},{"1":"0.3028919699","2":"0.127992878","3":"0.4077117658","4":"HAMBI_2494","5":"E","6":"2","7":"13","8":"451","9":"5324","10":"0.9087550","11":"0.9970946","12":"1"},{"1":"0.3160256892","2":"0.127992878","3":"0.4077117658","4":"HAMBI_2494","5":"C","6":"2","7":"13","8":"465","9":"5324","10":"0.9019636","11":"0.9970946","12":"1"},{"1":"0.3285873119","2":"0.127992878","3":"0.4600222367","4":"HAMBI_2159","5":"Q","6":"1","7":"15","8":"135","9":"5158","10":"0.9428462","11":"0.9944963","12":"1"},{"1":"0.3293719695","2":"0.127992878","3":"0.3293719695","4":"HAMBI_3237","5":"E","6":"1","7":"4","8":"555","9":"5840","10":"0.9524899","11":"0.9797402","12":"1"},{"1":"0.3815681185","2":"0.136430542","3":"0.4077117658","4":"HAMBI_2494","5":"L","6":"1","7":"13","8":"193","9":"5324","10":"0.9215469","11":"0.9970946","12":"1"},{"1":"0.3946641076","2":"0.136430542","3":"0.3946641076","4":"HAMBI_1842","5":"S","6":"1","7":"2","8":"965","9":"4348","10":"0.9507819","11":"0.9955351","12":"1"},{"1":"0.3992928350","2":"0.136430542","3":"0.3992928350","4":"HAMBI_2164","5":"S","6":"1","7":"2","8":"1220","9":"5424","10":"0.9494403","11":"0.9987271","12":"1"},{"1":"0.4046898919","2":"0.136430542","3":"0.4077117658","4":"HAMBI_2494","5":"I","6":"1","7":"13","8":"208","9":"5324","10":"0.9106938","11":"0.9970946","12":"1"},{"1":"0.4077117658","2":"0.136430542","3":"0.4077117658","4":"HAMBI_2494","5":"H","6":"1","7":"13","8":"210","9":"5324","10":"0.9092124","11":"0.9970946","12":"1"},{"1":"0.4947710422","2":"0.161088082","3":"0.4947710422","4":"HAMBI_1972","5":"S","6":"1","7":"3","8":"747","9":"3671","10":"0.8927095","11":"0.9999979","12":"1"},{"1":"0.6442922741","2":"0.204249118","3":"0.7516743198","4":"HAMBI_2159","5":"M","6":"1","7":"15","8":"343","9":"5158","10":"0.7369025","11":"0.9944963","12":"1"},{"1":"0.8468502370","2":"0.261578996","3":"0.8468502370","4":"HAMBI_2159","5":"S","6":"2","7":"15","8":"1063","9":"5158","10":"0.3752230","11":"0.9944963","12":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Here COG categories “D - Cell cycle control, cell division, chromosome partitioning”, “M - Cell wall/membrane/envelope biogenesis”, and “P - Inorganic ion transport and metabolism” are enriched with <span class="math inline">\(p_{adj} \leq 0.05\)</span>.</p>
</section>
<section id="nonparametric-simulation" class="level4" data-number="5.1.2.2">
<h4 data-number="5.1.2.2" class="anchored" data-anchor-id="nonparametric-simulation"><span class="header-section-number">5.1.2.2</span> Nonparametric simulation</h4>
<p>We can also look at different COG categories by simply counting the number of random samplings that have a COG count greater/less than that the observed COG count. Since we resampled the same number of genes as in the observed we can compare the two directly. Below we do this integrated over all species in the community to see if there was any pathway enriched across all species.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="fu">left_join</span>(COGs_distribution_simulated, COGs_distribution_observed,</span>
<span id="cb30-2"><a href="#cb30-2"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="#cb30-3"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="#cb30-4"></a>  <span class="fu">mutate</span>(<span class="at">gt =</span> <span class="fu">if_else</span>(nsim <span class="sc">&gt;=</span> nobs, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-5"><a href="#cb30-5"></a>  <span class="fu">group_by</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb30-6"><a href="#cb30-6"></a>  <span class="fu">summarize</span>(<span class="at">p =</span> <span class="fu">sum</span>(gt)<span class="sc">/</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb30-7"><a href="#cb30-7"></a>  <span class="fu">arrange</span>(p) <span class="sc">%&gt;%</span> </span>
<span id="cb30-8"><a href="#cb30-8"></a>  <span class="fu">mutate</span>(<span class="at">p_adj =</span> <span class="fu">p.adjust</span>(p, <span class="at">method =</span> <span class="st">"fdr"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["COG_category_single"],"name":[1],"type":["chr"],"align":["left"]},{"label":["p"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["p_adj"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"K","2":"0.01211754","3":"0.1696456"},{"1":"F","2":"0.24557156","3":"0.8085011"},{"1":"H","2":"0.24792213","3":"0.8085011"},{"1":"Q","2":"0.32269504","3":"0.8085011"},{"1":"D","2":"0.33742444","3":"0.8085011"},{"1":"C","2":"0.42970476","3":"0.8085011"},{"1":"T","2":"0.45785243","3":"0.8085011"},{"1":"M","2":"0.48430720","3":"0.8085011"},{"1":"P","2":"0.51975073","3":"0.8085011"},{"1":"O","2":"0.64808153","3":"0.9073141"},{"1":"S","2":"0.79180000","3":"1.0000000"},{"1":"E","2":"0.92625251","3":"1.0000000"},{"1":"I","2":"1.00000000","3":"1.0000000"},{"1":"L","2":"1.00000000","3":"1.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Only about 1% of the re-samples had a higher number of K COG genes compared to the observed number of K COG genes. No other COG category meets the standard alpha = 0.05 threshold. However, after correcting for multiple testing the 5% alpha threshold is no longer met.</p>
<p>This is just a sanity check that the randomly sampled COGs reflect the background distribution - i.e., a large fraction of the resamples are larger (and smaller) than the null. There is no systematic difference, and we observe the different COG categories in the resampels at basically the same rate as we observe the COG categories in the background.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="fu">left_join</span>(COGs_distribution_simulated, COGs_distribution_nullexpectation,</span>
<span id="cb31-2"><a href="#cb31-2"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4"></a>  <span class="fu">mutate</span>(<span class="at">fsim =</span> nsim<span class="sc">/</span><span class="fu">sum</span>(nsim), <span class="at">fnull =</span> nnull<span class="sc">/</span><span class="fu">sum</span>(nnull)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-5"><a href="#cb31-5"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-6"><a href="#cb31-6"></a>  <span class="fu">mutate</span>(<span class="at">gt =</span> <span class="fu">if_else</span>(fsim <span class="sc">&gt;=</span> fnull, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="#cb31-7"></a>  <span class="fu">group_by</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb31-8"><a href="#cb31-8"></a>  <span class="fu">summarize</span>(<span class="at">p =</span> <span class="fu">sum</span>(gt)<span class="sc">/</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb31-9"><a href="#cb31-9"></a>  <span class="fu">arrange</span>(p)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["COG_category_single"],"name":[1],"type":["chr"],"align":["left"]},{"label":["p"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"S","2":"0.0860000"},{"1":"K","2":"0.2496213"},{"1":"J","2":"0.4037548"},{"1":"N","2":"0.4520967"},{"1":"L","2":"0.4521211"},{"1":"I","2":"0.4525564"},{"1":"H","2":"0.4869860"},{"1":"P","2":"0.5160318"},{"1":"E","2":"0.5264529"},{"1":"G","2":"0.5326710"},{"1":"O","2":"0.5558753"},{"1":"F","2":"0.5704091"},{"1":"C","2":"0.6005222"},{"1":"M","2":"0.6048037"},{"1":"Q","2":"0.6526005"},{"1":"U","2":"0.6617682"},{"1":"T","2":"0.6637469"},{"1":"A","2":"1.0000000"},{"1":"B","2":"1.0000000"},{"1":"D","2":"1.0000000"},{"1":"V","2":"1.0000000"},{"1":"W","2":"1.0000000"},{"1":"Z","2":"1.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
<section id="cog-distribution-plot" class="level3" data-number="5.1.3">
<h3 data-number="5.1.3" class="anchored" data-anchor-id="cog-distribution-plot"><span class="header-section-number">5.1.3</span> COG Distribution Plot</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>cog_pmut <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb32-2"><a href="#cb32-2"></a>                      <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-3"><a href="#cb32-3"></a>  <span class="fu">count</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n),</span>
<span id="cb32-5"><a href="#cb32-5"></a>         <span class="at">n =</span> n<span class="sc">*</span><span class="dv">1000</span>,</span>
<span id="cb32-6"><a href="#cb32-6"></a>         <span class="at">type =</span> <span class="st">"parallel_mutations"</span>) </span>
<span id="cb32-7"><a href="#cb32-7"></a></span>
<span id="cb32-8"><a href="#cb32-8"></a>cog_everything <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb32-9"><a href="#cb32-9"></a>  <span class="fu">count</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb32-10"><a href="#cb32-10"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n), </span>
<span id="cb32-11"><a href="#cb32-11"></a>         <span class="at">type =</span> <span class="st">"all_genes"</span>) </span>
<span id="cb32-12"><a href="#cb32-12"></a></span>
<span id="cb32-13"><a href="#cb32-13"></a>pcogs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cog_pmut, cog_everything) <span class="sc">%&gt;%</span> </span>
<span id="cb32-14"><a href="#cb32-14"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-15"><a href="#cb32-15"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb32-16"><a href="#cb32-16"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(COG_category_single, f, <span class="at">.desc =</span> F), <span class="at">y=</span>n, <span class="at">fill =</span> type), <span class="at">stat=</span><span class="st">"identity"</span>,</span>
<span id="cb32-17"><a href="#cb32-17"></a>           <span class="at">position =</span> <span class="fu">position_dodge</span>( <span class="at">preserve =</span> <span class="st">"total"</span>)) <span class="sc">+</span></span>
<span id="cb32-18"><a href="#cb32-18"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb32-19"><a href="#cb32-19"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"COG Category"</span>, <span class="at">y =</span> <span class="st">"Fraction genes in COG category"</span>, <span class="at">fill =</span> <span class="st">"Subset"</span>) <span class="sc">+</span></span>
<span id="cb32-20"><a href="#cb32-20"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb32-21"><a href="#cb32-21"></a>    <span class="st">"N all genes"</span>, </span>
<span id="cb32-22"><a href="#cb32-22"></a>    <span class="at">sec.axis =</span> <span class="fu">sec_axis</span>(<span class="sc">~</span> . <span class="sc">*</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">1000</span>, <span class="at">name =</span> <span class="st">"N parallel genes"</span>)</span>
<span id="cb32-23"><a href="#cb32-23"></a>  ) <span class="sc">+</span></span>
<span id="cb32-24"><a href="#cb32-24"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb32-25"><a href="#cb32-25"></a>  <span class="fu">theme</span>(</span>
<span id="cb32-26"><a href="#cb32-26"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-27"><a href="#cb32-27"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-28"><a href="#cb32-28"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb32-29"><a href="#cb32-29"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-05" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_parallelism_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Number of genes in each Cluster of Orthologous Groups (COG) category for genes with a COG annotation across all 23 HAMBI population genomes (red bars, left y-axis) and across only genes exhibiting significant genomic parallelism (blue bars, right y-axis).
</figcaption>
</figure>
</div>
</section>
</section>
<section id="species-present-in-the-metagenomes" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="species-present-in-the-metagenomes"><span class="header-section-number">5.2</span> Species present in the metagenomes</h2>
<p>Now that I’ve done the analysis on the metagenomes we know that we can only detect HAMBI_2659, HAMBI_1972, HAMBI_1977, and HAMBI_1287 with high enough coverate to detect sequence variants. From the hypergeometric test above we know that COG categories D, M, and P are enriched in HAMBI_1972, HAMBI_1977, amd HAMBI_2659 respsectively. Here we will make a plot focusing only on those species</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="co"># species detected in the metagenomes</span></span>
<span id="cb33-2"><a href="#cb33-2"></a>mgsp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"HAMBI_2659"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_2443"</span>)</span>
<span id="cb33-3"><a href="#cb33-3"></a></span>
<span id="cb33-4"><a href="#cb33-4"></a>cog_pmut_mgsp <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb33-5"><a href="#cb33-5"></a>                      <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-6"><a href="#cb33-6"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> mgsp) <span class="sc">%&gt;%</span> </span>
<span id="cb33-7"><a href="#cb33-7"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb33-8"><a href="#cb33-8"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"tot"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-9"><a href="#cb33-9"></a>  <span class="co"># scaling factor to make ploting on 2 axes possible</span></span>
<span id="cb33-10"><a href="#cb33-10"></a>  <span class="fu">mutate</span>(<span class="at">n =</span> tot<span class="sc">*</span><span class="dv">250</span>,</span>
<span id="cb33-11"><a href="#cb33-11"></a>         <span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n),</span>
<span id="cb33-12"><a href="#cb33-12"></a>         <span class="at">type =</span> <span class="st">"parallel_mutations"</span>) </span>
<span id="cb33-13"><a href="#cb33-13"></a></span>
<span id="cb33-14"><a href="#cb33-14"></a>cog_everything_mgsp <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb33-15"><a href="#cb33-15"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> mgsp) <span class="sc">%&gt;%</span> </span>
<span id="cb33-16"><a href="#cb33-16"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb33-17"><a href="#cb33-17"></a>  <span class="fu">count</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb33-18"><a href="#cb33-18"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n),</span>
<span id="cb33-19"><a href="#cb33-19"></a>         <span class="at">type =</span> <span class="st">"all_genes"</span>) </span>
<span id="cb33-20"><a href="#cb33-20"></a></span>
<span id="cb33-21"><a href="#cb33-21"></a>pcogs_mgsp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cog_pmut_mgsp, cog_everything_mgsp) <span class="sc">%&gt;%</span> </span>
<span id="cb33-22"><a href="#cb33-22"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-23"><a href="#cb33-23"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb33-24"><a href="#cb33-24"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(COG_category_single, n, <span class="at">.desc =</span> F), <span class="at">y=</span>n, <span class="at">fill =</span> type), <span class="at">stat=</span><span class="st">"identity"</span>,</span>
<span id="cb33-25"><a href="#cb33-25"></a>           <span class="at">position =</span> <span class="fu">position_dodge</span>( <span class="at">preserve =</span> <span class="st">"total"</span>)) <span class="sc">+</span></span>
<span id="cb33-26"><a href="#cb33-26"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb33-27"><a href="#cb33-27"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">tibble</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="st">"D"</span>, <span class="st">"M"</span>, <span class="st">"M"</span>, <span class="st">"P"</span>), </span>
<span id="cb33-28"><a href="#cb33-28"></a>                           <span class="at">y =</span> <span class="fu">c</span>(<span class="dv">700</span>, <span class="dv">700</span>, <span class="dv">700</span>, <span class="dv">700</span>), </span>
<span id="cb33-29"><a href="#cb33-29"></a>                           <span class="at">strainID =</span> <span class="fu">c</span>(<span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_2443"</span>, <span class="st">"HAMBI_2659"</span>)),</span>
<span id="cb33-30"><a href="#cb33-30"></a>             <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y), <span class="at">shape =</span> <span class="dv">8</span>) <span class="sc">+</span></span>
<span id="cb33-31"><a href="#cb33-31"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> strainID, <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb33-32"><a href="#cb33-32"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"COG Category"</span>, <span class="at">y =</span> <span class="st">"Fraction genes in COG category"</span>, <span class="at">fill =</span> <span class="st">"Subset"</span>) <span class="sc">+</span></span>
<span id="cb33-33"><a href="#cb33-33"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb33-34"><a href="#cb33-34"></a>    <span class="st">"N all genes"</span>, </span>
<span id="cb33-35"><a href="#cb33-35"></a>    <span class="at">sec.axis =</span> <span class="fu">sec_axis</span>(<span class="sc">~</span> . <span class="sc">*</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">250</span>, <span class="at">name =</span> <span class="st">"N parallel genes"</span>)</span>
<span id="cb33-36"><a href="#cb33-36"></a>  ) <span class="sc">+</span></span>
<span id="cb33-37"><a href="#cb33-37"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb33-38"><a href="#cb33-38"></a>  <span class="fu">theme</span>(</span>
<span id="cb33-39"><a href="#cb33-39"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-40"><a href="#cb33-40"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-41"><a href="#cb33-41"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-42"><a href="#cb33-42"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb33-43"><a href="#cb33-43"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-06" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_parallelism_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Fraction of genes in each Cluster of Orthologous Groups (COG) category for genes with a COG annotation (red bars, left y-axis) and only genes exhibiting significant genomic parallelism (blue bars, right y-axis) for the four most abundant HAMBI genomes. COG categories “D - Cell cycle control, cell division, chromosome partitioning” for HAMBI_1972, “M - Cell wall/membrane/envelope biogenesis” for HAMBI_1977, and “P - Inorganic ion transport and metabolism” for HAMBI_2659 are significantly enriched (denoted with *) in the parallel mutations compared to the rest of the genome for these species.
</figcaption>
</figure>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"COG_enrich_wgs.svg"</span>), pcogs_mgsp, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">5</span>, </span>
<span id="cb34-2"><a href="#cb34-2"></a>       <span class="at">units=</span><span class="st">"in"</span>, <span class="at">device=</span><span class="st">"svg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<table class="table">
<colgroup>
<col style="width: 10%">
<col style="width: 22%">
<col style="width: 59%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th>strainID</th>
<th>Abundant in metagenomes</th>
<th>COG category</th>
<th>q value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HAMBI_1972</td>
<td>yes</td>
<td>D - Cell cycle control, cell division, chromosome partitioning</td>
<td>0.006</td>
</tr>
<tr class="even">
<td>HAMBI_2443</td>
<td>no</td>
<td>M - Cell wall/membrane/envelope biogenesis</td>
<td>0.013</td>
</tr>
<tr class="odd">
<td>HAMBI_1977</td>
<td>yes</td>
<td>M - Cell wall/membrane/envelope biogenesis</td>
<td>0.015</td>
</tr>
<tr class="even">
<td>HAMBI_2659</td>
<td>yes</td>
<td>P - Inorganic ion transport and metabolism</td>
<td>0.031</td>
</tr>
</tbody>
</table>


<!-- -->


</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-camargo2023" class="csl-entry" role="listitem">
Camargo, Antonio Pedro, Simon Roux, Frederik Schulz, Michal Babinski, Yan Xu, Bin Hu, Patrick S. G. Chain, Stephen Nayfach, and Nikos C. Kyrpides. 2023. <span>“Identification of Mobile Genetic Elements with geNomad.”</span> <em>Nature Biotechnology</em>, September. <a href="https://doi.org/10.1038/s41587-023-01953-y">https://doi.org/10.1038/s41587-023-01953-y</a>.
</div>
<div id="ref-durrant2020a" class="csl-entry" role="listitem">
Durrant, Matthew G., Michelle M. Li, Benjamin A. Siranosian, Stephen B. Montgomery, and Ami S. Bhatt. 2020. <span>“A Bioinformatic Analysis of Integrative Mobile Genetic Elements Highlights Their Role in Bacterial Adaptation.”</span> <em>Cell Host &amp; Microbe</em> 27 (1): 140–153.e9. <a href="https://doi.org/10.1016/j.chom.2019.10.022">https://doi.org/10.1016/j.chom.2019.10.022</a>.
</div>
<div id="ref-good2017" class="csl-entry" role="listitem">
Good, Benjamin H., Michael J. McDonald, Jeffrey E. Barrick, Richard E. Lenski, and Michael M. Desai. 2017. <span>“The Dynamics of Molecular Evolution over 60,000 Generations.”</span> <em>Nature</em> 551 (7678): 45–50. <a href="https://doi.org/10.1038/nature24287">https://doi.org/10.1038/nature24287</a>.
</div>
<div id="ref-shoemaker2021" class="csl-entry" role="listitem">
Shoemaker, William R, Evgeniya Polezhaeva, Kenzie B Givens, and Jay T Lennon. 2021. <span>“Molecular Evolutionary Dynamics of Energy Limited Microorganisms.”</span> Edited by Fabia Ursula Battistuzzi. <em>Molecular Biology and Evolution</em> 38 (10): 4532–45. <a href="https://doi.org/10.1093/molbev/msab195">https://doi.org/10.1093/molbev/msab195</a>.
</div>
<div id="ref-tenaillon2016" class="csl-entry" role="listitem">
Tenaillon, Olivier, Jeffrey E. Barrick, Noah Ribeck, Daniel E. Deatherage, Jeffrey L. Blanchard, Aurko Dasgupta, Gabriel C. Wu, et al. 2016. <span>“Tempo and Mode of Genome Evolution in a 50,000-Generation Experiment.”</span> <em>Nature</em> 536 (7615): 165–70. <a href="https://doi.org/10.1038/nature18959">https://doi.org/10.1038/nature18959</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb35" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb35-1"><a href="#cb35-1"></a><span class="co">---</span></span>
<span id="cb35-2"><a href="#cb35-2"></a><span class="an">title:</span><span class="co"> "WGS genetic parallelism analysis"</span></span>
<span id="cb35-3"><a href="#cb35-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb35-4"><a href="#cb35-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb35-5"><a href="#cb35-5"></a><span class="an">link-citations:</span><span class="co"> true</span></span>
<span id="cb35-6"><a href="#cb35-6"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb35-7"><a href="#cb35-7"></a><span class="an">abstract:</span><span class="co"> "This notebook reads in the formatted WGS variant frequencies and annotations. It then uses statistical tests to determine which mutations and which functional categories appeared across biological replicates within condition more often than would be expected by chance. It also looks for mutations/functions that arise within treatment categories more than expected by chance."</span></span>
<span id="cb35-8"><a href="#cb35-8"></a><span class="co">---</span></span>
<span id="cb35-9"><a href="#cb35-9"></a></span>
<span id="cb35-10"><a href="#cb35-10"></a><span class="fu"># Setup</span></span>
<span id="cb35-11"><a href="#cb35-11"></a></span>
<span id="cb35-12"><a href="#cb35-12"></a>Libraries and global variables</span>
<span id="cb35-13"><a href="#cb35-13"></a></span>
<span id="cb35-16"><a href="#cb35-16"></a><span class="in">```{r}</span></span>
<span id="cb35-17"><a href="#cb35-17"></a><span class="co">#| output: false</span></span>
<span id="cb35-18"><a href="#cb35-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb35-19"><a href="#cb35-19"></a><span class="fu">library</span>(here)</span>
<span id="cb35-20"><a href="#cb35-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb35-21"><a href="#cb35-21"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb35-22"><a href="#cb35-22"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb35-23"><a href="#cb35-23"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb35-24"><a href="#cb35-24"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb35-25"><a href="#cb35-25"></a></span>
<span id="cb35-26"><a href="#cb35-26"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb35-27"><a href="#cb35-27"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"wgs"</span>, <span class="st">"utils_parallelism.R"</span>))</span>
<span id="cb35-28"><a href="#cb35-28"></a><span class="in">```</span></span>
<span id="cb35-29"><a href="#cb35-29"></a></span>
<span id="cb35-30"><a href="#cb35-30"></a>Set up some directories</span>
<span id="cb35-31"><a href="#cb35-31"></a></span>
<span id="cb35-34"><a href="#cb35-34"></a><span class="in">```{r}</span></span>
<span id="cb35-35"><a href="#cb35-35"></a><span class="co">#| output: false</span></span>
<span id="cb35-36"><a href="#cb35-36"></a><span class="co">#| warning: false</span></span>
<span id="cb35-37"><a href="#cb35-37"></a><span class="co">#| error: false</span></span>
<span id="cb35-38"><a href="#cb35-38"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"wgs"</span>)</span>
<span id="cb35-39"><a href="#cb35-39"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb35-40"><a href="#cb35-40"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"wgs"</span>)</span>
<span id="cb35-41"><a href="#cb35-41"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"wgs"</span>)</span>
<span id="cb35-42"><a href="#cb35-42"></a></span>
<span id="cb35-43"><a href="#cb35-43"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb35-44"><a href="#cb35-44"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb35-45"><a href="#cb35-45"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span>
<span id="cb35-46"><a href="#cb35-46"></a><span class="in">```</span></span>
<span id="cb35-47"><a href="#cb35-47"></a></span>
<span id="cb35-48"><a href="#cb35-48"></a><span class="fu"># Read data</span></span>
<span id="cb35-49"><a href="#cb35-49"></a></span>
<span id="cb35-52"><a href="#cb35-52"></a><span class="in">```{r}</span></span>
<span id="cb35-53"><a href="#cb35-53"></a><span class="co">#| output: false</span></span>
<span id="cb35-54"><a href="#cb35-54"></a><span class="co">#| warning: false</span></span>
<span id="cb35-55"><a href="#cb35-55"></a><span class="co">#| error: false</span></span>
<span id="cb35-56"><a href="#cb35-56"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"mutect_parsed_annotated.rds"</span>))</span>
<span id="cb35-57"><a href="#cb35-57"></a>degentab <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb35-58"><a href="#cb35-58"></a>genome_len <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_genome_len.tsv"</span>))</span>
<span id="cb35-59"><a href="#cb35-59"></a><span class="in">```</span></span>
<span id="cb35-60"><a href="#cb35-60"></a></span>
<span id="cb35-61"><a href="#cb35-61"></a><span class="co">[</span><span class="ot">The mutect2 documentation</span><span class="co">](https://gatk.broadinstitute.org/hc/en-us/articles/360050722212-FAQ-for-Mutect2)</span> says that any variant without "PASS" is false positive and should be removed. We will follow this convention here and only focus on PASSing mutations.</span>
<span id="cb35-62"><a href="#cb35-62"></a></span>
<span id="cb35-65"><a href="#cb35-65"></a><span class="in">```{r}</span></span>
<span id="cb35-66"><a href="#cb35-66"></a><span class="co">#| output: false</span></span>
<span id="cb35-67"><a href="#cb35-67"></a><span class="co"># note that 1287 got an extra replicate of replicate "A" sequenced. It was called replicate "B" in the</span></span>
<span id="cb35-68"><a href="#cb35-68"></a><span class="co"># files and I will combine it with the other replicate "A"</span></span>
<span id="cb35-69"><a href="#cb35-69"></a>tofilter <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb35-70"><a href="#cb35-70"></a>  <span class="fu">filter</span>(strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span> <span class="sc">&amp;</span> replicate <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"A"</span>,<span class="st">"B"</span>))</span>
<span id="cb35-71"><a href="#cb35-71"></a></span>
<span id="cb35-72"><a href="#cb35-72"></a><span class="co"># this is to make sure we filter out duplicates by randomly taking one</span></span>
<span id="cb35-73"><a href="#cb35-73"></a><span class="co"># from each group BUT to make sure to select duplicates that have a PASS</span></span>
<span id="cb35-74"><a href="#cb35-74"></a>tocombine <span class="ot">&lt;-</span> tofilter <span class="sc">%&gt;%</span> </span>
<span id="cb35-75"><a href="#cb35-75"></a>  <span class="fu">group_by</span>(chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb35-76"><a href="#cb35-76"></a>  <span class="fu">mutate</span>(<span class="at">W =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb35-77"><a href="#cb35-77"></a>  <span class="fu">mutate</span>(<span class="at">W1 =</span> <span class="fu">case_when</span>(W <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> filter <span class="sc">==</span> <span class="st">"PASS"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb35-78"><a href="#cb35-78"></a>                       W <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> filter <span class="sc">!=</span> <span class="st">"PASS"</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb35-79"><a href="#cb35-79"></a>                       W <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb35-80"><a href="#cb35-80"></a>  <span class="fu">mutate</span>(<span class="at">W2 =</span> <span class="fu">if_else</span>(W <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(W1) <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-81"><a href="#cb35-81"></a>  <span class="fu">mutate</span>(<span class="at">W3 =</span> W2 <span class="sc">+</span> W1) <span class="sc">%&gt;%</span> </span>
<span id="cb35-82"><a href="#cb35-82"></a>  <span class="fu">relocate</span>(W, W1, W2, W3) <span class="sc">%&gt;%</span> </span>
<span id="cb35-83"><a href="#cb35-83"></a>  <span class="fu">filter</span>(W3 <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-84"><a href="#cb35-84"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-85"><a href="#cb35-85"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>W, <span class="sc">-</span>W1, <span class="sc">-</span>W2, <span class="sc">-</span>W3) <span class="sc">%&gt;%</span> </span>
<span id="cb35-86"><a href="#cb35-86"></a>  <span class="fu">mutate</span>(<span class="at">replicate =</span> <span class="st">"A"</span>)</span>
<span id="cb35-87"><a href="#cb35-87"></a></span>
<span id="cb35-88"><a href="#cb35-88"></a>mgvars_filt <span class="ot">&lt;-</span> <span class="fu">anti_join</span>(mgvars, tofilter) <span class="sc">%&gt;%</span></span>
<span id="cb35-89"><a href="#cb35-89"></a>  <span class="fu">bind_rows</span>(tocombine) <span class="sc">%&gt;%</span> </span>
<span id="cb35-90"><a href="#cb35-90"></a>  <span class="fu">mutate</span>(<span class="at">replicate =</span> <span class="fu">if_else</span>(replicate <span class="sc">==</span> <span class="st">"B"</span>, <span class="st">"C"</span>, replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-91"><a href="#cb35-91"></a>  <span class="co"># we are filtering by requiring a depth of at least 10 reads</span></span>
<span id="cb35-92"><a href="#cb35-92"></a>  <span class="co"># and that the variant passed the mutect2 internal filters</span></span>
<span id="cb35-93"><a href="#cb35-93"></a>  <span class="fu">filter</span>(filter <span class="sc">==</span> <span class="st">"PASS"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-94"><a href="#cb35-94"></a>  <span class="fu">group_by</span>(chrom, pos, ref, alt, replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb35-95"><a href="#cb35-95"></a>  <span class="fu">filter</span>(freq_alt <span class="sc">==</span> <span class="fu">max</span>(freq_alt)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-96"><a href="#cb35-96"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb35-97"><a href="#cb35-97"></a><span class="in">```</span></span>
<span id="cb35-98"><a href="#cb35-98"></a></span>
<span id="cb35-99"><a href="#cb35-99"></a><span class="fu"># Parallelism at nucleotide level</span></span>
<span id="cb35-100"><a href="#cb35-100"></a></span>
<span id="cb35-101"><a href="#cb35-101"></a>Looking at all mutations (inside and outside of coding sequences, synonymous and nonsynonymous)</span>
<span id="cb35-102"><a href="#cb35-102"></a></span>
<span id="cb35-105"><a href="#cb35-105"></a><span class="in">```{r}</span></span>
<span id="cb35-106"><a href="#cb35-106"></a>mgvars_filt <span class="sc">%&gt;%</span> </span>
<span id="cb35-107"><a href="#cb35-107"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb35-108"><a href="#cb35-108"></a>  <span class="fu">mutate</span>(<span class="at">mi=</span><span class="fu">n_distinct</span>(replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-109"><a href="#cb35-109"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-110"><a href="#cb35-110"></a>  <span class="fu">count</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb35-111"><a href="#cb35-111"></a>  <span class="fu">group_by</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb35-112"><a href="#cb35-112"></a>  <span class="fu">mutate</span>(<span class="at">tot =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-113"><a href="#cb35-113"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mi))</span>
<span id="cb35-114"><a href="#cb35-114"></a><span class="in">```</span></span>
<span id="cb35-115"><a href="#cb35-115"></a></span>
<span id="cb35-116"><a href="#cb35-116"></a>We identified 22 species with detectable mutations (basically everything, except for 1299 for which the sequencing failed). Interestingly, (as shown above) there are many species with a huge amount of parallel mutations across all three replicates. For example there are 1497 sites with the exact same mutation in all three sequenced replicates. Most of these are not fixed mutations (but some of them are), but what I realized when digging deeper is that most of these extremely parallel mutations reside on predicted plasmid sequences, in mobile genetic elements (like transposases), and in predicted prophages. Certainly something interesting is going on in these regions of these species, but I don't think that short read sequencing is really sufficient to tell us what is going on. Instead short reads seem to give lots of short indels in very small windows of the genome producing a lot of noise. So I think it is best to try and remove these kinds of mutations before proceeding with the analysis.</span>
<span id="cb35-117"><a href="#cb35-117"></a></span>
<span id="cb35-118"><a href="#cb35-118"></a>We used the tools MGEfinder v1.0.3 with database v1.0.2 <span class="co">[</span><span class="ot">@durrant2020a</span><span class="co">]</span> and geNomad v1.7.4 <span class="co">[</span><span class="ot">@camargo2023</span><span class="co">]</span> to identify mobile elements, plasmids and integrated viruses (prophages) in the genomes and excluded these regions from subsequent analyses.</span>
<span id="cb35-119"><a href="#cb35-119"></a></span>
<span id="cb35-122"><a href="#cb35-122"></a><span class="in">```{r}</span></span>
<span id="cb35-123"><a href="#cb35-123"></a><span class="co">#| output: false</span></span>
<span id="cb35-124"><a href="#cb35-124"></a><span class="co">#| warning: false</span></span>
<span id="cb35-125"><a href="#cb35-125"></a><span class="co">#| error: false</span></span>
<span id="cb35-126"><a href="#cb35-126"></a>mgefinder <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"MGE_finder_HAMBI_combined.tsv"</span>))</span>
<span id="cb35-127"><a href="#cb35-127"></a>genomad <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"genomad_HAMBI_combined.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-128"><a href="#cb35-128"></a>  <span class="co"># exclude very large plasmids over 1Mbp from the filtering</span></span>
<span id="cb35-129"><a href="#cb35-129"></a>  <span class="fu">filter</span>((end <span class="sc">-</span> start) <span class="sc">&lt;</span> <span class="dv">1000000</span>)</span>
<span id="cb35-130"><a href="#cb35-130"></a><span class="in">```</span></span>
<span id="cb35-131"><a href="#cb35-131"></a></span>
<span id="cb35-134"><a href="#cb35-134"></a><span class="in">```{r}</span></span>
<span id="cb35-135"><a href="#cb35-135"></a>mgvars_filt_mb <span class="ot">&lt;-</span> mgvars_filt <span class="sc">%&gt;%</span> </span>
<span id="cb35-136"><a href="#cb35-136"></a>  <span class="fu">left_join</span>(mgefinder, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-137"><a href="#cb35-137"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start.y <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end.y)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-138"><a href="#cb35-138"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start.y, <span class="sc">-</span>end.y, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb35-139"><a href="#cb35-139"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-140"><a href="#cb35-140"></a>  <span class="fu">left_join</span>(genomad, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-141"><a href="#cb35-141"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-142"><a href="#cb35-142"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start, <span class="sc">-</span>end, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb35-143"><a href="#cb35-143"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-144"><a href="#cb35-144"></a>  <span class="fu">rename</span>(<span class="at">start =</span> start.x, <span class="at">end =</span> end.x)</span>
<span id="cb35-145"><a href="#cb35-145"></a><span class="in">```</span></span>
<span id="cb35-146"><a href="#cb35-146"></a></span>
<span id="cb35-149"><a href="#cb35-149"></a><span class="in">```{r}</span></span>
<span id="cb35-150"><a href="#cb35-150"></a>mgvars_filt_mb <span class="sc">%&gt;%</span> </span>
<span id="cb35-151"><a href="#cb35-151"></a>  <span class="co"># remove the annotations for saving</span></span>
<span id="cb35-152"><a href="#cb35-152"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(feature_type<span class="sc">:</span><span class="fu">last_col</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb35-153"><a href="#cb35-153"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"wgsvars_filt_mb_nonsyn.tsv"</span>))</span>
<span id="cb35-154"><a href="#cb35-154"></a><span class="in">```</span></span>
<span id="cb35-155"><a href="#cb35-155"></a></span>
<span id="cb35-156"><a href="#cb35-156"></a></span>
<span id="cb35-159"><a href="#cb35-159"></a><span class="in">```{r}</span></span>
<span id="cb35-160"><a href="#cb35-160"></a>mgvars_filt_mb <span class="sc">%&gt;%</span> </span>
<span id="cb35-161"><a href="#cb35-161"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb35-162"><a href="#cb35-162"></a>  <span class="fu">mutate</span>(<span class="at">mi=</span><span class="fu">n_distinct</span>(replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-163"><a href="#cb35-163"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-164"><a href="#cb35-164"></a>  <span class="fu">count</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb35-165"><a href="#cb35-165"></a>  <span class="fu">group_by</span>(mi) <span class="sc">%&gt;%</span> </span>
<span id="cb35-166"><a href="#cb35-166"></a>  <span class="fu">mutate</span>(<span class="at">tot =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-167"><a href="#cb35-167"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(mi))</span>
<span id="cb35-168"><a href="#cb35-168"></a><span class="in">```</span></span>
<span id="cb35-169"><a href="#cb35-169"></a></span>
<span id="cb35-170"><a href="#cb35-170"></a>OK this seems a lot more reasonable here... Now there are much fewer mutations that are parallel across the replicates.</span>
<span id="cb35-171"><a href="#cb35-171"></a></span>
<span id="cb35-172"><a href="#cb35-172"></a>To put these observations in context, we can compare them to a simple null model in which mutations are uniformly distributed across the sites in the genome. We can then check how many sites with 1, 2, and 3-fold multiplicity that we would expect by chance. We define nucleotide parallelism as the number of mutations occurring at the same site in the genome in independent populations. For each site, we define the multiplicity, $m_{i}$, as the number of populations (i.e., replicates) with a mutation detected at that site, so that the multiplicity in this experiment can range from one to three.</span>
<span id="cb35-173"><a href="#cb35-173"></a></span>
<span id="cb35-174"><a href="#cb35-174"></a>The expected number of mutations with $m_{i} \geq m$ in a sample of total mutations size $n_{tot}$ is:</span>
<span id="cb35-175"><a href="#cb35-175"></a></span>
<span id="cb35-176"><a href="#cb35-176"></a>$$ S(m) \approx \sum_{n \geq m} \frac{n}{n_{tot}} \cdot L_{tot} \cdot \frac{ \left( \frac{n_{tot}}{L_{tot}} \right) ^n}{n!} e^{-n_{tot}/L_{tot}} $$</span>
<span id="cb35-177"><a href="#cb35-177"></a></span>
<span id="cb35-178"><a href="#cb35-178"></a>where $L_{tot}$ is the total number of bases in the genome.</span>
<span id="cb35-179"><a href="#cb35-179"></a></span>
<span id="cb35-180"><a href="#cb35-180"></a><span class="fu">## Nucleotide parallelism by treatment</span></span>
<span id="cb35-181"><a href="#cb35-181"></a></span>
<span id="cb35-182"><a href="#cb35-182"></a>Here we will do the estimation separately for each species $\times$ treatment combination.</span>
<span id="cb35-183"><a href="#cb35-183"></a></span>
<span id="cb35-186"><a href="#cb35-186"></a><span class="in">```{r}</span></span>
<span id="cb35-187"><a href="#cb35-187"></a>nuc_survival <span class="ot">&lt;-</span> mgvars_filt_mb <span class="sc">%&gt;%</span></span>
<span id="cb35-188"><a href="#cb35-188"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, pos, ref, alt, replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb35-189"><a href="#cb35-189"></a>  <span class="co"># multiplicity = number of replicate populations each unique mutation is</span></span>
<span id="cb35-190"><a href="#cb35-190"></a>  <span class="co"># observed in group by the mutation position, with alternative allele to the</span></span>
<span id="cb35-191"><a href="#cb35-191"></a>  <span class="co"># grouping</span></span>
<span id="cb35-192"><a href="#cb35-192"></a>  <span class="fu">summarize</span>(<span class="at">m =</span> <span class="fu">n_distinct</span>(replicate), <span class="at">.by =</span> <span class="fu">c</span>(strainID, pos, ref, alt)) <span class="sc">%&gt;%</span></span>
<span id="cb35-193"><a href="#cb35-193"></a>  <span class="co"># now calculate the total number of mutations across all replicates so we need</span></span>
<span id="cb35-194"><a href="#cb35-194"></a>  <span class="co"># to ungroup by mutation position/alt allele but because we still want to</span></span>
<span id="cb35-195"><a href="#cb35-195"></a>  <span class="co"># determine this value by treatment category we keep the treatment category</span></span>
<span id="cb35-196"><a href="#cb35-196"></a>  <span class="co"># grouping. However, this should be changed if you want to for example average</span></span>
<span id="cb35-197"><a href="#cb35-197"></a>  <span class="co"># over all the treatment conditions on a species basis</span></span>
<span id="cb35-198"><a href="#cb35-198"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span></span>
<span id="cb35-199"><a href="#cb35-199"></a>  <span class="fu">count</span>(m, <span class="at">name =</span> <span class="st">"m_count"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-200"><a href="#cb35-200"></a>  <span class="fu">mutate</span>(<span class="at">n =</span> m <span class="sc">*</span> m_count,</span>
<span id="cb35-201"><a href="#cb35-201"></a>         <span class="at">Ntot =</span> <span class="fu">sum</span>(n),</span>
<span id="cb35-202"><a href="#cb35-202"></a>         <span class="at">perc =</span> n <span class="sc">/</span> Ntot <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-203"><a href="#cb35-203"></a>  <span class="fu">left_join</span>(genome_len, <span class="at">by =</span> <span class="fu">join_by</span>(strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb35-204"><a href="#cb35-204"></a>  <span class="fu">arrange</span>(<span class="fu">cur_group_id</span>(), <span class="fu">desc</span>(m)) <span class="sc">%&gt;%</span></span>
<span id="cb35-205"><a href="#cb35-205"></a>  <span class="co">#  dpois() tells the probability mass at a given number of counts. Here we</span></span>
<span id="cb35-206"><a href="#cb35-206"></a>  <span class="co">#  want to get the probability of observing n mutations with multiplicity</span></span>
<span id="cb35-207"><a href="#cb35-207"></a>  <span class="co">#  = mi (i.e. the counts of mi in the observed data). We assume that</span></span>
<span id="cb35-208"><a href="#cb35-208"></a>  <span class="co">#  mutations independently occur on the genome of size Ltot at a rate of</span></span>
<span id="cb35-209"><a href="#cb35-209"></a>  <span class="co">#  lambda = Ntot/Ltot and that generally the events are rare. Thus this</span></span>
<span id="cb35-210"><a href="#cb35-210"></a>  <span class="co">#  situation can be modeled by the Poisson distribution. We can get the</span></span>
<span id="cb35-211"><a href="#cb35-211"></a>  <span class="co">#  binned number of mutations per level of multiplicity m by multiplying</span></span>
<span id="cb35-212"><a href="#cb35-212"></a>  <span class="co">#  the probability by the length of the genome and the binned mutations</span></span>
<span id="cb35-213"><a href="#cb35-213"></a>  <span class="co">#  divided by the total number of mutations.</span></span>
<span id="cb35-214"><a href="#cb35-214"></a>  <span class="fu">mutate</span>(<span class="at">m_count_expected =</span> <span class="fu">cumsum</span>((m_count <span class="sc">/</span> Ntot) <span class="sc">*</span></span>
<span id="cb35-215"><a href="#cb35-215"></a>                                     total_len <span class="sc">*</span></span>
<span id="cb35-216"><a href="#cb35-216"></a>                                     <span class="fu">dpois</span>(m, <span class="at">lambda =</span> Ntot <span class="sc">/</span> total_len))) <span class="sc">%&gt;%</span></span>
<span id="cb35-217"><a href="#cb35-217"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>num_contigs) <span class="sc">%&gt;%</span></span>
<span id="cb35-218"><a href="#cb35-218"></a>  <span class="fu">relocate</span>(m, n, Ntot, perc, m_count, m_count_expected) <span class="sc">%&gt;%</span></span>
<span id="cb35-219"><a href="#cb35-219"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb35-220"><a href="#cb35-220"></a></span>
<span id="cb35-221"><a href="#cb35-221"></a><span class="co"># setup for plotting</span></span>
<span id="cb35-222"><a href="#cb35-222"></a>nuc_survival_plot <span class="ot">&lt;-</span> nuc_survival <span class="sc">%&gt;%</span> </span>
<span id="cb35-223"><a href="#cb35-223"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-224"><a href="#cb35-224"></a>  <span class="co"># when there is only one multiplicity observed for a mutation filter such</span></span>
<span id="cb35-225"><a href="#cb35-225"></a>  <span class="co"># that the multiplicty of that mutation must be greater than one.</span></span>
<span id="cb35-226"><a href="#cb35-226"></a>  <span class="co"># Otherwise include all remaining mutations (m &gt; 0)</span></span>
<span id="cb35-227"><a href="#cb35-227"></a>  <span class="fu">filter</span>(<span class="fu">case_when</span>(<span class="fu">n</span>() <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> m <span class="sc">&gt;</span> <span class="dv">1</span>,</span>
<span id="cb35-228"><a href="#cb35-228"></a>                       <span class="cn">TRUE</span> <span class="sc">~</span> m <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-229"><a href="#cb35-229"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"m_count"</span>, <span class="st">"m_count_expected"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-230"><a href="#cb35-230"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">paste</span>(strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-231"><a href="#cb35-231"></a>  <span class="co"># and make final plot</span></span>
<span id="cb35-232"><a href="#cb35-232"></a>  <span class="fu">plot_nuc_survival</span>(., <span class="dv">5000</span>, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">1000</span>, <span class="dv">10000</span>), <span class="dv">4</span>)</span>
<span id="cb35-233"><a href="#cb35-233"></a><span class="in">```</span></span>
<span id="cb35-234"><a href="#cb35-234"></a></span>
<span id="cb35-235"><a href="#cb35-235"></a>::: {#fig-01}</span>
<span id="cb35-238"><a href="#cb35-238"></a><span class="in">```{r}</span></span>
<span id="cb35-239"><a href="#cb35-239"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb35-240"><a href="#cb35-240"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb35-241"><a href="#cb35-241"></a><span class="co">#| echo: false</span></span>
<span id="cb35-242"><a href="#cb35-242"></a><span class="co">#| warning: false</span></span>
<span id="cb35-243"><a href="#cb35-243"></a>nuc_survival_plot</span>
<span id="cb35-244"><a href="#cb35-244"></a><span class="in">```</span></span>
<span id="cb35-245"><a href="#cb35-245"></a></span>
<span id="cb35-246"><a href="#cb35-246"></a>Distribution of nucleotide multiplicity $(m_{i})$ - i.e., the number of replicates with with the same genomic mutation including site and alternative allele - for each species with multiplicities $\gt 1$. The observed data is shown in blue while the null expectation is shown in red. In all cases the observed $m_{i}$ exceeded the null expectation meaning that we must reject the simple null model of uniform mutation distribution in favor of the alternative model where mutations are nonrandom and cluster across replicate populations.</span>
<span id="cb35-247"><a href="#cb35-247"></a>:::</span>
<span id="cb35-248"><a href="#cb35-248"></a></span>
<span id="cb35-249"><a href="#cb35-249"></a><span class="fu">## Conclusion</span></span>
<span id="cb35-250"><a href="#cb35-250"></a></span>
<span id="cb35-251"><a href="#cb35-251"></a>Nucleotide parallelism results are presented in Figure @fig-01. For these species this very simple null model mostly predicts that we should expect fewer than two parallel mutations (same position, same alternative allele) across the three replicate populations. For 12 evolved species the observed data show an excess of nucleotide parallelism relative to this simple null expectation. In particular HAMBI_0006, HAMBI_1287, and HAMBI_2494 all have at least 1 mutation that is identical across all three replicate evolved populations. However, multi-hit sites are only a very small fraction of the total observed mutations across all species (\~ 3% across species with multi-hit sites). Thus, we will continue our analysis but looking at the gene level.</span>
<span id="cb35-252"><a href="#cb35-252"></a></span>
<span id="cb35-255"><a href="#cb35-255"></a><span class="in">```{r}</span></span>
<span id="cb35-256"><a href="#cb35-256"></a><span class="co">#| warning: false</span></span>
<span id="cb35-257"><a href="#cb35-257"></a>nuc_survival <span class="sc">%&gt;%</span> </span>
<span id="cb35-258"><a href="#cb35-258"></a>  <span class="fu">mutate</span>(<span class="at">Ntot_f =</span> <span class="fu">sum</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-259"><a href="#cb35-259"></a>  <span class="fu">summarize</span>(<span class="at">perc =</span> <span class="fu">sum</span>(n)<span class="sc">/</span>Ntot_f<span class="sc">*</span><span class="dv">100</span>, <span class="at">.by =</span> <span class="st">"m"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-260"><a href="#cb35-260"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-261"><a href="#cb35-261"></a>  <span class="fu">mutate</span>(<span class="at">cum_perc =</span> <span class="fu">cumsum</span>(perc))</span>
<span id="cb35-262"><a href="#cb35-262"></a><span class="in">```</span></span>
<span id="cb35-263"><a href="#cb35-263"></a></span>
<span id="cb35-264"><a href="#cb35-264"></a><span class="fu"># Parallelism at gene level</span></span>
<span id="cb35-265"><a href="#cb35-265"></a></span>
<span id="cb35-266"><a href="#cb35-266"></a>If selection pressures and mutation rates did not vary between genes, the number of mutations in each gene should be proportional to the target gene size. We assume that gene length is a primary driver of the target size. We then define a multiplicity for each gene according to:</span>
<span id="cb35-267"><a href="#cb35-267"></a></span>
<span id="cb35-268"><a href="#cb35-268"></a>$$m_{i} = n_{i} \cdot \frac{\overline{L}}{L_{i}}$$ {#eq-01}</span>
<span id="cb35-269"><a href="#cb35-269"></a></span>
<span id="cb35-270"><a href="#cb35-270"></a>where $n_{i}$ is the number of nonsynonymous mutations (i.e., amino acid changing) in $gene_{i}$ across all replicate populations per species, $L_{i}$ is the total number of nonsynonymous sites (i.e., excluding 4-fold degenerate sites) in $gene_{i}$, and $\overline{L}$ is the average value of $L_{i}$ across all genes in the genome. This definition ensures that under the null hypothesis, all genes have the same expected multiplicity $\overline{m} = n_{tot}/n_{genes}$.</span>
<span id="cb35-271"><a href="#cb35-271"></a></span>
<span id="cb35-272"><a href="#cb35-272"></a>We next define a null model for gene multiplicity that assumes mutations are assigned to genes with probability:</span>
<span id="cb35-273"><a href="#cb35-273"></a></span>
<span id="cb35-274"><a href="#cb35-274"></a>$$p_{i} \propto L_{i} r_{i}$$ {#eq-02}</span>
<span id="cb35-275"><a href="#cb35-275"></a></span>
<span id="cb35-276"><a href="#cb35-276"></a>for some set of enrichment factors $r_{i}$. In the alternate model, the maximum likelihood estimator of the enrichment factors $r_{i}$ is the ratio of observed to expected gene multiplicities, $r_{i} = m_{i}/\overline{m}$. The net increase of the log-likelihood relative to the null model ($r_{i} = 1$) is then given by:</span>
<span id="cb35-277"><a href="#cb35-277"></a></span>
<span id="cb35-278"><a href="#cb35-278"></a>$$\Delta \ell = \sum_{i} n_{i} \log \left ( \frac{m_{i}}{\overline{m}} \right )$$ {#eq-03}</span>
<span id="cb35-279"><a href="#cb35-279"></a></span>
<span id="cb35-280"><a href="#cb35-280"></a>This likelihood ratio estimator is equivalent to the total G-score introduced in earlier work <span class="co">[</span><span class="ot">@tenaillon2016</span><span class="co">]</span>. Here we assess statistical significance using permutation tests following Shoemaker *et al.* <span class="co">[</span><span class="ot">@shoemaker2021</span><span class="co">]</span>. For permutations tests (n = 10,000) we randomly resample mutations to the observed $n_{tot}$ set size from each species using the multinomial distribution, where the probability of sampling a non-synonymous mutation at $gene_{i}$ was given by $p_{i} = {L_{i}} / {L_{tot}}$ where $L_{i}$ is the total number of nonsynonymous sites in $gene_{i}$ and $L_{tot}$ is the total number of nonsynonymous sites in the genome.</span>
<span id="cb35-281"><a href="#cb35-281"></a></span>
<span id="cb35-282"><a href="#cb35-282"></a>We next identify specific genes enriched for mutations following the criteria of Good *et. al* . <span class="co">[</span><span class="ot">@good2017</span><span class="co">]</span>. We calculate a \textit{p} value for each gene using the equation:</span>
<span id="cb35-283"><a href="#cb35-283"></a></span>
<span id="cb35-284"><a href="#cb35-284"></a>$$P_{i} = \sum_{n \geq n_{i}} \frac{ \left ( \frac{n_{tot} L_{i}}{ \overline{L} N_{genes}} \right )^{n}}{n!} e^{- \frac{n_{tot} L_{i} }{\overline{L} N_{genes}}}$$ {#eq-04}</span>
<span id="cb35-285"><a href="#cb35-285"></a></span>
<span id="cb35-286"><a href="#cb35-286"></a>Here we only consider genes with $n_{i} \geq 3$ to exclude low \textit{p} values driven primarily by gene length. Under the null hypothesis, the expected number of genes with $P_{i} \geq p$ can be found using a Poisson survival curve given by:</span>
<span id="cb35-287"><a href="#cb35-287"></a></span>
<span id="cb35-288"><a href="#cb35-288"></a>$$\overline{N}(P) \approx \sum_{i=1}^{N_{genes}} \sum_{n=3}^{\infty} \theta (P - P_{i}(n, L_{i}) \cdot \frac{ \left( \frac{n_{tot} L_{i}}{\overline{L} N_{genes}} \right)^{n}}{n!} e^{-\left( \frac{n_{tot} L_{i}}{\overline{L} N_{genes}} \right)}$$ {#eq-05}</span>
<span id="cb35-289"><a href="#cb35-289"></a></span>
<span id="cb35-290"><a href="#cb35-290"></a>We can compare this expected number to the observed number of genes $N(P)$ using a critical \textit{p} value ($P^{\ast}$) such that</span>
<span id="cb35-291"><a href="#cb35-291"></a></span>
<span id="cb35-292"><a href="#cb35-292"></a>$$\frac{\overline{N}(P^{\ast})}{N(P^{\ast})} \leq \alpha$$ {#eq-06}</span>
<span id="cb35-293"><a href="#cb35-293"></a></span>
<span id="cb35-294"><a href="#cb35-294"></a>for a given FDR $\alpha$ value of 0.05. For this value we then define the set of significantly enriched genes as:</span>
<span id="cb35-295"><a href="#cb35-295"></a></span>
<span id="cb35-296"><a href="#cb35-296"></a>$$I = \left<span class="sc">\{</span> i : P_{i} \leq P^{\ast} \left(\alpha \right) \right<span class="sc">\}</span>$$ {#eq-07}</span>
<span id="cb35-297"><a href="#cb35-297"></a></span>
<span id="cb35-298"><a href="#cb35-298"></a><span class="fu">## Prepare input data for non-syn mutations only</span></span>
<span id="cb35-299"><a href="#cb35-299"></a></span>
<span id="cb35-302"><a href="#cb35-302"></a><span class="in">```{r}</span></span>
<span id="cb35-303"><a href="#cb35-303"></a>mgvars_filt_ns <span class="ot">&lt;-</span> mgvars_filt_mb <span class="sc">%&gt;%</span> </span>
<span id="cb35-304"><a href="#cb35-304"></a>  <span class="co"># exclude intragenic, intergenic, and synonymous mutations. Also exclude</span></span>
<span id="cb35-305"><a href="#cb35-305"></a>  <span class="co"># fusion functions because these are weird and also rare. Excluding the</span></span>
<span id="cb35-306"><a href="#cb35-306"></a>  <span class="co"># modifier category also ensures that we filter out any tRNAs with mutations</span></span>
<span id="cb35-307"><a href="#cb35-307"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(effect, <span class="st">"intergenic|intragenic|synonymous|fusion"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-308"><a href="#cb35-308"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(impact, <span class="st">"MODIFIER"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-309"><a href="#cb35-309"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, replicate, chrom<span class="sc">:</span>alt, locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb35-310"><a href="#cb35-310"></a>  <span class="fu">relocate</span>(strainID, locus_tag)</span>
<span id="cb35-311"><a href="#cb35-311"></a><span class="in">```</span></span>
<span id="cb35-312"><a href="#cb35-312"></a></span>
<span id="cb35-313"><a href="#cb35-313"></a><span class="fu">## G score</span></span>
<span id="cb35-314"><a href="#cb35-314"></a></span>
<span id="cb35-315"><a href="#cb35-315"></a>Warning! this takes some time</span>
<span id="cb35-316"><a href="#cb35-316"></a></span>
<span id="cb35-319"><a href="#cb35-319"></a><span class="in">```{r}</span></span>
<span id="cb35-320"><a href="#cb35-320"></a><span class="co">#| eval: false</span></span>
<span id="cb35-321"><a href="#cb35-321"></a><span class="co">#| echo: true</span></span>
<span id="cb35-322"><a href="#cb35-322"></a>muts_by_sp <span class="ot">&lt;-</span> mgvars_filt_ns <span class="sc">%&gt;%</span> </span>
<span id="cb35-323"><a href="#cb35-323"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-324"><a href="#cb35-324"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-325"><a href="#cb35-325"></a>  <span class="fu">mutate</span>(<span class="at">groupid =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb35-326"><a href="#cb35-326"></a>  <span class="fu">group_by</span>(groupid, strainID)</span>
<span id="cb35-327"><a href="#cb35-327"></a></span>
<span id="cb35-328"><a href="#cb35-328"></a><span class="co"># doing 10000 permutations</span></span>
<span id="cb35-329"><a href="#cb35-329"></a>result_sp <span class="ot">&lt;-</span> <span class="fu">run_full_gene_parallelism_pipeline</span>(muts_by_sp, degentab, <span class="dv">10000</span>)</span>
<span id="cb35-330"><a href="#cb35-330"></a></span>
<span id="cb35-331"><a href="#cb35-331"></a><span class="co"># Save this for later</span></span>
<span id="cb35-332"><a href="#cb35-332"></a><span class="fu">write_rds</span>(result_sp, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"parallelism_results_sp_nomobile.rds"</span>))</span>
<span id="cb35-333"><a href="#cb35-333"></a><span class="in">```</span></span>
<span id="cb35-334"><a href="#cb35-334"></a></span>
<span id="cb35-337"><a href="#cb35-337"></a><span class="in">```{r}</span></span>
<span id="cb35-338"><a href="#cb35-338"></a><span class="co"># Read in previous results</span></span>
<span id="cb35-339"><a href="#cb35-339"></a>result_sp <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"parallelism_results_sp_nomobile.rds"</span>))</span>
<span id="cb35-340"><a href="#cb35-340"></a><span class="in">```</span></span>
<span id="cb35-341"><a href="#cb35-341"></a></span>
<span id="cb35-342"><a href="#cb35-342"></a><span class="fu">### Simulating the number of replicates with a parallel gene hit</span></span>
<span id="cb35-343"><a href="#cb35-343"></a></span>
<span id="cb35-344"><a href="#cb35-344"></a>As another sanity check, we took the resamplings from the null distribution and calculated the number of replicates with a parallel gene hit (i.e. the same gene hit in $\geq 2$ experimental replicates) and compared that to the observed distribution. The null distribution reflected the process of sampling a mutation under a multinomial distribution with probability proportional to the total number of nonsynonymous sites $L_{i}$ in $gene_{i}$. We observed that in 9/16 cases the null distribution produced fewer parallel gene hits than observed in the data from 2 experimental replicates. In all 16 cases the null distribution produced fewer parallel gene hits than observed in the data from 3 experimental replicates. Thus, for some suspicious genes with very high mutational density (see below) we required these genes to be present in all three replicates to include them in our final significant gene lists.</span>
<span id="cb35-345"><a href="#cb35-345"></a></span>
<span id="cb35-346"><a href="#cb35-346"></a>::: {#fig-03a}</span>
<span id="cb35-349"><a href="#cb35-349"></a><span class="in">```{r}</span></span>
<span id="cb35-350"><a href="#cb35-350"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb35-351"><a href="#cb35-351"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb35-352"><a href="#cb35-352"></a><span class="co">#| echo: false</span></span>
<span id="cb35-353"><a href="#cb35-353"></a><span class="co">#| warning: false</span></span>
<span id="cb35-354"><a href="#cb35-354"></a><span class="fu">plot_sims</span>(result_sp, <span class="dv">1</span>)</span>
<span id="cb35-355"><a href="#cb35-355"></a><span class="in">```</span></span>
<span id="cb35-356"><a href="#cb35-356"></a></span>
<span id="cb35-357"><a href="#cb35-357"></a>Number of genes (y axis) with nonsynonymous mutation hits in $N_{replicates} = <span class="sc">\{</span>n : 1 \leq n \leq 3<span class="sc">\}</span>$ (x axis). Blue bars depict the distribution observed in the data, while red bars depict the null distribution from resampling mutations to the observed $n_{tot}$ set size from each species using the multinomial distribution, where the probability of sampling a non-synonymous mutation at $gene_{i}$ was dependent only on the number of nonsynonymous sites in the gene.</span>
<span id="cb35-358"><a href="#cb35-358"></a>:::</span>
<span id="cb35-359"><a href="#cb35-359"></a></span>
<span id="cb35-360"><a href="#cb35-360"></a>::: {#fig-03b}</span>
<span id="cb35-363"><a href="#cb35-363"></a><span class="in">```{r}</span></span>
<span id="cb35-364"><a href="#cb35-364"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb35-365"><a href="#cb35-365"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb35-366"><a href="#cb35-366"></a><span class="co">#| echo: false</span></span>
<span id="cb35-367"><a href="#cb35-367"></a><span class="co">#| warning: false</span></span>
<span id="cb35-368"><a href="#cb35-368"></a><span class="fu">plot_sims</span>(result_sp, <span class="dv">2</span>)</span>
<span id="cb35-369"><a href="#cb35-369"></a><span class="in">```</span></span>
<span id="cb35-370"><a href="#cb35-370"></a></span>
<span id="cb35-371"><a href="#cb35-371"></a>Same as in @fig-03a but for different treatment combinations.</span>
<span id="cb35-372"><a href="#cb35-372"></a>:::</span>
<span id="cb35-373"><a href="#cb35-373"></a></span>
<span id="cb35-374"><a href="#cb35-374"></a><span class="fu">### Visualize the survival curves for each treatment combination</span></span>
<span id="cb35-375"><a href="#cb35-375"></a></span>
<span id="cb35-378"><a href="#cb35-378"></a><span class="in">```{r}</span></span>
<span id="cb35-379"><a href="#cb35-379"></a>survcurv <span class="ot">&lt;-</span> result_sp<span class="sc">$</span>output_df2plot <span class="sc">%&gt;%</span> </span>
<span id="cb35-380"><a href="#cb35-380"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(xend)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-381"><a href="#cb35-381"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-382"><a href="#cb35-382"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb35-383"><a href="#cb35-383"></a>    <span class="fu">geom_segment</span>(</span>
<span id="cb35-384"><a href="#cb35-384"></a>      <span class="fu">aes</span>(</span>
<span id="cb35-385"><a href="#cb35-385"></a>        <span class="at">x =</span> x,</span>
<span id="cb35-386"><a href="#cb35-386"></a>        <span class="at">y =</span> y,</span>
<span id="cb35-387"><a href="#cb35-387"></a>        <span class="at">xend =</span> xend,</span>
<span id="cb35-388"><a href="#cb35-388"></a>        <span class="at">yend =</span> yend</span>
<span id="cb35-389"><a href="#cb35-389"></a>      ),</span>
<span id="cb35-390"><a href="#cb35-390"></a>      <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb35-391"><a href="#cb35-391"></a>      <span class="at">linewidth =</span> <span class="fl">0.25</span>, <span class="at">color =</span> <span class="st">"red"</span></span>
<span id="cb35-392"><a href="#cb35-392"></a>    ) <span class="sc">+</span></span>
<span id="cb35-393"><a href="#cb35-393"></a>    <span class="fu">geom_step</span>( <span class="fu">aes</span>(<span class="at">x =</span> obs_p, <span class="at">y =</span> value, <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb35-394"><a href="#cb35-394"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> label, <span class="at">ncol =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb35-395"><a href="#cb35-395"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"$-log_{10}P$"</span>), <span class="at">y =</span> <span class="st">"Number of genes"</span>) <span class="sc">+</span></span>
<span id="cb35-396"><a href="#cb35-396"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"grey50"</span>)) <span class="sc">+</span></span>
<span id="cb35-397"><a href="#cb35-397"></a>    <span class="fu">scale_y_log10</span>(</span>
<span id="cb35-398"><a href="#cb35-398"></a>      <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb35-399"><a href="#cb35-399"></a>      <span class="at">labels =</span> scales<span class="sc">::</span><span class="fu">trans_format</span>(<span class="st">"log10"</span>, scales<span class="sc">::</span><span class="fu">math_format</span>(<span class="dv">10</span> <span class="sc">^</span>.x))</span>
<span id="cb35-400"><a href="#cb35-400"></a>    ) <span class="sc">+</span></span>
<span id="cb35-401"><a href="#cb35-401"></a>    <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb35-402"><a href="#cb35-402"></a>    <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">200</span>)) <span class="sc">+</span></span>
<span id="cb35-403"><a href="#cb35-403"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb35-404"><a href="#cb35-404"></a>    <span class="fu">theme</span>(</span>
<span id="cb35-405"><a href="#cb35-405"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-406"><a href="#cb35-406"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-407"><a href="#cb35-407"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb35-408"><a href="#cb35-408"></a>    )</span>
<span id="cb35-409"><a href="#cb35-409"></a><span class="in">```</span></span>
<span id="cb35-410"><a href="#cb35-410"></a></span>
<span id="cb35-411"><a href="#cb35-411"></a>::: {#fig-04}</span>
<span id="cb35-414"><a href="#cb35-414"></a><span class="in">```{r}</span></span>
<span id="cb35-415"><a href="#cb35-415"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb35-416"><a href="#cb35-416"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb35-417"><a href="#cb35-417"></a><span class="co">#| echo: false</span></span>
<span id="cb35-418"><a href="#cb35-418"></a><span class="co">#| warning: false</span></span>
<span id="cb35-419"><a href="#cb35-419"></a>survcurv</span>
<span id="cb35-420"><a href="#cb35-420"></a><span class="in">```</span></span>
<span id="cb35-421"><a href="#cb35-421"></a></span>
<span id="cb35-422"><a href="#cb35-422"></a>The observed $(N)$ (blue line) and expected number $(\overline{N})$ (grey line) of genes with $P_{i} \leq P$, as a function of $P$. The red dashed line denotes the genome-wide significance threshold $P^{*}$ for $\alpha = 0.05$ defined in Equation @eq-06.</span>
<span id="cb35-423"><a href="#cb35-423"></a>:::</span>
<span id="cb35-424"><a href="#cb35-424"></a></span>
<span id="cb35-425"><a href="#cb35-425"></a><span class="fu">### Inspecting individual genes</span></span>
<span id="cb35-426"><a href="#cb35-426"></a></span>
<span id="cb35-429"><a href="#cb35-429"></a><span class="in">```{r}</span></span>
<span id="cb35-430"><a href="#cb35-430"></a>result_sp<span class="sc">$</span>output_gscore <span class="sc">%&gt;%</span> </span>
<span id="cb35-431"><a href="#cb35-431"></a>  <span class="fu">filter</span>(pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span>
<span id="cb35-432"><a href="#cb35-432"></a><span class="in">```</span></span>
<span id="cb35-433"><a href="#cb35-433"></a></span>
<span id="cb35-434"><a href="#cb35-434"></a>Also some genes tend to have a large number of consecutive mutations which is probably due to incorrect read mapping and not real mutations so we will try and filter these out. We'll do this by filtering on genes with a mutational density $d_{g} \gt 0.15$, where the mutational density in gene $g$ was defined as</span>
<span id="cb35-435"><a href="#cb35-435"></a></span>
<span id="cb35-436"><a href="#cb35-436"></a>$$d_{g} = \frac{|A|_{g}}{b_{g} - a_{g}}$$</span>
<span id="cb35-437"><a href="#cb35-437"></a></span>
<span id="cb35-438"><a href="#cb35-438"></a>for all genome positions $x$ on the interval $<span class="sc">\{</span>x | a_{g} \leq x \leq b_{g}<span class="sc">\}</span>$ where $b_{g}$ is the greatest position and $a_{g}$ is the smallest position of observed mutations in gene $g$ and $|A|_{g}$ is the cardinality of the mutation set.</span>
<span id="cb35-439"><a href="#cb35-439"></a></span>
<span id="cb35-442"><a href="#cb35-442"></a><span class="in">```{r}</span></span>
<span id="cb35-443"><a href="#cb35-443"></a>high_density_genes <span class="ot">&lt;-</span> mgvars_filt_ns <span class="sc">%&gt;%</span> </span>
<span id="cb35-444"><a href="#cb35-444"></a>  <span class="fu">group_by</span>(strainID, locus_tag, replicate) <span class="sc">%&gt;%</span> </span>
<span id="cb35-445"><a href="#cb35-445"></a>  <span class="fu">mutate</span>(<span class="at">mutwindox =</span> pos <span class="sc">-</span> <span class="fu">lag</span>(pos),</span>
<span id="cb35-446"><a href="#cb35-446"></a>         <span class="at">mut_num =</span> <span class="fu">n</span>(),</span>
<span id="cb35-447"><a href="#cb35-447"></a>         <span class="at">mut_dens =</span> <span class="fu">n</span>()<span class="sc">/</span>(<span class="fu">max</span>(pos) <span class="sc">-</span>  <span class="fu">min</span>(pos))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-448"><a href="#cb35-448"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-449"><a href="#cb35-449"></a>  <span class="fu">filter</span>(mut_dens <span class="sc">!=</span> <span class="cn">Inf</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-450"><a href="#cb35-450"></a>  <span class="fu">filter</span>(mut_dens <span class="sc">&gt;</span> <span class="fl">0.15</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-451"><a href="#cb35-451"></a>  <span class="fu">distinct</span>(locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb35-452"><a href="#cb35-452"></a>  <span class="fu">pull</span>()</span>
<span id="cb35-453"><a href="#cb35-453"></a><span class="in">```</span></span>
<span id="cb35-454"><a href="#cb35-454"></a></span>
<span id="cb35-457"><a href="#cb35-457"></a><span class="in">```{r}</span></span>
<span id="cb35-458"><a href="#cb35-458"></a>output_gene_table_filt <span class="ot">&lt;-</span> result_sp<span class="sc">$</span>output_gene_table <span class="sc">%&gt;%</span> </span>
<span id="cb35-459"><a href="#cb35-459"></a>  <span class="co"># only genes exceeding the critical pstar value from the survival curves above</span></span>
<span id="cb35-460"><a href="#cb35-460"></a>  <span class="fu">filter</span>(neg_log10P <span class="sc">&gt;=</span> pstar) <span class="sc">%&gt;%</span> </span>
<span id="cb35-461"><a href="#cb35-461"></a>  <span class="co"># only include genes with at least 3 hits in the same gene or found in more than one replicate.</span></span>
<span id="cb35-462"><a href="#cb35-462"></a>  <span class="fu">filter</span>(observed_hits_n_i <span class="sc">&gt;=</span> <span class="dv">3</span> <span class="sc">|</span> n_replicate <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-463"><a href="#cb35-463"></a>  <span class="co"># filters out the problematic genes above unless the number of replicates detected is 2 or more</span></span>
<span id="cb35-464"><a href="#cb35-464"></a>  <span class="fu">filter</span>(<span class="fu">case_when</span>(n_replicate <span class="sc">&lt;</span> <span class="dv">2</span> <span class="sc">~</span> <span class="sc">!</span>(locus_tag <span class="sc">%in%</span> high_density_genes), </span>
<span id="cb35-465"><a href="#cb35-465"></a>                   <span class="cn">TRUE</span> <span class="sc">~</span> n_replicate <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-466"><a href="#cb35-466"></a>  <span class="fu">left_join</span>(<span class="fu">distinct</span>(<span class="fu">select</span>(mgvars_filt, locus_tag, product, cds_length<span class="sc">:</span>gene)),</span>
<span id="cb35-467"><a href="#cb35-467"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag))</span>
<span id="cb35-468"><a href="#cb35-468"></a></span>
<span id="cb35-469"><a href="#cb35-469"></a>output_gene_table_filt</span>
<span id="cb35-470"><a href="#cb35-470"></a><span class="in">```</span></span>
<span id="cb35-471"><a href="#cb35-471"></a></span>
<span id="cb35-472"><a href="#cb35-472"></a><span class="fu"># Convergent/divergent evolution</span></span>
<span id="cb35-473"><a href="#cb35-473"></a></span>
<span id="cb35-474"><a href="#cb35-474"></a>So now we have a list of enriched genes for each species. We can see that, for example, some gene functions pop up multiple times (like transcription factors and sensory kinases). Are these different functions popping up across species more that we would expect by chance? We'll try and test this using a simulation based approach</span>
<span id="cb35-475"><a href="#cb35-475"></a></span>
<span id="cb35-478"><a href="#cb35-478"></a><span class="in">```{r}</span></span>
<span id="cb35-479"><a href="#cb35-479"></a><span class="co">#| eval: false</span></span>
<span id="cb35-480"><a href="#cb35-480"></a><span class="co">#| echo: true</span></span>
<span id="cb35-481"><a href="#cb35-481"></a><span class="co"># Read in results of eggnogmapper which maps a COG category to every gene it can in each genome</span></span>
<span id="cb35-482"><a href="#cb35-482"></a>eggnograw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_all.emapper.annotations.xz"</span>)) </span>
<span id="cb35-483"><a href="#cb35-483"></a><span class="in">```</span></span>
<span id="cb35-484"><a href="#cb35-484"></a></span>
<span id="cb35-487"><a href="#cb35-487"></a><span class="in">```{r}</span></span>
<span id="cb35-488"><a href="#cb35-488"></a>cog_description <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb35-489"><a href="#cb35-489"></a>  <span class="sc">~</span>COG_category_single,                                                  <span class="sc">~</span>COG_category_long,</span>
<span id="cb35-490"><a href="#cb35-490"></a>            <span class="st">"J"</span>,               <span class="st">"J - Translation, ribosomal structure and biogenesis"</span>,</span>
<span id="cb35-491"><a href="#cb35-491"></a>            <span class="st">"A"</span>,                               <span class="st">"A - RNA processing and modification"</span>,</span>
<span id="cb35-492"><a href="#cb35-492"></a>            <span class="st">"K"</span>,                                                 <span class="st">"K – Transcription"</span>,</span>
<span id="cb35-493"><a href="#cb35-493"></a>            <span class="st">"L"</span>,                         <span class="st">"L - Replication, recombination and repair"</span>,</span>
<span id="cb35-494"><a href="#cb35-494"></a>            <span class="st">"B"</span>,                              <span class="st">"B - Chromatin structure and dynamics"</span>,</span>
<span id="cb35-495"><a href="#cb35-495"></a>            <span class="st">"D"</span>,    <span class="st">"D - Cell cycle control, cell division, chromosome partitioning"</span>,</span>
<span id="cb35-496"><a href="#cb35-496"></a>            <span class="st">"Y"</span>,                                             <span class="st">"Y - Nuclear structure"</span>,</span>
<span id="cb35-497"><a href="#cb35-497"></a>            <span class="st">"V"</span>,                                            <span class="st">"V - Defense mechanisms"</span>,</span>
<span id="cb35-498"><a href="#cb35-498"></a>            <span class="st">"T"</span>,                                <span class="st">"T - Signal transduction mechanisms"</span>,</span>
<span id="cb35-499"><a href="#cb35-499"></a>            <span class="st">"M"</span>,                        <span class="st">"M - Cell wall/membrane/envelope biogenesis"</span>,</span>
<span id="cb35-500"><a href="#cb35-500"></a>            <span class="st">"N"</span>,                                                 <span class="st">"N - Cell motility"</span>,</span>
<span id="cb35-501"><a href="#cb35-501"></a>            <span class="st">"Z"</span>,                                                  <span class="st">"Z – Cytoskeleton"</span>,</span>
<span id="cb35-502"><a href="#cb35-502"></a>            <span class="st">"W"</span>,                                      <span class="st">"W - Extracellular structures"</span>,</span>
<span id="cb35-503"><a href="#cb35-503"></a>            <span class="st">"U"</span>, <span class="st">"U - Intracellular trafficking, secretion, and vesicular transport"</span>,</span>
<span id="cb35-504"><a href="#cb35-504"></a>            <span class="st">"O"</span>,  <span class="st">"O - Posttranslational modification, protein turnover, chaperones"</span>,</span>
<span id="cb35-505"><a href="#cb35-505"></a>            <span class="st">"X"</span>,                              <span class="st">"X - Mobilome: prophages, transposons"</span>,</span>
<span id="cb35-506"><a href="#cb35-506"></a>            <span class="st">"C"</span>,                              <span class="st">"C - Energy production and conversion"</span>,</span>
<span id="cb35-507"><a href="#cb35-507"></a>            <span class="st">"G"</span>,                         <span class="st">"G - Carbohydrate transport and metabolism"</span>,</span>
<span id="cb35-508"><a href="#cb35-508"></a>            <span class="st">"E"</span>,                           <span class="st">"E - Amino acid transport and metabolism"</span>,</span>
<span id="cb35-509"><a href="#cb35-509"></a>            <span class="st">"F"</span>,                           <span class="st">"F - Nucleotide transport and metabolism"</span>,</span>
<span id="cb35-510"><a href="#cb35-510"></a>            <span class="st">"H"</span>,                             <span class="st">"H - Coenzyme transport and metabolism"</span>,</span>
<span id="cb35-511"><a href="#cb35-511"></a>            <span class="st">"I"</span>,                                <span class="st">"I - Lipid transport and metabolism"</span>,</span>
<span id="cb35-512"><a href="#cb35-512"></a>            <span class="st">"P"</span>,                        <span class="st">"P - Inorganic ion transport and metabolism"</span>,</span>
<span id="cb35-513"><a href="#cb35-513"></a>            <span class="st">"Q"</span>,  <span class="st">"Q - Secondary metabolites biosynthesis, transport and catabolism"</span>,</span>
<span id="cb35-514"><a href="#cb35-514"></a>            <span class="st">"R"</span>,                              <span class="st">"R - General function prediction only"</span>,</span>
<span id="cb35-515"><a href="#cb35-515"></a>            <span class="st">"S"</span>,                                              <span class="st">"S - Function unknown"</span></span>
<span id="cb35-516"><a href="#cb35-516"></a>  )</span>
<span id="cb35-517"><a href="#cb35-517"></a><span class="in">```</span></span>
<span id="cb35-518"><a href="#cb35-518"></a></span>
<span id="cb35-521"><a href="#cb35-521"></a><span class="in">```{r}</span></span>
<span id="cb35-522"><a href="#cb35-522"></a><span class="co">#| eval: false</span></span>
<span id="cb35-523"><a href="#cb35-523"></a><span class="co">#| echo: true</span></span>
<span id="cb35-524"><a href="#cb35-524"></a><span class="co"># formatting the eggnog dataframe for downstream use</span></span>
<span id="cb35-525"><a href="#cb35-525"></a>enrichedspprefix <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">paste0</span>(<span class="st">"H"</span>, <span class="fu">str_extract</span>(output_gene_table_filt<span class="sc">$</span>locus_tag, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>)))</span>
<span id="cb35-526"><a href="#cb35-526"></a>enrichedsp <span class="ot">&lt;-</span> <span class="fu">unique</span>(output_gene_table_filt<span class="sc">$</span>strainID)</span>
<span id="cb35-527"><a href="#cb35-527"></a><span class="fu">names</span>(enrichedsp) <span class="ot">&lt;-</span> enrichedspprefix</span>
<span id="cb35-528"><a href="#cb35-528"></a></span>
<span id="cb35-529"><a href="#cb35-529"></a><span class="co"># use a seed for reproducibility because we are randomly sampling the split COGs</span></span>
<span id="cb35-530"><a href="#cb35-530"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">6783543</span>, eggnog <span class="ot">&lt;-</span> eggnograw <span class="sc">%&gt;%</span>  </span>
<span id="cb35-531"><a href="#cb35-531"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(query, <span class="fu">paste</span>(enrichedspprefix, <span class="at">collapse=</span><span class="st">"|"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-532"><a href="#cb35-532"></a>  <span class="fu">mutate</span>(<span class="at">COG_category =</span> <span class="fu">if_else</span>(COG_category <span class="sc">==</span> <span class="st">"-"</span>, <span class="cn">NA_character_</span>, COG_category)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-533"><a href="#cb35-533"></a>  <span class="co"># If multiple cog categories, split them into a character vector then randomly sample</span></span>
<span id="cb35-534"><a href="#cb35-534"></a>  <span class="co"># one of them</span></span>
<span id="cb35-535"><a href="#cb35-535"></a>  <span class="fu">mutate</span>(<span class="at">COG_category1 =</span> <span class="fu">strsplit</span>(COG_category, <span class="at">split =</span> <span class="st">""</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-536"><a href="#cb35-536"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-537"><a href="#cb35-537"></a>  <span class="fu">mutate</span>(<span class="at">COG_category_single =</span> <span class="fu">sample</span>(COG_category1, <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-538"><a href="#cb35-538"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-539"><a href="#cb35-539"></a>  <span class="fu">mutate</span>(<span class="at">strmatchvar =</span> <span class="fu">str_extract</span>(query, <span class="st">"^[:alnum:]*"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-540"><a href="#cb35-540"></a>  <span class="fu">mutate</span>(<span class="at">strainID =</span> <span class="fu">str_replace_all</span>(strmatchvar, enrichedsp)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-541"><a href="#cb35-541"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>strmatchvar, <span class="sc">-</span>COG_category1) <span class="sc">%&gt;%</span> </span>
<span id="cb35-542"><a href="#cb35-542"></a>  <span class="fu">rename</span>(<span class="at">locus_tag =</span> query) <span class="sc">%&gt;%</span> </span>
<span id="cb35-543"><a href="#cb35-543"></a>  <span class="fu">left_join</span>(cog_description, <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-544"><a href="#cb35-544"></a>  <span class="fu">relocate</span>(strainID, locus_tag, COG_category_single, COG_category_long))</span>
<span id="cb35-545"><a href="#cb35-545"></a></span>
<span id="cb35-546"><a href="#cb35-546"></a><span class="fu">write_tsv</span>(eggnog, here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_all_eggnog_formatted.tsv.xz"</span>)) </span>
<span id="cb35-547"><a href="#cb35-547"></a><span class="in">```</span></span>
<span id="cb35-548"><a href="#cb35-548"></a></span>
<span id="cb35-551"><a href="#cb35-551"></a><span class="in">```{r}</span></span>
<span id="cb35-552"><a href="#cb35-552"></a><span class="co">#| output: false</span></span>
<span id="cb35-553"><a href="#cb35-553"></a><span class="co"># read back the formatted eggnog data</span></span>
<span id="cb35-554"><a href="#cb35-554"></a>eggnog <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_all_eggnog_formatted.tsv.xz"</span>))</span>
<span id="cb35-555"><a href="#cb35-555"></a><span class="in">```</span></span>
<span id="cb35-556"><a href="#cb35-556"></a></span>
<span id="cb35-557"><a href="#cb35-557"></a>save a copy of significantly parallel genes as a table</span>
<span id="cb35-558"><a href="#cb35-558"></a></span>
<span id="cb35-561"><a href="#cb35-561"></a><span class="in">```{r}</span></span>
<span id="cb35-562"><a href="#cb35-562"></a><span class="fu">left_join</span>(output_gene_table_filt, eggnog, </span>
<span id="cb35-563"><a href="#cb35-563"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-564"><a href="#cb35-564"></a>  <span class="fu">rename</span>(<span class="at">prokka_annotation =</span> product) <span class="sc">%&gt;%</span> </span>
<span id="cb35-565"><a href="#cb35-565"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"enriched_parallel_genes.tsv"</span>))</span>
<span id="cb35-566"><a href="#cb35-566"></a><span class="in">```</span></span>
<span id="cb35-567"><a href="#cb35-567"></a></span>
<span id="cb35-568"><a href="#cb35-568"></a></span>
<span id="cb35-569"><a href="#cb35-569"></a><span class="fu">## All species</span></span>
<span id="cb35-570"><a href="#cb35-570"></a></span>
<span id="cb35-571"><a href="#cb35-571"></a>Calculate the probability of each gene (ns_sites/total_ns_sites) and link to COG categories</span>
<span id="cb35-572"><a href="#cb35-572"></a></span>
<span id="cb35-575"><a href="#cb35-575"></a><span class="in">```{r}</span></span>
<span id="cb35-576"><a href="#cb35-576"></a>eggnog_nslen <span class="ot">&lt;-</span> <span class="fu">left_join</span>(eggnog, degentab,</span>
<span id="cb35-577"><a href="#cb35-577"></a>                          <span class="at">by =</span> <span class="fu">join_by</span>(strainID, locus_tag)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-578"><a href="#cb35-578"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, locus_tag, COG_category_single, ns_length) <span class="sc">%&gt;%</span> </span>
<span id="cb35-579"><a href="#cb35-579"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-580"><a href="#cb35-580"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-581"><a href="#cb35-581"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> ns_length<span class="sc">/</span><span class="fu">sum</span>(ns_length), </span>
<span id="cb35-582"><a href="#cb35-582"></a>         <span class="at">cog_genes =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb35-583"><a href="#cb35-583"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb35-584"><a href="#cb35-584"></a><span class="in">```</span></span>
<span id="cb35-585"><a href="#cb35-585"></a></span>
<span id="cb35-586"><a href="#cb35-586"></a><span class="fu">### Jensen Shannon Divergence</span></span>
<span id="cb35-587"><a href="#cb35-587"></a></span>
<span id="cb35-588"><a href="#cb35-588"></a>This function draws n genes from each species with a probability proportional to gene length, where n = the number of significant parallel genes observed from the data. We repeat this simulation 10 000 times and below we check if 1) the Jensen Shannon divergence between the distribution of the COG categories drawn randomly and the background (i.e., the distribution of COG categories in the whole genome) is greater than the Jensen Shannon divergence between the background and the observed distribution of COG categories in significantly parallel genes. The idea is that if some COG categories are enriched/depleted in the parallel genes, then the JSD divergence between the observed and the background should be larger than the JSD between the random sampled genes (sampling probability based only on gene length) and the background. We calculate an empirical P value by "counting" how many times the simulated JSD exceeds the observed.</span>
<span id="cb35-589"><a href="#cb35-589"></a></span>
<span id="cb35-592"><a href="#cb35-592"></a><span class="in">```{r}</span></span>
<span id="cb35-593"><a href="#cb35-593"></a><span class="co">#| eval: false</span></span>
<span id="cb35-594"><a href="#cb35-594"></a><span class="co">#| echo: true</span></span>
<span id="cb35-595"><a href="#cb35-595"></a>gene_draw <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb35-596"><a href="#cb35-596"></a>  output_gene_table_filt <span class="sc">%&gt;%</span></span>
<span id="cb35-597"><a href="#cb35-597"></a>    <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span></span>
<span id="cb35-598"><a href="#cb35-598"></a>    <span class="fu">summarize</span>(<span class="at">nsiggenes =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb35-599"><a href="#cb35-599"></a>    <span class="fu">left_join</span>(eggnog_nslen,</span>
<span id="cb35-600"><a href="#cb35-600"></a>              <span class="at">by =</span> <span class="fu">join_by</span>(strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb35-601"><a href="#cb35-601"></a>    <span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span><span class="fu">c</span>(strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb35-602"><a href="#cb35-602"></a>    <span class="fu">mutate</span>(<span class="at">locus_tag =</span> <span class="fu">map</span>(</span>
<span id="cb35-603"><a href="#cb35-603"></a>      data,</span>
<span id="cb35-604"><a href="#cb35-604"></a>      \(x) <span class="fu">sample</span>(</span>
<span id="cb35-605"><a href="#cb35-605"></a>        <span class="at">x =</span> x<span class="sc">$</span>locus_tag,</span>
<span id="cb35-606"><a href="#cb35-606"></a>        <span class="at">size =</span> <span class="fu">unique</span>(x<span class="sc">$</span>nsiggenes),</span>
<span id="cb35-607"><a href="#cb35-607"></a>        <span class="at">prob =</span> x<span class="sc">$</span>p,</span>
<span id="cb35-608"><a href="#cb35-608"></a>        <span class="at">replace =</span> <span class="cn">FALSE</span></span>
<span id="cb35-609"><a href="#cb35-609"></a>      )</span>
<span id="cb35-610"><a href="#cb35-610"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb35-611"><a href="#cb35-611"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>data) <span class="sc">%&gt;%</span></span>
<span id="cb35-612"><a href="#cb35-612"></a>    <span class="fu">unnest</span>(<span class="at">cols =</span> locus_tag) </span>
<span id="cb35-613"><a href="#cb35-613"></a>}</span>
<span id="cb35-614"><a href="#cb35-614"></a></span>
<span id="cb35-615"><a href="#cb35-615"></a><span class="co"># take 10 000 random draws of the genes. Use a seed for reproducibility</span></span>
<span id="cb35-616"><a href="#cb35-616"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">12367</span>,</span>
<span id="cb35-617"><a href="#cb35-617"></a>                 genes_simulated <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10000</span>, <span class="sc">~</span><span class="fu">gene_draw</span>(), <span class="at">.progress=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-618"><a href="#cb35-618"></a>                   <span class="fu">list_rbind</span>(<span class="at">names_to =</span> <span class="st">"id"</span>))</span>
<span id="cb35-619"><a href="#cb35-619"></a></span>
<span id="cb35-620"><a href="#cb35-620"></a><span class="co"># save the output so we don't need to run again</span></span>
<span id="cb35-621"><a href="#cb35-621"></a><span class="fu">write_rds</span>(genes_simulated, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"genes_simulated_nomobile.rds"</span>))</span>
<span id="cb35-622"><a href="#cb35-622"></a><span class="in">```</span></span>
<span id="cb35-623"><a href="#cb35-623"></a></span>
<span id="cb35-624"><a href="#cb35-624"></a>Some data formatting</span>
<span id="cb35-625"><a href="#cb35-625"></a></span>
<span id="cb35-628"><a href="#cb35-628"></a><span class="in">```{r}</span></span>
<span id="cb35-629"><a href="#cb35-629"></a><span class="co"># read data back</span></span>
<span id="cb35-630"><a href="#cb35-630"></a>genes_simulated <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"genes_simulated_nomobile.rds"</span>))</span>
<span id="cb35-631"><a href="#cb35-631"></a></span>
<span id="cb35-632"><a href="#cb35-632"></a><span class="co"># map the locus_tags to COG categories for the simulated</span></span>
<span id="cb35-633"><a href="#cb35-633"></a><span class="co"># gene draws</span></span>
<span id="cb35-634"><a href="#cb35-634"></a>COGs_distribution_simulated <span class="ot">&lt;-</span> genes_simulated <span class="sc">%&gt;%</span></span>
<span id="cb35-635"><a href="#cb35-635"></a>  <span class="fu">left_join</span>(eggnog_nslen, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, locus_tag)) <span class="sc">%&gt;%</span></span>
<span id="cb35-636"><a href="#cb35-636"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb35-637"><a href="#cb35-637"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"nsim"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-638"><a href="#cb35-638"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb35-639"><a href="#cb35-639"></a></span>
<span id="cb35-640"><a href="#cb35-640"></a><span class="co"># get the discrete distribution of COGs in the significant parallel genes</span></span>
<span id="cb35-641"><a href="#cb35-641"></a>COGs_distribution_observed <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb35-642"><a href="#cb35-642"></a>                                        <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-643"><a href="#cb35-643"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"nobs"</span>) </span>
<span id="cb35-644"><a href="#cb35-644"></a></span>
<span id="cb35-645"><a href="#cb35-645"></a><span class="co"># get the discrete distribution of COGs across the whole genome of all species</span></span>
<span id="cb35-646"><a href="#cb35-646"></a><span class="co"># this is the null case (i.e. the background that we would expect if there was no</span></span>
<span id="cb35-647"><a href="#cb35-647"></a><span class="co"># enrichment)</span></span>
<span id="cb35-648"><a href="#cb35-648"></a>COGs_distribution_nullexpectation <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb35-649"><a href="#cb35-649"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"nnull"</span>) </span>
<span id="cb35-650"><a href="#cb35-650"></a><span class="in">```</span></span>
<span id="cb35-651"><a href="#cb35-651"></a></span>
<span id="cb35-652"><a href="#cb35-652"></a>Now we calculate the Jensen Shannon divergence between the background and the simulated data to estimate an empirical P value</span>
<span id="cb35-653"><a href="#cb35-653"></a></span>
<span id="cb35-656"><a href="#cb35-656"></a><span class="in">```{r}</span></span>
<span id="cb35-657"><a href="#cb35-657"></a><span class="co"># function to calculate the Jensen Shannon divergene of a pair of</span></span>
<span id="cb35-658"><a href="#cb35-658"></a><span class="co"># distributions</span></span>
<span id="cb35-659"><a href="#cb35-659"></a>JSD_pair <span class="ot">&lt;-</span> <span class="cf">function</span>(p, q){</span>
<span id="cb35-660"><a href="#cb35-660"></a>  m <span class="ot">&lt;-</span> <span class="fl">0.5</span> <span class="sc">*</span> (p <span class="sc">+</span> q)</span>
<span id="cb35-661"><a href="#cb35-661"></a>  JS <span class="ot">&lt;-</span>  <span class="fl">0.5</span> <span class="sc">*</span> (<span class="fu">sum</span>(p <span class="sc">*</span> <span class="fu">log</span>(p<span class="sc">/</span>m)) <span class="sc">+</span> <span class="fu">sum</span>(q <span class="sc">*</span> <span class="fu">log</span>(q<span class="sc">/</span>m)))</span>
<span id="cb35-662"><a href="#cb35-662"></a>  <span class="fu">return</span>(JS)</span>
<span id="cb35-663"><a href="#cb35-663"></a>}</span>
<span id="cb35-664"><a href="#cb35-664"></a></span>
<span id="cb35-665"><a href="#cb35-665"></a><span class="co"># Calculate the Jensen Shannon divergence between the COG distribution observed in </span></span>
<span id="cb35-666"><a href="#cb35-666"></a><span class="co"># enriched genes and the COG distribution in the genomic background</span></span>
<span id="cb35-667"><a href="#cb35-667"></a>JSD_observed <span class="ot">&lt;-</span> <span class="fu">left_join</span>(COGs_distribution_observed, COGs_distribution_nullexpectation,</span>
<span id="cb35-668"><a href="#cb35-668"></a>                          <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-669"><a href="#cb35-669"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> nobs<span class="sc">/</span><span class="fu">sum</span>(nobs), </span>
<span id="cb35-670"><a href="#cb35-670"></a>         <span class="at">q =</span> nnull<span class="sc">/</span><span class="fu">sum</span>(nnull)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-671"><a href="#cb35-671"></a>  <span class="fu">summarize</span>(<span class="at">JSD =</span> <span class="fu">JSD_pair</span>(p, q)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-672"><a href="#cb35-672"></a>  <span class="fu">pull</span>()</span>
<span id="cb35-673"><a href="#cb35-673"></a></span>
<span id="cb35-674"><a href="#cb35-674"></a><span class="co"># now calculate JSD between the simulated draws and the backgound distribution</span></span>
<span id="cb35-675"><a href="#cb35-675"></a>JSD_simulated <span class="ot">&lt;-</span> <span class="fu">left_join</span>(COGs_distribution_simulated, </span>
<span id="cb35-676"><a href="#cb35-676"></a>                           COGs_distribution_nullexpectation,</span>
<span id="cb35-677"><a href="#cb35-677"></a>                           <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-678"><a href="#cb35-678"></a>  <span class="co"># drops from calculation categories that aren't in both the observed data</span></span>
<span id="cb35-679"><a href="#cb35-679"></a>  <span class="co"># and the resampled data. Need to do this because JSD can't handle zero values</span></span>
<span id="cb35-680"><a href="#cb35-680"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-681"><a href="#cb35-681"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb35-682"><a href="#cb35-682"></a>  <span class="fu">mutate</span>(<span class="at">p =</span> nsim<span class="sc">/</span><span class="fu">sum</span>(nsim), </span>
<span id="cb35-683"><a href="#cb35-683"></a>         <span class="at">q =</span> nnull<span class="sc">/</span><span class="fu">sum</span>(nnull)) <span class="sc">%&gt;%</span></span>
<span id="cb35-684"><a href="#cb35-684"></a>  <span class="fu">summarize</span>(<span class="at">JSD =</span> <span class="fu">JSD_pair</span>(p, q)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-685"><a href="#cb35-685"></a>  <span class="fu">pull</span>(JSD)</span>
<span id="cb35-686"><a href="#cb35-686"></a></span>
<span id="cb35-687"><a href="#cb35-687"></a><span class="co"># proportion of JSD measures between simulated and the background that are greater</span></span>
<span id="cb35-688"><a href="#cb35-688"></a><span class="co"># than the JSD between the observed and the background</span></span>
<span id="cb35-689"><a href="#cb35-689"></a>global_p <span class="ot">&lt;-</span> <span class="fu">length</span>(JSD_simulated[JSD_simulated <span class="sc">&gt;=</span> JSD_observed])<span class="sc">/</span><span class="fu">length</span>(JSD_simulated)</span>
<span id="cb35-690"><a href="#cb35-690"></a></span>
<span id="cb35-691"><a href="#cb35-691"></a>global_p</span>
<span id="cb35-692"><a href="#cb35-692"></a><span class="in">```</span></span>
<span id="cb35-693"><a href="#cb35-693"></a></span>
<span id="cb35-694"><a href="#cb35-694"></a><span class="fu">### Enrichment of individual COG categories</span></span>
<span id="cb35-695"><a href="#cb35-695"></a></span>
<span id="cb35-696"><a href="#cb35-696"></a>OK so the overall distribution of COG categories in the parallel genes is not significantly different than in the rest of the genome - ~40% of the time the JSD between the COG distribution in a random sample and the background is greater than the observed COG distribution and the background.</span>
<span id="cb35-697"><a href="#cb35-697"></a></span>
<span id="cb35-698"><a href="#cb35-698"></a><span class="fu">#### Hypergeometric Test</span></span>
<span id="cb35-699"><a href="#cb35-699"></a></span>
<span id="cb35-700"><a href="#cb35-700"></a>We can test enrichment in individual COG categories in individual species using the hypergeometric test</span>
<span id="cb35-703"><a href="#cb35-703"></a><span class="in">```{r}</span></span>
<span id="cb35-704"><a href="#cb35-704"></a><span class="fu">library</span>(qvalue)</span>
<span id="cb35-705"><a href="#cb35-705"></a></span>
<span id="cb35-706"><a href="#cb35-706"></a>par2test <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb35-707"><a href="#cb35-707"></a>                      <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span></span>
<span id="cb35-708"><a href="#cb35-708"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-709"><a href="#cb35-709"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-710"><a href="#cb35-710"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"n_cog_par"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-711"><a href="#cb35-711"></a>  <span class="fu">mutate</span>(<span class="at">n_par =</span> <span class="fu">sum</span>(n_cog_par))</span>
<span id="cb35-712"><a href="#cb35-712"></a></span>
<span id="cb35-713"><a href="#cb35-713"></a>back2test <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb35-714"><a href="#cb35-714"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-715"><a href="#cb35-715"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-716"><a href="#cb35-716"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"n_cog_background"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-717"><a href="#cb35-717"></a>  <span class="fu">mutate</span>(<span class="at">n_background =</span> <span class="fu">sum</span>(n_cog_background))</span>
<span id="cb35-718"><a href="#cb35-718"></a></span>
<span id="cb35-719"><a href="#cb35-719"></a>phypres <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par2test, back2test, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-720"><a href="#cb35-720"></a>  <span class="fu">mutate</span>(<span class="at">p_enrich =</span> <span class="fu">enricher</span>(n_background, n_par, n_cog_background, n_cog_par, <span class="at">over=</span><span class="cn">TRUE</span>),</span>
<span id="cb35-721"><a href="#cb35-721"></a>         <span class="at">p_deplete =</span> <span class="fu">enricher</span>(n_background, n_par, n_cog_background, n_cog_par, <span class="at">over=</span><span class="cn">FALSE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-722"><a href="#cb35-722"></a>  <span class="fu">mutate</span>(<span class="at">p_enrich_adj =</span> <span class="fu">p.adjust</span>(p_enrich, <span class="at">method =</span> <span class="st">"fdr"</span>),</span>
<span id="cb35-723"><a href="#cb35-723"></a>         <span class="at">p_deplete_adj =</span> <span class="fu">p.adjust</span>(p_deplete, <span class="at">method =</span> <span class="st">"fdr"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-724"><a href="#cb35-724"></a>  <span class="fu">arrange</span>(p_enrich_adj)</span>
<span id="cb35-725"><a href="#cb35-725"></a></span>
<span id="cb35-726"><a href="#cb35-726"></a>qenrich <span class="ot">&lt;-</span> <span class="fu">qvalue_truncp</span>(phypres<span class="sc">$</span>p_enrich)</span>
<span id="cb35-727"><a href="#cb35-727"></a>qdeplete <span class="ot">&lt;-</span> <span class="fu">qvalue_truncp</span>(phypres<span class="sc">$</span>p_deplete)</span>
<span id="cb35-728"><a href="#cb35-728"></a></span>
<span id="cb35-729"><a href="#cb35-729"></a>phypres<span class="sc">$</span>q_enrich <span class="ot">&lt;-</span> qenrich<span class="sc">$</span>qvalues</span>
<span id="cb35-730"><a href="#cb35-730"></a>phypres<span class="sc">$</span>q_deplete <span class="ot">&lt;-</span> qdeplete<span class="sc">$</span>qvalues</span>
<span id="cb35-731"><a href="#cb35-731"></a></span>
<span id="cb35-732"><a href="#cb35-732"></a>phypres <span class="sc">%&gt;%</span> </span>
<span id="cb35-733"><a href="#cb35-733"></a>  <span class="fu">relocate</span>(p_enrich, q_enrich, p_enrich_adj) <span class="sc">%&gt;%</span> </span>
<span id="cb35-734"><a href="#cb35-734"></a>  <span class="fu">arrange</span>(p_enrich)</span>
<span id="cb35-735"><a href="#cb35-735"></a><span class="in">```</span></span>
<span id="cb35-736"><a href="#cb35-736"></a></span>
<span id="cb35-737"><a href="#cb35-737"></a>Here COG categories "D - Cell cycle control, cell division, chromosome partitioning", "M - Cell wall/membrane/envelope biogenesis", and "P - Inorganic ion transport and metabolism" are enriched with $p_{adj} \leq 0.05$.</span>
<span id="cb35-738"><a href="#cb35-738"></a>                   </span>
<span id="cb35-739"><a href="#cb35-739"></a><span class="in">            </span></span>
<span id="cb35-740"><a href="#cb35-740"></a><span class="fu">#### Nonparametric simulation</span></span>
<span id="cb35-741"><a href="#cb35-741"></a></span>
<span id="cb35-742"><a href="#cb35-742"></a>We can also look at different COG categories by simply counting the number of random samplings that have a COG count greater/less than that the observed COG count. Since we resampled the same number of genes as in the observed we can compare the two directly. Below we do this integrated over all species in the community to see if there was any pathway enriched across all species.</span>
<span id="cb35-743"><a href="#cb35-743"></a></span>
<span id="cb35-746"><a href="#cb35-746"></a><span class="in">```{r}</span></span>
<span id="cb35-747"><a href="#cb35-747"></a><span class="fu">left_join</span>(COGs_distribution_simulated, COGs_distribution_observed,</span>
<span id="cb35-748"><a href="#cb35-748"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-749"><a href="#cb35-749"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-750"><a href="#cb35-750"></a>  <span class="fu">mutate</span>(<span class="at">gt =</span> <span class="fu">if_else</span>(nsim <span class="sc">&gt;=</span> nobs, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-751"><a href="#cb35-751"></a>  <span class="fu">group_by</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb35-752"><a href="#cb35-752"></a>  <span class="fu">summarize</span>(<span class="at">p =</span> <span class="fu">sum</span>(gt)<span class="sc">/</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb35-753"><a href="#cb35-753"></a>  <span class="fu">arrange</span>(p) <span class="sc">%&gt;%</span> </span>
<span id="cb35-754"><a href="#cb35-754"></a>  <span class="fu">mutate</span>(<span class="at">p_adj =</span> <span class="fu">p.adjust</span>(p, <span class="at">method =</span> <span class="st">"fdr"</span>))</span>
<span id="cb35-755"><a href="#cb35-755"></a><span class="in">```</span></span>
<span id="cb35-756"><a href="#cb35-756"></a></span>
<span id="cb35-757"><a href="#cb35-757"></a>Only about 1% of the re-samples had a higher number of K COG genes compared to the observed number of K COG genes. No other COG category meets the standard alpha = 0.05 threshold. However, after correcting for multiple testing the 5% alpha threshold is no longer met.</span>
<span id="cb35-758"><a href="#cb35-758"></a></span>
<span id="cb35-759"><a href="#cb35-759"></a>This is just a sanity check that the randomly sampled COGs reflect the background distribution - i.e., a large fraction of the resamples are larger (and smaller) than the null. There is no systematic difference, and we observe the different COG categories in the resampels at basically the same rate as we observe the COG categories in the background.</span>
<span id="cb35-760"><a href="#cb35-760"></a></span>
<span id="cb35-763"><a href="#cb35-763"></a><span class="in">```{r}</span></span>
<span id="cb35-764"><a href="#cb35-764"></a><span class="fu">left_join</span>(COGs_distribution_simulated, COGs_distribution_nullexpectation,</span>
<span id="cb35-765"><a href="#cb35-765"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(COG_category_single)) <span class="sc">%&gt;%</span></span>
<span id="cb35-766"><a href="#cb35-766"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb35-767"><a href="#cb35-767"></a>  <span class="fu">mutate</span>(<span class="at">fsim =</span> nsim<span class="sc">/</span><span class="fu">sum</span>(nsim), <span class="at">fnull =</span> nnull<span class="sc">/</span><span class="fu">sum</span>(nnull)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-768"><a href="#cb35-768"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-769"><a href="#cb35-769"></a>  <span class="fu">mutate</span>(<span class="at">gt =</span> <span class="fu">if_else</span>(fsim <span class="sc">&gt;=</span> fnull, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-770"><a href="#cb35-770"></a>  <span class="fu">group_by</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb35-771"><a href="#cb35-771"></a>  <span class="fu">summarize</span>(<span class="at">p =</span> <span class="fu">sum</span>(gt)<span class="sc">/</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb35-772"><a href="#cb35-772"></a>  <span class="fu">arrange</span>(p)</span>
<span id="cb35-773"><a href="#cb35-773"></a><span class="in">```</span></span>
<span id="cb35-774"><a href="#cb35-774"></a></span>
<span id="cb35-775"><a href="#cb35-775"></a><span class="fu">### COG Distribution Plot</span></span>
<span id="cb35-776"><a href="#cb35-776"></a></span>
<span id="cb35-779"><a href="#cb35-779"></a><span class="in">```{r}</span></span>
<span id="cb35-780"><a href="#cb35-780"></a>cog_pmut <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb35-781"><a href="#cb35-781"></a>                      <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-782"><a href="#cb35-782"></a>  <span class="fu">count</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb35-783"><a href="#cb35-783"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n),</span>
<span id="cb35-784"><a href="#cb35-784"></a>         <span class="at">n =</span> n<span class="sc">*</span><span class="dv">1000</span>,</span>
<span id="cb35-785"><a href="#cb35-785"></a>         <span class="at">type =</span> <span class="st">"parallel_mutations"</span>) </span>
<span id="cb35-786"><a href="#cb35-786"></a></span>
<span id="cb35-787"><a href="#cb35-787"></a>cog_everything <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb35-788"><a href="#cb35-788"></a>  <span class="fu">count</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb35-789"><a href="#cb35-789"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n), </span>
<span id="cb35-790"><a href="#cb35-790"></a>         <span class="at">type =</span> <span class="st">"all_genes"</span>) </span>
<span id="cb35-791"><a href="#cb35-791"></a></span>
<span id="cb35-792"><a href="#cb35-792"></a>pcogs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cog_pmut, cog_everything) <span class="sc">%&gt;%</span> </span>
<span id="cb35-793"><a href="#cb35-793"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb35-794"><a href="#cb35-794"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb35-795"><a href="#cb35-795"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(COG_category_single, f, <span class="at">.desc =</span> F), <span class="at">y=</span>n, <span class="at">fill =</span> type), <span class="at">stat=</span><span class="st">"identity"</span>,</span>
<span id="cb35-796"><a href="#cb35-796"></a>           <span class="at">position =</span> <span class="fu">position_dodge</span>( <span class="at">preserve =</span> <span class="st">"total"</span>)) <span class="sc">+</span></span>
<span id="cb35-797"><a href="#cb35-797"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb35-798"><a href="#cb35-798"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"COG Category"</span>, <span class="at">y =</span> <span class="st">"Fraction genes in COG category"</span>, <span class="at">fill =</span> <span class="st">"Subset"</span>) <span class="sc">+</span></span>
<span id="cb35-799"><a href="#cb35-799"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb35-800"><a href="#cb35-800"></a>    <span class="st">"N all genes"</span>, </span>
<span id="cb35-801"><a href="#cb35-801"></a>    <span class="at">sec.axis =</span> <span class="fu">sec_axis</span>(<span class="sc">~</span> . <span class="sc">*</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">1000</span>, <span class="at">name =</span> <span class="st">"N parallel genes"</span>)</span>
<span id="cb35-802"><a href="#cb35-802"></a>  ) <span class="sc">+</span></span>
<span id="cb35-803"><a href="#cb35-803"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb35-804"><a href="#cb35-804"></a>  <span class="fu">theme</span>(</span>
<span id="cb35-805"><a href="#cb35-805"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-806"><a href="#cb35-806"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-807"><a href="#cb35-807"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb35-808"><a href="#cb35-808"></a>  )</span>
<span id="cb35-809"><a href="#cb35-809"></a><span class="in">```</span></span>
<span id="cb35-810"><a href="#cb35-810"></a></span>
<span id="cb35-811"><a href="#cb35-811"></a>::: {#fig-05}</span>
<span id="cb35-814"><a href="#cb35-814"></a><span class="in">```{r}</span></span>
<span id="cb35-815"><a href="#cb35-815"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb35-816"><a href="#cb35-816"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb35-817"><a href="#cb35-817"></a><span class="co">#| echo: false</span></span>
<span id="cb35-818"><a href="#cb35-818"></a>pcogs</span>
<span id="cb35-819"><a href="#cb35-819"></a><span class="in">```</span></span>
<span id="cb35-820"><a href="#cb35-820"></a></span>
<span id="cb35-821"><a href="#cb35-821"></a>Number of genes in each Cluster of Orthologous Groups (COG) category for genes with a COG annotation across all 23 HAMBI population genomes (red bars, left y-axis) and across only genes exhibiting significant genomic parallelism (blue bars, right y-axis).</span>
<span id="cb35-822"><a href="#cb35-822"></a>:::</span>
<span id="cb35-823"><a href="#cb35-823"></a></span>
<span id="cb35-824"><a href="#cb35-824"></a><span class="fu">## Species present in the metagenomes</span></span>
<span id="cb35-825"><a href="#cb35-825"></a></span>
<span id="cb35-826"><a href="#cb35-826"></a>Now that I've done the analysis on the metagenomes we know that we can only detect HAMBI_2659, HAMBI_1972, HAMBI_1977, and HAMBI_1287 with high enough coverate to detect sequence variants. From the hypergeometric test above we know that COG categories D, M, and P are enriched in HAMBI_1972, HAMBI_1977, amd HAMBI_2659 respsectively. Here we will make a plot focusing only on those species</span>
<span id="cb35-827"><a href="#cb35-827"></a></span>
<span id="cb35-830"><a href="#cb35-830"></a><span class="in">```{r}</span></span>
<span id="cb35-831"><a href="#cb35-831"></a><span class="co"># species detected in the metagenomes</span></span>
<span id="cb35-832"><a href="#cb35-832"></a>mgsp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"HAMBI_2659"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_2443"</span>)</span>
<span id="cb35-833"><a href="#cb35-833"></a></span>
<span id="cb35-834"><a href="#cb35-834"></a>cog_pmut_mgsp <span class="ot">&lt;-</span> <span class="fu">left_join</span>(output_gene_table_filt, eggnog,</span>
<span id="cb35-835"><a href="#cb35-835"></a>                      <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-836"><a href="#cb35-836"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> mgsp) <span class="sc">%&gt;%</span> </span>
<span id="cb35-837"><a href="#cb35-837"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-838"><a href="#cb35-838"></a>  <span class="fu">count</span>(COG_category_single, <span class="at">name =</span> <span class="st">"tot"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-839"><a href="#cb35-839"></a>  <span class="co"># scaling factor to make ploting on 2 axes possible</span></span>
<span id="cb35-840"><a href="#cb35-840"></a>  <span class="fu">mutate</span>(<span class="at">n =</span> tot<span class="sc">*</span><span class="dv">250</span>,</span>
<span id="cb35-841"><a href="#cb35-841"></a>         <span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n),</span>
<span id="cb35-842"><a href="#cb35-842"></a>         <span class="at">type =</span> <span class="st">"parallel_mutations"</span>) </span>
<span id="cb35-843"><a href="#cb35-843"></a></span>
<span id="cb35-844"><a href="#cb35-844"></a>cog_everything_mgsp <span class="ot">&lt;-</span> eggnog <span class="sc">%&gt;%</span> </span>
<span id="cb35-845"><a href="#cb35-845"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> mgsp) <span class="sc">%&gt;%</span> </span>
<span id="cb35-846"><a href="#cb35-846"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb35-847"><a href="#cb35-847"></a>  <span class="fu">count</span>(COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb35-848"><a href="#cb35-848"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> n<span class="sc">/</span><span class="fu">sum</span>(n),</span>
<span id="cb35-849"><a href="#cb35-849"></a>         <span class="at">type =</span> <span class="st">"all_genes"</span>) </span>
<span id="cb35-850"><a href="#cb35-850"></a></span>
<span id="cb35-851"><a href="#cb35-851"></a>pcogs_mgsp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cog_pmut_mgsp, cog_everything_mgsp) <span class="sc">%&gt;%</span> </span>
<span id="cb35-852"><a href="#cb35-852"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(COG_category_single)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-853"><a href="#cb35-853"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb35-854"><a href="#cb35-854"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">fct_reorder</span>(COG_category_single, n, <span class="at">.desc =</span> F), <span class="at">y=</span>n, <span class="at">fill =</span> type), <span class="at">stat=</span><span class="st">"identity"</span>,</span>
<span id="cb35-855"><a href="#cb35-855"></a>           <span class="at">position =</span> <span class="fu">position_dodge</span>( <span class="at">preserve =</span> <span class="st">"total"</span>)) <span class="sc">+</span></span>
<span id="cb35-856"><a href="#cb35-856"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb35-857"><a href="#cb35-857"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">tibble</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="st">"D"</span>, <span class="st">"M"</span>, <span class="st">"M"</span>, <span class="st">"P"</span>), </span>
<span id="cb35-858"><a href="#cb35-858"></a>                           <span class="at">y =</span> <span class="fu">c</span>(<span class="dv">700</span>, <span class="dv">700</span>, <span class="dv">700</span>, <span class="dv">700</span>), </span>
<span id="cb35-859"><a href="#cb35-859"></a>                           <span class="at">strainID =</span> <span class="fu">c</span>(<span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_2443"</span>, <span class="st">"HAMBI_2659"</span>)),</span>
<span id="cb35-860"><a href="#cb35-860"></a>             <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y), <span class="at">shape =</span> <span class="dv">8</span>) <span class="sc">+</span></span>
<span id="cb35-861"><a href="#cb35-861"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> strainID, <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb35-862"><a href="#cb35-862"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"COG Category"</span>, <span class="at">y =</span> <span class="st">"Fraction genes in COG category"</span>, <span class="at">fill =</span> <span class="st">"Subset"</span>) <span class="sc">+</span></span>
<span id="cb35-863"><a href="#cb35-863"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb35-864"><a href="#cb35-864"></a>    <span class="st">"N all genes"</span>, </span>
<span id="cb35-865"><a href="#cb35-865"></a>    <span class="at">sec.axis =</span> <span class="fu">sec_axis</span>(<span class="sc">~</span> . <span class="sc">*</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">250</span>, <span class="at">name =</span> <span class="st">"N parallel genes"</span>)</span>
<span id="cb35-866"><a href="#cb35-866"></a>  ) <span class="sc">+</span></span>
<span id="cb35-867"><a href="#cb35-867"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb35-868"><a href="#cb35-868"></a>  <span class="fu">theme</span>(</span>
<span id="cb35-869"><a href="#cb35-869"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-870"><a href="#cb35-870"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-871"><a href="#cb35-871"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-872"><a href="#cb35-872"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb35-873"><a href="#cb35-873"></a>  )</span>
<span id="cb35-874"><a href="#cb35-874"></a><span class="in">```</span></span>
<span id="cb35-875"><a href="#cb35-875"></a></span>
<span id="cb35-876"><a href="#cb35-876"></a>::: {#fig-06}</span>
<span id="cb35-879"><a href="#cb35-879"></a><span class="in">```{r}</span></span>
<span id="cb35-880"><a href="#cb35-880"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb35-881"><a href="#cb35-881"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb35-882"><a href="#cb35-882"></a><span class="co">#| echo: false</span></span>
<span id="cb35-883"><a href="#cb35-883"></a>pcogs_mgsp</span>
<span id="cb35-884"><a href="#cb35-884"></a><span class="in">```</span></span>
<span id="cb35-885"><a href="#cb35-885"></a></span>
<span id="cb35-886"><a href="#cb35-886"></a>Fraction of genes in each Cluster of Orthologous Groups (COG) category for genes with a COG annotation (red bars, left y-axis) and only genes exhibiting significant genomic parallelism (blue bars, right y-axis) for the four most abundant HAMBI genomes. COG categories "D - Cell cycle control, cell division, chromosome partitioning" for HAMBI_1972, "M - Cell wall/membrane/envelope biogenesis" for HAMBI_1977, and "P - Inorganic ion transport and metabolism" for HAMBI_2659 are significantly enriched (denoted with *) in the parallel mutations compared to the rest of the genome for these species.</span>
<span id="cb35-887"><a href="#cb35-887"></a>:::</span>
<span id="cb35-888"><a href="#cb35-888"></a></span>
<span id="cb35-889"><a href="#cb35-889"></a></span>
<span id="cb35-892"><a href="#cb35-892"></a><span class="in">```{r}</span></span>
<span id="cb35-893"><a href="#cb35-893"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"COG_enrich_wgs.svg"</span>), pcogs_mgsp, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">5</span>, </span>
<span id="cb35-894"><a href="#cb35-894"></a>       <span class="at">units=</span><span class="st">"in"</span>, <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb35-895"><a href="#cb35-895"></a><span class="in">```</span></span>
<span id="cb35-896"><a href="#cb35-896"></a></span>
<span id="cb35-897"><a href="#cb35-897"></a>| strainID   | Abundant in metagenomes | COG category                                                      | q value |</span>
<span id="cb35-898"><a href="#cb35-898"></a>|------------|-------------------------|-------------------------------------------------------------------|---------|</span>
<span id="cb35-899"><a href="#cb35-899"></a>| HAMBI_1972 | yes                     | D - Cell cycle control, cell division, chromosome partitioning    | 0.006   |</span>
<span id="cb35-900"><a href="#cb35-900"></a>| HAMBI_2443 | no                      | M - Cell wall/membrane/envelope biogenesis                        | 0.013   |</span>
<span id="cb35-901"><a href="#cb35-901"></a>| HAMBI_1977 | yes                     | M - Cell wall/membrane/envelope biogenesis                        | 0.015   |</span>
<span id="cb35-902"><a href="#cb35-902"></a>| HAMBI_2659 | yes                     | P - Inorganic ion transport and metabolism                        | 0.031   |</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>