<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-10-30">

<title>hambiDoubleEvo - Analysis of community composition</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/illumina_v3v4/01_rpkm2tab.html">Community composition amplicon</a></li><li class="breadcrumb-item"><a href="../../R/illumina_v3v4/02_analysis.html">2. Analyze and plot</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">hambiDoubleEvo</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/01_rpkm2tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/02_analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. Analyze and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Population WGS</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Metagenome sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Variant plotting</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/04_mutation_dynamics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Mutational dynamics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a>
  <ul class="collapse">
  <li><a href="#some-functions" id="toc-some-functions" class="nav-link" data-scroll-target="#some-functions"><span class="header-section-number">1.1</span> Some functions</a></li>
  </ul></li>
  <li><a href="#plot-community-composition" id="toc-plot-community-composition" class="nav-link" data-scroll-target="#plot-community-composition"><span class="header-section-number">2</span> Plot Community Composition</a></li>
  <li><a href="#ordination-of-community-composition" id="toc-ordination-of-community-composition" class="nav-link" data-scroll-target="#ordination-of-community-composition"><span class="header-section-number">3</span> Ordination of community composition</a>
  <ul class="collapse">
  <li><a href="#transform-data" id="toc-transform-data" class="nav-link" data-scroll-target="#transform-data"><span class="header-section-number">3.1</span> Transform data</a></li>
  <li><a href="#replace-zeros" id="toc-replace-zeros" class="nav-link" data-scroll-target="#replace-zeros"><span class="header-section-number">3.2</span> Replace zeros</a></li>
  <li><a href="#calculate-bray-curtis-dissimilarity" id="toc-calculate-bray-curtis-dissimilarity" class="nav-link" data-scroll-target="#calculate-bray-curtis-dissimilarity"><span class="header-section-number">3.3</span> Calculate Bray-curtis dissimilarity</a></li>
  <li><a href="#calculate-with-aitchison-distance" id="toc-calculate-with-aitchison-distance" class="nav-link" data-scroll-target="#calculate-with-aitchison-distance"><span class="header-section-number">3.4</span> Calculate with Aitchison distance</a></li>
  <li><a href="#compare-aitchison-distance-with-clr" id="toc-compare-aitchison-distance-with-clr" class="nav-link" data-scroll-target="#compare-aitchison-distance-with-clr"><span class="header-section-number">3.5</span> Compare Aitchison distance with CLR</a></li>
  <li><a href="#environment-vectors" id="toc-environment-vectors" class="nav-link" data-scroll-target="#environment-vectors"><span class="header-section-number">3.6</span> Environment vectors</a>
  <ul class="collapse">
  <li><a href="#significance-of-the-environmental-covariates" id="toc-significance-of-the-environmental-covariates" class="nav-link" data-scroll-target="#significance-of-the-environmental-covariates"><span class="header-section-number">3.6.1</span> Significance of the environmental covariates</a></li>
  <li><a href="#plot-with-prey-history-highlighted" id="toc-plot-with-prey-history-highlighted" class="nav-link" data-scroll-target="#plot-with-prey-history-highlighted"><span class="header-section-number">3.6.2</span> Plot with prey history highlighted</a></li>
  <li><a href="#plot-with-predator-history-highlighted" id="toc-plot-with-predator-history-highlighted" class="nav-link" data-scroll-target="#plot-with-predator-history-highlighted"><span class="header-section-number">3.6.3</span> Plot with predator history highlighted</a></li>
  <li><a href="#permanova" id="toc-permanova" class="nav-link" data-scroll-target="#permanova"><span class="header-section-number">3.6.4</span> PERMANOVA</a></li>
  </ul></li>
  <li><a href="#geometric-analysis-of-temporal-trajectories-in-pca" id="toc-geometric-analysis-of-temporal-trajectories-in-pca" class="nav-link" data-scroll-target="#geometric-analysis-of-temporal-trajectories-in-pca"><span class="header-section-number">3.7</span> Geometric analysis of temporal trajectories in PCA</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/illumina_v3v4/01_rpkm2tab.html">Community composition amplicon</a></li><li class="breadcrumb-item"><a href="../../R/illumina_v3v4/02_analysis.html">2. Analyze and plot</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Analysis of community composition</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 30, 2024</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook takes formatted 16S rRNA amplicon counts and plots and analyzes them in different ways to look at community composition
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Libraries and global variables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(fishualize)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(withr)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-9"><a href="#cb1-9"></a></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Set up some directories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"illumina_v3v4"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-4"><a href="#cb2-4"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"illumina_v3v4"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"illumina_v3v4"</span>)</span>
<span id="cb2-6"><a href="#cb2-6"></a></span>
<span id="cb2-7"><a href="#cb2-7"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-8"><a href="#cb2-8"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="some-functions" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="some-functions"><span class="header-section-number">1.1</span> Some functions</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>myarplot <span class="ot">&lt;-</span> <span class="cf">function</span>(.data){</span>
<span id="cb3-2"><a href="#cb3-2"></a>  cil <span class="ot">&lt;-</span> <span class="fu">unique</span>(.data<span class="sc">$</span>predator_history)</span>
<span id="cb3-3"><a href="#cb3-3"></a>  bac <span class="ot">&lt;-</span> <span class="fu">unique</span>(.data<span class="sc">$</span>prey_history)</span>
<span id="cb3-4"><a href="#cb3-4"></a>  </span>
<span id="cb3-5"><a href="#cb3-5"></a>  mytitle <span class="ot">&lt;-</span> <span class="fu">case_when</span>(cil <span class="sc">==</span> <span class="st">"anc"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"anc bac + anc cil"</span>,</span>
<span id="cb3-6"><a href="#cb3-6"></a>                       cil <span class="sc">==</span> <span class="st">"anc"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"evo"</span> <span class="sc">~</span> <span class="st">"evo bac + anc cil"</span>,</span>
<span id="cb3-7"><a href="#cb3-7"></a>                       cil <span class="sc">==</span> <span class="st">"evo"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"anc bac + evo cil"</span>,</span>
<span id="cb3-8"><a href="#cb3-8"></a>                       cil <span class="sc">==</span> <span class="st">"evo"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"evo"</span> <span class="sc">~</span> <span class="st">"evo bac + evo cil"</span>,</span>
<span id="cb3-9"><a href="#cb3-9"></a>                       cil <span class="sc">==</span> <span class="st">"nopredator"</span> <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"anc bac"</span>,</span>
<span id="cb3-10"><a href="#cb3-10"></a>                       cil <span class="sc">==</span> <span class="st">"nopredator"</span> <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"evo"</span> <span class="sc">~</span> <span class="st">"evo bac"</span>,</span>
<span id="cb3-11"><a href="#cb3-11"></a>                       <span class="co">#cil == "nopredator" &amp; bac == "evo" ~ "starting anc bac",</span></span>
<span id="cb3-12"><a href="#cb3-12"></a>                       <span class="co">#cil == "nopredator" &amp; bac == "evo" ~ "starting evo bac"</span></span>
<span id="cb3-13"><a href="#cb3-13"></a>  )</span>
<span id="cb3-14"><a href="#cb3-14"></a>  </span>
<span id="cb3-15"><a href="#cb3-15"></a>  <span class="fu">ggplot</span>(.data) <span class="sc">+</span></span>
<span id="cb3-16"><a href="#cb3-16"></a>    <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>f, <span class="at">fill=</span>strainID),</span>
<span id="cb3-17"><a href="#cb3-17"></a>              <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb3-18"><a href="#cb3-18"></a>    <span class="fu">facet_wrap</span>( <span class="sc">~</span> replicate, <span class="at">strip.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb3-19"><a href="#cb3-19"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb3-20"><a href="#cb3-20"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb3-21"><a href="#cb3-21"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">60</span>), <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">40</span>, <span class="dv">60</span>)) <span class="sc">+</span></span>
<span id="cb3-22"><a href="#cb3-22"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>, <span class="at">fill=</span><span class="st">""</span>, <span class="at">title=</span>mytitle) <span class="sc">+</span> </span>
<span id="cb3-23"><a href="#cb3-23"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb3-24"><a href="#cb3-24"></a>    <span class="fu">myartheme</span>()</span>
<span id="cb3-25"><a href="#cb3-25"></a>  </span>
<span id="cb3-26"><a href="#cb3-26"></a>}</span>
<span id="cb3-27"><a href="#cb3-27"></a></span>
<span id="cb3-28"><a href="#cb3-28"></a>myartheme <span class="ot">&lt;-</span> <span class="cf">function</span>(...){</span>
<span id="cb3-29"><a href="#cb3-29"></a>  <span class="fu">theme</span>(</span>
<span id="cb3-30"><a href="#cb3-30"></a>    <span class="at">panel.spacing.x =</span> <span class="fu">unit</span>(<span class="fl">0.05</span>,<span class="st">"line"</span>),</span>
<span id="cb3-31"><a href="#cb3-31"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb3-32"><a href="#cb3-32"></a>    <span class="at">strip.background.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-33"><a href="#cb3-33"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-34"><a href="#cb3-34"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-35"><a href="#cb3-35"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-36"><a href="#cb3-36"></a>    <span class="co">#axis.text.x = element_blank(),</span></span>
<span id="cb3-37"><a href="#cb3-37"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb3-38"><a href="#cb3-38"></a>    <span class="at">axis.line.y =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb3-39"><a href="#cb3-39"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-40"><a href="#cb3-40"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-41"><a href="#cb3-41"></a>    <span class="at">legend.key =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-42"><a href="#cb3-42"></a>    ...)</span>
<span id="cb3-43"><a href="#cb3-43"></a>}</span>
<span id="cb3-44"><a href="#cb3-44"></a></span>
<span id="cb3-45"><a href="#cb3-45"></a>strain.order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"HAMBI_2443"</span>, <span class="st">"HAMBI_3031"</span>, <span class="st">"HAMBI_1988"</span>, <span class="st">"HAMBI_3237"</span>, <span class="st">"HAMBI_0262"</span>, </span>
<span id="cb3-46"><a href="#cb3-46"></a>                  <span class="st">"HAMBI_2159"</span>, <span class="st">"HAMBI_1842"</span>, <span class="st">"HAMBI_2164"</span>, <span class="st">"HAMBI_0006"</span>, <span class="st">"HAMBI_0097"</span>, </span>
<span id="cb3-47"><a href="#cb3-47"></a>                  <span class="st">"HAMBI_2494"</span>, <span class="st">"HAMBI_1299"</span>, <span class="st">"HAMBI_2160"</span>, <span class="st">"HAMBI_1279"</span>, <span class="st">"HAMBI_2792"</span>, </span>
<span id="cb3-48"><a href="#cb3-48"></a>                  <span class="st">"HAMBI_1896"</span>, <span class="st">"HAMBI_0105"</span>, <span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_1923"</span>, </span>
<span id="cb3-49"><a href="#cb3-49"></a>                  <span class="st">"HAMBI_2659"</span>, <span class="st">"HAMBI_1292"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1287"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Read data</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"species_counts.tsv"</span>))</span>
<span id="cb4-2"><a href="#cb4-2"></a>metadf <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"shared"</span>, <span class="st">"metadata_formatted.rds"</span>))</span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a>counts_f <span class="ot">&lt;-</span> <span class="fu">left_join</span>(metadf, counts) <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="fu">mutate</span>(<span class="at">f=</span>count<span class="sc">/</span><span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span>  </span>
<span id="cb4-7"><a href="#cb4-7"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(strainID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="plot-community-composition" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Plot Community Composition</h1>
<p>community composition of the initial conditions</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>pinit <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2"></a>  <span class="fu">mutate</span>(<span class="at">strainID=</span><span class="fu">factor</span>(strainID, <span class="at">levels=</span>strain.order)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(condition_prey_pred, <span class="st">"inoculum"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> f, <span class="at">x=</span>replicate, <span class="at">fill =</span> strainID),</span>
<span id="cb5-6"><a href="#cb5-6"></a>           <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> prey_history, <span class="at">scales=</span><span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb5-8"><a href="#cb5-8"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb5-9"><a href="#cb5-9"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb5-10"><a href="#cb5-10"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"% abundance"</span>, <span class="at">fill=</span><span class="st">""</span>, <span class="at">title=</span><span class="st">"Community composition in starting inocula"</span>) <span class="sc">+</span> </span>
<span id="cb5-11"><a href="#cb5-11"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb5-12"><a href="#cb5-12"></a>  <span class="fu">myartheme</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>pinit</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Community composition of the inoculae used to start the experiment
</figcaption>
</figure>
</div>
<p>Plot community compositions during the experiment</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>phi <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  <span class="fu">mutate</span>(<span class="at">strainID=</span><span class="fu">factor</span>(strainID, <span class="at">levels=</span>strain.order),</span>
<span id="cb8-3"><a href="#cb8-3"></a>         <span class="at">time_days =</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(time_days))) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_days)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5"></a>  <span class="fu">group_by</span>(prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="fu">group_split</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-7"><a href="#cb8-7"></a>  <span class="fu">map</span>(myarplot) <span class="sc">%&gt;%</span> </span>
<span id="cb8-8"><a href="#cb8-8"></a>  <span class="fu">wrap_plots</span>(., <span class="at">ncol =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">'collect'</span>) <span class="sc">+</span></span>
<span id="cb8-10"><a href="#cb8-10"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb8-11"><a href="#cb8-11"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'bottom'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>phi</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-02" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Community composition temporal trajectories during the experiment
</figcaption>
</figure>
</div>
</section>
<section id="ordination-of-community-composition" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Ordination of community composition</h1>
<p>Need some additional libraries here</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="fu">library</span>(zCompositions)</span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="fu">library</span>(vegan)</span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="fu">library</span>(ape)</span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="fu">library</span>(corrr)</span>
<span id="cb10-6"><a href="#cb10-6"></a><span class="fu">library</span>(ggrepel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="transform-data" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="transform-data"><span class="header-section-number">3.1</span> Transform data</h2>
<p>zCompositions has problems with species with &lt; 2 observations so we need to filter these out</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>lowstrains <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(condition_prey_pred, <span class="st">"inoculum"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">summarize</span>(<span class="at">n_samples =</span> <span class="fu">n</span>(),</span>
<span id="cb11-6"><a href="#cb11-6"></a>            <span class="at">n_gt0 =</span> <span class="fu">sum</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb11-7"><a href="#cb11-7"></a>            <span class="at">p_gt0 =</span> n_gt0 <span class="sc">/</span> n_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb11-8"><a href="#cb11-8"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here we remove strains present in &lt; 50 samples across transfer categories and present in &lt; 20 samples in at least 2/3 transfer categories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>lowstrainsv <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="st">"HAMBI_0097"</span>,</span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="st">"HAMBI_0262"</span>,</span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="st">"HAMBI_1842"</span>,</span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="st">"HAMBI_1988"</span>,</span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="st">"HAMBI_2443"</span>,</span>
<span id="cb12-7"><a href="#cb12-7"></a>  <span class="st">"HAMBI_2792"</span>,</span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="st">"HAMBI_3031"</span>,</span>
<span id="cb12-9"><a href="#cb12-9"></a>  <span class="st">"HAMBI_3237"</span></span>
<span id="cb12-10"><a href="#cb12-10"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>transform to matrix</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>mymat <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">filter</span>(strainID <span class="sc">%nin%</span> lowstrainsv) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(condition_prey_pred, <span class="st">"inoculum"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count) <span class="sc">%&gt;%</span> </span>
<span id="cb13-6"><a href="#cb13-6"></a>  <span class="co"># important to arrange by sample as this makes some later joins easier</span></span>
<span id="cb13-7"><a href="#cb13-7"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb13-8"><a href="#cb13-8"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"strainID"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-9"><a href="#cb13-9"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-10"><a href="#cb13-10"></a>  <span class="fu">data.frame</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="replace-zeros" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="replace-zeros"><span class="header-section-number">3.2</span> Replace zeros</h2>
<p>Compositional analysis with the centered log-ratio can’t handle zero values. Some people just replace them with a pseudocount. Another way is to impute them based on various different strategies.</p>
<p>Literature:</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6755255/">A field guide for the compositional analysis of any-omics data</a>
<ul>
<li><a href="https://zenodo.org/record/3270954#.Y1KastJBxhE">Supplemental material</a></li>
</ul></li>
<li><a href="https://doi.org/10.1016/j.chemolab.2015.02.019">zCompositions — R package for multivariate imputation of left-censored data under a compositional approach</a></li>
</ul>
<p>Here we will uses a Geometric Bayesian-multiplicative replacement strategy that preserves the ratios between the non-zero components. The “prop” option returns relative abundances.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="fu">set.seed</span>(<span class="dv">12378</span>)</span>
<span id="cb14-2"><a href="#cb14-2"></a>comp <span class="ot">&lt;-</span> zCompositions<span class="sc">::</span><span class="fu">cmultRepl</span>(mymat, <span class="at">method =</span> <span class="st">"GBM"</span>, <span class="at">output =</span> <span class="st">"prop"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>No. adjusted imputations:  77 </code></pre>
</div>
</div>
</section>
<section id="calculate-bray-curtis-dissimilarity" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="calculate-bray-curtis-dissimilarity"><span class="header-section-number">3.3</span> Calculate Bray-curtis dissimilarity</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="fu">set.seed</span>(<span class="dv">12353</span>)</span>
<span id="cb16-2"><a href="#cb16-2"></a>bray_dist      <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(comp, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb16-3"><a href="#cb16-3"></a>pcoa_ord_bray  <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">pcoa</span>(bray_dist)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="calculate-with-aitchison-distance" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="calculate-with-aitchison-distance"><span class="header-section-number">3.4</span> Calculate with Aitchison distance</h2>
<p>Aitchison distance is the Euclidean distance of the centered log-ratio transform (clr). This distance (unlike Euclidean distance on read counts) has scale invariance, perturbation invariance, permutation invariance and sub-compositional dominance.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">set.seed</span>(<span class="dv">12354</span>)</span>
<span id="cb17-2"><a href="#cb17-2"></a>balclr     <span class="ot">&lt;-</span> compositions<span class="sc">::</span><span class="fu">clr</span>(comp)</span>
<span id="cb17-3"><a href="#cb17-3"></a>aitc_dist  <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(balclr, <span class="at">method =</span> <span class="st">"euclidean"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="compare-aitchison-distance-with-clr" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="compare-aitchison-distance-with-clr"><span class="header-section-number">3.5</span> Compare Aitchison distance with CLR</h2>
<p>When the Aitchison distance is used in Principle co-ordinate Analysis (PCoA) it is equivalent to standard Principle Component Analyis (PCA) on the clr transformed data</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="fu">set.seed</span>(<span class="dv">12355</span>)</span>
<span id="cb18-2"><a href="#cb18-2"></a>pcoa_ord_aitc  <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">pcoa</span>(aitc_dist)</span>
<span id="cb18-3"><a href="#cb18-3"></a>pca_ord_aitc   <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(balclr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>For example, these ordinations are the same, just that Axis2 is the mirrorimage between. Since the rotation is arbitrary this does not matter.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="fu">plot</span>(pcoa_ord_aitc<span class="sc">$</span>vectors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-03" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Plot of Principal Coordinate Analysis (PCoA) done in the Aitchison geometry of the simplex (i.e.&nbsp;PCoA on the euclidean distance of the centered log-ratio transform species compositions).
</figcaption>
</figure>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="fu">plot</span>(pca_ord_aitc<span class="sc">$</span>x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Plot of Principal Component Analysis (PCA) done in the Aitchison geometry of the simplex (i.e.&nbsp;PCA on the centered log-ratio transform species compositions). Notice that the PCoA using the euclidean distance of centered log-ratio transformed species frequencies is equivalent to the PCA directly using clr-transformed values in <a href="#fig-03" class="quarto-xref">Figure&nbsp;3</a>.
</figcaption>
</figure>
</div>
</section>
<section id="environment-vectors" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="environment-vectors"><span class="header-section-number">3.6</span> Environment vectors</h2>
<p>left_join with metadata</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>pca2plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pca_ord_aitc<span class="sc">$</span>x) <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3"></a>  <span class="fu">left_join</span>(metadf) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5"></a>  <span class="fu">mutate</span>(<span class="at">expcombo=</span><span class="fu">interaction</span>(condition_prey_pred, time_days, <span class="at">sep =</span> <span class="st">"_"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>About 85% of variance explained in first 5 PCs</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>pca_ord_aitc_importance <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">summary</span>(pca_ord_aitc)<span class="sc">$</span>importance) <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"parameter"</span>)</span>
<span id="cb22-3"><a href="#cb22-3"></a></span>
<span id="cb22-4"><a href="#cb22-4"></a>pca_ord_aitc_importance</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["parameter"],"name":[1],"type":["chr"],"align":["left"]},{"label":["PC1"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["PC2"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["PC3"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["PC4"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["PC5"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["PC6"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["PC7"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["PC8"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["PC9"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["PC10"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["PC11"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["PC12"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["PC13"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["PC14"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["PC15"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["PC16"],"name":[17],"type":["dbl"],"align":["right"]}],"data":[{"1":"Standard deviation","2":"4.330732","3":"3.311528","4":"2.509542","5":"2.116339","6":"1.77754","7":"1.549158","8":"1.258983","9":"0.964737","10":"0.8166996","11":"0.7054134","12":"0.6401701","13":"0.5483365","14":"0.4717219","15":"0.3846169","16":"0.2833165","17":"2.519812e-15"},{"1":"Proportion of Variance","2":"0.368480","3":"0.215450","4":"0.123730","5":"0.088000","6":"0.06208","7":"0.047150","8":"0.031140","9":"0.018290","10":"0.0131000","11":"0.0097800","12":"0.0080500","13":"0.0059100","14":"0.0043700","15":"0.0029100","16":"0.0015800","17":"0.000000e+00"},{"1":"Cumulative Proportion","2":"0.368480","3":"0.583930","4":"0.707660","5":"0.795650","6":"0.85773","7":"0.904880","8":"0.936020","9":"0.954310","10":"0.9674100","11":"0.9771900","12":"0.9852400","13":"0.9911400","14":"0.9955200","15":"0.9984200","16":"1.0000000","17":"1.000000e+00"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Environmental/experimental variables associated with ordinatoion</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>ef <span class="ot">&lt;-</span></span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">envfit</span>(</span>
<span id="cb23-3"><a href="#cb23-3"></a>    pca_ord_aitc <span class="sc">~</span> prey_history <span class="sc">+</span> predator_history <span class="sc">+</span> ciliates_ml <span class="sc">+</span> time_days,</span>
<span id="cb23-4"><a href="#cb23-4"></a>    <span class="at">data =</span> <span class="fu">semi_join</span>(metadf, dplyr<span class="sc">::</span><span class="fu">select</span>(pca2plot, sample)),</span>
<span id="cb23-5"><a href="#cb23-5"></a>    <span class="at">na.rm =</span> T,</span>
<span id="cb23-6"><a href="#cb23-6"></a>    <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb23-7"><a href="#cb23-7"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="fu">biplot</span>(pca_ord_aitc, <span class="at">choices=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">labSize=</span><span class="dv">0</span>)</span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="fu">plot</span>(ef)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-05" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Environmental vectors plotted onto ordination in <a href="#fig-04" class="quarto-xref">Figure&nbsp;4</a>
</figcaption>
</figure>
</div>
<section id="significance-of-the-environmental-covariates" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="significance-of-the-environmental-covariates"><span class="header-section-number">3.6.1</span> Significance of the environmental covariates</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>ef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
***VECTORS

                PC1     PC2     r2 Pr(&gt;r)    
ciliates_ml 0.84491 0.53492 0.1262  0.001 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 999

***FACTORS:

Centroids:
                        PC1     PC2
prey_historyanc      4.3571 -0.5277
prey_historyevo     -3.0257  1.4454
predator_historyanc  1.1182 -1.1211
predator_historyevo  0.2132  2.0388
time_days4          -0.8610 -4.2785
time_days8           0.8755 -1.0156
time_days12          1.1134  0.1314
time_days16          1.2332  0.0357
time_days20         -0.0521  0.3658
time_days24          1.8596  1.4475
time_days28          1.5366  1.0170
time_days32          0.4066  0.7041
time_days36          0.9413  1.1857
time_days40          0.4137  1.0675
time_days44          0.4505  1.2408
time_days48          0.2291  1.2173
time_days52          0.9245  1.6736
time_days56          0.8231  0.6831
time_days60          0.0920  1.4073

Goodness of fit:
                     r2 Pr(&gt;r)    
prey_history     0.5357  0.001 ***
predator_history 0.0991  0.001 ***
time_days        0.0911  0.246    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
Permutation: free
Number of permutations: 999

24 observations deleted due to missingness</code></pre>
</div>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>con_scrs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(ef, <span class="at">display =</span> <span class="st">"vectors"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"var"</span>) <span class="sc">%&gt;%</span> <span class="fu">tibble</span>()</span>
<span id="cb27-3"><a href="#cb27-3"></a></span>
<span id="cb27-4"><a href="#cb27-4"></a>fct_scrs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(ef, <span class="at">display =</span> <span class="st">"factors"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"var"</span>) <span class="sc">%&gt;%</span>  <span class="fu">tibble</span>()</span>
<span id="cb27-6"><a href="#cb27-6"></a></span>
<span id="cb27-7"><a href="#cb27-7"></a>scale_factor <span class="ot">&lt;-</span> <span class="dv">15</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plot-with-prey-history-highlighted" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="plot-with-prey-history-highlighted"><span class="header-section-number">3.6.2</span> Plot with prey history highlighted</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>ppca <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca2plot) <span class="sc">+</span></span>
<span id="cb28-2"><a href="#cb28-2"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb28-3"><a href="#cb28-3"></a>    <span class="at">x =</span> PC1,</span>
<span id="cb28-4"><a href="#cb28-4"></a>    <span class="at">y =</span> PC2,</span>
<span id="cb28-5"><a href="#cb28-5"></a>    <span class="at">color =</span> prey_history,</span>
<span id="cb28-6"><a href="#cb28-6"></a>    <span class="at">shape =</span> predator_history), <span class="at">size=</span><span class="dv">3</span> ) <span class="sc">+</span></span>
<span id="cb28-7"><a href="#cb28-7"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> con_scrs,</span>
<span id="cb28-8"><a href="#cb28-8"></a>               <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> PC2<span class="sc">*</span>scale_factor),</span>
<span id="cb28-9"><a href="#cb28-9"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">"cm"</span>)), <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> con_scrs, <span class="fu">aes</span>(<span class="at">x =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> PC2<span class="sc">*</span>scale_factor, <span class="at">label =</span> var),</span>
<span id="cb28-11"><a href="#cb28-11"></a>                  <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb28-12"><a href="#cb28-12"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>), </span>
<span id="cb28-13"><a href="#cb28-13"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">3</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb28-14"><a href="#cb28-14"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> prey_history)) <span class="sc">+</span> </span>
<span id="cb28-15"><a href="#cb28-15"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb28-16"><a href="#cb28-16"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#88CCEE"</span>, <span class="st">"#CC6677"</span>)) <span class="sc">+</span></span>
<span id="cb28-17"><a href="#cb28-17"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb28-18"><a href="#cb28-18"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb28-19"><a href="#cb28-19"></a>  <span class="fu">theme</span>(</span>
<span id="cb28-20"><a href="#cb28-20"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-21"><a href="#cb28-21"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-22"><a href="#cb28-22"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb28-23"><a href="#cb28-23"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-06" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Plot of principal component analysis (PCA) done in the Aitchison geometry of the simplex (i.e.&nbsp;centered log-ratio transform species compositions). Color shows evolutionary history of the prey and shape shows evolutionary history of the predator. Arrows and ellipses depict statistically significant (P &lt; 0.05) experimental variables (continuous and categorical, resp) fit to the ordination axes via regression (<code>vegan::envfit</code>) and projected onto the ordination plot. Significance is assessed by permutation.
</figcaption>
</figure>
</div>
</section>
<section id="plot-with-predator-history-highlighted" class="level3" data-number="3.6.3">
<h3 data-number="3.6.3" class="anchored" data-anchor-id="plot-with-predator-history-highlighted"><span class="header-section-number">3.6.3</span> Plot with predator history highlighted</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="fu">ggplot</span>(pca2plot) <span class="sc">+</span></span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb29-3"><a href="#cb29-3"></a>    <span class="at">x =</span> PC1,</span>
<span id="cb29-4"><a href="#cb29-4"></a>    <span class="at">y =</span> PC2,</span>
<span id="cb29-5"><a href="#cb29-5"></a>    <span class="at">color =</span> predator_history,</span>
<span id="cb29-6"><a href="#cb29-6"></a>    <span class="at">shape =</span> prey_history), <span class="at">size=</span><span class="dv">3</span> ) <span class="sc">+</span></span>
<span id="cb29-7"><a href="#cb29-7"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> con_scrs,</span>
<span id="cb29-8"><a href="#cb29-8"></a>               <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> PC2<span class="sc">*</span>scale_factor),</span>
<span id="cb29-9"><a href="#cb29-9"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">"cm"</span>)), <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb29-10"><a href="#cb29-10"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> con_scrs, <span class="fu">aes</span>(<span class="at">x =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> PC2<span class="sc">*</span>scale_factor, <span class="at">label =</span> var),</span>
<span id="cb29-11"><a href="#cb29-11"></a>                  <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb29-12"><a href="#cb29-12"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>), </span>
<span id="cb29-13"><a href="#cb29-13"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">3</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb29-14"><a href="#cb29-14"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> predator_history)) <span class="sc">+</span> </span>
<span id="cb29-15"><a href="#cb29-15"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb29-16"><a href="#cb29-16"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb29-17"><a href="#cb29-17"></a>  <span class="fu">theme</span>(</span>
<span id="cb29-18"><a href="#cb29-18"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb29-19"><a href="#cb29-19"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb29-20"><a href="#cb29-20"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb29-21"><a href="#cb29-21"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-07" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-07-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-07-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: As in <a href="#fig-06" class="quarto-xref">Figure&nbsp;6</a> but with predator evolutionary history highlighted in color.
</figcaption>
</figure>
</div>
<p>The evolved predator seems to “tighten” up the variation and make samplles with ancestral prey “look” more like the samples with evolved prey</p>
</section>
<section id="permanova" class="level3" data-number="3.6.4">
<h3 data-number="3.6.4" class="anchored" data-anchor-id="permanova"><span class="header-section-number">3.6.4</span> PERMANOVA</h3>
<p>PERMANOVA suggests significant effect of predator and prey evolutionary history</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="fu">set.seed</span>(<span class="dv">45781</span>)</span>
<span id="cb30-2"><a href="#cb30-2"></a><span class="fu">adonis2</span>(aitc_dist <span class="sc">~</span> pca2plot<span class="sc">$</span>prey_history <span class="sc">+</span> pca2plot<span class="sc">$</span>predator_history,</span>
<span id="cb30-3"><a href="#cb30-3"></a>       <span class="at">permutations =</span> <span class="fl">1e3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["SumOfSqs"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["R2"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"3","2":"4200.465","3":"0.4065275","4":"45.66653","5":"0.000999001","_rn_":"Model"},{"1":"200","2":"6132.084","3":"0.5934725","4":"NA","5":"NA","_rn_":"Residual"},{"1":"203","2":"10332.549","3":"1.0000000","4":"NA","5":"NA","_rn_":"Total"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Check for homogeneity of variance between these two categories. Suggests that they are different although maybe not by very much…</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="fu">set.seed</span>(<span class="dv">45782</span>)</span>
<span id="cb31-2"><a href="#cb31-2"></a>bdprey <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitc_dist, pca2plot<span class="sc">$</span>prey_history)</span>
<span id="cb31-3"><a href="#cb31-3"></a><span class="fu">permutest</span>(bdprey)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
           Df Sum Sq Mean Sq      F N.Perm Pr(&gt;F)   
Groups      1  38.61  38.613 9.4672    999  0.002 **
Residuals 202 823.88   4.079                        
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="fu">set.seed</span>(<span class="dv">45783</span>)</span>
<span id="cb33-2"><a href="#cb33-2"></a>bdpred <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitc_dist, pca2plot<span class="sc">$</span>predator_history)</span>
<span id="cb33-3"><a href="#cb33-3"></a><span class="fu">permutest</span>(bdpred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
           Df Sum Sq Mean Sq     F N.Perm Pr(&gt;F)    
Groups      2 136.53  68.263 34.83    999  0.001 ***
Residuals 201 393.94   1.960                        
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
</section>
<section id="geometric-analysis-of-temporal-trajectories-in-pca" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="geometric-analysis-of-temporal-trajectories-in-pca"><span class="header-section-number">3.7</span> Geometric analysis of temporal trajectories in PCA</h2>
<p>Function to calculate euclidean distance between points</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>euclidean_dist <span class="ot">&lt;-</span> <span class="cf">function</span>(x1, x2, y1, y2){</span>
<span id="cb35-2"><a href="#cb35-2"></a>  <span class="fu">sqrt</span>((x1 <span class="sc">-</span> x2)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> (y1 <span class="sc">-</span> y2)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb35-3"><a href="#cb35-3"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>eucdist <span class="ot">&lt;-</span> pca2plot <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="#cb36-2"></a>  <span class="fu">filter</span>(prey_history <span class="sc">!=</span> <span class="st">"none"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3"></a>  <span class="fu">filter</span>(predator_history <span class="sc">!=</span> <span class="st">"nopredator"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="#cb36-4"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PC1, PC2, replicate, prey_history, predator_history, time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5"></a>  <span class="fu">arrange</span>(replicate, prey_history, predator_history, time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb36-6"><a href="#cb36-6"></a>  <span class="fu">group_by</span>(replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb36-7"><a href="#cb36-7"></a>  <span class="fu">mutate</span>(<span class="at">d =</span> <span class="fu">euclidean_dist</span>(PC1, <span class="fu">lag</span>(PC1), PC2, <span class="fu">lag</span>(PC2)),</span>
<span id="cb36-8"><a href="#cb36-8"></a>         <span class="at">range =</span> <span class="fu">paste0</span>(<span class="fu">lag</span>(time_days),<span class="st">"-"</span>,time_days)) <span class="sc">%&gt;%</span></span>
<span id="cb36-9"><a href="#cb36-9"></a>  <span class="fu">mutate</span>(<span class="at">range =</span> forcats<span class="sc">::</span><span class="fu">fct_reorder</span>(range, <span class="fu">as.numeric</span>(<span class="fu">as.character</span>((time_days))))) <span class="sc">%&gt;%</span></span>
<span id="cb36-10"><a href="#cb36-10"></a>  <span class="fu">filter</span>(range <span class="sc">!=</span> <span class="st">"NA-4"</span>)</span>
<span id="cb36-11"><a href="#cb36-11"></a></span>
<span id="cb36-12"><a href="#cb36-12"></a><span class="fu">write_tsv</span>(eucdist, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"aitchison_pca_distances.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="fu">ggplot</span>(eucdist) <span class="sc">+</span></span>
<span id="cb37-2"><a href="#cb37-2"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> range, <span class="at">y =</span> d, <span class="at">color =</span> replicate, <span class="at">group=</span>replicate)) <span class="sc">+</span></span>
<span id="cb37-3"><a href="#cb37-3"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Time range (days)"</span>, <span class="at">y =</span> <span class="st">"Euclidean distance between days"</span>) <span class="sc">+</span></span>
<span id="cb37-4"><a href="#cb37-4"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7"></a>  <span class="fu">theme</span>(</span>
<span id="cb37-8"><a href="#cb37-8"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb37-9"><a href="#cb37-9"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb37-10"><a href="#cb37-10"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-11"><a href="#cb37-11"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-08" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-08-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-08-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Geometric analysis of community trajectories based on PCA. Segment lengths between consecutive sampling days in the two PCA dimensions were calculated and plotted chronologically for each biological replicate in each treatment combination. As sampling intervals were of equal duration (except day 16 - 24), segment length represents relative speed of compositional change.
</figcaption>
</figure>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb38" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb38-1"><a href="#cb38-1"></a><span class="co">---</span></span>
<span id="cb38-2"><a href="#cb38-2"></a><span class="an">title:</span><span class="co"> "Analysis of community composition"</span></span>
<span id="cb38-3"><a href="#cb38-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb38-4"><a href="#cb38-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb38-5"><a href="#cb38-5"></a><span class="an">abstract:</span><span class="co"> "This notebook takes formatted 16S rRNA amplicon counts and plots and analyzes them in different ways to look at community composition"</span></span>
<span id="cb38-6"><a href="#cb38-6"></a><span class="co">---</span></span>
<span id="cb38-7"><a href="#cb38-7"></a></span>
<span id="cb38-8"><a href="#cb38-8"></a><span class="fu"># Setup</span></span>
<span id="cb38-9"><a href="#cb38-9"></a></span>
<span id="cb38-10"><a href="#cb38-10"></a>Libraries and global variables</span>
<span id="cb38-11"><a href="#cb38-11"></a></span>
<span id="cb38-14"><a href="#cb38-14"></a><span class="in">```{r}</span></span>
<span id="cb38-15"><a href="#cb38-15"></a><span class="co">#| output: false</span></span>
<span id="cb38-16"><a href="#cb38-16"></a><span class="co">#| warning: false</span></span>
<span id="cb38-17"><a href="#cb38-17"></a><span class="co">#| error: false</span></span>
<span id="cb38-18"><a href="#cb38-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb38-19"><a href="#cb38-19"></a><span class="fu">library</span>(here)</span>
<span id="cb38-20"><a href="#cb38-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb38-21"><a href="#cb38-21"></a><span class="fu">library</span>(fishualize)</span>
<span id="cb38-22"><a href="#cb38-22"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb38-23"><a href="#cb38-23"></a><span class="fu">library</span>(withr)</span>
<span id="cb38-24"><a href="#cb38-24"></a><span class="fu">library</span>(scales)</span>
<span id="cb38-25"><a href="#cb38-25"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb38-26"><a href="#cb38-26"></a></span>
<span id="cb38-27"><a href="#cb38-27"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb38-28"><a href="#cb38-28"></a><span class="in">```</span></span>
<span id="cb38-29"><a href="#cb38-29"></a></span>
<span id="cb38-30"><a href="#cb38-30"></a>Set up some directories</span>
<span id="cb38-31"><a href="#cb38-31"></a></span>
<span id="cb38-34"><a href="#cb38-34"></a><span class="in">```{r}</span></span>
<span id="cb38-35"><a href="#cb38-35"></a><span class="co">#| output: false</span></span>
<span id="cb38-36"><a href="#cb38-36"></a><span class="co">#| warning: false</span></span>
<span id="cb38-37"><a href="#cb38-37"></a><span class="co">#| error: false</span></span>
<span id="cb38-38"><a href="#cb38-38"></a><span class="co">#| </span></span>
<span id="cb38-39"><a href="#cb38-39"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"illumina_v3v4"</span>)</span>
<span id="cb38-40"><a href="#cb38-40"></a></span>
<span id="cb38-41"><a href="#cb38-41"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb38-42"><a href="#cb38-42"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"illumina_v3v4"</span>)</span>
<span id="cb38-43"><a href="#cb38-43"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"illumina_v3v4"</span>)</span>
<span id="cb38-44"><a href="#cb38-44"></a></span>
<span id="cb38-45"><a href="#cb38-45"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb38-46"><a href="#cb38-46"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span>
<span id="cb38-47"><a href="#cb38-47"></a><span class="in">```</span></span>
<span id="cb38-48"><a href="#cb38-48"></a></span>
<span id="cb38-49"><a href="#cb38-49"></a><span class="fu">## Some functions</span></span>
<span id="cb38-50"><a href="#cb38-50"></a></span>
<span id="cb38-53"><a href="#cb38-53"></a><span class="in">```{r}</span></span>
<span id="cb38-54"><a href="#cb38-54"></a>myarplot <span class="ot">&lt;-</span> <span class="cf">function</span>(.data){</span>
<span id="cb38-55"><a href="#cb38-55"></a>  cil <span class="ot">&lt;-</span> <span class="fu">unique</span>(.data<span class="sc">$</span>predator_history)</span>
<span id="cb38-56"><a href="#cb38-56"></a>  bac <span class="ot">&lt;-</span> <span class="fu">unique</span>(.data<span class="sc">$</span>prey_history)</span>
<span id="cb38-57"><a href="#cb38-57"></a>  </span>
<span id="cb38-58"><a href="#cb38-58"></a>  mytitle <span class="ot">&lt;-</span> <span class="fu">case_when</span>(cil <span class="sc">==</span> <span class="st">"anc"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"anc bac + anc cil"</span>,</span>
<span id="cb38-59"><a href="#cb38-59"></a>                       cil <span class="sc">==</span> <span class="st">"anc"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"evo"</span> <span class="sc">~</span> <span class="st">"evo bac + anc cil"</span>,</span>
<span id="cb38-60"><a href="#cb38-60"></a>                       cil <span class="sc">==</span> <span class="st">"evo"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"anc bac + evo cil"</span>,</span>
<span id="cb38-61"><a href="#cb38-61"></a>                       cil <span class="sc">==</span> <span class="st">"evo"</span>  <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"evo"</span> <span class="sc">~</span> <span class="st">"evo bac + evo cil"</span>,</span>
<span id="cb38-62"><a href="#cb38-62"></a>                       cil <span class="sc">==</span> <span class="st">"nopredator"</span> <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"anc"</span> <span class="sc">~</span> <span class="st">"anc bac"</span>,</span>
<span id="cb38-63"><a href="#cb38-63"></a>                       cil <span class="sc">==</span> <span class="st">"nopredator"</span> <span class="sc">&amp;</span> bac <span class="sc">==</span> <span class="st">"evo"</span> <span class="sc">~</span> <span class="st">"evo bac"</span>,</span>
<span id="cb38-64"><a href="#cb38-64"></a>                       <span class="co">#cil == "nopredator" &amp; bac == "evo" ~ "starting anc bac",</span></span>
<span id="cb38-65"><a href="#cb38-65"></a>                       <span class="co">#cil == "nopredator" &amp; bac == "evo" ~ "starting evo bac"</span></span>
<span id="cb38-66"><a href="#cb38-66"></a>  )</span>
<span id="cb38-67"><a href="#cb38-67"></a>  </span>
<span id="cb38-68"><a href="#cb38-68"></a>  <span class="fu">ggplot</span>(.data) <span class="sc">+</span></span>
<span id="cb38-69"><a href="#cb38-69"></a>    <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>f, <span class="at">fill=</span>strainID),</span>
<span id="cb38-70"><a href="#cb38-70"></a>              <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb38-71"><a href="#cb38-71"></a>    <span class="fu">facet_wrap</span>( <span class="sc">~</span> replicate, <span class="at">strip.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb38-72"><a href="#cb38-72"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb38-73"><a href="#cb38-73"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb38-74"><a href="#cb38-74"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">60</span>), <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">40</span>, <span class="dv">60</span>)) <span class="sc">+</span></span>
<span id="cb38-75"><a href="#cb38-75"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">""</span>, <span class="at">fill=</span><span class="st">""</span>, <span class="at">title=</span>mytitle) <span class="sc">+</span> </span>
<span id="cb38-76"><a href="#cb38-76"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb38-77"><a href="#cb38-77"></a>    <span class="fu">myartheme</span>()</span>
<span id="cb38-78"><a href="#cb38-78"></a>  </span>
<span id="cb38-79"><a href="#cb38-79"></a>}</span>
<span id="cb38-80"><a href="#cb38-80"></a></span>
<span id="cb38-81"><a href="#cb38-81"></a>myartheme <span class="ot">&lt;-</span> <span class="cf">function</span>(...){</span>
<span id="cb38-82"><a href="#cb38-82"></a>  <span class="fu">theme</span>(</span>
<span id="cb38-83"><a href="#cb38-83"></a>    <span class="at">panel.spacing.x =</span> <span class="fu">unit</span>(<span class="fl">0.05</span>,<span class="st">"line"</span>),</span>
<span id="cb38-84"><a href="#cb38-84"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb38-85"><a href="#cb38-85"></a>    <span class="at">strip.background.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-86"><a href="#cb38-86"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-87"><a href="#cb38-87"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-88"><a href="#cb38-88"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-89"><a href="#cb38-89"></a>    <span class="co">#axis.text.x = element_blank(),</span></span>
<span id="cb38-90"><a href="#cb38-90"></a>    <span class="at">axis.line.x =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb38-91"><a href="#cb38-91"></a>    <span class="at">axis.line.y =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb38-92"><a href="#cb38-92"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-93"><a href="#cb38-93"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-94"><a href="#cb38-94"></a>    <span class="at">legend.key =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-95"><a href="#cb38-95"></a>    ...)</span>
<span id="cb38-96"><a href="#cb38-96"></a>}</span>
<span id="cb38-97"><a href="#cb38-97"></a></span>
<span id="cb38-98"><a href="#cb38-98"></a>strain.order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"HAMBI_2443"</span>, <span class="st">"HAMBI_3031"</span>, <span class="st">"HAMBI_1988"</span>, <span class="st">"HAMBI_3237"</span>, <span class="st">"HAMBI_0262"</span>, </span>
<span id="cb38-99"><a href="#cb38-99"></a>                  <span class="st">"HAMBI_2159"</span>, <span class="st">"HAMBI_1842"</span>, <span class="st">"HAMBI_2164"</span>, <span class="st">"HAMBI_0006"</span>, <span class="st">"HAMBI_0097"</span>, </span>
<span id="cb38-100"><a href="#cb38-100"></a>                  <span class="st">"HAMBI_2494"</span>, <span class="st">"HAMBI_1299"</span>, <span class="st">"HAMBI_2160"</span>, <span class="st">"HAMBI_1279"</span>, <span class="st">"HAMBI_2792"</span>, </span>
<span id="cb38-101"><a href="#cb38-101"></a>                  <span class="st">"HAMBI_1896"</span>, <span class="st">"HAMBI_0105"</span>, <span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_1923"</span>, </span>
<span id="cb38-102"><a href="#cb38-102"></a>                  <span class="st">"HAMBI_2659"</span>, <span class="st">"HAMBI_1292"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1287"</span>)</span>
<span id="cb38-103"><a href="#cb38-103"></a><span class="in">```</span></span>
<span id="cb38-104"><a href="#cb38-104"></a></span>
<span id="cb38-105"><a href="#cb38-105"></a>Read data</span>
<span id="cb38-106"><a href="#cb38-106"></a></span>
<span id="cb38-109"><a href="#cb38-109"></a><span class="in">```{r}</span></span>
<span id="cb38-110"><a href="#cb38-110"></a><span class="co">#| output: false</span></span>
<span id="cb38-111"><a href="#cb38-111"></a><span class="co">#| warning: false</span></span>
<span id="cb38-112"><a href="#cb38-112"></a><span class="co">#| error: false</span></span>
<span id="cb38-113"><a href="#cb38-113"></a>counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"species_counts.tsv"</span>))</span>
<span id="cb38-114"><a href="#cb38-114"></a>metadf <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"shared"</span>, <span class="st">"metadata_formatted.rds"</span>))</span>
<span id="cb38-115"><a href="#cb38-115"></a></span>
<span id="cb38-116"><a href="#cb38-116"></a>counts_f <span class="ot">&lt;-</span> <span class="fu">left_join</span>(metadf, counts) <span class="sc">%&gt;%</span> </span>
<span id="cb38-117"><a href="#cb38-117"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb38-118"><a href="#cb38-118"></a>  <span class="fu">mutate</span>(<span class="at">f=</span>count<span class="sc">/</span><span class="fu">sum</span>(count)) <span class="sc">%&gt;%</span>  </span>
<span id="cb38-119"><a href="#cb38-119"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-120"><a href="#cb38-120"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(strainID))</span>
<span id="cb38-121"><a href="#cb38-121"></a><span class="in">```</span></span>
<span id="cb38-122"><a href="#cb38-122"></a></span>
<span id="cb38-123"><a href="#cb38-123"></a><span class="fu"># Plot Community Composition</span></span>
<span id="cb38-124"><a href="#cb38-124"></a></span>
<span id="cb38-125"><a href="#cb38-125"></a>community composition of the initial conditions</span>
<span id="cb38-126"><a href="#cb38-126"></a></span>
<span id="cb38-129"><a href="#cb38-129"></a><span class="in">```{r}</span></span>
<span id="cb38-130"><a href="#cb38-130"></a>pinit <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb38-131"><a href="#cb38-131"></a>  <span class="fu">mutate</span>(<span class="at">strainID=</span><span class="fu">factor</span>(strainID, <span class="at">levels=</span>strain.order)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-132"><a href="#cb38-132"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(condition_prey_pred, <span class="st">"inoculum"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-133"><a href="#cb38-133"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb38-134"><a href="#cb38-134"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> f, <span class="at">x=</span>replicate, <span class="at">fill =</span> strainID),</span>
<span id="cb38-135"><a href="#cb38-135"></a>           <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb38-136"><a href="#cb38-136"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> prey_history, <span class="at">scales=</span><span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb38-137"><a href="#cb38-137"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb38-138"><a href="#cb38-138"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb38-139"><a href="#cb38-139"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"% abundance"</span>, <span class="at">fill=</span><span class="st">""</span>, <span class="at">title=</span><span class="st">"Community composition in starting inocula"</span>) <span class="sc">+</span> </span>
<span id="cb38-140"><a href="#cb38-140"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb38-141"><a href="#cb38-141"></a>  <span class="fu">myartheme</span>()</span>
<span id="cb38-142"><a href="#cb38-142"></a><span class="in">```</span></span>
<span id="cb38-143"><a href="#cb38-143"></a></span>
<span id="cb38-144"><a href="#cb38-144"></a>::: {#fig-01}</span>
<span id="cb38-147"><a href="#cb38-147"></a><span class="in">```{r}</span></span>
<span id="cb38-148"><a href="#cb38-148"></a><span class="co">#| warning: false</span></span>
<span id="cb38-149"><a href="#cb38-149"></a><span class="co">#| error: false</span></span>
<span id="cb38-150"><a href="#cb38-150"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb38-151"><a href="#cb38-151"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb38-152"><a href="#cb38-152"></a>pinit</span>
<span id="cb38-153"><a href="#cb38-153"></a><span class="in">```</span></span>
<span id="cb38-154"><a href="#cb38-154"></a>Community composition of the inoculae used to start the experiment</span>
<span id="cb38-155"><a href="#cb38-155"></a>:::</span>
<span id="cb38-156"><a href="#cb38-156"></a></span>
<span id="cb38-159"><a href="#cb38-159"></a><span class="in">```{r}</span></span>
<span id="cb38-160"><a href="#cb38-160"></a><span class="co">#| echo: false</span></span>
<span id="cb38-161"><a href="#cb38-161"></a><span class="fu">ggsave</span>(</span>
<span id="cb38-162"><a href="#cb38-162"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"species_comp_init.png"</span>),</span>
<span id="cb38-163"><a href="#cb38-163"></a>  pinit,</span>
<span id="cb38-164"><a href="#cb38-164"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb38-165"><a href="#cb38-165"></a>  <span class="at">height =</span> <span class="dv">5</span>,</span>
<span id="cb38-166"><a href="#cb38-166"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb38-167"><a href="#cb38-167"></a>  <span class="at">device =</span> <span class="st">"png"</span>,</span>
<span id="cb38-168"><a href="#cb38-168"></a>  <span class="at">dpi =</span> <span class="dv">320</span></span>
<span id="cb38-169"><a href="#cb38-169"></a>)</span>
<span id="cb38-170"><a href="#cb38-170"></a></span>
<span id="cb38-171"><a href="#cb38-171"></a><span class="fu">ggsave</span>(</span>
<span id="cb38-172"><a href="#cb38-172"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"species_comp_init.pdf"</span>),</span>
<span id="cb38-173"><a href="#cb38-173"></a>  pinit,</span>
<span id="cb38-174"><a href="#cb38-174"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb38-175"><a href="#cb38-175"></a>  <span class="at">height =</span> <span class="dv">5</span>,</span>
<span id="cb38-176"><a href="#cb38-176"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb38-177"><a href="#cb38-177"></a>  <span class="at">device =</span> <span class="st">"pdf"</span></span>
<span id="cb38-178"><a href="#cb38-178"></a>)</span>
<span id="cb38-179"><a href="#cb38-179"></a><span class="in">```</span></span>
<span id="cb38-180"><a href="#cb38-180"></a></span>
<span id="cb38-181"><a href="#cb38-181"></a>Plot community compositions during the experiment</span>
<span id="cb38-182"><a href="#cb38-182"></a></span>
<span id="cb38-185"><a href="#cb38-185"></a><span class="in">```{r}</span></span>
<span id="cb38-186"><a href="#cb38-186"></a>phi <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb38-187"><a href="#cb38-187"></a>  <span class="fu">mutate</span>(<span class="at">strainID=</span><span class="fu">factor</span>(strainID, <span class="at">levels=</span>strain.order),</span>
<span id="cb38-188"><a href="#cb38-188"></a>         <span class="at">time_days =</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(time_days))) <span class="sc">%&gt;%</span> </span>
<span id="cb38-189"><a href="#cb38-189"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(time_days)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-190"><a href="#cb38-190"></a>  <span class="fu">group_by</span>(prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb38-191"><a href="#cb38-191"></a>  <span class="fu">group_split</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-192"><a href="#cb38-192"></a>  <span class="fu">map</span>(myarplot) <span class="sc">%&gt;%</span> </span>
<span id="cb38-193"><a href="#cb38-193"></a>  <span class="fu">wrap_plots</span>(., <span class="at">ncol =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb38-194"><a href="#cb38-194"></a>  <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">'collect'</span>) <span class="sc">+</span></span>
<span id="cb38-195"><a href="#cb38-195"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">'A'</span>) <span class="sc">&amp;</span> </span>
<span id="cb38-196"><a href="#cb38-196"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'bottom'</span>)</span>
<span id="cb38-197"><a href="#cb38-197"></a><span class="in">```</span></span>
<span id="cb38-198"><a href="#cb38-198"></a></span>
<span id="cb38-199"><a href="#cb38-199"></a>::: {#fig-02}</span>
<span id="cb38-202"><a href="#cb38-202"></a><span class="in">```{r}</span></span>
<span id="cb38-203"><a href="#cb38-203"></a><span class="co">#| warning: false</span></span>
<span id="cb38-204"><a href="#cb38-204"></a><span class="co">#| error: false</span></span>
<span id="cb38-205"><a href="#cb38-205"></a><span class="co">#| fig.width: 14</span></span>
<span id="cb38-206"><a href="#cb38-206"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb38-207"><a href="#cb38-207"></a>phi</span>
<span id="cb38-208"><a href="#cb38-208"></a><span class="in">```</span></span>
<span id="cb38-209"><a href="#cb38-209"></a>Community composition temporal trajectories during the experiment</span>
<span id="cb38-210"><a href="#cb38-210"></a>:::</span>
<span id="cb38-211"><a href="#cb38-211"></a></span>
<span id="cb38-214"><a href="#cb38-214"></a><span class="in">```{r}</span></span>
<span id="cb38-215"><a href="#cb38-215"></a><span class="co">#| echo: false</span></span>
<span id="cb38-216"><a href="#cb38-216"></a><span class="fu">ggsave</span>(</span>
<span id="cb38-217"><a href="#cb38-217"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"species_comp_full.png"</span>),</span>
<span id="cb38-218"><a href="#cb38-218"></a>  phi,</span>
<span id="cb38-219"><a href="#cb38-219"></a>  <span class="at">width =</span> <span class="dv">14</span>,</span>
<span id="cb38-220"><a href="#cb38-220"></a>  <span class="at">height =</span> <span class="dv">8</span>,</span>
<span id="cb38-221"><a href="#cb38-221"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb38-222"><a href="#cb38-222"></a>  <span class="at">device =</span> <span class="st">"png"</span>,</span>
<span id="cb38-223"><a href="#cb38-223"></a>  <span class="at">dpi =</span> <span class="dv">320</span></span>
<span id="cb38-224"><a href="#cb38-224"></a>)</span>
<span id="cb38-225"><a href="#cb38-225"></a></span>
<span id="cb38-226"><a href="#cb38-226"></a><span class="fu">ggsave</span>(</span>
<span id="cb38-227"><a href="#cb38-227"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"species_comp_full.pdf"</span>),</span>
<span id="cb38-228"><a href="#cb38-228"></a>  phi,</span>
<span id="cb38-229"><a href="#cb38-229"></a>  <span class="at">width =</span> <span class="dv">14</span>,</span>
<span id="cb38-230"><a href="#cb38-230"></a>  <span class="at">height =</span> <span class="dv">8</span>,</span>
<span id="cb38-231"><a href="#cb38-231"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb38-232"><a href="#cb38-232"></a>  <span class="at">device =</span> <span class="st">"pdf"</span></span>
<span id="cb38-233"><a href="#cb38-233"></a>)</span>
<span id="cb38-234"><a href="#cb38-234"></a><span class="in">```</span></span>
<span id="cb38-235"><a href="#cb38-235"></a></span>
<span id="cb38-236"><a href="#cb38-236"></a><span class="fu"># Ordination of community composition</span></span>
<span id="cb38-237"><a href="#cb38-237"></a></span>
<span id="cb38-238"><a href="#cb38-238"></a>Need some additional libraries here</span>
<span id="cb38-239"><a href="#cb38-239"></a></span>
<span id="cb38-242"><a href="#cb38-242"></a><span class="in">```{r}</span></span>
<span id="cb38-243"><a href="#cb38-243"></a><span class="co">#| output: false</span></span>
<span id="cb38-244"><a href="#cb38-244"></a><span class="co">#| warning: false</span></span>
<span id="cb38-245"><a href="#cb38-245"></a><span class="co">#| error: false</span></span>
<span id="cb38-246"><a href="#cb38-246"></a><span class="fu">library</span>(compositions)</span>
<span id="cb38-247"><a href="#cb38-247"></a><span class="fu">library</span>(zCompositions)</span>
<span id="cb38-248"><a href="#cb38-248"></a><span class="fu">library</span>(vegan)</span>
<span id="cb38-249"><a href="#cb38-249"></a><span class="fu">library</span>(ape)</span>
<span id="cb38-250"><a href="#cb38-250"></a><span class="fu">library</span>(corrr)</span>
<span id="cb38-251"><a href="#cb38-251"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb38-252"><a href="#cb38-252"></a><span class="in">```</span></span>
<span id="cb38-253"><a href="#cb38-253"></a></span>
<span id="cb38-254"><a href="#cb38-254"></a><span class="fu">## Transform data</span></span>
<span id="cb38-255"><a href="#cb38-255"></a></span>
<span id="cb38-256"><a href="#cb38-256"></a>zCompositions has problems with species with <span class="sc">\&lt;</span> 2 observations so we need to filter these out</span>
<span id="cb38-257"><a href="#cb38-257"></a></span>
<span id="cb38-260"><a href="#cb38-260"></a><span class="in">```{r}</span></span>
<span id="cb38-261"><a href="#cb38-261"></a><span class="co">#| output: false</span></span>
<span id="cb38-262"><a href="#cb38-262"></a><span class="co">#| warning: false</span></span>
<span id="cb38-263"><a href="#cb38-263"></a><span class="co">#| error: false</span></span>
<span id="cb38-264"><a href="#cb38-264"></a>lowstrains <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb38-265"><a href="#cb38-265"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(condition_prey_pred, <span class="st">"inoculum"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-266"><a href="#cb38-266"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-267"><a href="#cb38-267"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb38-268"><a href="#cb38-268"></a>  <span class="fu">summarize</span>(<span class="at">n_samples =</span> <span class="fu">n</span>(),</span>
<span id="cb38-269"><a href="#cb38-269"></a>            <span class="at">n_gt0 =</span> <span class="fu">sum</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb38-270"><a href="#cb38-270"></a>            <span class="at">p_gt0 =</span> n_gt0 <span class="sc">/</span> n_samples) <span class="sc">%&gt;%</span> </span>
<span id="cb38-271"><a href="#cb38-271"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb38-272"><a href="#cb38-272"></a><span class="in">```</span></span>
<span id="cb38-273"><a href="#cb38-273"></a></span>
<span id="cb38-274"><a href="#cb38-274"></a>Here we remove strains present in <span class="sc">\&lt;</span> 50 samples across transfer categories and present in <span class="sc">\&lt;</span> 20 samples in at least 2/3 transfer categories</span>
<span id="cb38-275"><a href="#cb38-275"></a></span>
<span id="cb38-278"><a href="#cb38-278"></a><span class="in">```{r}</span></span>
<span id="cb38-279"><a href="#cb38-279"></a>lowstrainsv <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb38-280"><a href="#cb38-280"></a>  <span class="st">"HAMBI_0097"</span>,</span>
<span id="cb38-281"><a href="#cb38-281"></a>  <span class="st">"HAMBI_0262"</span>,</span>
<span id="cb38-282"><a href="#cb38-282"></a>  <span class="st">"HAMBI_1842"</span>,</span>
<span id="cb38-283"><a href="#cb38-283"></a>  <span class="st">"HAMBI_1988"</span>,</span>
<span id="cb38-284"><a href="#cb38-284"></a>  <span class="st">"HAMBI_2443"</span>,</span>
<span id="cb38-285"><a href="#cb38-285"></a>  <span class="st">"HAMBI_2792"</span>,</span>
<span id="cb38-286"><a href="#cb38-286"></a>  <span class="st">"HAMBI_3031"</span>,</span>
<span id="cb38-287"><a href="#cb38-287"></a>  <span class="st">"HAMBI_3237"</span></span>
<span id="cb38-288"><a href="#cb38-288"></a>)</span>
<span id="cb38-289"><a href="#cb38-289"></a><span class="in">```</span></span>
<span id="cb38-290"><a href="#cb38-290"></a></span>
<span id="cb38-291"><a href="#cb38-291"></a>transform to matrix</span>
<span id="cb38-292"><a href="#cb38-292"></a></span>
<span id="cb38-295"><a href="#cb38-295"></a><span class="in">```{r}</span></span>
<span id="cb38-296"><a href="#cb38-296"></a>mymat <span class="ot">&lt;-</span> counts_f <span class="sc">%&gt;%</span> </span>
<span id="cb38-297"><a href="#cb38-297"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(count)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-298"><a href="#cb38-298"></a>  <span class="fu">filter</span>(strainID <span class="sc">%nin%</span> lowstrainsv) <span class="sc">%&gt;%</span> </span>
<span id="cb38-299"><a href="#cb38-299"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(condition_prey_pred, <span class="st">"inoculum"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-300"><a href="#cb38-300"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count) <span class="sc">%&gt;%</span> </span>
<span id="cb38-301"><a href="#cb38-301"></a>  <span class="co"># important to arrange by sample as this makes some later joins easier</span></span>
<span id="cb38-302"><a href="#cb38-302"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb38-303"><a href="#cb38-303"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"strainID"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-304"><a href="#cb38-304"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-305"><a href="#cb38-305"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb38-306"><a href="#cb38-306"></a><span class="in">```</span></span>
<span id="cb38-307"><a href="#cb38-307"></a></span>
<span id="cb38-308"><a href="#cb38-308"></a><span class="fu">## Replace zeros</span></span>
<span id="cb38-309"><a href="#cb38-309"></a></span>
<span id="cb38-310"><a href="#cb38-310"></a>Compositional analysis with the centered log-ratio can't handle zero values. Some people just replace them with a pseudocount. Another way is to impute them based on various different strategies.</span>
<span id="cb38-311"><a href="#cb38-311"></a></span>
<span id="cb38-312"><a href="#cb38-312"></a>Literature:</span>
<span id="cb38-313"><a href="#cb38-313"></a></span>
<span id="cb38-314"><a href="#cb38-314"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">A field guide for the compositional analysis of any-omics data</span><span class="co">](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6755255/)</span></span>
<span id="cb38-315"><a href="#cb38-315"></a><span class="ss">    -   </span><span class="co">[</span><span class="ot">Supplemental material</span><span class="co">](https://zenodo.org/record/3270954#.Y1KastJBxhE)</span></span>
<span id="cb38-316"><a href="#cb38-316"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">zCompositions --- R package for multivariate imputation of left-censored data under a compositional approach</span><span class="co">](https://doi.org/10.1016/j.chemolab.2015.02.019)</span></span>
<span id="cb38-317"><a href="#cb38-317"></a></span>
<span id="cb38-318"><a href="#cb38-318"></a>Here we will uses a Geometric Bayesian-multiplicative replacement strategy that preserves the ratios between the non-zero components. The "prop" option returns relative abundances.</span>
<span id="cb38-319"><a href="#cb38-319"></a></span>
<span id="cb38-322"><a href="#cb38-322"></a><span class="in">```{r}</span></span>
<span id="cb38-323"><a href="#cb38-323"></a><span class="fu">set.seed</span>(<span class="dv">12378</span>)</span>
<span id="cb38-324"><a href="#cb38-324"></a>comp <span class="ot">&lt;-</span> zCompositions<span class="sc">::</span><span class="fu">cmultRepl</span>(mymat, <span class="at">method =</span> <span class="st">"GBM"</span>, <span class="at">output =</span> <span class="st">"prop"</span>)</span>
<span id="cb38-325"><a href="#cb38-325"></a><span class="in">```</span></span>
<span id="cb38-326"><a href="#cb38-326"></a></span>
<span id="cb38-327"><a href="#cb38-327"></a><span class="fu">## Calculate Bray-curtis dissimilarity</span></span>
<span id="cb38-328"><a href="#cb38-328"></a></span>
<span id="cb38-331"><a href="#cb38-331"></a><span class="in">```{r}</span></span>
<span id="cb38-332"><a href="#cb38-332"></a><span class="fu">set.seed</span>(<span class="dv">12353</span>)</span>
<span id="cb38-333"><a href="#cb38-333"></a>bray_dist      <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(comp, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb38-334"><a href="#cb38-334"></a>pcoa_ord_bray  <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">pcoa</span>(bray_dist)</span>
<span id="cb38-335"><a href="#cb38-335"></a><span class="in">```</span></span>
<span id="cb38-336"><a href="#cb38-336"></a></span>
<span id="cb38-337"><a href="#cb38-337"></a><span class="fu">## Calculate with Aitchison distance</span></span>
<span id="cb38-338"><a href="#cb38-338"></a></span>
<span id="cb38-339"><a href="#cb38-339"></a>Aitchison distance is the Euclidean distance of the centered log-ratio transform (clr). This distance (unlike Euclidean distance on read counts) has scale invariance, perturbation invariance, permutation invariance and sub-compositional dominance.</span>
<span id="cb38-340"><a href="#cb38-340"></a></span>
<span id="cb38-343"><a href="#cb38-343"></a><span class="in">```{r}</span></span>
<span id="cb38-344"><a href="#cb38-344"></a><span class="fu">set.seed</span>(<span class="dv">12354</span>)</span>
<span id="cb38-345"><a href="#cb38-345"></a>balclr     <span class="ot">&lt;-</span> compositions<span class="sc">::</span><span class="fu">clr</span>(comp)</span>
<span id="cb38-346"><a href="#cb38-346"></a>aitc_dist  <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(balclr, <span class="at">method =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb38-347"><a href="#cb38-347"></a><span class="in">```</span></span>
<span id="cb38-348"><a href="#cb38-348"></a></span>
<span id="cb38-349"><a href="#cb38-349"></a><span class="fu">## Compare Aitchison distance with CLR</span></span>
<span id="cb38-350"><a href="#cb38-350"></a></span>
<span id="cb38-351"><a href="#cb38-351"></a>When the Aitchison distance is used in Principle co-ordinate Analysis (PCoA) it is equivalent to standard Principle Component Analyis (PCA) on the clr transformed data</span>
<span id="cb38-352"><a href="#cb38-352"></a></span>
<span id="cb38-355"><a href="#cb38-355"></a><span class="in">```{r}</span></span>
<span id="cb38-356"><a href="#cb38-356"></a><span class="fu">set.seed</span>(<span class="dv">12355</span>)</span>
<span id="cb38-357"><a href="#cb38-357"></a>pcoa_ord_aitc  <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">pcoa</span>(aitc_dist)</span>
<span id="cb38-358"><a href="#cb38-358"></a>pca_ord_aitc   <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(balclr)</span>
<span id="cb38-359"><a href="#cb38-359"></a><span class="in">```</span></span>
<span id="cb38-360"><a href="#cb38-360"></a></span>
<span id="cb38-361"><a href="#cb38-361"></a>For example, these ordinations are the same, just that Axis2 is the mirrorimage between. Since the rotation is arbitrary this does not matter.</span>
<span id="cb38-362"><a href="#cb38-362"></a></span>
<span id="cb38-363"><a href="#cb38-363"></a>::: {#fig-03}</span>
<span id="cb38-366"><a href="#cb38-366"></a><span class="in">```{r}</span></span>
<span id="cb38-367"><a href="#cb38-367"></a><span class="co">#| warning: false</span></span>
<span id="cb38-368"><a href="#cb38-368"></a><span class="co">#| error: false</span></span>
<span id="cb38-369"><a href="#cb38-369"></a><span class="co">#| fig.width: 6</span></span>
<span id="cb38-370"><a href="#cb38-370"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb38-371"><a href="#cb38-371"></a><span class="fu">plot</span>(pcoa_ord_aitc<span class="sc">$</span>vectors)</span>
<span id="cb38-372"><a href="#cb38-372"></a><span class="in">```</span></span>
<span id="cb38-373"><a href="#cb38-373"></a></span>
<span id="cb38-374"><a href="#cb38-374"></a>Plot of Principal Coordinate Analysis (PCoA) done in the Aitchison geometry of the simplex (i.e. PCoA on the euclidean distance of the centered log-ratio transform species compositions). </span>
<span id="cb38-375"><a href="#cb38-375"></a>:::</span>
<span id="cb38-376"><a href="#cb38-376"></a></span>
<span id="cb38-377"><a href="#cb38-377"></a>::: {#fig-04}</span>
<span id="cb38-380"><a href="#cb38-380"></a><span class="in">```{r}</span></span>
<span id="cb38-381"><a href="#cb38-381"></a><span class="co">#| warning: false</span></span>
<span id="cb38-382"><a href="#cb38-382"></a><span class="co">#| error: false</span></span>
<span id="cb38-383"><a href="#cb38-383"></a><span class="co">#| fig.width: 6</span></span>
<span id="cb38-384"><a href="#cb38-384"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb38-385"><a href="#cb38-385"></a><span class="fu">plot</span>(pca_ord_aitc<span class="sc">$</span>x)</span>
<span id="cb38-386"><a href="#cb38-386"></a><span class="in">```</span></span>
<span id="cb38-387"><a href="#cb38-387"></a>Plot of Principal Component Analysis (PCA) done in the Aitchison geometry of the simplex (i.e. PCA on the centered log-ratio transform species compositions). Notice that the PCoA using the euclidean distance of centered log-ratio transformed species frequencies is equivalent to the PCA directly using clr-transformed values in @fig-03.</span>
<span id="cb38-388"><a href="#cb38-388"></a>:::</span>
<span id="cb38-389"><a href="#cb38-389"></a></span>
<span id="cb38-390"><a href="#cb38-390"></a><span class="fu">## Environment vectors</span></span>
<span id="cb38-391"><a href="#cb38-391"></a></span>
<span id="cb38-392"><a href="#cb38-392"></a>left_join with metadata</span>
<span id="cb38-393"><a href="#cb38-393"></a></span>
<span id="cb38-396"><a href="#cb38-396"></a><span class="in">```{r}</span></span>
<span id="cb38-397"><a href="#cb38-397"></a><span class="co">#| output: false</span></span>
<span id="cb38-398"><a href="#cb38-398"></a><span class="co">#| warning: false</span></span>
<span id="cb38-399"><a href="#cb38-399"></a><span class="co">#| error: false</span></span>
<span id="cb38-400"><a href="#cb38-400"></a>pca2plot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pca_ord_aitc<span class="sc">$</span>x) <span class="sc">%&gt;%</span> </span>
<span id="cb38-401"><a href="#cb38-401"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-402"><a href="#cb38-402"></a>  <span class="fu">left_join</span>(metadf) <span class="sc">%&gt;%</span> </span>
<span id="cb38-403"><a href="#cb38-403"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb38-404"><a href="#cb38-404"></a>  <span class="fu">mutate</span>(<span class="at">expcombo=</span><span class="fu">interaction</span>(condition_prey_pred, time_days, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb38-405"><a href="#cb38-405"></a><span class="in">```</span></span>
<span id="cb38-406"><a href="#cb38-406"></a></span>
<span id="cb38-407"><a href="#cb38-407"></a>About 85% of variance explained in first 5 PCs</span>
<span id="cb38-408"><a href="#cb38-408"></a></span>
<span id="cb38-411"><a href="#cb38-411"></a><span class="in">```{r}</span></span>
<span id="cb38-412"><a href="#cb38-412"></a>pca_ord_aitc_importance <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">summary</span>(pca_ord_aitc)<span class="sc">$</span>importance) <span class="sc">%&gt;%</span> </span>
<span id="cb38-413"><a href="#cb38-413"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"parameter"</span>)</span>
<span id="cb38-414"><a href="#cb38-414"></a></span>
<span id="cb38-415"><a href="#cb38-415"></a>pca_ord_aitc_importance</span>
<span id="cb38-416"><a href="#cb38-416"></a><span class="in">```</span></span>
<span id="cb38-417"><a href="#cb38-417"></a></span>
<span id="cb38-418"><a href="#cb38-418"></a>Environmental/experimental variables associated with ordinatoion</span>
<span id="cb38-419"><a href="#cb38-419"></a></span>
<span id="cb38-422"><a href="#cb38-422"></a><span class="in">```{r}</span></span>
<span id="cb38-423"><a href="#cb38-423"></a><span class="co">#| warning: false</span></span>
<span id="cb38-424"><a href="#cb38-424"></a><span class="co">#| error: false</span></span>
<span id="cb38-425"><a href="#cb38-425"></a>ef <span class="ot">&lt;-</span></span>
<span id="cb38-426"><a href="#cb38-426"></a>  <span class="fu">envfit</span>(</span>
<span id="cb38-427"><a href="#cb38-427"></a>    pca_ord_aitc <span class="sc">~</span> prey_history <span class="sc">+</span> predator_history <span class="sc">+</span> ciliates_ml <span class="sc">+</span> time_days,</span>
<span id="cb38-428"><a href="#cb38-428"></a>    <span class="at">data =</span> <span class="fu">semi_join</span>(metadf, dplyr<span class="sc">::</span><span class="fu">select</span>(pca2plot, sample)),</span>
<span id="cb38-429"><a href="#cb38-429"></a>    <span class="at">na.rm =</span> T,</span>
<span id="cb38-430"><a href="#cb38-430"></a>    <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)</span>
<span id="cb38-431"><a href="#cb38-431"></a>  )</span>
<span id="cb38-432"><a href="#cb38-432"></a><span class="in">```</span></span>
<span id="cb38-433"><a href="#cb38-433"></a></span>
<span id="cb38-434"><a href="#cb38-434"></a>::: {#fig-05}</span>
<span id="cb38-437"><a href="#cb38-437"></a><span class="in">```{r}</span></span>
<span id="cb38-438"><a href="#cb38-438"></a><span class="co">#| warning: false</span></span>
<span id="cb38-439"><a href="#cb38-439"></a><span class="co">#| error: false</span></span>
<span id="cb38-440"><a href="#cb38-440"></a><span class="co">#| fig.width: 6</span></span>
<span id="cb38-441"><a href="#cb38-441"></a><span class="co">#| fig.height: 6</span></span>
<span id="cb38-442"><a href="#cb38-442"></a><span class="fu">biplot</span>(pca_ord_aitc, <span class="at">choices=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">labSize=</span><span class="dv">0</span>)</span>
<span id="cb38-443"><a href="#cb38-443"></a><span class="fu">plot</span>(ef)</span>
<span id="cb38-444"><a href="#cb38-444"></a><span class="in">```</span></span>
<span id="cb38-445"><a href="#cb38-445"></a>Environmental vectors plotted onto ordination in @fig-04</span>
<span id="cb38-446"><a href="#cb38-446"></a>:::</span>
<span id="cb38-447"><a href="#cb38-447"></a></span>
<span id="cb38-448"><a href="#cb38-448"></a><span class="fu">### Significance of the environmental covariates</span></span>
<span id="cb38-449"><a href="#cb38-449"></a></span>
<span id="cb38-452"><a href="#cb38-452"></a><span class="in">```{r}</span></span>
<span id="cb38-453"><a href="#cb38-453"></a>ef</span>
<span id="cb38-454"><a href="#cb38-454"></a><span class="in">```</span></span>
<span id="cb38-455"><a href="#cb38-455"></a></span>
<span id="cb38-458"><a href="#cb38-458"></a><span class="in">```{r}</span></span>
<span id="cb38-459"><a href="#cb38-459"></a>con_scrs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(ef, <span class="at">display =</span> <span class="st">"vectors"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-460"><a href="#cb38-460"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"var"</span>) <span class="sc">%&gt;%</span> <span class="fu">tibble</span>()</span>
<span id="cb38-461"><a href="#cb38-461"></a></span>
<span id="cb38-462"><a href="#cb38-462"></a>fct_scrs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(ef, <span class="at">display =</span> <span class="st">"factors"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-463"><a href="#cb38-463"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"var"</span>) <span class="sc">%&gt;%</span>  <span class="fu">tibble</span>()</span>
<span id="cb38-464"><a href="#cb38-464"></a></span>
<span id="cb38-465"><a href="#cb38-465"></a>scale_factor <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb38-466"><a href="#cb38-466"></a><span class="in">```</span></span>
<span id="cb38-467"><a href="#cb38-467"></a></span>
<span id="cb38-468"><a href="#cb38-468"></a><span class="fu">### Plot with prey history highlighted</span></span>
<span id="cb38-469"><a href="#cb38-469"></a></span>
<span id="cb38-472"><a href="#cb38-472"></a><span class="in">```{r}</span></span>
<span id="cb38-473"><a href="#cb38-473"></a>ppca <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca2plot) <span class="sc">+</span></span>
<span id="cb38-474"><a href="#cb38-474"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb38-475"><a href="#cb38-475"></a>    <span class="at">x =</span> PC1,</span>
<span id="cb38-476"><a href="#cb38-476"></a>    <span class="at">y =</span> PC2,</span>
<span id="cb38-477"><a href="#cb38-477"></a>    <span class="at">color =</span> prey_history,</span>
<span id="cb38-478"><a href="#cb38-478"></a>    <span class="at">shape =</span> predator_history), <span class="at">size=</span><span class="dv">3</span> ) <span class="sc">+</span></span>
<span id="cb38-479"><a href="#cb38-479"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> con_scrs,</span>
<span id="cb38-480"><a href="#cb38-480"></a>               <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> PC2<span class="sc">*</span>scale_factor),</span>
<span id="cb38-481"><a href="#cb38-481"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">"cm"</span>)), <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb38-482"><a href="#cb38-482"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> con_scrs, <span class="fu">aes</span>(<span class="at">x =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> PC2<span class="sc">*</span>scale_factor, <span class="at">label =</span> var),</span>
<span id="cb38-483"><a href="#cb38-483"></a>                  <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb38-484"><a href="#cb38-484"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>), </span>
<span id="cb38-485"><a href="#cb38-485"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">3</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb38-486"><a href="#cb38-486"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> prey_history)) <span class="sc">+</span> </span>
<span id="cb38-487"><a href="#cb38-487"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb38-488"><a href="#cb38-488"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#88CCEE"</span>, <span class="st">"#CC6677"</span>)) <span class="sc">+</span></span>
<span id="cb38-489"><a href="#cb38-489"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb38-490"><a href="#cb38-490"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb38-491"><a href="#cb38-491"></a>  <span class="fu">theme</span>(</span>
<span id="cb38-492"><a href="#cb38-492"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-493"><a href="#cb38-493"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-494"><a href="#cb38-494"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-495"><a href="#cb38-495"></a>  )</span>
<span id="cb38-496"><a href="#cb38-496"></a><span class="in">```</span></span>
<span id="cb38-497"><a href="#cb38-497"></a></span>
<span id="cb38-498"><a href="#cb38-498"></a>::: {#fig-06}</span>
<span id="cb38-501"><a href="#cb38-501"></a><span class="in">```{r}</span></span>
<span id="cb38-502"><a href="#cb38-502"></a><span class="co">#| echo: false</span></span>
<span id="cb38-503"><a href="#cb38-503"></a>ppca</span>
<span id="cb38-504"><a href="#cb38-504"></a><span class="in">```</span></span>
<span id="cb38-505"><a href="#cb38-505"></a>Plot of principal component analysis (PCA) done in the Aitchison geometry of the simplex (i.e. centered log-ratio transform species compositions). Color shows evolutionary history of the prey and shape shows evolutionary history of the predator. Arrows and ellipses depict statistically significant (P &lt; 0.05) experimental variables (continuous and categorical, resp) fit to the ordination axes via regression (<span class="in">`vegan::envfit`</span>) and projected onto the ordination plot. Significance is assessed by permutation.</span>
<span id="cb38-506"><a href="#cb38-506"></a>:::</span>
<span id="cb38-507"><a href="#cb38-507"></a></span>
<span id="cb38-510"><a href="#cb38-510"></a><span class="in">```{r}</span></span>
<span id="cb38-511"><a href="#cb38-511"></a><span class="co">#| echo: false</span></span>
<span id="cb38-512"><a href="#cb38-512"></a><span class="fu">ggsave</span>(</span>
<span id="cb38-513"><a href="#cb38-513"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"ordination_final.png"</span>),</span>
<span id="cb38-514"><a href="#cb38-514"></a>  ppca,</span>
<span id="cb38-515"><a href="#cb38-515"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb38-516"><a href="#cb38-516"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb38-517"><a href="#cb38-517"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb38-518"><a href="#cb38-518"></a>  <span class="at">device =</span> <span class="st">"png"</span>,</span>
<span id="cb38-519"><a href="#cb38-519"></a>  <span class="at">dpi =</span> <span class="dv">320</span></span>
<span id="cb38-520"><a href="#cb38-520"></a>)</span>
<span id="cb38-521"><a href="#cb38-521"></a></span>
<span id="cb38-522"><a href="#cb38-522"></a><span class="fu">ggsave</span>(</span>
<span id="cb38-523"><a href="#cb38-523"></a>  here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"ordination_final.pdf"</span>),</span>
<span id="cb38-524"><a href="#cb38-524"></a>  ppca,</span>
<span id="cb38-525"><a href="#cb38-525"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb38-526"><a href="#cb38-526"></a>  <span class="at">height =</span> <span class="dv">6</span>,</span>
<span id="cb38-527"><a href="#cb38-527"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb38-528"><a href="#cb38-528"></a>  <span class="at">device =</span> <span class="st">"pdf"</span></span>
<span id="cb38-529"><a href="#cb38-529"></a>)</span>
<span id="cb38-530"><a href="#cb38-530"></a><span class="in">```</span></span>
<span id="cb38-531"><a href="#cb38-531"></a></span>
<span id="cb38-532"><a href="#cb38-532"></a><span class="fu">### Plot with predator history highlighted</span></span>
<span id="cb38-533"><a href="#cb38-533"></a></span>
<span id="cb38-534"><a href="#cb38-534"></a>::: {#fig-07}</span>
<span id="cb38-537"><a href="#cb38-537"></a><span class="in">```{r}</span></span>
<span id="cb38-538"><a href="#cb38-538"></a><span class="fu">ggplot</span>(pca2plot) <span class="sc">+</span></span>
<span id="cb38-539"><a href="#cb38-539"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(</span>
<span id="cb38-540"><a href="#cb38-540"></a>    <span class="at">x =</span> PC1,</span>
<span id="cb38-541"><a href="#cb38-541"></a>    <span class="at">y =</span> PC2,</span>
<span id="cb38-542"><a href="#cb38-542"></a>    <span class="at">color =</span> predator_history,</span>
<span id="cb38-543"><a href="#cb38-543"></a>    <span class="at">shape =</span> prey_history), <span class="at">size=</span><span class="dv">3</span> ) <span class="sc">+</span></span>
<span id="cb38-544"><a href="#cb38-544"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> con_scrs,</span>
<span id="cb38-545"><a href="#cb38-545"></a>               <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> PC2<span class="sc">*</span>scale_factor),</span>
<span id="cb38-546"><a href="#cb38-546"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.25</span>, <span class="st">"cm"</span>)), <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb38-547"><a href="#cb38-547"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> con_scrs, <span class="fu">aes</span>(<span class="at">x =</span> PC1<span class="sc">*</span>scale_factor, <span class="at">y =</span> PC2<span class="sc">*</span>scale_factor, <span class="at">label =</span> var),</span>
<span id="cb38-548"><a href="#cb38-548"></a>                  <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb38-549"><a href="#cb38-549"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>), </span>
<span id="cb38-550"><a href="#cb38-550"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, <span class="fu">round</span>(pca_ord_aitc_importance[<span class="dv">2</span>,<span class="dv">3</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>),<span class="st">"%)"</span>)) <span class="sc">+</span></span>
<span id="cb38-551"><a href="#cb38-551"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> predator_history)) <span class="sc">+</span> </span>
<span id="cb38-552"><a href="#cb38-552"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb38-553"><a href="#cb38-553"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb38-554"><a href="#cb38-554"></a>  <span class="fu">theme</span>(</span>
<span id="cb38-555"><a href="#cb38-555"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-556"><a href="#cb38-556"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-557"><a href="#cb38-557"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-558"><a href="#cb38-558"></a>  )</span>
<span id="cb38-559"><a href="#cb38-559"></a></span>
<span id="cb38-560"><a href="#cb38-560"></a><span class="in">```</span></span>
<span id="cb38-561"><a href="#cb38-561"></a>As in @Fig-06 but with predator evolutionary history highlighted in color.</span>
<span id="cb38-562"><a href="#cb38-562"></a>:::</span>
<span id="cb38-563"><a href="#cb38-563"></a></span>
<span id="cb38-564"><a href="#cb38-564"></a>The evolved predator seems to "tighten" up the variation and make samplles with ancestral prey "look" more like the samples with evolved prey</span>
<span id="cb38-565"><a href="#cb38-565"></a></span>
<span id="cb38-566"><a href="#cb38-566"></a><span class="fu">### PERMANOVA</span></span>
<span id="cb38-567"><a href="#cb38-567"></a></span>
<span id="cb38-568"><a href="#cb38-568"></a>PERMANOVA suggests significant effect of predator and prey evolutionary history</span>
<span id="cb38-569"><a href="#cb38-569"></a></span>
<span id="cb38-572"><a href="#cb38-572"></a><span class="in">```{r}</span></span>
<span id="cb38-573"><a href="#cb38-573"></a><span class="fu">set.seed</span>(<span class="dv">45781</span>)</span>
<span id="cb38-574"><a href="#cb38-574"></a><span class="fu">adonis2</span>(aitc_dist <span class="sc">~</span> pca2plot<span class="sc">$</span>prey_history <span class="sc">+</span> pca2plot<span class="sc">$</span>predator_history,</span>
<span id="cb38-575"><a href="#cb38-575"></a>       <span class="at">permutations =</span> <span class="fl">1e3</span>)</span>
<span id="cb38-576"><a href="#cb38-576"></a><span class="in">```</span></span>
<span id="cb38-577"><a href="#cb38-577"></a></span>
<span id="cb38-578"><a href="#cb38-578"></a>Check for homogeneity of variance between these two categories. Suggests that they are different although maybe not by very much...</span>
<span id="cb38-579"><a href="#cb38-579"></a></span>
<span id="cb38-582"><a href="#cb38-582"></a><span class="in">```{r}</span></span>
<span id="cb38-583"><a href="#cb38-583"></a><span class="fu">set.seed</span>(<span class="dv">45782</span>)</span>
<span id="cb38-584"><a href="#cb38-584"></a>bdprey <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitc_dist, pca2plot<span class="sc">$</span>prey_history)</span>
<span id="cb38-585"><a href="#cb38-585"></a><span class="fu">permutest</span>(bdprey)</span>
<span id="cb38-586"><a href="#cb38-586"></a><span class="in">```</span></span>
<span id="cb38-587"><a href="#cb38-587"></a></span>
<span id="cb38-590"><a href="#cb38-590"></a><span class="in">```{r}</span></span>
<span id="cb38-591"><a href="#cb38-591"></a><span class="fu">set.seed</span>(<span class="dv">45783</span>)</span>
<span id="cb38-592"><a href="#cb38-592"></a>bdpred <span class="ot">&lt;-</span> <span class="fu">betadisper</span>(aitc_dist, pca2plot<span class="sc">$</span>predator_history)</span>
<span id="cb38-593"><a href="#cb38-593"></a><span class="fu">permutest</span>(bdpred)</span>
<span id="cb38-594"><a href="#cb38-594"></a><span class="in">```</span></span>
<span id="cb38-595"><a href="#cb38-595"></a></span>
<span id="cb38-596"><a href="#cb38-596"></a></span>
<span id="cb38-597"><a href="#cb38-597"></a><span class="fu">## Geometric analysis of temporal trajectories in PCA</span></span>
<span id="cb38-598"><a href="#cb38-598"></a></span>
<span id="cb38-599"><a href="#cb38-599"></a>Function to calculate euclidean distance between points</span>
<span id="cb38-602"><a href="#cb38-602"></a><span class="in">```{r}</span></span>
<span id="cb38-603"><a href="#cb38-603"></a>euclidean_dist <span class="ot">&lt;-</span> <span class="cf">function</span>(x1, x2, y1, y2){</span>
<span id="cb38-604"><a href="#cb38-604"></a>  <span class="fu">sqrt</span>((x1 <span class="sc">-</span> x2)<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> (y1 <span class="sc">-</span> y2)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb38-605"><a href="#cb38-605"></a>}</span>
<span id="cb38-606"><a href="#cb38-606"></a><span class="in">```</span></span>
<span id="cb38-607"><a href="#cb38-607"></a></span>
<span id="cb38-610"><a href="#cb38-610"></a><span class="in">```{r}</span></span>
<span id="cb38-611"><a href="#cb38-611"></a>eucdist <span class="ot">&lt;-</span> pca2plot <span class="sc">%&gt;%</span> </span>
<span id="cb38-612"><a href="#cb38-612"></a>  <span class="fu">filter</span>(prey_history <span class="sc">!=</span> <span class="st">"none"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-613"><a href="#cb38-613"></a>  <span class="fu">filter</span>(predator_history <span class="sc">!=</span> <span class="st">"nopredator"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-614"><a href="#cb38-614"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(PC1, PC2, replicate, prey_history, predator_history, time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb38-615"><a href="#cb38-615"></a>  <span class="fu">arrange</span>(replicate, prey_history, predator_history, time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb38-616"><a href="#cb38-616"></a>  <span class="fu">group_by</span>(replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb38-617"><a href="#cb38-617"></a>  <span class="fu">mutate</span>(<span class="at">d =</span> <span class="fu">euclidean_dist</span>(PC1, <span class="fu">lag</span>(PC1), PC2, <span class="fu">lag</span>(PC2)),</span>
<span id="cb38-618"><a href="#cb38-618"></a>         <span class="at">range =</span> <span class="fu">paste0</span>(<span class="fu">lag</span>(time_days),<span class="st">"-"</span>,time_days)) <span class="sc">%&gt;%</span></span>
<span id="cb38-619"><a href="#cb38-619"></a>  <span class="fu">mutate</span>(<span class="at">range =</span> forcats<span class="sc">::</span><span class="fu">fct_reorder</span>(range, <span class="fu">as.numeric</span>(<span class="fu">as.character</span>((time_days))))) <span class="sc">%&gt;%</span></span>
<span id="cb38-620"><a href="#cb38-620"></a>  <span class="fu">filter</span>(range <span class="sc">!=</span> <span class="st">"NA-4"</span>)</span>
<span id="cb38-621"><a href="#cb38-621"></a></span>
<span id="cb38-622"><a href="#cb38-622"></a><span class="fu">write_tsv</span>(eucdist, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"aitchison_pca_distances.tsv"</span>))</span>
<span id="cb38-623"><a href="#cb38-623"></a><span class="in">```</span></span>
<span id="cb38-624"><a href="#cb38-624"></a></span>
<span id="cb38-625"><a href="#cb38-625"></a>::: {#fig-08}</span>
<span id="cb38-628"><a href="#cb38-628"></a><span class="in">```{r}</span></span>
<span id="cb38-629"><a href="#cb38-629"></a><span class="fu">ggplot</span>(eucdist) <span class="sc">+</span></span>
<span id="cb38-630"><a href="#cb38-630"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> range, <span class="at">y =</span> d, <span class="at">color =</span> replicate, <span class="at">group=</span>replicate)) <span class="sc">+</span></span>
<span id="cb38-631"><a href="#cb38-631"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Time range (days)"</span>, <span class="at">y =</span> <span class="st">"Euclidean distance between days"</span>) <span class="sc">+</span></span>
<span id="cb38-632"><a href="#cb38-632"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb38-633"><a href="#cb38-633"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span></span>
<span id="cb38-634"><a href="#cb38-634"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb38-635"><a href="#cb38-635"></a>  <span class="fu">theme</span>(</span>
<span id="cb38-636"><a href="#cb38-636"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb38-637"><a href="#cb38-637"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb38-638"><a href="#cb38-638"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb38-639"><a href="#cb38-639"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb38-640"><a href="#cb38-640"></a><span class="in">```</span></span>
<span id="cb38-641"><a href="#cb38-641"></a>Geometric analysis of community trajectories based on PCA. Segment lengths between consecutive sampling days in the two PCA dimensions were calculated and plotted chronologically for each biological replicate in each treatment combination. As sampling intervals were of equal duration (except day 16 - 24), segment length represents relative speed of compositional change.</span>
<span id="cb38-642"><a href="#cb38-642"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>