<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-11-03">

<title>hambiDoubleEvo - Format metagenome variants</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">Metagenome sequencing</a></li><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">1. Read and format</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">hambiDoubleEvo</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/01_rpkm2tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/02_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Analyze and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Population WGS</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Metagenome sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/01_format_variants.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Variant plotting</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/04_mutation_dynamics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Mutational dynamics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a>
  <ul class="collapse">
  <li><a href="#untar-and-decompress" id="toc-untar-and-decompress" class="nav-link" data-scroll-target="#untar-and-decompress"><span class="header-section-number">1.1</span> Untar and decompress</a></li>
  </ul></li>
  <li><a href="#reading-and-small-formatting-of-data" id="toc-reading-and-small-formatting-of-data" class="nav-link" data-scroll-target="#reading-and-small-formatting-of-data"><span class="header-section-number">2</span> Reading and small formatting of data</a>
  <ul class="collapse">
  <li><a href="#coverage-data" id="toc-coverage-data" class="nav-link" data-scroll-target="#coverage-data"><span class="header-section-number">2.1</span> Coverage data</a></li>
  <li><a href="#variants" id="toc-variants" class="nav-link" data-scroll-target="#variants"><span class="header-section-number">2.2</span> Variants</a>
  <ul class="collapse">
  <li><a href="#snpeff-annotations" id="toc-snpeff-annotations" class="nav-link" data-scroll-target="#snpeff-annotations"><span class="header-section-number">2.2.1</span> SnpEff annotations</a></li>
  <li><a href="#allele-frequencies" id="toc-allele-frequencies" class="nav-link" data-scroll-target="#allele-frequencies"><span class="header-section-number">2.2.2</span> Allele frequencies</a></li>
  <li><a href="#clean-up-tmp-directories" id="toc-clean-up-tmp-directories" class="nav-link" data-scroll-target="#clean-up-tmp-directories"><span class="header-section-number">2.2.3</span> Clean up tmp directories</a></li>
  </ul></li>
  <li><a href="#mutations-from-wgs-analysis" id="toc-mutations-from-wgs-analysis" class="nav-link" data-scroll-target="#mutations-from-wgs-analysis"><span class="header-section-number">2.3</span> Mutations from WGS analysis</a></li>
  <li><a href="#annotations" id="toc-annotations" class="nav-link" data-scroll-target="#annotations"><span class="header-section-number">2.4</span> Annotations</a></li>
  </ul></li>
  <li><a href="#formatting-and-filtering" id="toc-formatting-and-filtering" class="nav-link" data-scroll-target="#formatting-and-filtering"><span class="header-section-number">3</span> Formatting and filtering</a>
  <ul class="collapse">
  <li><a href="#exclude-regions-with-mobile-elements-in-metagenomes" id="toc-exclude-regions-with-mobile-elements-in-metagenomes" class="nav-link" data-scroll-target="#exclude-regions-with-mobile-elements-in-metagenomes"><span class="header-section-number">3.1</span> Exclude regions with mobile elements in metagenomes</a></li>
  <li><a href="#filter-wgs-on-species-only-included-in-metag" id="toc-filter-wgs-on-species-only-included-in-metag" class="nav-link" data-scroll-target="#filter-wgs-on-species-only-included-in-metag"><span class="header-section-number">3.2</span> Filter WGS on species only included in metaG</a></li>
  <li><a href="#combine-wgs-and-metag" id="toc-combine-wgs-and-metag" class="nav-link" data-scroll-target="#combine-wgs-and-metag"><span class="header-section-number">3.3</span> Combine WGS and metaG</a></li>
  <li><a href="#expand-grid-for-variants" id="toc-expand-grid-for-variants" class="nav-link" data-scroll-target="#expand-grid-for-variants"><span class="header-section-number">3.4</span> Expand grid for variants</a></li>
  <li><a href="#format-metagenome-coverage-data" id="toc-format-metagenome-coverage-data" class="nav-link" data-scroll-target="#format-metagenome-coverage-data"><span class="header-section-number">3.5</span> Format metagenome coverage data</a></li>
  </ul></li>
  <li><a href="#combine-variants-into-complete-time-series" id="toc-combine-variants-into-complete-time-series" class="nav-link" data-scroll-target="#combine-variants-into-complete-time-series"><span class="header-section-number">4</span> Combine variants into complete time series</a></li>
  <li><a href="#export" id="toc-export" class="nav-link" data-scroll-target="#export"><span class="header-section-number">5</span> Export</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">Metagenome sequencing</a></li><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">1. Read and format</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Format metagenome variants</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 3, 2024</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook formats for downstream use the table and VCF output from mutect2 called on metagenome data. It does some filtering to remove variants that are likely spurious and connects variants to functional annotations and gene identities. Finally it joins the WGS variants formatted earlier as the initial sampling point so that we can make time series from 0, 8, 28, and 60 days.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Libraries and global variables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(withr)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(archive)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Set up some directories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>vcftar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"tables.tar.gz"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a>covtar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"coverage.tar.gz"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span>
<span id="cb2-8"><a href="#cb2-8"></a></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-10"><a href="#cb2-10"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-11"><a href="#cb2-11"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="untar-and-decompress" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="untar-and-decompress"><span class="header-section-number">1.1</span> Untar and decompress</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># untar directory containing variant tables </span></span>
<span id="cb3-2"><a href="#cb3-2"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb3-3"><a href="#cb3-3"></a>  vcftar,</span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb3-5"><a href="#cb3-5"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb3-6"><a href="#cb3-6"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb3-7"><a href="#cb3-7"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb3-8"><a href="#cb3-8"></a>)</span>
<span id="cb3-9"><a href="#cb3-9"></a></span>
<span id="cb3-10"><a href="#cb3-10"></a>vcfdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"tables"</span>)</span>
<span id="cb3-11"><a href="#cb3-11"></a></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co"># untar directory containing variant tables </span></span>
<span id="cb3-13"><a href="#cb3-13"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb3-14"><a href="#cb3-14"></a>  covtar,</span>
<span id="cb3-15"><a href="#cb3-15"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb3-16"><a href="#cb3-16"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb3-17"><a href="#cb3-17"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb3-18"><a href="#cb3-18"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb3-19"><a href="#cb3-19"></a>)</span>
<span id="cb3-20"><a href="#cb3-20"></a></span>
<span id="cb3-21"><a href="#cb3-21"></a>covdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"coverage"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="reading-and-small-formatting-of-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Reading and small formatting of data</h1>
<section id="coverage-data" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="coverage-data"><span class="header-section-number">2.1</span> Coverage data</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>covfiles <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="at">path =</span> covdir,</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="at">all =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="at">glob =</span> <span class="st">"*.coverage.tsv"</span>,</span>
<span id="cb4-7"><a href="#cb4-7"></a>  <span class="at">regexp =</span> <span class="cn">NULL</span>,</span>
<span id="cb4-8"><a href="#cb4-8"></a>  <span class="at">invert =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-9"><a href="#cb4-9"></a>  <span class="at">fail =</span> <span class="cn">TRUE</span></span>
<span id="cb4-10"><a href="#cb4-10"></a>)</span>
<span id="cb4-11"><a href="#cb4-11"></a></span>
<span id="cb4-12"><a href="#cb4-12"></a>covslurped <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(</span>
<span id="cb4-13"><a href="#cb4-13"></a>  covfiles,</span>
<span id="cb4-14"><a href="#cb4-14"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb4-15"><a href="#cb4-15"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(</span>
<span id="cb4-16"><a href="#cb4-16"></a>    <span class="st">"scaffold"</span>,</span>
<span id="cb4-17"><a href="#cb4-17"></a>    <span class="st">"mean"</span>,</span>
<span id="cb4-18"><a href="#cb4-18"></a>    <span class="st">"trimmed_mean"</span>,</span>
<span id="cb4-19"><a href="#cb4-19"></a>    <span class="st">"covered_fraction"</span>,</span>
<span id="cb4-20"><a href="#cb4-20"></a>    <span class="st">"covered_bases"</span>,</span>
<span id="cb4-21"><a href="#cb4-21"></a>    <span class="st">"variance"</span>,</span>
<span id="cb4-22"><a href="#cb4-22"></a>    <span class="st">"length"</span>,</span>
<span id="cb4-23"><a href="#cb4-23"></a>    <span class="st">"read_count"</span>,</span>
<span id="cb4-24"><a href="#cb4-24"></a>    <span class="st">"reads_per_base"</span></span>
<span id="cb4-25"><a href="#cb4-25"></a>  ),</span>
<span id="cb4-26"><a href="#cb4-26"></a>  <span class="at">col_types =</span> <span class="st">"cdddddddd"</span>,</span>
<span id="cb4-27"><a href="#cb4-27"></a>  <span class="at">id =</span> <span class="st">"file_name"</span></span>
<span id="cb4-28"><a href="#cb4-28"></a>)</span>
<span id="cb4-29"><a href="#cb4-29"></a></span>
<span id="cb4-30"><a href="#cb4-30"></a>covslurpedfmt <span class="ot">&lt;-</span> covslurped <span class="sc">%&gt;%</span></span>
<span id="cb4-31"><a href="#cb4-31"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_extract</span>(file_name, <span class="st">"SH-MET-[:digit:]{3}"</span>),</span>
<span id="cb4-32"><a href="#cb4-32"></a>         <span class="at">strainID   =</span> <span class="fu">str_extract</span>(file_name, <span class="st">"HAMBI_[:digit:]{4}"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-33"><a href="#cb4-33"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>file_name) <span class="sc">%&gt;%</span></span>
<span id="cb4-34"><a href="#cb4-34"></a>  <span class="fu">relocate</span>(strainID, sample)</span>
<span id="cb4-35"><a href="#cb4-35"></a></span>
<span id="cb4-36"><a href="#cb4-36"></a>covslurpedfmtflt <span class="ot">&lt;-</span> covslurpedfmt <span class="sc">%&gt;%</span></span>
<span id="cb4-37"><a href="#cb4-37"></a>  <span class="fu">mutate</span>(<span class="at">bamname =</span> <span class="fu">paste0</span>(strainID,<span class="st">"-"</span>,sample)) <span class="sc">%&gt;%</span></span>
<span id="cb4-38"><a href="#cb4-38"></a>  <span class="co"># exclude species with trimmed mean (Dt) &lt; 5</span></span>
<span id="cb4-39"><a href="#cb4-39"></a>  <span class="fu">mutate</span>(<span class="at">FILTER =</span> <span class="fu">if_else</span>(trimmed_mean <span class="sc">&gt;=</span> <span class="dv">5</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb4-40"><a href="#cb4-40"></a></span>
<span id="cb4-41"><a href="#cb4-41"></a><span class="fu">write_tsv</span>(covslurpedfmtflt, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"coverage.tsv"</span>))</span>
<span id="cb4-42"><a href="#cb4-42"></a></span>
<span id="cb4-43"><a href="#cb4-43"></a><span class="co"># cleanup</span></span>
<span id="cb4-44"><a href="#cb4-44"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(covdir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>covslurpedfmtflt <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"coverage.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="variants" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="variants"><span class="header-section-number">2.2</span> Variants</h2>
<section id="snpeff-annotations" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" class="anchored" data-anchor-id="snpeff-annotations"><span class="header-section-number">2.2.1</span> SnpEff annotations</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>snpefffiles <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="at">path =</span> vcfdir,</span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="at">all =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-5"><a href="#cb6-5"></a>  <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="at">glob =</span> <span class="st">"*.snpeff.tsv"</span>,</span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="at">regexp =</span> <span class="cn">NULL</span>,</span>
<span id="cb6-8"><a href="#cb6-8"></a>  <span class="at">invert =</span> <span class="cn">FALSE</span>,</span>
<span id="cb6-9"><a href="#cb6-9"></a>  <span class="at">fail =</span> <span class="cn">TRUE</span></span>
<span id="cb6-10"><a href="#cb6-10"></a>)</span>
<span id="cb6-11"><a href="#cb6-11"></a></span>
<span id="cb6-12"><a href="#cb6-12"></a>snpeffslurped <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(</span>
<span id="cb6-13"><a href="#cb6-13"></a>  snpefffiles,</span>
<span id="cb6-14"><a href="#cb6-14"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb6-15"><a href="#cb6-15"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(</span>
<span id="cb6-16"><a href="#cb6-16"></a>    <span class="st">"chrom"</span>,</span>
<span id="cb6-17"><a href="#cb6-17"></a>    <span class="st">"pos"</span>,</span>
<span id="cb6-18"><a href="#cb6-18"></a>    <span class="st">"ref"</span>,</span>
<span id="cb6-19"><a href="#cb6-19"></a>    <span class="st">"alt"</span>,</span>
<span id="cb6-20"><a href="#cb6-20"></a>    <span class="st">"filter"</span>,</span>
<span id="cb6-21"><a href="#cb6-21"></a>    <span class="co"># Annotated using Sequence Ontology terms. Multiple effects can be concatenated using '&amp;'.</span></span>
<span id="cb6-22"><a href="#cb6-22"></a>    <span class="st">"effect"</span>,</span>
<span id="cb6-23"><a href="#cb6-23"></a>    <span class="co"># A simple estimation of putative impact / deleteriousness : HIGH, MODERATE, LOW, or MODIFIER</span></span>
<span id="cb6-24"><a href="#cb6-24"></a>    <span class="st">"impact"</span>,</span>
<span id="cb6-25"><a href="#cb6-25"></a>    <span class="co"># gene id from prokka</span></span>
<span id="cb6-26"><a href="#cb6-26"></a>    <span class="st">"locus_tag"</span>,</span>
<span id="cb6-27"><a href="#cb6-27"></a>    <span class="co"># whether variant is coding or noncoding</span></span>
<span id="cb6-28"><a href="#cb6-28"></a>    <span class="st">"biotype"</span>,</span>
<span id="cb6-29"><a href="#cb6-29"></a>    <span class="co"># Variant using HGVS notation (DNA level)</span></span>
<span id="cb6-30"><a href="#cb6-30"></a>    <span class="st">"hgvs_c"</span>,</span>
<span id="cb6-31"><a href="#cb6-31"></a>    <span class="co"># Variant using HGVS notation (Protein level).</span></span>
<span id="cb6-32"><a href="#cb6-32"></a>    <span class="st">"hgvs_p"</span>,</span>
<span id="cb6-33"><a href="#cb6-33"></a>    <span class="co"># Position in coding bases (one based includes START and STOP codons).</span></span>
<span id="cb6-34"><a href="#cb6-34"></a>    <span class="st">"cds_pos"</span>,</span>
<span id="cb6-35"><a href="#cb6-35"></a>    <span class="co"># Total number of coding bases (one based includes START and STOP codons).</span></span>
<span id="cb6-36"><a href="#cb6-36"></a>    <span class="st">"cds_len"</span>,</span>
<span id="cb6-37"><a href="#cb6-37"></a>    <span class="co"># Position in translated product (one based includes START and STOP codons).</span></span>
<span id="cb6-38"><a href="#cb6-38"></a>    <span class="st">"aa_pos"</span>,</span>
<span id="cb6-39"><a href="#cb6-39"></a>    <span class="co"># Total length of translated product (one based includes START and STOP codons).</span></span>
<span id="cb6-40"><a href="#cb6-40"></a>    <span class="st">"aa_len"</span></span>
<span id="cb6-41"><a href="#cb6-41"></a>  ),</span>
<span id="cb6-42"><a href="#cb6-42"></a>  <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">"cdcccccccccdddd"</span>),</span>
<span id="cb6-43"><a href="#cb6-43"></a>  <span class="at">id =</span> <span class="st">"sample"</span></span>
<span id="cb6-44"><a href="#cb6-44"></a>)</span>
<span id="cb6-45"><a href="#cb6-45"></a></span>
<span id="cb6-46"><a href="#cb6-46"></a>snpeffslurpedfmt <span class="ot">&lt;-</span> snpeffslurped <span class="sc">%&gt;%</span> </span>
<span id="cb6-47"><a href="#cb6-47"></a>  <span class="co"># this is weird, but joining on column like filter that contains characters like ';' doesnt work</span></span>
<span id="cb6-48"><a href="#cb6-48"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>filter) <span class="sc">%&gt;%</span> </span>
<span id="cb6-49"><a href="#cb6-49"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_extract</span>(sample, </span>
<span id="cb6-50"><a href="#cb6-50"></a>                              <span class="fu">regex</span>(<span class="st">"(HAMBI_[:digit:]{4}-SH-MET-[:digit:]{3})"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb6-51"><a href="#cb6-51"></a>  <span class="fu">separate</span>(sample, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"strainID"</span>, <span class="st">"sample"</span>), <span class="at">sep=</span><span class="st">"-"</span>, <span class="at">extra =</span> <span class="st">"merge"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="allele-frequencies" class="level3" data-number="2.2.2">
<h3 data-number="2.2.2" class="anchored" data-anchor-id="allele-frequencies"><span class="header-section-number">2.2.2</span> Allele frequencies</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>varfiles <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="at">path =</span> vcfdir,</span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="at">all =</span> <span class="cn">FALSE</span>,</span>
<span id="cb7-4"><a href="#cb7-4"></a>  <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb7-6"><a href="#cb7-6"></a>  <span class="at">glob =</span> <span class="st">"*.variants.tsv"</span>,</span>
<span id="cb7-7"><a href="#cb7-7"></a>  <span class="at">regexp =</span> <span class="cn">NULL</span>,</span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="at">invert =</span> <span class="cn">FALSE</span>,</span>
<span id="cb7-9"><a href="#cb7-9"></a>  <span class="at">fail =</span> <span class="cn">TRUE</span></span>
<span id="cb7-10"><a href="#cb7-10"></a>)</span>
<span id="cb7-11"><a href="#cb7-11"></a></span>
<span id="cb7-12"><a href="#cb7-12"></a>varslurped <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(</span>
<span id="cb7-13"><a href="#cb7-13"></a>  varfiles,</span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb7-15"><a href="#cb7-15"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(</span>
<span id="cb7-16"><a href="#cb7-16"></a>    <span class="st">"chrom"</span>,</span>
<span id="cb7-17"><a href="#cb7-17"></a>    <span class="st">"pos"</span>,</span>
<span id="cb7-18"><a href="#cb7-18"></a>    <span class="st">"ref"</span>,</span>
<span id="cb7-19"><a href="#cb7-19"></a>    <span class="st">"alt"</span>,</span>
<span id="cb7-20"><a href="#cb7-20"></a>    <span class="st">"filter"</span>,</span>
<span id="cb7-21"><a href="#cb7-21"></a>    <span class="st">"type"</span>,</span>
<span id="cb7-22"><a href="#cb7-22"></a>    <span class="co"># Allelic depths for the ref and alt alleles in the order listed</span></span>
<span id="cb7-23"><a href="#cb7-23"></a>    <span class="st">"ad"</span>,</span>
<span id="cb7-24"><a href="#cb7-24"></a>    <span class="co"># Allele fractions of alternate alleles. Excludes filtering</span></span>
<span id="cb7-25"><a href="#cb7-25"></a>    <span class="st">"freq_alt_raw"</span>,</span>
<span id="cb7-26"><a href="#cb7-26"></a>    <span class="co"># Approximate read depth (reads are filtered if MQ=255 or with bad mates)</span></span>
<span id="cb7-27"><a href="#cb7-27"></a>    <span class="st">"depth_total"</span>,</span>
<span id="cb7-28"><a href="#cb7-28"></a>    <span class="co"># Count of fragments supporting each allele.</span></span>
<span id="cb7-29"><a href="#cb7-29"></a>    <span class="st">"frag_allele_depth"</span>,</span>
<span id="cb7-30"><a href="#cb7-30"></a>    <span class="co"># Genotype Quality</span></span>
<span id="cb7-31"><a href="#cb7-31"></a>    <span class="st">"genotype_qual"</span></span>
<span id="cb7-32"><a href="#cb7-32"></a>  ),</span>
<span id="cb7-33"><a href="#cb7-33"></a>  <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">"cdccccccccc"</span>),</span>
<span id="cb7-34"><a href="#cb7-34"></a>  <span class="at">id =</span> <span class="st">"sample"</span></span>
<span id="cb7-35"><a href="#cb7-35"></a>)</span>
<span id="cb7-36"><a href="#cb7-36"></a></span>
<span id="cb7-37"><a href="#cb7-37"></a>varslurpedfmt <span class="ot">&lt;-</span> varslurped <span class="sc">%&gt;%</span> </span>
<span id="cb7-38"><a href="#cb7-38"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_extract</span>(sample, </span>
<span id="cb7-39"><a href="#cb7-39"></a>                              <span class="fu">regex</span>(<span class="st">"(HAMBI_[:digit:]{4}-SH-MET-[:digit:]{3})"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb7-40"><a href="#cb7-40"></a>  <span class="fu">separate</span>(sample, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"strainID"</span>, <span class="st">"sample"</span>), <span class="at">sep=</span><span class="st">"-"</span>, <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-41"><a href="#cb7-41"></a>  <span class="co"># taken only first two most abundant alleles</span></span>
<span id="cb7-42"><a href="#cb7-42"></a>  <span class="fu">separate</span>(ad, <span class="at">into =</span>  <span class="fu">c</span>(<span class="st">"depth_ref"</span>, <span class="st">"depth_alt"</span>), </span>
<span id="cb7-43"><a href="#cb7-43"></a>           <span class="at">sep=</span><span class="st">","</span>, <span class="at">extra=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-44"><a href="#cb7-44"></a>  <span class="co"># format depths to numberic</span></span>
<span id="cb7-45"><a href="#cb7-45"></a>  <span class="fu">mutate</span>(<span class="at">depth_ref =</span> <span class="fu">as.numeric</span>(depth_ref), </span>
<span id="cb7-46"><a href="#cb7-46"></a>         <span class="at">depth_alt =</span> <span class="fu">as.numeric</span>(depth_alt)) <span class="sc">%&gt;%</span></span>
<span id="cb7-47"><a href="#cb7-47"></a>  <span class="fu">select</span>(<span class="sc">-</span>frag_allele_depth, <span class="sc">-</span>genotype_qual) <span class="sc">%&gt;%</span> </span>
<span id="cb7-48"><a href="#cb7-48"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt =</span> depth_alt<span class="sc">/</span>(depth_ref <span class="sc">+</span> depth_alt),</span>
<span id="cb7-49"><a href="#cb7-49"></a>         <span class="at">freq_ref =</span> <span class="dv">1</span> <span class="sc">-</span> freq_alt) <span class="sc">%&gt;%</span> </span>
<span id="cb7-50"><a href="#cb7-50"></a>  <span class="co"># final format depths to numeric</span></span>
<span id="cb7-51"><a href="#cb7-51"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt_raw =</span> <span class="fu">as.numeric</span>(freq_alt_raw),</span>
<span id="cb7-52"><a href="#cb7-52"></a>         <span class="at">depth_total =</span> <span class="fu">as.numeric</span>(depth_total))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `freq_alt_raw = as.numeric(freq_alt_raw)`.
Caused by warning:
! NAs introduced by coercion</code></pre>
</div>
</div>
</section>
<section id="clean-up-tmp-directories" class="level3" data-number="2.2.3">
<h3 data-number="2.2.3" class="anchored" data-anchor-id="clean-up-tmp-directories"><span class="header-section-number">2.2.3</span> Clean up tmp directories</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># remove decompressed vcf and tables directory from temp location</span></span>
<span id="cb9-2"><a href="#cb9-2"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(vcfdir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="mutations-from-wgs-analysis" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="mutations-from-wgs-analysis"><span class="header-section-number">2.3</span> Mutations from WGS analysis</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>wgs_freq <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"wgs"</span>, <span class="st">"wgsvars_filt_mb_nonsyn.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="annotations" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="annotations"><span class="header-section-number">2.4</span> Annotations</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">#annotations &lt;- read_rds(here::here(data_raw, "annotations_codon_degeneracy.rds"))</span></span>
<span id="cb11-2"><a href="#cb11-2"></a>mgefinder <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"MGE_finder_HAMBI_combined.tsv"</span>))</span>
<span id="cb11-3"><a href="#cb11-3"></a>genomad <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"genomad_HAMBI_combined.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="co"># exclude very large plasmids over 1Mbp from the filtering</span></span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">filter</span>((end <span class="sc">-</span> start) <span class="sc">&lt;</span> <span class="dv">1000000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="formatting-and-filtering" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Formatting and filtering</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co"># Combine snpeff and var freqs</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">left_join</span>(varslurpedfmt, snpeffslurpedfmt,</span>
<span id="cb12-3"><a href="#cb12-3"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(strainID, sample, chrom, pos, ref, alt)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># get species called in metagenomes</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>sps <span class="ot">&lt;-</span> <span class="fu">pull</span>(mgvars, chrom) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">str_extract</span>(<span class="st">"HAMBI_</span><span class="sc">\\</span><span class="st">d{4}"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">unique</span>()</span>
<span id="cb13-5"><a href="#cb13-5"></a></span>
<span id="cb13-6"><a href="#cb13-6"></a>sps</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "HAMBI_0403" "HAMBI_1287" "HAMBI_1292" "HAMBI_1972" "HAMBI_1977"
[6] "HAMBI_2659"</code></pre>
</div>
</div>
<section id="exclude-regions-with-mobile-elements-in-metagenomes" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="exclude-regions-with-mobile-elements-in-metagenomes"><span class="header-section-number">3.1</span> Exclude regions with mobile elements in metagenomes</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>mgvars_filt_mb <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="fu">left_join</span>(mgefinder, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start, <span class="sc">-</span>end, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6"></a>  <span class="fu">left_join</span>(genomad, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-7"><a href="#cb15-7"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-8"><a href="#cb15-8"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start, <span class="sc">-</span>end, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9"></a>  <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="filter-wgs-on-species-only-included-in-metag" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="filter-wgs-on-species-only-included-in-metag"><span class="header-section-number">3.2</span> Filter WGS on species only included in metaG</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>wgs_filt_mb <span class="ot">&lt;-</span> wgs_freq <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> sps) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3"></a>  <span class="fu">mutate</span>(<span class="at">anc =</span> <span class="st">"anc"</span>, <span class="at">evo =</span> <span class="st">"evo"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(anc, evo), <span class="at">names_to =</span> <span class="st">"predator_history"</span>, <span class="at">values_to =</span> <span class="st">"tmp"</span>) <span class="sc">%&gt;%</span>  <span class="fu">select</span>(<span class="sc">-</span>tmp) <span class="sc">%&gt;%</span> </span>
<span id="cb16-5"><a href="#cb16-5"></a>  <span class="fu">mutate</span>(<span class="at">prey_history =</span> <span class="st">"evo"</span>, </span>
<span id="cb16-6"><a href="#cb16-6"></a>         <span class="at">transfer_category =</span> <span class="st">"hi"</span>,</span>
<span id="cb16-7"><a href="#cb16-7"></a>         <span class="at">transfer_volume_ul =</span> <span class="dv">1800</span>,</span>
<span id="cb16-8"><a href="#cb16-8"></a>         <span class="at">time_days =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="combine-wgs-and-metag" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="combine-wgs-and-metag"><span class="header-section-number">3.3</span> Combine WGS and metaG</h2>
<p>This includes all variants - both amino acid changing and non-changing</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>freq_mg_wgs <span class="ot">&lt;-</span> <span class="fu">left_join</span>(mgvars_filt_mb, </span>
<span id="cb17-2"><a href="#cb17-2"></a>                         <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metadata.tsv"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="fu">filter</span>(transfer_category <span class="sc">==</span> <span class="st">"hi"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="fu">bind_rows</span>(wgs_filt_mb) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="fu">mutate</span>(<span class="at">pos =</span> <span class="fu">as.numeric</span>(pos),</span>
<span id="cb17-6"><a href="#cb17-6"></a>         <span class="at">depth_total =</span> <span class="fu">as.numeric</span>(depth_total)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-7"><a href="#cb17-7"></a>  <span class="fu">select</span>(<span class="sc">-</span>hgvs_c, <span class="sc">-</span>aa_pos, <span class="sc">-</span>aa_len)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 36 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (5): sample, transfer_category, replicate, prey_history, predator_history
dbl (2): transfer_volume_ul, time_days

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Joining with `by = join_by(sample)`</code></pre>
</div>
</div>
</section>
<section id="expand-grid-for-variants" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="expand-grid-for-variants"><span class="header-section-number">3.4</span> Expand grid for variants</h2>
<p>Make a dataframe where all times/species/replicates/genomic position/experimental conditions are present</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>vars_united <span class="ot">&lt;-</span> freq_mg_wgs <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="fu">select</span>(chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="fu">unite</span>(<span class="st">"combo"</span>, <span class="fu">c</span>(chrom, pos, ref, alt), <span class="at">sep =</span> <span class="st">"|"</span>)</span>
<span id="cb19-5"><a href="#cb19-5"></a></span>
<span id="cb19-6"><a href="#cb19-6"></a>times_full <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(<span class="at">combo =</span> vars_united<span class="sc">$</span>combo, </span>
<span id="cb19-7"><a href="#cb19-7"></a>              <span class="at">time_days =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">28</span>, <span class="dv">60</span>), </span>
<span id="cb19-8"><a href="#cb19-8"></a>              <span class="at">replicate =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"C"</span>, <span class="st">"E"</span>),</span>
<span id="cb19-9"><a href="#cb19-9"></a>              <span class="at">prey_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>),</span>
<span id="cb19-10"><a href="#cb19-10"></a>              <span class="at">predator_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-11"><a href="#cb19-11"></a>  <span class="fu">separate_wider_delim</span>(combo, <span class="at">delim =</span> <span class="st">"|"</span>, <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"chrom"</span>, <span class="st">"pos"</span>, <span class="st">"ref"</span>, <span class="st">"alt"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-12"><a href="#cb19-12"></a>  <span class="fu">mutate</span>(<span class="at">pos =</span> <span class="fu">as.numeric</span>(pos),</span>
<span id="cb19-13"><a href="#cb19-13"></a>         <span class="at">transfer_category =</span> <span class="st">"hi"</span>,</span>
<span id="cb19-14"><a href="#cb19-14"></a>         <span class="at">transfer_volume_ul =</span> <span class="dv">1800</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="format-metagenome-coverage-data" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="format-metagenome-coverage-data"><span class="header-section-number">3.5</span> Format metagenome coverage data</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>mg_cov <span class="ot">&lt;-</span> covslurpedfmtflt <span class="sc">%&gt;%</span>  </span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> sps) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3"></a>  <span class="fu">left_join</span>(<span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metadata.tsv"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="fu">filter</span>(transfer_category <span class="sc">==</span> <span class="st">"hi"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="co"># samples with trimmed mean coverage &gt;=10 with over 99% of genome covered</span></span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="co"># are to be trusted</span></span>
<span id="cb20-7"><a href="#cb20-7"></a>  <span class="fu">mutate</span>(<span class="at">cov_thresh_met =</span> <span class="fu">if_else</span>(trimmed_mean <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">&amp;</span> covered_fraction <span class="sc">&gt;</span> <span class="fl">0.99</span>, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-8"><a href="#cb20-8"></a>  <span class="fu">select</span>(sample, strainID, <span class="at">chrom=</span>scaffold, trimmed_mean, covered_fraction, transfer_category<span class="sc">:</span>cov_thresh_met) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="combine-variants-into-complete-time-series" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Combine variants into complete time series</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>freq_full <span class="ot">&lt;-</span> <span class="fu">left_join</span>(times_full, freq_mg_wgs, <span class="at">relationship =</span> <span class="st">"one-to-one"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="fu">arrange</span>(chrom, pos, ref, alt, replicate, predator_history, prey_history, time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3"></a>  <span class="co"># group_by(chrom) %&gt;% </span></span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="fu">fill</span>(strainID, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6"></a>  <span class="fu">group_by</span>(time_days, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb21-7"><a href="#cb21-7"></a>  <span class="fu">fill</span>(sample, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-8"><a href="#cb21-8"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9"></a>  <span class="fu">left_join</span>(mg_cov) <span class="sc">%&gt;%</span> </span>
<span id="cb21-10"><a href="#cb21-10"></a>  <span class="fu">mutate</span>(<span class="at">cov_thresh_met =</span> <span class="fu">if_else</span>(time_days <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, cov_thresh_met)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-11"><a href="#cb21-11"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt_complete =</span> <span class="fu">case_when</span>(cov_thresh_met <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(freq_alt) <span class="sc">~</span> freq_alt,</span>
<span id="cb21-12"><a href="#cb21-12"></a>                                       cov_thresh_met <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(freq_alt) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb21-13"><a href="#cb21-13"></a>                                       <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_real_</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(chrom, pos, ref, alt, time_days, replicate,
prey_history, predator_history, transfer_category, transfer_volume_ul)`
Joining with `by = join_by(chrom, time_days, replicate, prey_history,
predator_history, transfer_category, transfer_volume_ul, strainID, sample)`</code></pre>
</div>
</div>
<p>Some filtering to reduce size. Also record the number of mutect2 passing variants and add a time-lagged variable. We remove variants that only have an observation at T0 and never in any metagenome aample due to low coverage (i.e.&nbsp;they are NA). This will reduce the size of the dataset and make sure we focus on more than just signals present in the WGS analysis</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>freq_full_sm01 <span class="ot">&lt;-</span> freq_full <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">group_by</span>(chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb23-6"><a href="#cb23-6"></a>  <span class="co"># can't be all zeros</span></span>
<span id="cb23-7"><a href="#cb23-7"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(freq_alt_complete, <span class="at">na.rm=</span><span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-8"><a href="#cb23-8"></a>  <span class="co"># exclude only NAs</span></span>
<span id="cb23-9"><a href="#cb23-9"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(freq_alt_complete[time_days <span class="sc">!=</span> <span class="dv">0</span>])) <span class="sc">&lt;</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-10"><a href="#cb23-10"></a>  <span class="co"># mark some statistics that may be useful for later filtering</span></span>
<span id="cb23-11"><a href="#cb23-11"></a>  <span class="fu">mutate</span>(<span class="at">n_gt_fthresh =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> T),</span>
<span id="cb23-12"><a href="#cb23-12"></a>         <span class="at">n_pass   =</span> <span class="fu">sum</span>(filter <span class="sc">==</span> <span class="st">"PASS"</span>, <span class="at">na.rm =</span> T),</span>
<span id="cb23-13"><a href="#cb23-13"></a>         <span class="at">n_cov_gt =</span> <span class="fu">sum</span>(cov_thresh_met, <span class="at">na.rm =</span>T)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-14"><a href="#cb23-14"></a>  <span class="co"># freq_alt_complete_t1=lag(freq_alt_complete, order_by = time_days, n=1)</span></span>
<span id="cb23-15"><a href="#cb23-15"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here we require at least one observation with greater than 0.05 alt frequency and at least one variant passing mutect filters in the variant time series</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>freq_full_sm02 <span class="ot">&lt;-</span> freq_full_sm01 <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2"></a>  <span class="fu">filter</span>(n_gt_fthresh <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> n_pass <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="fu">fill</span>(<span class="fu">c</span>(type, effect, impact, locus_tag, biotype, hgvs_p, cds_pos, cds_len), <span class="at">.direction =</span> <span class="st">"updown"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="export" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Export</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="co"># save table for later use</span></span>
<span id="cb25-2"><a href="#cb25-2"></a><span class="fu">write_tsv</span>(freq_full_sm02, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metagenome_variant_timeseries.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb26" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb26-1"><a href="#cb26-1"></a><span class="co">---</span></span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="an">title:</span><span class="co"> "Format metagenome variants"</span></span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb26-4"><a href="#cb26-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb26-5"><a href="#cb26-5"></a><span class="an">abstract:</span><span class="co"> "This notebook formats for downstream use the table and VCF output from mutect2 called on metagenome data. It does some filtering to remove variants that are likely spurious and connects variants to functional annotations and gene identities. Finally it joins the WGS variants formatted earlier as the initial sampling point so that we can make time series from 0, 8, 28, and 60 days."</span></span>
<span id="cb26-6"><a href="#cb26-6"></a><span class="co">---</span></span>
<span id="cb26-7"><a href="#cb26-7"></a></span>
<span id="cb26-8"><a href="#cb26-8"></a><span class="fu"># Setup</span></span>
<span id="cb26-9"><a href="#cb26-9"></a></span>
<span id="cb26-10"><a href="#cb26-10"></a>Libraries and global variables</span>
<span id="cb26-11"><a href="#cb26-11"></a></span>
<span id="cb26-14"><a href="#cb26-14"></a><span class="in">```{r}</span></span>
<span id="cb26-15"><a href="#cb26-15"></a><span class="co">#| output: false</span></span>
<span id="cb26-16"><a href="#cb26-16"></a><span class="fu">library</span>(here)</span>
<span id="cb26-17"><a href="#cb26-17"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb26-18"><a href="#cb26-18"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb26-19"><a href="#cb26-19"></a><span class="fu">library</span>(withr)</span>
<span id="cb26-20"><a href="#cb26-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb26-21"><a href="#cb26-21"></a><span class="fu">library</span>(archive)</span>
<span id="cb26-22"><a href="#cb26-22"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb26-23"><a href="#cb26-23"></a><span class="in">```</span></span>
<span id="cb26-24"><a href="#cb26-24"></a></span>
<span id="cb26-25"><a href="#cb26-25"></a>Set up some directories</span>
<span id="cb26-26"><a href="#cb26-26"></a></span>
<span id="cb26-29"><a href="#cb26-29"></a><span class="in">```{r}</span></span>
<span id="cb26-30"><a href="#cb26-30"></a><span class="co">#| output: false</span></span>
<span id="cb26-31"><a href="#cb26-31"></a><span class="co">#| warning: false</span></span>
<span id="cb26-32"><a href="#cb26-32"></a><span class="co">#| error: false</span></span>
<span id="cb26-33"><a href="#cb26-33"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb26-34"><a href="#cb26-34"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb26-35"><a href="#cb26-35"></a>vcftar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"tables.tar.gz"</span>)</span>
<span id="cb26-36"><a href="#cb26-36"></a>covtar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"coverage.tar.gz"</span>)</span>
<span id="cb26-37"><a href="#cb26-37"></a></span>
<span id="cb26-38"><a href="#cb26-38"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb26-39"><a href="#cb26-39"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span>
<span id="cb26-40"><a href="#cb26-40"></a></span>
<span id="cb26-41"><a href="#cb26-41"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb26-42"><a href="#cb26-42"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb26-43"><a href="#cb26-43"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb26-44"><a href="#cb26-44"></a><span class="in">```</span></span>
<span id="cb26-45"><a href="#cb26-45"></a></span>
<span id="cb26-46"><a href="#cb26-46"></a><span class="fu">## Untar and decompress</span></span>
<span id="cb26-47"><a href="#cb26-47"></a></span>
<span id="cb26-50"><a href="#cb26-50"></a><span class="in">```{r}</span></span>
<span id="cb26-51"><a href="#cb26-51"></a><span class="co"># untar directory containing variant tables </span></span>
<span id="cb26-52"><a href="#cb26-52"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb26-53"><a href="#cb26-53"></a>  vcftar,</span>
<span id="cb26-54"><a href="#cb26-54"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb26-55"><a href="#cb26-55"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb26-56"><a href="#cb26-56"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb26-57"><a href="#cb26-57"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb26-58"><a href="#cb26-58"></a>)</span>
<span id="cb26-59"><a href="#cb26-59"></a></span>
<span id="cb26-60"><a href="#cb26-60"></a>vcfdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"tables"</span>)</span>
<span id="cb26-61"><a href="#cb26-61"></a></span>
<span id="cb26-62"><a href="#cb26-62"></a><span class="co"># untar directory containing variant tables </span></span>
<span id="cb26-63"><a href="#cb26-63"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb26-64"><a href="#cb26-64"></a>  covtar,</span>
<span id="cb26-65"><a href="#cb26-65"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb26-66"><a href="#cb26-66"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb26-67"><a href="#cb26-67"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb26-68"><a href="#cb26-68"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb26-69"><a href="#cb26-69"></a>)</span>
<span id="cb26-70"><a href="#cb26-70"></a></span>
<span id="cb26-71"><a href="#cb26-71"></a>covdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"coverage"</span>)</span>
<span id="cb26-72"><a href="#cb26-72"></a><span class="in">```</span></span>
<span id="cb26-73"><a href="#cb26-73"></a></span>
<span id="cb26-74"><a href="#cb26-74"></a></span>
<span id="cb26-75"><a href="#cb26-75"></a><span class="fu"># Reading and small formatting of data</span></span>
<span id="cb26-76"><a href="#cb26-76"></a></span>
<span id="cb26-77"><a href="#cb26-77"></a><span class="fu">## Coverage data </span></span>
<span id="cb26-80"><a href="#cb26-80"></a><span class="in">```{r}</span></span>
<span id="cb26-81"><a href="#cb26-81"></a><span class="co">#| eval: false</span></span>
<span id="cb26-82"><a href="#cb26-82"></a><span class="co">#| echo: true</span></span>
<span id="cb26-83"><a href="#cb26-83"></a>covfiles <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb26-84"><a href="#cb26-84"></a>  <span class="at">path =</span> covdir,</span>
<span id="cb26-85"><a href="#cb26-85"></a>  <span class="at">all =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-86"><a href="#cb26-86"></a>  <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-87"><a href="#cb26-87"></a>  <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb26-88"><a href="#cb26-88"></a>  <span class="at">glob =</span> <span class="st">"*.coverage.tsv"</span>,</span>
<span id="cb26-89"><a href="#cb26-89"></a>  <span class="at">regexp =</span> <span class="cn">NULL</span>,</span>
<span id="cb26-90"><a href="#cb26-90"></a>  <span class="at">invert =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-91"><a href="#cb26-91"></a>  <span class="at">fail =</span> <span class="cn">TRUE</span></span>
<span id="cb26-92"><a href="#cb26-92"></a>)</span>
<span id="cb26-93"><a href="#cb26-93"></a></span>
<span id="cb26-94"><a href="#cb26-94"></a>covslurped <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(</span>
<span id="cb26-95"><a href="#cb26-95"></a>  covfiles,</span>
<span id="cb26-96"><a href="#cb26-96"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb26-97"><a href="#cb26-97"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(</span>
<span id="cb26-98"><a href="#cb26-98"></a>    <span class="st">"scaffold"</span>,</span>
<span id="cb26-99"><a href="#cb26-99"></a>    <span class="st">"mean"</span>,</span>
<span id="cb26-100"><a href="#cb26-100"></a>    <span class="st">"trimmed_mean"</span>,</span>
<span id="cb26-101"><a href="#cb26-101"></a>    <span class="st">"covered_fraction"</span>,</span>
<span id="cb26-102"><a href="#cb26-102"></a>    <span class="st">"covered_bases"</span>,</span>
<span id="cb26-103"><a href="#cb26-103"></a>    <span class="st">"variance"</span>,</span>
<span id="cb26-104"><a href="#cb26-104"></a>    <span class="st">"length"</span>,</span>
<span id="cb26-105"><a href="#cb26-105"></a>    <span class="st">"read_count"</span>,</span>
<span id="cb26-106"><a href="#cb26-106"></a>    <span class="st">"reads_per_base"</span></span>
<span id="cb26-107"><a href="#cb26-107"></a>  ),</span>
<span id="cb26-108"><a href="#cb26-108"></a>  <span class="at">col_types =</span> <span class="st">"cdddddddd"</span>,</span>
<span id="cb26-109"><a href="#cb26-109"></a>  <span class="at">id =</span> <span class="st">"file_name"</span></span>
<span id="cb26-110"><a href="#cb26-110"></a>)</span>
<span id="cb26-111"><a href="#cb26-111"></a></span>
<span id="cb26-112"><a href="#cb26-112"></a>covslurpedfmt <span class="ot">&lt;-</span> covslurped <span class="sc">%&gt;%</span></span>
<span id="cb26-113"><a href="#cb26-113"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_extract</span>(file_name, <span class="st">"SH-MET-[:digit:]{3}"</span>),</span>
<span id="cb26-114"><a href="#cb26-114"></a>         <span class="at">strainID   =</span> <span class="fu">str_extract</span>(file_name, <span class="st">"HAMBI_[:digit:]{4}"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-115"><a href="#cb26-115"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>file_name) <span class="sc">%&gt;%</span></span>
<span id="cb26-116"><a href="#cb26-116"></a>  <span class="fu">relocate</span>(strainID, sample)</span>
<span id="cb26-117"><a href="#cb26-117"></a></span>
<span id="cb26-118"><a href="#cb26-118"></a>covslurpedfmtflt <span class="ot">&lt;-</span> covslurpedfmt <span class="sc">%&gt;%</span></span>
<span id="cb26-119"><a href="#cb26-119"></a>  <span class="fu">mutate</span>(<span class="at">bamname =</span> <span class="fu">paste0</span>(strainID,<span class="st">"-"</span>,sample)) <span class="sc">%&gt;%</span></span>
<span id="cb26-120"><a href="#cb26-120"></a>  <span class="co"># exclude species with trimmed mean (Dt) &lt; 5</span></span>
<span id="cb26-121"><a href="#cb26-121"></a>  <span class="fu">mutate</span>(<span class="at">FILTER =</span> <span class="fu">if_else</span>(trimmed_mean <span class="sc">&gt;=</span> <span class="dv">5</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>))</span>
<span id="cb26-122"><a href="#cb26-122"></a></span>
<span id="cb26-123"><a href="#cb26-123"></a><span class="fu">write_tsv</span>(covslurpedfmtflt, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"coverage.tsv"</span>))</span>
<span id="cb26-124"><a href="#cb26-124"></a></span>
<span id="cb26-125"><a href="#cb26-125"></a><span class="co"># cleanup</span></span>
<span id="cb26-126"><a href="#cb26-126"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(covdir)</span>
<span id="cb26-127"><a href="#cb26-127"></a><span class="in">```</span></span>
<span id="cb26-128"><a href="#cb26-128"></a></span>
<span id="cb26-131"><a href="#cb26-131"></a><span class="in">```{r}</span></span>
<span id="cb26-132"><a href="#cb26-132"></a><span class="co">#| output: false</span></span>
<span id="cb26-133"><a href="#cb26-133"></a><span class="co">#| warning: false</span></span>
<span id="cb26-134"><a href="#cb26-134"></a><span class="co">#| error: false</span></span>
<span id="cb26-135"><a href="#cb26-135"></a>covslurpedfmtflt <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"coverage.tsv"</span>))</span>
<span id="cb26-136"><a href="#cb26-136"></a><span class="in">```</span></span>
<span id="cb26-137"><a href="#cb26-137"></a></span>
<span id="cb26-138"><a href="#cb26-138"></a><span class="fu">## Variants </span></span>
<span id="cb26-139"><a href="#cb26-139"></a></span>
<span id="cb26-140"><a href="#cb26-140"></a><span class="fu">### SnpEff annotations </span></span>
<span id="cb26-141"><a href="#cb26-141"></a></span>
<span id="cb26-144"><a href="#cb26-144"></a><span class="in">```{r}</span></span>
<span id="cb26-145"><a href="#cb26-145"></a>snpefffiles <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb26-146"><a href="#cb26-146"></a>  <span class="at">path =</span> vcfdir,</span>
<span id="cb26-147"><a href="#cb26-147"></a>  <span class="at">all =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-148"><a href="#cb26-148"></a>  <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-149"><a href="#cb26-149"></a>  <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb26-150"><a href="#cb26-150"></a>  <span class="at">glob =</span> <span class="st">"*.snpeff.tsv"</span>,</span>
<span id="cb26-151"><a href="#cb26-151"></a>  <span class="at">regexp =</span> <span class="cn">NULL</span>,</span>
<span id="cb26-152"><a href="#cb26-152"></a>  <span class="at">invert =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-153"><a href="#cb26-153"></a>  <span class="at">fail =</span> <span class="cn">TRUE</span></span>
<span id="cb26-154"><a href="#cb26-154"></a>)</span>
<span id="cb26-155"><a href="#cb26-155"></a></span>
<span id="cb26-156"><a href="#cb26-156"></a>snpeffslurped <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(</span>
<span id="cb26-157"><a href="#cb26-157"></a>  snpefffiles,</span>
<span id="cb26-158"><a href="#cb26-158"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb26-159"><a href="#cb26-159"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(</span>
<span id="cb26-160"><a href="#cb26-160"></a>    <span class="st">"chrom"</span>,</span>
<span id="cb26-161"><a href="#cb26-161"></a>    <span class="st">"pos"</span>,</span>
<span id="cb26-162"><a href="#cb26-162"></a>    <span class="st">"ref"</span>,</span>
<span id="cb26-163"><a href="#cb26-163"></a>    <span class="st">"alt"</span>,</span>
<span id="cb26-164"><a href="#cb26-164"></a>    <span class="st">"filter"</span>,</span>
<span id="cb26-165"><a href="#cb26-165"></a>    <span class="co"># Annotated using Sequence Ontology terms. Multiple effects can be concatenated using '&amp;'.</span></span>
<span id="cb26-166"><a href="#cb26-166"></a>    <span class="st">"effect"</span>,</span>
<span id="cb26-167"><a href="#cb26-167"></a>    <span class="co"># A simple estimation of putative impact / deleteriousness : HIGH, MODERATE, LOW, or MODIFIER</span></span>
<span id="cb26-168"><a href="#cb26-168"></a>    <span class="st">"impact"</span>,</span>
<span id="cb26-169"><a href="#cb26-169"></a>    <span class="co"># gene id from prokka</span></span>
<span id="cb26-170"><a href="#cb26-170"></a>    <span class="st">"locus_tag"</span>,</span>
<span id="cb26-171"><a href="#cb26-171"></a>    <span class="co"># whether variant is coding or noncoding</span></span>
<span id="cb26-172"><a href="#cb26-172"></a>    <span class="st">"biotype"</span>,</span>
<span id="cb26-173"><a href="#cb26-173"></a>    <span class="co"># Variant using HGVS notation (DNA level)</span></span>
<span id="cb26-174"><a href="#cb26-174"></a>    <span class="st">"hgvs_c"</span>,</span>
<span id="cb26-175"><a href="#cb26-175"></a>    <span class="co"># Variant using HGVS notation (Protein level).</span></span>
<span id="cb26-176"><a href="#cb26-176"></a>    <span class="st">"hgvs_p"</span>,</span>
<span id="cb26-177"><a href="#cb26-177"></a>    <span class="co"># Position in coding bases (one based includes START and STOP codons).</span></span>
<span id="cb26-178"><a href="#cb26-178"></a>    <span class="st">"cds_pos"</span>,</span>
<span id="cb26-179"><a href="#cb26-179"></a>    <span class="co"># Total number of coding bases (one based includes START and STOP codons).</span></span>
<span id="cb26-180"><a href="#cb26-180"></a>    <span class="st">"cds_len"</span>,</span>
<span id="cb26-181"><a href="#cb26-181"></a>    <span class="co"># Position in translated product (one based includes START and STOP codons).</span></span>
<span id="cb26-182"><a href="#cb26-182"></a>    <span class="st">"aa_pos"</span>,</span>
<span id="cb26-183"><a href="#cb26-183"></a>    <span class="co"># Total length of translated product (one based includes START and STOP codons).</span></span>
<span id="cb26-184"><a href="#cb26-184"></a>    <span class="st">"aa_len"</span></span>
<span id="cb26-185"><a href="#cb26-185"></a>  ),</span>
<span id="cb26-186"><a href="#cb26-186"></a>  <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">"cdcccccccccdddd"</span>),</span>
<span id="cb26-187"><a href="#cb26-187"></a>  <span class="at">id =</span> <span class="st">"sample"</span></span>
<span id="cb26-188"><a href="#cb26-188"></a>)</span>
<span id="cb26-189"><a href="#cb26-189"></a></span>
<span id="cb26-190"><a href="#cb26-190"></a>snpeffslurpedfmt <span class="ot">&lt;-</span> snpeffslurped <span class="sc">%&gt;%</span> </span>
<span id="cb26-191"><a href="#cb26-191"></a>  <span class="co"># this is weird, but joining on column like filter that contains characters like ';' doesnt work</span></span>
<span id="cb26-192"><a href="#cb26-192"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>filter) <span class="sc">%&gt;%</span> </span>
<span id="cb26-193"><a href="#cb26-193"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_extract</span>(sample, </span>
<span id="cb26-194"><a href="#cb26-194"></a>                              <span class="fu">regex</span>(<span class="st">"(HAMBI_[:digit:]{4}-SH-MET-[:digit:]{3})"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-195"><a href="#cb26-195"></a>  <span class="fu">separate</span>(sample, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"strainID"</span>, <span class="st">"sample"</span>), <span class="at">sep=</span><span class="st">"-"</span>, <span class="at">extra =</span> <span class="st">"merge"</span>)</span>
<span id="cb26-196"><a href="#cb26-196"></a><span class="in">```</span></span>
<span id="cb26-197"><a href="#cb26-197"></a></span>
<span id="cb26-198"><a href="#cb26-198"></a></span>
<span id="cb26-199"><a href="#cb26-199"></a><span class="fu">### Allele frequencies </span></span>
<span id="cb26-200"><a href="#cb26-200"></a></span>
<span id="cb26-203"><a href="#cb26-203"></a><span class="in">```{r}</span></span>
<span id="cb26-204"><a href="#cb26-204"></a>varfiles <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(</span>
<span id="cb26-205"><a href="#cb26-205"></a>  <span class="at">path =</span> vcfdir,</span>
<span id="cb26-206"><a href="#cb26-206"></a>  <span class="at">all =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-207"><a href="#cb26-207"></a>  <span class="at">recurse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-208"><a href="#cb26-208"></a>  <span class="at">type =</span> <span class="st">"file"</span>,</span>
<span id="cb26-209"><a href="#cb26-209"></a>  <span class="at">glob =</span> <span class="st">"*.variants.tsv"</span>,</span>
<span id="cb26-210"><a href="#cb26-210"></a>  <span class="at">regexp =</span> <span class="cn">NULL</span>,</span>
<span id="cb26-211"><a href="#cb26-211"></a>  <span class="at">invert =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-212"><a href="#cb26-212"></a>  <span class="at">fail =</span> <span class="cn">TRUE</span></span>
<span id="cb26-213"><a href="#cb26-213"></a>)</span>
<span id="cb26-214"><a href="#cb26-214"></a></span>
<span id="cb26-215"><a href="#cb26-215"></a>varslurped <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(</span>
<span id="cb26-216"><a href="#cb26-216"></a>  varfiles,</span>
<span id="cb26-217"><a href="#cb26-217"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb26-218"><a href="#cb26-218"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(</span>
<span id="cb26-219"><a href="#cb26-219"></a>    <span class="st">"chrom"</span>,</span>
<span id="cb26-220"><a href="#cb26-220"></a>    <span class="st">"pos"</span>,</span>
<span id="cb26-221"><a href="#cb26-221"></a>    <span class="st">"ref"</span>,</span>
<span id="cb26-222"><a href="#cb26-222"></a>    <span class="st">"alt"</span>,</span>
<span id="cb26-223"><a href="#cb26-223"></a>    <span class="st">"filter"</span>,</span>
<span id="cb26-224"><a href="#cb26-224"></a>    <span class="st">"type"</span>,</span>
<span id="cb26-225"><a href="#cb26-225"></a>    <span class="co"># Allelic depths for the ref and alt alleles in the order listed</span></span>
<span id="cb26-226"><a href="#cb26-226"></a>    <span class="st">"ad"</span>,</span>
<span id="cb26-227"><a href="#cb26-227"></a>    <span class="co"># Allele fractions of alternate alleles. Excludes filtering</span></span>
<span id="cb26-228"><a href="#cb26-228"></a>    <span class="st">"freq_alt_raw"</span>,</span>
<span id="cb26-229"><a href="#cb26-229"></a>    <span class="co"># Approximate read depth (reads are filtered if MQ=255 or with bad mates)</span></span>
<span id="cb26-230"><a href="#cb26-230"></a>    <span class="st">"depth_total"</span>,</span>
<span id="cb26-231"><a href="#cb26-231"></a>    <span class="co"># Count of fragments supporting each allele.</span></span>
<span id="cb26-232"><a href="#cb26-232"></a>    <span class="st">"frag_allele_depth"</span>,</span>
<span id="cb26-233"><a href="#cb26-233"></a>    <span class="co"># Genotype Quality</span></span>
<span id="cb26-234"><a href="#cb26-234"></a>    <span class="st">"genotype_qual"</span></span>
<span id="cb26-235"><a href="#cb26-235"></a>  ),</span>
<span id="cb26-236"><a href="#cb26-236"></a>  <span class="at">col_types =</span> <span class="fu">c</span>(<span class="st">"cdccccccccc"</span>),</span>
<span id="cb26-237"><a href="#cb26-237"></a>  <span class="at">id =</span> <span class="st">"sample"</span></span>
<span id="cb26-238"><a href="#cb26-238"></a>)</span>
<span id="cb26-239"><a href="#cb26-239"></a></span>
<span id="cb26-240"><a href="#cb26-240"></a>varslurpedfmt <span class="ot">&lt;-</span> varslurped <span class="sc">%&gt;%</span> </span>
<span id="cb26-241"><a href="#cb26-241"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_extract</span>(sample, </span>
<span id="cb26-242"><a href="#cb26-242"></a>                              <span class="fu">regex</span>(<span class="st">"(HAMBI_[:digit:]{4}-SH-MET-[:digit:]{3})"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-243"><a href="#cb26-243"></a>  <span class="fu">separate</span>(sample, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"strainID"</span>, <span class="st">"sample"</span>), <span class="at">sep=</span><span class="st">"-"</span>, <span class="at">extra =</span> <span class="st">"merge"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-244"><a href="#cb26-244"></a>  <span class="co"># taken only first two most abundant alleles</span></span>
<span id="cb26-245"><a href="#cb26-245"></a>  <span class="fu">separate</span>(ad, <span class="at">into =</span>  <span class="fu">c</span>(<span class="st">"depth_ref"</span>, <span class="st">"depth_alt"</span>), </span>
<span id="cb26-246"><a href="#cb26-246"></a>           <span class="at">sep=</span><span class="st">","</span>, <span class="at">extra=</span><span class="st">"drop"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-247"><a href="#cb26-247"></a>  <span class="co"># format depths to numberic</span></span>
<span id="cb26-248"><a href="#cb26-248"></a>  <span class="fu">mutate</span>(<span class="at">depth_ref =</span> <span class="fu">as.numeric</span>(depth_ref), </span>
<span id="cb26-249"><a href="#cb26-249"></a>         <span class="at">depth_alt =</span> <span class="fu">as.numeric</span>(depth_alt)) <span class="sc">%&gt;%</span></span>
<span id="cb26-250"><a href="#cb26-250"></a>  <span class="fu">select</span>(<span class="sc">-</span>frag_allele_depth, <span class="sc">-</span>genotype_qual) <span class="sc">%&gt;%</span> </span>
<span id="cb26-251"><a href="#cb26-251"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt =</span> depth_alt<span class="sc">/</span>(depth_ref <span class="sc">+</span> depth_alt),</span>
<span id="cb26-252"><a href="#cb26-252"></a>         <span class="at">freq_ref =</span> <span class="dv">1</span> <span class="sc">-</span> freq_alt) <span class="sc">%&gt;%</span> </span>
<span id="cb26-253"><a href="#cb26-253"></a>  <span class="co"># final format depths to numeric</span></span>
<span id="cb26-254"><a href="#cb26-254"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt_raw =</span> <span class="fu">as.numeric</span>(freq_alt_raw),</span>
<span id="cb26-255"><a href="#cb26-255"></a>         <span class="at">depth_total =</span> <span class="fu">as.numeric</span>(depth_total))</span>
<span id="cb26-256"><a href="#cb26-256"></a><span class="in">```</span></span>
<span id="cb26-257"><a href="#cb26-257"></a></span>
<span id="cb26-258"><a href="#cb26-258"></a><span class="fu">### Clean up tmp directories</span></span>
<span id="cb26-261"><a href="#cb26-261"></a><span class="in">```{r}</span></span>
<span id="cb26-262"><a href="#cb26-262"></a><span class="co"># remove decompressed vcf and tables directory from temp location</span></span>
<span id="cb26-263"><a href="#cb26-263"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(vcfdir)</span>
<span id="cb26-264"><a href="#cb26-264"></a><span class="in">```</span></span>
<span id="cb26-265"><a href="#cb26-265"></a></span>
<span id="cb26-266"><a href="#cb26-266"></a><span class="fu">## Mutations from WGS analysis</span></span>
<span id="cb26-267"><a href="#cb26-267"></a></span>
<span id="cb26-270"><a href="#cb26-270"></a><span class="in">```{r}</span></span>
<span id="cb26-271"><a href="#cb26-271"></a><span class="co">#| output: false</span></span>
<span id="cb26-272"><a href="#cb26-272"></a><span class="co">#| warning: false</span></span>
<span id="cb26-273"><a href="#cb26-273"></a><span class="co">#| error: false</span></span>
<span id="cb26-274"><a href="#cb26-274"></a>wgs_freq <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"wgs"</span>, <span class="st">"wgsvars_filt_mb_nonsyn.tsv"</span>))</span>
<span id="cb26-275"><a href="#cb26-275"></a><span class="in">```</span></span>
<span id="cb26-276"><a href="#cb26-276"></a></span>
<span id="cb26-277"><a href="#cb26-277"></a><span class="fu">## Annotations</span></span>
<span id="cb26-278"><a href="#cb26-278"></a></span>
<span id="cb26-281"><a href="#cb26-281"></a><span class="in">```{r}</span></span>
<span id="cb26-282"><a href="#cb26-282"></a><span class="co">#| output: false</span></span>
<span id="cb26-283"><a href="#cb26-283"></a><span class="co">#| warning: false</span></span>
<span id="cb26-284"><a href="#cb26-284"></a><span class="co">#| error: false</span></span>
<span id="cb26-285"><a href="#cb26-285"></a><span class="co">#annotations &lt;- read_rds(here::here(data_raw, "annotations_codon_degeneracy.rds"))</span></span>
<span id="cb26-286"><a href="#cb26-286"></a>mgefinder <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"MGE_finder_HAMBI_combined.tsv"</span>))</span>
<span id="cb26-287"><a href="#cb26-287"></a>genomad <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"genomad_HAMBI_combined.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-288"><a href="#cb26-288"></a>  <span class="co"># exclude very large plasmids over 1Mbp from the filtering</span></span>
<span id="cb26-289"><a href="#cb26-289"></a>  <span class="fu">filter</span>((end <span class="sc">-</span> start) <span class="sc">&lt;</span> <span class="dv">1000000</span>)</span>
<span id="cb26-290"><a href="#cb26-290"></a><span class="in">```</span></span>
<span id="cb26-291"><a href="#cb26-291"></a></span>
<span id="cb26-292"><a href="#cb26-292"></a><span class="fu"># Formatting and filtering</span></span>
<span id="cb26-293"><a href="#cb26-293"></a></span>
<span id="cb26-296"><a href="#cb26-296"></a><span class="in">```{r}</span></span>
<span id="cb26-297"><a href="#cb26-297"></a><span class="co"># Combine snpeff and var freqs</span></span>
<span id="cb26-298"><a href="#cb26-298"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">left_join</span>(varslurpedfmt, snpeffslurpedfmt,</span>
<span id="cb26-299"><a href="#cb26-299"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(strainID, sample, chrom, pos, ref, alt)) </span>
<span id="cb26-300"><a href="#cb26-300"></a><span class="in">```</span></span>
<span id="cb26-301"><a href="#cb26-301"></a></span>
<span id="cb26-304"><a href="#cb26-304"></a><span class="in">```{r}</span></span>
<span id="cb26-305"><a href="#cb26-305"></a><span class="co"># get species called in metagenomes</span></span>
<span id="cb26-306"><a href="#cb26-306"></a>sps <span class="ot">&lt;-</span> <span class="fu">pull</span>(mgvars, chrom) <span class="sc">%&gt;%</span> </span>
<span id="cb26-307"><a href="#cb26-307"></a>  <span class="fu">str_extract</span>(<span class="st">"HAMBI_</span><span class="sc">\\</span><span class="st">d{4}"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-308"><a href="#cb26-308"></a>  <span class="fu">unique</span>()</span>
<span id="cb26-309"><a href="#cb26-309"></a></span>
<span id="cb26-310"><a href="#cb26-310"></a>sps</span>
<span id="cb26-311"><a href="#cb26-311"></a><span class="in">```</span></span>
<span id="cb26-312"><a href="#cb26-312"></a></span>
<span id="cb26-313"><a href="#cb26-313"></a><span class="fu">## Exclude regions with mobile elements in metagenomes</span></span>
<span id="cb26-316"><a href="#cb26-316"></a><span class="in">```{r}</span></span>
<span id="cb26-317"><a href="#cb26-317"></a>mgvars_filt_mb <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb26-318"><a href="#cb26-318"></a>  <span class="fu">left_join</span>(mgefinder, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-319"><a href="#cb26-319"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-320"><a href="#cb26-320"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start, <span class="sc">-</span>end, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb26-321"><a href="#cb26-321"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-322"><a href="#cb26-322"></a>  <span class="fu">left_join</span>(genomad, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, chrom), <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-323"><a href="#cb26-323"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(pos <span class="sc">&gt;=</span> start <span class="sc">&amp;</span> pos <span class="sc">&lt;=</span> end)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-324"><a href="#cb26-324"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>name, <span class="sc">-</span>start, <span class="sc">-</span>end, <span class="sc">-</span>prediction_tool) <span class="sc">%&gt;%</span> </span>
<span id="cb26-325"><a href="#cb26-325"></a>  <span class="fu">distinct</span>()</span>
<span id="cb26-326"><a href="#cb26-326"></a><span class="in">```</span></span>
<span id="cb26-327"><a href="#cb26-327"></a></span>
<span id="cb26-328"><a href="#cb26-328"></a><span class="fu">## Filter WGS on species only included in metaG</span></span>
<span id="cb26-331"><a href="#cb26-331"></a><span class="in">```{r}</span></span>
<span id="cb26-332"><a href="#cb26-332"></a>wgs_filt_mb <span class="ot">&lt;-</span> wgs_freq <span class="sc">%&gt;%</span> </span>
<span id="cb26-333"><a href="#cb26-333"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> sps) <span class="sc">%&gt;%</span> </span>
<span id="cb26-334"><a href="#cb26-334"></a>  <span class="fu">mutate</span>(<span class="at">anc =</span> <span class="st">"anc"</span>, <span class="at">evo =</span> <span class="st">"evo"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-335"><a href="#cb26-335"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(anc, evo), <span class="at">names_to =</span> <span class="st">"predator_history"</span>, <span class="at">values_to =</span> <span class="st">"tmp"</span>) <span class="sc">%&gt;%</span>  <span class="fu">select</span>(<span class="sc">-</span>tmp) <span class="sc">%&gt;%</span> </span>
<span id="cb26-336"><a href="#cb26-336"></a>  <span class="fu">mutate</span>(<span class="at">prey_history =</span> <span class="st">"evo"</span>, </span>
<span id="cb26-337"><a href="#cb26-337"></a>         <span class="at">transfer_category =</span> <span class="st">"hi"</span>,</span>
<span id="cb26-338"><a href="#cb26-338"></a>         <span class="at">transfer_volume_ul =</span> <span class="dv">1800</span>,</span>
<span id="cb26-339"><a href="#cb26-339"></a>         <span class="at">time_days =</span> <span class="dv">0</span>)</span>
<span id="cb26-340"><a href="#cb26-340"></a><span class="in">```</span></span>
<span id="cb26-341"><a href="#cb26-341"></a></span>
<span id="cb26-342"><a href="#cb26-342"></a><span class="fu">## Combine WGS and metaG</span></span>
<span id="cb26-343"><a href="#cb26-343"></a></span>
<span id="cb26-344"><a href="#cb26-344"></a>This includes all variants - both amino acid changing and non-changing</span>
<span id="cb26-347"><a href="#cb26-347"></a><span class="in">```{r}</span></span>
<span id="cb26-348"><a href="#cb26-348"></a>freq_mg_wgs <span class="ot">&lt;-</span> <span class="fu">left_join</span>(mgvars_filt_mb, </span>
<span id="cb26-349"><a href="#cb26-349"></a>                         <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metadata.tsv"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-350"><a href="#cb26-350"></a>  <span class="fu">filter</span>(transfer_category <span class="sc">==</span> <span class="st">"hi"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-351"><a href="#cb26-351"></a>  <span class="fu">bind_rows</span>(wgs_filt_mb) <span class="sc">%&gt;%</span> </span>
<span id="cb26-352"><a href="#cb26-352"></a>  <span class="fu">mutate</span>(<span class="at">pos =</span> <span class="fu">as.numeric</span>(pos),</span>
<span id="cb26-353"><a href="#cb26-353"></a>         <span class="at">depth_total =</span> <span class="fu">as.numeric</span>(depth_total)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-354"><a href="#cb26-354"></a>  <span class="fu">select</span>(<span class="sc">-</span>hgvs_c, <span class="sc">-</span>aa_pos, <span class="sc">-</span>aa_len)</span>
<span id="cb26-355"><a href="#cb26-355"></a><span class="in">```</span></span>
<span id="cb26-356"><a href="#cb26-356"></a></span>
<span id="cb26-357"><a href="#cb26-357"></a><span class="fu">## Expand grid for variants</span></span>
<span id="cb26-358"><a href="#cb26-358"></a></span>
<span id="cb26-359"><a href="#cb26-359"></a>Make a dataframe where all times/species/replicates/genomic position/experimental conditions are present</span>
<span id="cb26-362"><a href="#cb26-362"></a><span class="in">```{r}</span></span>
<span id="cb26-363"><a href="#cb26-363"></a>vars_united <span class="ot">&lt;-</span> freq_mg_wgs <span class="sc">%&gt;%</span> </span>
<span id="cb26-364"><a href="#cb26-364"></a>  <span class="fu">select</span>(chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb26-365"><a href="#cb26-365"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-366"><a href="#cb26-366"></a>  <span class="fu">unite</span>(<span class="st">"combo"</span>, <span class="fu">c</span>(chrom, pos, ref, alt), <span class="at">sep =</span> <span class="st">"|"</span>)</span>
<span id="cb26-367"><a href="#cb26-367"></a></span>
<span id="cb26-368"><a href="#cb26-368"></a>times_full <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(<span class="at">combo =</span> vars_united<span class="sc">$</span>combo, </span>
<span id="cb26-369"><a href="#cb26-369"></a>              <span class="at">time_days =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">28</span>, <span class="dv">60</span>), </span>
<span id="cb26-370"><a href="#cb26-370"></a>              <span class="at">replicate =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"C"</span>, <span class="st">"E"</span>),</span>
<span id="cb26-371"><a href="#cb26-371"></a>              <span class="at">prey_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>),</span>
<span id="cb26-372"><a href="#cb26-372"></a>              <span class="at">predator_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-373"><a href="#cb26-373"></a>  <span class="fu">separate_wider_delim</span>(combo, <span class="at">delim =</span> <span class="st">"|"</span>, <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"chrom"</span>, <span class="st">"pos"</span>, <span class="st">"ref"</span>, <span class="st">"alt"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-374"><a href="#cb26-374"></a>  <span class="fu">mutate</span>(<span class="at">pos =</span> <span class="fu">as.numeric</span>(pos),</span>
<span id="cb26-375"><a href="#cb26-375"></a>         <span class="at">transfer_category =</span> <span class="st">"hi"</span>,</span>
<span id="cb26-376"><a href="#cb26-376"></a>         <span class="at">transfer_volume_ul =</span> <span class="dv">1800</span>)</span>
<span id="cb26-377"><a href="#cb26-377"></a><span class="in">```</span></span>
<span id="cb26-378"><a href="#cb26-378"></a></span>
<span id="cb26-379"><a href="#cb26-379"></a><span class="fu">## Format metagenome coverage data</span></span>
<span id="cb26-380"><a href="#cb26-380"></a></span>
<span id="cb26-383"><a href="#cb26-383"></a><span class="in">```{r}</span></span>
<span id="cb26-384"><a href="#cb26-384"></a><span class="co">#| output: false</span></span>
<span id="cb26-385"><a href="#cb26-385"></a><span class="co">#| warning: false</span></span>
<span id="cb26-386"><a href="#cb26-386"></a><span class="co">#| error: false</span></span>
<span id="cb26-387"><a href="#cb26-387"></a>mg_cov <span class="ot">&lt;-</span> covslurpedfmtflt <span class="sc">%&gt;%</span>  </span>
<span id="cb26-388"><a href="#cb26-388"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> sps) <span class="sc">%&gt;%</span> </span>
<span id="cb26-389"><a href="#cb26-389"></a>  <span class="fu">left_join</span>(<span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metadata.tsv"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-390"><a href="#cb26-390"></a>  <span class="fu">filter</span>(transfer_category <span class="sc">==</span> <span class="st">"hi"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-391"><a href="#cb26-391"></a>  <span class="co"># samples with trimmed mean coverage &gt;=10 with over 99% of genome covered</span></span>
<span id="cb26-392"><a href="#cb26-392"></a>  <span class="co"># are to be trusted</span></span>
<span id="cb26-393"><a href="#cb26-393"></a>  <span class="fu">mutate</span>(<span class="at">cov_thresh_met =</span> <span class="fu">if_else</span>(trimmed_mean <span class="sc">&gt;=</span> <span class="dv">10</span> <span class="sc">&amp;</span> covered_fraction <span class="sc">&gt;</span> <span class="fl">0.99</span>, <span class="dv">1</span>, <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-394"><a href="#cb26-394"></a>  <span class="fu">select</span>(sample, strainID, <span class="at">chrom=</span>scaffold, trimmed_mean, covered_fraction, transfer_category<span class="sc">:</span>cov_thresh_met) </span>
<span id="cb26-395"><a href="#cb26-395"></a><span class="in">```</span></span>
<span id="cb26-396"><a href="#cb26-396"></a></span>
<span id="cb26-397"><a href="#cb26-397"></a><span class="fu"># Combine variants into complete time series</span></span>
<span id="cb26-400"><a href="#cb26-400"></a><span class="in">```{r}</span></span>
<span id="cb26-401"><a href="#cb26-401"></a>freq_full <span class="ot">&lt;-</span> <span class="fu">left_join</span>(times_full, freq_mg_wgs, <span class="at">relationship =</span> <span class="st">"one-to-one"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-402"><a href="#cb26-402"></a>  <span class="fu">arrange</span>(chrom, pos, ref, alt, replicate, predator_history, prey_history, time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb26-403"><a href="#cb26-403"></a>  <span class="co"># group_by(chrom) %&gt;% </span></span>
<span id="cb26-404"><a href="#cb26-404"></a>  <span class="fu">fill</span>(strainID, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-405"><a href="#cb26-405"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-406"><a href="#cb26-406"></a>  <span class="fu">group_by</span>(time_days, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb26-407"><a href="#cb26-407"></a>  <span class="fu">fill</span>(sample, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-408"><a href="#cb26-408"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-409"><a href="#cb26-409"></a>  <span class="fu">left_join</span>(mg_cov) <span class="sc">%&gt;%</span> </span>
<span id="cb26-410"><a href="#cb26-410"></a>  <span class="fu">mutate</span>(<span class="at">cov_thresh_met =</span> <span class="fu">if_else</span>(time_days <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, cov_thresh_met)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-411"><a href="#cb26-411"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt_complete =</span> <span class="fu">case_when</span>(cov_thresh_met <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(freq_alt) <span class="sc">~</span> freq_alt,</span>
<span id="cb26-412"><a href="#cb26-412"></a>                                       cov_thresh_met <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(freq_alt) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb26-413"><a href="#cb26-413"></a>                                       <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA_real_</span>))</span>
<span id="cb26-414"><a href="#cb26-414"></a><span class="in">```</span></span>
<span id="cb26-415"><a href="#cb26-415"></a></span>
<span id="cb26-416"><a href="#cb26-416"></a>Some filtering to reduce size. Also record the number of mutect2 passing variants and add a time-lagged variable. We remove variants that only have an observation at T0 and never in any metagenome aample due to low coverage (i.e. they are NA). This will reduce the size of the dataset and make sure we focus on more than just signals present in the WGS analysis</span>
<span id="cb26-419"><a href="#cb26-419"></a><span class="in">```{r}</span></span>
<span id="cb26-420"><a href="#cb26-420"></a>freq_full_sm01 <span class="ot">&lt;-</span> freq_full <span class="sc">%&gt;%</span> </span>
<span id="cb26-421"><a href="#cb26-421"></a>  <span class="fu">group_by</span>(chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb26-422"><a href="#cb26-422"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb26-423"><a href="#cb26-423"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-424"><a href="#cb26-424"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb26-425"><a href="#cb26-425"></a>  <span class="co"># can't be all zeros</span></span>
<span id="cb26-426"><a href="#cb26-426"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(freq_alt_complete, <span class="at">na.rm=</span><span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-427"><a href="#cb26-427"></a>  <span class="co"># exclude only NAs</span></span>
<span id="cb26-428"><a href="#cb26-428"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(freq_alt_complete[time_days <span class="sc">!=</span> <span class="dv">0</span>])) <span class="sc">&lt;</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-429"><a href="#cb26-429"></a>  <span class="co"># mark some statistics that may be useful for later filtering</span></span>
<span id="cb26-430"><a href="#cb26-430"></a>  <span class="fu">mutate</span>(<span class="at">n_gt_fthresh =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> T),</span>
<span id="cb26-431"><a href="#cb26-431"></a>         <span class="at">n_pass   =</span> <span class="fu">sum</span>(filter <span class="sc">==</span> <span class="st">"PASS"</span>, <span class="at">na.rm =</span> T),</span>
<span id="cb26-432"><a href="#cb26-432"></a>         <span class="at">n_cov_gt =</span> <span class="fu">sum</span>(cov_thresh_met, <span class="at">na.rm =</span>T)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-433"><a href="#cb26-433"></a>  <span class="co"># freq_alt_complete_t1=lag(freq_alt_complete, order_by = time_days, n=1)</span></span>
<span id="cb26-434"><a href="#cb26-434"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb26-435"><a href="#cb26-435"></a><span class="in">```</span></span>
<span id="cb26-436"><a href="#cb26-436"></a></span>
<span id="cb26-437"><a href="#cb26-437"></a>Here we require at least one observation with greater than 0.05 alt frequency and at least one variant passing mutect filters in the variant time series</span>
<span id="cb26-438"><a href="#cb26-438"></a></span>
<span id="cb26-441"><a href="#cb26-441"></a><span class="in">```{r}</span></span>
<span id="cb26-442"><a href="#cb26-442"></a>freq_full_sm02 <span class="ot">&lt;-</span> freq_full_sm01 <span class="sc">%&gt;%</span> </span>
<span id="cb26-443"><a href="#cb26-443"></a>  <span class="fu">filter</span>(n_gt_fthresh <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> n_pass <span class="sc">&gt;=</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-444"><a href="#cb26-444"></a>  <span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span> </span>
<span id="cb26-445"><a href="#cb26-445"></a>  <span class="fu">fill</span>(<span class="fu">c</span>(type, effect, impact, locus_tag, biotype, hgvs_p, cds_pos, cds_len), <span class="at">.direction =</span> <span class="st">"updown"</span>)</span>
<span id="cb26-446"><a href="#cb26-446"></a><span class="in">```</span></span>
<span id="cb26-447"><a href="#cb26-447"></a></span>
<span id="cb26-448"><a href="#cb26-448"></a><span class="fu"># Export</span></span>
<span id="cb26-451"><a href="#cb26-451"></a><span class="in">```{r}</span></span>
<span id="cb26-452"><a href="#cb26-452"></a><span class="co"># save table for later use</span></span>
<span id="cb26-453"><a href="#cb26-453"></a><span class="fu">write_tsv</span>(freq_full_sm02, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metagenome_variant_timeseries.tsv"</span>))</span>
<span id="cb26-454"><a href="#cb26-454"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>