<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-10-30">

<title>hambiDoubleEvo - Plot variant time series</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">Metagenome sequencing</a></li><li class="breadcrumb-item"><a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html">3. Variant plotting</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">hambiDoubleEvo</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/01_rpkm2tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/02_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Analyze and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Population WGS</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Metagenome sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">3. Variant plotting</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/04_mutation_dynamics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Mutational dynamics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">2</span> Read data</a></li>
  <li><a href="#clustering" id="toc-clustering" class="nav-link" data-scroll-target="#clustering"><span class="header-section-number">3</span> Clustering</a></li>
  <li><a href="#main-text-figure" id="toc-main-text-figure" class="nav-link" data-scroll-target="#main-text-figure"><span class="header-section-number">4</span> Main text figure</a>
  <ul class="collapse">
  <li><a href="#heatmap-all-species-together" id="toc-heatmap-all-species-together" class="nav-link" data-scroll-target="#heatmap-all-species-together"><span class="header-section-number">4.1</span> Heatmap all species together</a>
  <ul class="collapse">
  <li><a href="#all-alleles" id="toc-all-alleles" class="nav-link" data-scroll-target="#all-alleles"><span class="header-section-number">4.1.1</span> All alleles</a></li>
  <li><a href="#parallel-genes" id="toc-parallel-genes" class="nav-link" data-scroll-target="#parallel-genes"><span class="header-section-number">4.1.2</span> Parallel genes</a></li>
  </ul></li>
  <li><a href="#individual-allelle-trajectories" id="toc-individual-allelle-trajectories" class="nav-link" data-scroll-target="#individual-allelle-trajectories"><span class="header-section-number">4.2</span> Individual allelle trajectories</a>
  <ul class="collapse">
  <li><a href="#hambi_1287" id="toc-hambi_1287" class="nav-link" data-scroll-target="#hambi_1287"><span class="header-section-number">4.2.1</span> HAMBI_1287</a></li>
  <li><a href="#hambi_1972" id="toc-hambi_1972" class="nav-link" data-scroll-target="#hambi_1972"><span class="header-section-number">4.2.2</span> HAMBI_1972</a></li>
  <li><a href="#hambi_1977" id="toc-hambi_1977" class="nav-link" data-scroll-target="#hambi_1977"><span class="header-section-number">4.2.3</span> HAMBI_1977</a></li>
  <li><a href="#hambi_2659" id="toc-hambi_2659" class="nav-link" data-scroll-target="#hambi_2659"><span class="header-section-number">4.2.4</span> HAMBI_2659</a></li>
  <li><a href="#selection-of-parallel-genes-to-plot" id="toc-selection-of-parallel-genes-to-plot" class="nav-link" data-scroll-target="#selection-of-parallel-genes-to-plot"><span class="header-section-number">4.2.5</span> Selection of parallel genes to plot</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">Metagenome sequencing</a></li><li class="breadcrumb-item"><a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html">3. Variant plotting</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Plot variant time series</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 30, 2024</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook reads in the formatted metagenome variant frequencies and annotations. It makes some plots of the variant trajectories through time.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Libraries and global variables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(fishualize)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(withr)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb1-9"><a href="#cb1-9"></a></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Set up some directories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-8"><a href="#cb2-8"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read data</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># these were already filtered in the last step</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metagenome_variant_timeseries.tsv"</span>))</span>
<span id="cb3-3"><a href="#cb3-3"></a>degentab <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb3-4"><a href="#cb3-4"></a>genome_len <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_genome_len.tsv"</span>))</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co"># annotations</span></span>
<span id="cb3-6"><a href="#cb3-6"></a>annotations <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co"># significantly parallel genes</span></span>
<span id="cb3-8"><a href="#cb3-8"></a>par_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"enriched_parallel_genes.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>cog_description <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="sc">~</span>COG_category_single,                                                  <span class="sc">~</span>COG_category_long,</span>
<span id="cb4-3"><a href="#cb4-3"></a>            <span class="st">"J"</span>,               <span class="st">"J - Translation, ribosomal structure and biogenesis"</span>,</span>
<span id="cb4-4"><a href="#cb4-4"></a>            <span class="st">"A"</span>,                               <span class="st">"A - RNA processing and modification"</span>,</span>
<span id="cb4-5"><a href="#cb4-5"></a>            <span class="st">"K"</span>,                                                 <span class="st">"K – Transcription"</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>            <span class="st">"L"</span>,                         <span class="st">"L - Replication, recombination and repair"</span>,</span>
<span id="cb4-7"><a href="#cb4-7"></a>            <span class="st">"B"</span>,                              <span class="st">"B - Chromatin structure and dynamics"</span>,</span>
<span id="cb4-8"><a href="#cb4-8"></a>            <span class="st">"D"</span>,    <span class="st">"D - Cell cycle control, cell division, chromosome partitioning"</span>,</span>
<span id="cb4-9"><a href="#cb4-9"></a>            <span class="st">"Y"</span>,                                             <span class="st">"Y - Nuclear structure"</span>,</span>
<span id="cb4-10"><a href="#cb4-10"></a>            <span class="st">"V"</span>,                                            <span class="st">"V - Defense mechanisms"</span>,</span>
<span id="cb4-11"><a href="#cb4-11"></a>            <span class="st">"T"</span>,                                <span class="st">"T - Signal transduction mechanisms"</span>,</span>
<span id="cb4-12"><a href="#cb4-12"></a>            <span class="st">"M"</span>,                        <span class="st">"M - Cell wall/membrane/envelope biogenesis"</span>,</span>
<span id="cb4-13"><a href="#cb4-13"></a>            <span class="st">"N"</span>,                                                 <span class="st">"N - Cell motility"</span>,</span>
<span id="cb4-14"><a href="#cb4-14"></a>            <span class="st">"Z"</span>,                                                  <span class="st">"Z – Cytoskeleton"</span>,</span>
<span id="cb4-15"><a href="#cb4-15"></a>            <span class="st">"W"</span>,                                      <span class="st">"W - Extracellular structures"</span>,</span>
<span id="cb4-16"><a href="#cb4-16"></a>            <span class="st">"U"</span>, <span class="st">"U - Intracellular trafficking, secretion, and vesicular transport"</span>,</span>
<span id="cb4-17"><a href="#cb4-17"></a>            <span class="st">"O"</span>,  <span class="st">"O - Posttranslational modification, protein turnover, chaperones"</span>,</span>
<span id="cb4-18"><a href="#cb4-18"></a>            <span class="st">"X"</span>,                              <span class="st">"X - Mobilome: prophages, transposons"</span>,</span>
<span id="cb4-19"><a href="#cb4-19"></a>            <span class="st">"C"</span>,                              <span class="st">"C - Energy production and conversion"</span>,</span>
<span id="cb4-20"><a href="#cb4-20"></a>            <span class="st">"G"</span>,                         <span class="st">"G - Carbohydrate transport and metabolism"</span>,</span>
<span id="cb4-21"><a href="#cb4-21"></a>            <span class="st">"E"</span>,                           <span class="st">"E - Amino acid transport and metabolism"</span>,</span>
<span id="cb4-22"><a href="#cb4-22"></a>            <span class="st">"F"</span>,                           <span class="st">"F - Nucleotide transport and metabolism"</span>,</span>
<span id="cb4-23"><a href="#cb4-23"></a>            <span class="st">"H"</span>,                             <span class="st">"H - Coenzyme transport and metabolism"</span>,</span>
<span id="cb4-24"><a href="#cb4-24"></a>            <span class="st">"I"</span>,                                <span class="st">"I - Lipid transport and metabolism"</span>,</span>
<span id="cb4-25"><a href="#cb4-25"></a>            <span class="st">"P"</span>,                        <span class="st">"P - Inorganic ion transport and metabolism"</span>,</span>
<span id="cb4-26"><a href="#cb4-26"></a>            <span class="st">"Q"</span>,  <span class="st">"Q - Secondary metabolites biosynthesis, transport and catabolism"</span>,</span>
<span id="cb4-27"><a href="#cb4-27"></a>            <span class="st">"R"</span>,                              <span class="st">"R - General function prediction only"</span>,</span>
<span id="cb4-28"><a href="#cb4-28"></a>            <span class="st">"S"</span>,                                              <span class="st">"S - Function unknown"</span></span>
<span id="cb4-29"><a href="#cb4-29"></a>  )</span>
<span id="cb4-30"><a href="#cb4-30"></a></span>
<span id="cb4-31"><a href="#cb4-31"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">12367</span>,</span>
<span id="cb4-32"><a href="#cb4-32"></a>                 cogpal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(cog_description<span class="sc">$</span>COG_category_single)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb4-33"><a href="#cb4-33"></a>)</span>
<span id="cb4-34"><a href="#cb4-34"></a><span class="fu">names</span>(cogpal) <span class="ot">&lt;-</span> cog_description<span class="sc">$</span>COG_category_long</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="clustering" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Clustering</h1>
<p><del>One clustering approach is with the kml package - <a href="https://github.com/keithmcnulty/longitudinal_clustering">see tutorial here</a>. Another option would be to use the <code>curveRep</code> function from <code>Hmisc</code> - <a href="https://stats.stackexchange.com/a/17804">see more here</a>.</del></p>
<p>Classic old hierarchical clustering from hclust works best and is easiest to use</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># create unique group id as a convenience for later plotting</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>df_grouped <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="co">#filter(!str_detect(effect, "intergenic|intragenic|synonymous|fusion")) %&gt;% </span></span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="co">#filter(!str_detect(impact, "MODIFIER")) %&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="fu">relocate</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb5-8"><a href="#cb5-8"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb5-9"><a href="#cb5-9"></a></span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="co"># format the grouped dataframe in a way that can be plotted</span></span>
<span id="cb5-11"><a href="#cb5-11"></a>df2clust <span class="ot">&lt;-</span> df_grouped <span class="sc">%&gt;%</span> </span>
<span id="cb5-12"><a href="#cb5-12"></a>  <span class="fu">mutate</span>(<span class="at">day =</span> <span class="fu">paste0</span>(<span class="st">"day"</span>, time_days))  <span class="sc">%&gt;%</span> </span>
<span id="cb5-13"><a href="#cb5-13"></a>  <span class="fu">select</span>(group_id, day, freq_alt_complete) <span class="sc">%&gt;%</span> </span>
<span id="cb5-14"><a href="#cb5-14"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"day"</span>, <span class="at">values_from =</span> <span class="st">"freq_alt_complete"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-15"><a href="#cb5-15"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-16"><a href="#cb5-16"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"group_id"</span>)</span>
<span id="cb5-17"><a href="#cb5-17"></a></span>
<span id="cb5-18"><a href="#cb5-18"></a><span class="co"># scale the dataframe for clustering</span></span>
<span id="cb5-19"><a href="#cb5-19"></a>df2clust_scaled <span class="ot">&lt;-</span> <span class="fu">scale</span>(df2clust)</span>
<span id="cb5-20"><a href="#cb5-20"></a></span>
<span id="cb5-21"><a href="#cb5-21"></a><span class="co"># perform the hierarchcical clustering using euclidean distance and Ward's D</span></span>
<span id="cb5-22"><a href="#cb5-22"></a>hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(df2clust_scaled, <span class="at">method =</span> <span class="st">"euclidean"</span>), <span class="at">method =</span> <span class="st">"ward.D2"</span> )</span>
<span id="cb5-23"><a href="#cb5-23"></a></span>
<span id="cb5-24"><a href="#cb5-24"></a><span class="co"># get order of the clustered observations. The order is for the groups</span></span>
<span id="cb5-25"><a href="#cb5-25"></a>ord <span class="ot">&lt;-</span> hc<span class="sc">$</span>order</span>
<span id="cb5-26"><a href="#cb5-26"></a></span>
<span id="cb5-27"><a href="#cb5-27"></a><span class="co"># get cluster membership. Again memebership is for the groups</span></span>
<span id="cb5-28"><a href="#cb5-28"></a>myclust <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc, <span class="at">k =</span> <span class="dv">6</span>)</span>
<span id="cb5-29"><a href="#cb5-29"></a></span>
<span id="cb5-30"><a href="#cb5-30"></a>df2plot <span class="ot">&lt;-</span> df_grouped <span class="sc">%&gt;%</span> </span>
<span id="cb5-31"><a href="#cb5-31"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> myclust[group_id],</span>
<span id="cb5-32"><a href="#cb5-32"></a>         <span class="at">order =</span> ord[group_id]) <span class="sc">%&gt;%</span> </span>
<span id="cb5-33"><a href="#cb5-33"></a>  <span class="co"># convert group_id into a factor that is ordered by the hierarchical clustering</span></span>
<span id="cb5-34"><a href="#cb5-34"></a>  <span class="co"># done above</span></span>
<span id="cb5-35"><a href="#cb5-35"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">factor</span>(group_id, <span class="at">levels =</span> ord),</span>
<span id="cb5-36"><a href="#cb5-36"></a>         <span class="at">cluster =</span> <span class="fu">factor</span>(cluster)) <span class="sc">%&gt;%</span></span>
<span id="cb5-37"><a href="#cb5-37"></a>  <span class="fu">mutate</span>(<span class="at">day =</span> <span class="fu">factor</span>(time_days), </span>
<span id="cb5-38"><a href="#cb5-38"></a>         <span class="at">pos =</span> <span class="fu">factor</span>(pos),</span>
<span id="cb5-39"><a href="#cb5-39"></a>         <span class="at">mylab =</span> <span class="fu">interaction</span>(replicate, prey_history, predator_history, <span class="fu">paste0</span>(locus_tag, <span class="st">"_"</span>, pos)),</span>
<span id="cb5-40"><a href="#cb5-40"></a>         <span class="at">strainID2 =</span> <span class="fu">case_when</span>(strainID <span class="sc">==</span> <span class="st">"HAMBI_0403"</span> <span class="sc">~</span> <span class="dv">5</span>,</span>
<span id="cb5-41"><a href="#cb5-41"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span> <span class="sc">~</span> <span class="dv">4</span>,</span>
<span id="cb5-42"><a href="#cb5-42"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span> <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb5-43"><a href="#cb5-43"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span> <span class="sc">~</span> <span class="dv">3</span>,</span>
<span id="cb5-44"><a href="#cb5-44"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span> <span class="sc">~</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-45"><a href="#cb5-45"></a>  <span class="fu">mutate</span>(<span class="at">mylab2 =</span> <span class="fu">fct_reorder</span>(group_id, strainID2)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-46"><a href="#cb5-46"></a>  <span class="fu">relocate</span>(mylab2, order, strainID2) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="main-text-figure" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Main text figure</h1>
<p>Idea is to have a one heatmap for the alt allele frequencies for every detected mutation that passes our thresholds and then have some horiztonal bars which designate the different treatment combinations, the longitudinal clusters, and the species identieis stacked on top. Then below this we will have some simple line plots show the highly parallel mutations identified in the prior analysis.</p>
<section id="heatmap-all-species-together" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="heatmap-all-species-together"><span class="header-section-number">4.1</span> Heatmap all species together</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>blank_theme01 <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="fu">theme</span>(</span>
<span id="cb6-3"><a href="#cb6-3"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-4"><a href="#cb6-4"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-5"><a href="#cb6-5"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-6"><a href="#cb6-6"></a>    <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-7"><a href="#cb6-7"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-8"><a href="#cb6-8"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-9"><a href="#cb6-9"></a>    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">"cm"</span>),</span>
<span id="cb6-10"><a href="#cb6-10"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-11"><a href="#cb6-11"></a>    <span class="at">strip.text =</span> <span class="fu">element_blank</span>()</span>
<span id="cb6-12"><a href="#cb6-12"></a>  )</span>
<span id="cb6-13"><a href="#cb6-13"></a>}</span>
<span id="cb6-14"><a href="#cb6-14"></a></span>
<span id="cb6-15"><a href="#cb6-15"></a>blank_theme02 <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb6-16"><a href="#cb6-16"></a>  <span class="fu">theme</span>(</span>
<span id="cb6-17"><a href="#cb6-17"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-18"><a href="#cb6-18"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-19"><a href="#cb6-19"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-20"><a href="#cb6-20"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-21"><a href="#cb6-21"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-22"><a href="#cb6-22"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-23"><a href="#cb6-23"></a>    <span class="at">strip.text =</span> <span class="fu">element_blank</span>()</span>
<span id="cb6-24"><a href="#cb6-24"></a>  )</span>
<span id="cb6-25"><a href="#cb6-25"></a>}</span>
<span id="cb6-26"><a href="#cb6-26"></a></span>
<span id="cb6-27"><a href="#cb6-27"></a>phoriz_bar <span class="ot">&lt;-</span> <span class="cf">function</span>(df, fillvar, mypal){</span>
<span id="cb6-28"><a href="#cb6-28"></a>  <span class="co"># only take the first time point so to make one bar</span></span>
<span id="cb6-29"><a href="#cb6-29"></a>  <span class="fu">filter</span>(df, day <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-30"><a href="#cb6-30"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> day,  <span class="at">x=</span>mylab2, <span class="at">fill =</span> {{ fillvar }})) <span class="sc">+</span></span>
<span id="cb6-31"><a href="#cb6-31"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb6-32"><a href="#cb6-32"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb6-33"><a href="#cb6-33"></a>    <span class="fu">facet_wrap</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb6-34"><a href="#cb6-34"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> {{ mypal }}) <span class="sc">+</span> </span>
<span id="cb6-35"><a href="#cb6-35"></a>    <span class="fu">blank_theme01</span>()</span>
<span id="cb6-36"><a href="#cb6-36"></a>}</span>
<span id="cb6-37"><a href="#cb6-37"></a></span>
<span id="cb6-38"><a href="#cb6-38"></a>pheat_nolb <span class="ot">&lt;-</span> <span class="cf">function</span>(df, fillvar){</span>
<span id="cb6-39"><a href="#cb6-39"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> day,  <span class="at">x=</span>mylab2, <span class="at">fill =</span> {{ fillvar }} )) <span class="sc">+</span></span>
<span id="cb6-40"><a href="#cb6-40"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb6-41"><a href="#cb6-41"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb6-42"><a href="#cb6-42"></a>    <span class="fu">facet_wrap</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb6-43"><a href="#cb6-43"></a>    <span class="fu">scale_fill_viridis_c</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">trans =</span> <span class="st">"sqrt"</span>) <span class="sc">+</span></span>
<span id="cb6-44"><a href="#cb6-44"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb6-45"><a href="#cb6-45"></a>    <span class="fu">blank_theme01</span>()</span>
<span id="cb6-46"><a href="#cb6-46"></a>}</span>
<span id="cb6-47"><a href="#cb6-47"></a></span>
<span id="cb6-48"><a href="#cb6-48"></a>pheat_labs <span class="ot">&lt;-</span> <span class="cf">function</span>(df, fillvar){</span>
<span id="cb6-49"><a href="#cb6-49"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> day,  <span class="at">x=</span>mylab2, <span class="at">fill =</span> {{ fillvar }})) <span class="sc">+</span></span>
<span id="cb6-50"><a href="#cb6-50"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb6-51"><a href="#cb6-51"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb6-52"><a href="#cb6-52"></a>    <span class="fu">facet_wrap</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb6-53"><a href="#cb6-53"></a>    <span class="fu">scale_fill_viridis_c</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">trans =</span> <span class="st">"sqrt"</span>) <span class="sc">+</span></span>
<span id="cb6-54"><a href="#cb6-54"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb6-55"><a href="#cb6-55"></a>    <span class="fu">blank_theme02</span>()</span>
<span id="cb6-56"><a href="#cb6-56"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="all-alleles" class="level3" data-number="4.1.1">
<h3 data-number="4.1.1" class="anchored" data-anchor-id="all-alleles"><span class="header-section-number">4.1.1</span> All alleles</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">#clupal &lt;- unname(createPalette(length(unique(myclust)), c("#F3874AFF", "#FCD125FF"), M=5000))</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>clupal <span class="ot">&lt;-</span> <span class="fu">gray.colors</span>(<span class="fu">length</span>(<span class="fu">unique</span>(myclust)), <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="fl">0.9</span>, <span class="at">gamma =</span> <span class="fl">2.2</span>, <span class="dv">1</span>, <span class="at">rev =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="fu">names</span>(clupal) <span class="ot">&lt;-</span> <span class="fu">unique</span>(myclust)</span>
<span id="cb7-4"><a href="#cb7-4"></a></span>
<span id="cb7-5"><a href="#cb7-5"></a>pheat_all <span class="ot">&lt;-</span> <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), cluster, clupal) <span class="sc">+</span></span>
<span id="cb7-6"><a href="#cb7-6"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), freq_alt_complete) <span class="sc">+</span></span>
<span id="cb7-7"><a href="#cb7-7"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb7-9"><a href="#cb7-9"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb7-10"><a href="#cb7-10"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb7-11"><a href="#cb7-11"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb7-12"><a href="#cb7-12"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb7-13"><a href="#cb7-13"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_0403"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb7-14"><a href="#cb7-14"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_0403"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb7-15"><a href="#cb7-15"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">10</span>,</span>
<span id="cb7-16"><a href="#cb7-16"></a>              <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>),</span>
<span id="cb7-17"><a href="#cb7-17"></a>              <span class="at">guides =</span> <span class="st">"collect"</span>)</span>
<span id="cb7-18"><a href="#cb7-18"></a></span>
<span id="cb7-19"><a href="#cb7-19"></a>pheat_all</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<section id="save" class="level4" data-number="4.1.1.1">
<h4 data-number="4.1.1.1" class="anchored" data-anchor-id="save"><span class="header-section-number">4.1.1.1</span> Save</h4>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"heatmap_all_alleles.svg"</span>), pheat_all, <span class="at">width=</span><span class="dv">11</span>, <span class="at">height=</span><span class="dv">4</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb8-2"><a href="#cb8-2"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="parallel-genes" class="level3" data-number="4.1.2">
<h3 data-number="4.1.2" class="anchored" data-anchor-id="parallel-genes"><span class="header-section-number">4.1.2</span> Parallel genes</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>pdfr_evo <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, df2plot, </span>
<span id="cb9-2"><a href="#cb9-2"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="fu">filter</span>(prey_history <span class="sc">==</span> <span class="st">"evo"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4"></a>  <span class="co"># some manual renaming</span></span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">case_when</span>(locus_tag <span class="sc">==</span> <span class="st">"H1287_02172"</span> <span class="sc">~</span> <span class="st">"yddV_2"</span>,</span>
<span id="cb9-6"><a href="#cb9-6"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1977_02612"</span> <span class="sc">~</span> <span class="st">"dppA/oppA"</span>,</span>
<span id="cb9-7"><a href="#cb9-7"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00299"</span> <span class="sc">~</span> <span class="st">"PAP2_like"</span>,</span>
<span id="cb9-8"><a href="#cb9-8"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02826"</span> <span class="sc">~</span> <span class="st">"ynjC"</span>,</span>
<span id="cb9-9"><a href="#cb9-9"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00671"</span> <span class="sc">~</span> <span class="st">"yqaA"</span>,</span>
<span id="cb9-10"><a href="#cb9-10"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00421"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb9-11"><a href="#cb9-11"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02723"</span> <span class="sc">~</span> <span class="st">"yifE"</span>,</span>
<span id="cb9-12"><a href="#cb9-12"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H2659_01789"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb9-13"><a href="#cb9-13"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-14"><a href="#cb9-14"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-15"><a href="#cb9-15"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, locus_tag, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-16"><a href="#cb9-16"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">paste0</span>(gene_lab, <span class="st">" | "</span>, replicate, <span class="st">" | "</span>,  cluster, <span class="st">"|"</span>, group_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-17"><a href="#cb9-17"></a>  <span class="fu">arrange</span>(COG_category_long, gene_lab) <span class="sc">%&gt;%</span> </span>
<span id="cb9-18"><a href="#cb9-18"></a>  <span class="fu">mutate</span>(<span class="at">mylab2 =</span> <span class="fu">factor</span>(gene_lab, <span class="fu">unique</span>(gene_lab)))</span>
<span id="cb9-19"><a href="#cb9-19"></a></span>
<span id="cb9-20"><a href="#cb9-20"></a>pdfr_anc <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, df2plot, </span>
<span id="cb9-21"><a href="#cb9-21"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-22"><a href="#cb9-22"></a>  <span class="fu">filter</span>(prey_history <span class="sc">==</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-23"><a href="#cb9-23"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">case_when</span>(locus_tag <span class="sc">==</span> <span class="st">"H1972_02826"</span> <span class="sc">~</span> <span class="st">"ynjC"</span>,</span>
<span id="cb9-24"><a href="#cb9-24"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00421"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb9-25"><a href="#cb9-25"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-26"><a href="#cb9-26"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-27"><a href="#cb9-27"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, locus_tag, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-28"><a href="#cb9-28"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">paste0</span>(gene_lab, <span class="st">" | "</span>, replicate, <span class="st">" | "</span>,  cluster, <span class="st">"|"</span>, group_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-29"><a href="#cb9-29"></a>  <span class="fu">arrange</span>(COG_category_long, gene_lab) <span class="sc">%&gt;%</span> </span>
<span id="cb9-30"><a href="#cb9-30"></a>  <span class="fu">mutate</span>(<span class="at">mylab2 =</span> <span class="fu">factor</span>(gene_lab, <span class="fu">unique</span>(gene_lab)))</span>
<span id="cb9-31"><a href="#cb9-31"></a>  </span>
<span id="cb9-32"><a href="#cb9-32"></a><span class="co"># evolved</span></span>
<span id="cb9-33"><a href="#cb9-33"></a>pheat_par <span class="ot">&lt;-</span> <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb9-34"><a href="#cb9-34"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb9-35"><a href="#cb9-35"></a>  <span class="co"># evolved 1972</span></span>
<span id="cb9-36"><a href="#cb9-36"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb9-37"><a href="#cb9-37"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb9-38"><a href="#cb9-38"></a>  <span class="co"># ancestral 1972 - the only species that has significantly parallel genes in the ancestral treatment</span></span>
<span id="cb9-39"><a href="#cb9-39"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_anc, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb9-40"><a href="#cb9-40"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_anc, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb9-41"><a href="#cb9-41"></a>  <span class="co"># evolved 1977</span></span>
<span id="cb9-42"><a href="#cb9-42"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb9-43"><a href="#cb9-43"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb9-44"><a href="#cb9-44"></a>  <span class="co"># evolved 1287</span></span>
<span id="cb9-45"><a href="#cb9-45"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb9-46"><a href="#cb9-46"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb9-47"><a href="#cb9-47"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">10</span>,</span>
<span id="cb9-48"><a href="#cb9-48"></a>              <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>),</span>
<span id="cb9-49"><a href="#cb9-49"></a>              <span class="at">guides =</span> <span class="st">"collect"</span>)</span>
<span id="cb9-50"><a href="#cb9-50"></a></span>
<span id="cb9-51"><a href="#cb9-51"></a>pheat_par</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<section id="save-1" class="level4" data-number="4.1.2.1">
<h4 data-number="4.1.2.1" class="anchored" data-anchor-id="save-1"><span class="header-section-number">4.1.2.1</span> Save</h4>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"heatmap_par_alleles.svg"</span>), pheat_par, <span class="at">width=</span><span class="dv">12</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb10-2"><a href="#cb10-2"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="individual-allelle-trajectories" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="individual-allelle-trajectories"><span class="header-section-number">4.2</span> Individual allelle trajectories</h2>
<p>Read data from parallelism analysis</p>
<p>Function for plotting trajectories for indivdidual genes</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>plotpargenes <span class="ot">&lt;-</span> <span class="cf">function</span>(strainID){</span>
<span id="cb11-2"><a href="#cb11-2"></a>  pdfr <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, df2plot, </span>
<span id="cb11-3"><a href="#cb11-3"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="co"># to reduce size only include genes that are mutated in at least two of the replicates</span></span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">filter</span>(n_replicate <span class="sc">&gt;=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="co"># name the genes for plotting</span></span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene),</span>
<span id="cb11-8"><a href="#cb11-8"></a>         <span class="at">treat =</span> <span class="fu">paste0</span>(<span class="st">"prey:"</span>, prey_history, <span class="st">" | pred:"</span>, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-9"><a href="#cb11-9"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, Description, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-10"><a href="#cb11-10"></a>  <span class="fu">filter</span>(strainID <span class="sc">==</span> {{ strainID }})</span>
<span id="cb11-11"><a href="#cb11-11"></a>  </span>
<span id="cb11-12"><a href="#cb11-12"></a>  <span class="co"># make random color set to help differentiate alleles </span></span>
<span id="cb11-13"><a href="#cb11-13"></a>  mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(pdfr<span class="sc">$</span>mylab2)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb11-14"><a href="#cb11-14"></a>  </span>
<span id="cb11-15"><a href="#cb11-15"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb11-16"><a href="#cb11-16"></a>  <span class="fu">geom_line</span>(<span class="at">data=</span>pdfr[<span class="sc">!</span><span class="fu">is.na</span>(pdfr<span class="sc">$</span>freq_alt_complete), ],</span>
<span id="cb11-17"><a href="#cb11-17"></a>            <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete,</span>
<span id="cb11-18"><a href="#cb11-18"></a>                <span class="at">group =</span> mylab2,</span>
<span id="cb11-19"><a href="#cb11-19"></a>                <span class="at">color =</span> hgvs_p)) <span class="sc">+</span></span>
<span id="cb11-20"><a href="#cb11-20"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> pdfr, </span>
<span id="cb11-21"><a href="#cb11-21"></a>             <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete, <span class="at">shape =</span> replicate), </span>
<span id="cb11-22"><a href="#cb11-22"></a>             <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb11-23"><a href="#cb11-23"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb11-24"><a href="#cb11-24"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal) <span class="sc">+</span> </span>
<span id="cb11-25"><a href="#cb11-25"></a>  <span class="fu">facet_wrap</span>(gene_lab <span class="sc">~</span> treat) <span class="sc">+</span></span>
<span id="cb11-26"><a href="#cb11-26"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Days"</span>, <span class="at">y =</span> <span class="st">"Allele frequency"</span>) <span class="sc">+</span></span>
<span id="cb11-27"><a href="#cb11-27"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb11-28"><a href="#cb11-28"></a>  <span class="fu">theme</span>(</span>
<span id="cb11-29"><a href="#cb11-29"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb11-30"><a href="#cb11-30"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb11-31"><a href="#cb11-31"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-32"><a href="#cb11-32"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb11-33"><a href="#cb11-33"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="hambi_1287" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="hambi_1287"><span class="header-section-number">4.2.1</span> HAMBI_1287</h3>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Nonsynonymous variant frequencies (vertical axis) over time (horizontal axis) for combinations of individual genes (title bar top row) and treatment combinations (title bar bottom row). Point shape denotes the biological replicate from the treatment and line colors indicate the amino acid changing variant (some genes have multiple non-synonymous variants at different positions). Focal genes are from <em>Citrobacter koserii</em> HAMBI_1287 and are those exhibiting significant parallelism for the duration of the experiment.
</figcaption>
</figure>
</div>
</section>
<section id="hambi_1972" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="hambi_1972"><span class="header-section-number">4.2.2</span> HAMBI_1972</h3>
<div id="fig-02" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: As in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> but for <em>Aeromonas caviae</em> HAMBI_1972.
</figcaption>
</figure>
</div>
</section>
<section id="hambi_1977" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="hambi_1977"><span class="header-section-number">4.2.3</span> HAMBI_1977</h3>
<div id="fig-03" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 9 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: As in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> but for <em>Pseudomonas chlororaphis</em> HAMBI_1977.
</figcaption>
</figure>
</div>
</section>
<section id="hambi_2659" class="level3" data-number="4.2.4">
<h3 data-number="4.2.4" class="anchored" data-anchor-id="hambi_2659"><span class="header-section-number">4.2.4</span> HAMBI_2659</h3>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 12 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: As in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> but for <em>Stenotrophomonas maltophilia</em> HAMBI_2659.
</figcaption>
</figure>
</div>
</section>
<section id="selection-of-parallel-genes-to-plot" class="level3" data-number="4.2.5">
<h3 data-number="4.2.5" class="anchored" data-anchor-id="selection-of-parallel-genes-to-plot"><span class="header-section-number">4.2.5</span> Selection of parallel genes to plot</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>pdfr <span class="ot">&lt;-</span> par_genes <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2"></a>  <span class="fu">filter</span>(n_replicate <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(gene_multiplicity_m_i)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4"></a>  <span class="co"># take top 20 highest multiplicitiies</span></span>
<span id="cb14-5"><a href="#cb14-5"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-6"><a href="#cb14-6"></a>  <span class="fu">left_join</span>(df2plot, <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7"></a>  <span class="co"># name the genes for plotting</span></span>
<span id="cb14-8"><a href="#cb14-8"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene),</span>
<span id="cb14-9"><a href="#cb14-9"></a>         <span class="at">treat =</span> <span class="fu">paste0</span>(<span class="st">"prey:"</span>, prey_history, <span class="st">" | pred:"</span>, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-10"><a href="#cb14-10"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, Description, gene_lab))</span>
<span id="cb14-11"><a href="#cb14-11"></a></span>
<span id="cb14-12"><a href="#cb14-12"></a></span>
<span id="cb14-13"><a href="#cb14-13"></a>mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(pdfr<span class="sc">$</span>mylab2)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb14-14"><a href="#cb14-14"></a>  </span>
<span id="cb14-15"><a href="#cb14-15"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb14-16"><a href="#cb14-16"></a>  <span class="fu">geom_line</span>(<span class="at">data=</span>pdfr[<span class="sc">!</span><span class="fu">is.na</span>(pdfr<span class="sc">$</span>freq_alt_complete), ],</span>
<span id="cb14-17"><a href="#cb14-17"></a>            <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete,</span>
<span id="cb14-18"><a href="#cb14-18"></a>                <span class="at">group =</span> mylab2,</span>
<span id="cb14-19"><a href="#cb14-19"></a>                <span class="at">color =</span> hgvs_p)) <span class="sc">+</span></span>
<span id="cb14-20"><a href="#cb14-20"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> pdfr, </span>
<span id="cb14-21"><a href="#cb14-21"></a>             <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete, <span class="at">shape =</span> replicate), </span>
<span id="cb14-22"><a href="#cb14-22"></a>             <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb14-23"><a href="#cb14-23"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb14-24"><a href="#cb14-24"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal) <span class="sc">+</span> </span>
<span id="cb14-25"><a href="#cb14-25"></a>  <span class="fu">facet_wrap</span>(gene_lab <span class="sc">~</span> treat) <span class="sc">+</span></span>
<span id="cb14-26"><a href="#cb14-26"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Days"</span>, <span class="at">y =</span> <span class="st">"Allele frequency"</span>) <span class="sc">+</span></span>
<span id="cb14-27"><a href="#cb14-27"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb14-28"><a href="#cb14-28"></a>  <span class="fu">theme</span>(</span>
<span id="cb14-29"><a href="#cb14-29"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb14-30"><a href="#cb14-30"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb14-31"><a href="#cb14-31"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb14-32"><a href="#cb14-32"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-05" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 11 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03_analyze_metagenome_variant_timeseries_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: As in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> but for the genes with the top 20 higest gene multiplicities across HAMBI_1287, HAMBI_1972, HAMBI_1977, and HAMBI_2659.
</figcaption>
</figure>
</div>


<!-- -->

</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb16" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb16-1"><a href="#cb16-1"></a><span class="co">---</span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="an">title:</span><span class="co"> "Plot variant time series"</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb16-5"><a href="#cb16-5"></a><span class="an">abstract:</span><span class="co"> "This notebook reads in the formatted metagenome variant frequencies and annotations. It makes some plots of the variant trajectories through time."</span></span>
<span id="cb16-6"><a href="#cb16-6"></a><span class="co">---</span></span>
<span id="cb16-7"><a href="#cb16-7"></a></span>
<span id="cb16-8"><a href="#cb16-8"></a><span class="fu"># Setup</span></span>
<span id="cb16-9"><a href="#cb16-9"></a></span>
<span id="cb16-10"><a href="#cb16-10"></a>Libraries and global variables</span>
<span id="cb16-11"><a href="#cb16-11"></a></span>
<span id="cb16-14"><a href="#cb16-14"></a><span class="in">```{r}</span></span>
<span id="cb16-15"><a href="#cb16-15"></a><span class="co">#| output: false</span></span>
<span id="cb16-16"><a href="#cb16-16"></a><span class="co">#| warning: false</span></span>
<span id="cb16-17"><a href="#cb16-17"></a><span class="co">#| error: false</span></span>
<span id="cb16-18"><a href="#cb16-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb16-19"><a href="#cb16-19"></a><span class="fu">library</span>(here)</span>
<span id="cb16-20"><a href="#cb16-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb16-21"><a href="#cb16-21"></a><span class="fu">library</span>(fishualize)</span>
<span id="cb16-22"><a href="#cb16-22"></a><span class="fu">library</span>(withr)</span>
<span id="cb16-23"><a href="#cb16-23"></a><span class="fu">library</span>(scales)</span>
<span id="cb16-24"><a href="#cb16-24"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb16-25"><a href="#cb16-25"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb16-26"><a href="#cb16-26"></a></span>
<span id="cb16-27"><a href="#cb16-27"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb16-28"><a href="#cb16-28"></a><span class="in">```</span></span>
<span id="cb16-29"><a href="#cb16-29"></a></span>
<span id="cb16-30"><a href="#cb16-30"></a>Set up some directories</span>
<span id="cb16-31"><a href="#cb16-31"></a></span>
<span id="cb16-34"><a href="#cb16-34"></a><span class="in">```{r}</span></span>
<span id="cb16-35"><a href="#cb16-35"></a><span class="co">#| output: false</span></span>
<span id="cb16-36"><a href="#cb16-36"></a><span class="co">#| warning: false</span></span>
<span id="cb16-37"><a href="#cb16-37"></a><span class="co">#| error: false</span></span>
<span id="cb16-38"><a href="#cb16-38"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb16-39"><a href="#cb16-39"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb16-40"><a href="#cb16-40"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb16-41"><a href="#cb16-41"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb16-42"><a href="#cb16-42"></a></span>
<span id="cb16-43"><a href="#cb16-43"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb16-44"><a href="#cb16-44"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb16-45"><a href="#cb16-45"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span>
<span id="cb16-46"><a href="#cb16-46"></a><span class="in">```</span></span>
<span id="cb16-47"><a href="#cb16-47"></a></span>
<span id="cb16-48"><a href="#cb16-48"></a><span class="fu"># Read data</span></span>
<span id="cb16-49"><a href="#cb16-49"></a></span>
<span id="cb16-52"><a href="#cb16-52"></a><span class="in">```{r}</span></span>
<span id="cb16-53"><a href="#cb16-53"></a><span class="co">#| output: false</span></span>
<span id="cb16-54"><a href="#cb16-54"></a><span class="co">#| warning: false</span></span>
<span id="cb16-55"><a href="#cb16-55"></a><span class="co">#| error: false</span></span>
<span id="cb16-56"><a href="#cb16-56"></a></span>
<span id="cb16-57"><a href="#cb16-57"></a><span class="co"># these were already filtered in the last step</span></span>
<span id="cb16-58"><a href="#cb16-58"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metagenome_variant_timeseries.tsv"</span>))</span>
<span id="cb16-59"><a href="#cb16-59"></a>degentab <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb16-60"><a href="#cb16-60"></a>genome_len <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_genome_len.tsv"</span>))</span>
<span id="cb16-61"><a href="#cb16-61"></a><span class="co"># annotations</span></span>
<span id="cb16-62"><a href="#cb16-62"></a>annotations <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb16-63"><a href="#cb16-63"></a><span class="co"># significantly parallel genes</span></span>
<span id="cb16-64"><a href="#cb16-64"></a>par_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"enriched_parallel_genes.tsv"</span>))</span>
<span id="cb16-65"><a href="#cb16-65"></a></span>
<span id="cb16-66"><a href="#cb16-66"></a><span class="in">```</span></span>
<span id="cb16-67"><a href="#cb16-67"></a></span>
<span id="cb16-68"><a href="#cb16-68"></a></span>
<span id="cb16-71"><a href="#cb16-71"></a><span class="in">```{r}</span></span>
<span id="cb16-72"><a href="#cb16-72"></a>cog_description <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb16-73"><a href="#cb16-73"></a>  <span class="sc">~</span>COG_category_single,                                                  <span class="sc">~</span>COG_category_long,</span>
<span id="cb16-74"><a href="#cb16-74"></a>            <span class="st">"J"</span>,               <span class="st">"J - Translation, ribosomal structure and biogenesis"</span>,</span>
<span id="cb16-75"><a href="#cb16-75"></a>            <span class="st">"A"</span>,                               <span class="st">"A - RNA processing and modification"</span>,</span>
<span id="cb16-76"><a href="#cb16-76"></a>            <span class="st">"K"</span>,                                                 <span class="st">"K – Transcription"</span>,</span>
<span id="cb16-77"><a href="#cb16-77"></a>            <span class="st">"L"</span>,                         <span class="st">"L - Replication, recombination and repair"</span>,</span>
<span id="cb16-78"><a href="#cb16-78"></a>            <span class="st">"B"</span>,                              <span class="st">"B - Chromatin structure and dynamics"</span>,</span>
<span id="cb16-79"><a href="#cb16-79"></a>            <span class="st">"D"</span>,    <span class="st">"D - Cell cycle control, cell division, chromosome partitioning"</span>,</span>
<span id="cb16-80"><a href="#cb16-80"></a>            <span class="st">"Y"</span>,                                             <span class="st">"Y - Nuclear structure"</span>,</span>
<span id="cb16-81"><a href="#cb16-81"></a>            <span class="st">"V"</span>,                                            <span class="st">"V - Defense mechanisms"</span>,</span>
<span id="cb16-82"><a href="#cb16-82"></a>            <span class="st">"T"</span>,                                <span class="st">"T - Signal transduction mechanisms"</span>,</span>
<span id="cb16-83"><a href="#cb16-83"></a>            <span class="st">"M"</span>,                        <span class="st">"M - Cell wall/membrane/envelope biogenesis"</span>,</span>
<span id="cb16-84"><a href="#cb16-84"></a>            <span class="st">"N"</span>,                                                 <span class="st">"N - Cell motility"</span>,</span>
<span id="cb16-85"><a href="#cb16-85"></a>            <span class="st">"Z"</span>,                                                  <span class="st">"Z – Cytoskeleton"</span>,</span>
<span id="cb16-86"><a href="#cb16-86"></a>            <span class="st">"W"</span>,                                      <span class="st">"W - Extracellular structures"</span>,</span>
<span id="cb16-87"><a href="#cb16-87"></a>            <span class="st">"U"</span>, <span class="st">"U - Intracellular trafficking, secretion, and vesicular transport"</span>,</span>
<span id="cb16-88"><a href="#cb16-88"></a>            <span class="st">"O"</span>,  <span class="st">"O - Posttranslational modification, protein turnover, chaperones"</span>,</span>
<span id="cb16-89"><a href="#cb16-89"></a>            <span class="st">"X"</span>,                              <span class="st">"X - Mobilome: prophages, transposons"</span>,</span>
<span id="cb16-90"><a href="#cb16-90"></a>            <span class="st">"C"</span>,                              <span class="st">"C - Energy production and conversion"</span>,</span>
<span id="cb16-91"><a href="#cb16-91"></a>            <span class="st">"G"</span>,                         <span class="st">"G - Carbohydrate transport and metabolism"</span>,</span>
<span id="cb16-92"><a href="#cb16-92"></a>            <span class="st">"E"</span>,                           <span class="st">"E - Amino acid transport and metabolism"</span>,</span>
<span id="cb16-93"><a href="#cb16-93"></a>            <span class="st">"F"</span>,                           <span class="st">"F - Nucleotide transport and metabolism"</span>,</span>
<span id="cb16-94"><a href="#cb16-94"></a>            <span class="st">"H"</span>,                             <span class="st">"H - Coenzyme transport and metabolism"</span>,</span>
<span id="cb16-95"><a href="#cb16-95"></a>            <span class="st">"I"</span>,                                <span class="st">"I - Lipid transport and metabolism"</span>,</span>
<span id="cb16-96"><a href="#cb16-96"></a>            <span class="st">"P"</span>,                        <span class="st">"P - Inorganic ion transport and metabolism"</span>,</span>
<span id="cb16-97"><a href="#cb16-97"></a>            <span class="st">"Q"</span>,  <span class="st">"Q - Secondary metabolites biosynthesis, transport and catabolism"</span>,</span>
<span id="cb16-98"><a href="#cb16-98"></a>            <span class="st">"R"</span>,                              <span class="st">"R - General function prediction only"</span>,</span>
<span id="cb16-99"><a href="#cb16-99"></a>            <span class="st">"S"</span>,                                              <span class="st">"S - Function unknown"</span></span>
<span id="cb16-100"><a href="#cb16-100"></a>  )</span>
<span id="cb16-101"><a href="#cb16-101"></a></span>
<span id="cb16-102"><a href="#cb16-102"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">12367</span>,</span>
<span id="cb16-103"><a href="#cb16-103"></a>                 cogpal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(cog_description<span class="sc">$</span>COG_category_single)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb16-104"><a href="#cb16-104"></a>)</span>
<span id="cb16-105"><a href="#cb16-105"></a><span class="fu">names</span>(cogpal) <span class="ot">&lt;-</span> cog_description<span class="sc">$</span>COG_category_long</span>
<span id="cb16-106"><a href="#cb16-106"></a><span class="in">```</span></span>
<span id="cb16-107"><a href="#cb16-107"></a></span>
<span id="cb16-108"><a href="#cb16-108"></a><span class="fu"># Clustering</span></span>
<span id="cb16-109"><a href="#cb16-109"></a></span>
<span id="cb16-110"><a href="#cb16-110"></a>~~One clustering approach is with the kml package - [see tutorial here](https://github.com/keithmcnulty/longitudinal_clustering). Another option would be to use the `curveRep` function from `Hmisc` - [see more here](https://stats.stackexchange.com/a/17804).~~</span>
<span id="cb16-111"><a href="#cb16-111"></a></span>
<span id="cb16-112"><a href="#cb16-112"></a>Classic old hierarchical clustering from hclust works best and is easiest to use</span>
<span id="cb16-113"><a href="#cb16-113"></a></span>
<span id="cb16-116"><a href="#cb16-116"></a><span class="in">```{r}</span></span>
<span id="cb16-117"><a href="#cb16-117"></a><span class="co"># create unique group id as a convenience for later plotting</span></span>
<span id="cb16-118"><a href="#cb16-118"></a>df_grouped <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb16-119"><a href="#cb16-119"></a>  <span class="co">#filter(!str_detect(effect, "intergenic|intragenic|synonymous|fusion")) %&gt;% </span></span>
<span id="cb16-120"><a href="#cb16-120"></a>  <span class="co">#filter(!str_detect(impact, "MODIFIER")) %&gt;%</span></span>
<span id="cb16-121"><a href="#cb16-121"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb16-122"><a href="#cb16-122"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb16-123"><a href="#cb16-123"></a>  <span class="fu">relocate</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb16-124"><a href="#cb16-124"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb16-125"><a href="#cb16-125"></a></span>
<span id="cb16-126"><a href="#cb16-126"></a><span class="co"># format the grouped dataframe in a way that can be plotted</span></span>
<span id="cb16-127"><a href="#cb16-127"></a>df2clust <span class="ot">&lt;-</span> df_grouped <span class="sc">%&gt;%</span> </span>
<span id="cb16-128"><a href="#cb16-128"></a>  <span class="fu">mutate</span>(<span class="at">day =</span> <span class="fu">paste0</span>(<span class="st">"day"</span>, time_days))  <span class="sc">%&gt;%</span> </span>
<span id="cb16-129"><a href="#cb16-129"></a>  <span class="fu">select</span>(group_id, day, freq_alt_complete) <span class="sc">%&gt;%</span> </span>
<span id="cb16-130"><a href="#cb16-130"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"day"</span>, <span class="at">values_from =</span> <span class="st">"freq_alt_complete"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-131"><a href="#cb16-131"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb16-132"><a href="#cb16-132"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"group_id"</span>)</span>
<span id="cb16-133"><a href="#cb16-133"></a></span>
<span id="cb16-134"><a href="#cb16-134"></a><span class="co"># scale the dataframe for clustering</span></span>
<span id="cb16-135"><a href="#cb16-135"></a>df2clust_scaled <span class="ot">&lt;-</span> <span class="fu">scale</span>(df2clust)</span>
<span id="cb16-136"><a href="#cb16-136"></a></span>
<span id="cb16-137"><a href="#cb16-137"></a><span class="co"># perform the hierarchcical clustering using euclidean distance and Ward's D</span></span>
<span id="cb16-138"><a href="#cb16-138"></a>hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(df2clust_scaled, <span class="at">method =</span> <span class="st">"euclidean"</span>), <span class="at">method =</span> <span class="st">"ward.D2"</span> )</span>
<span id="cb16-139"><a href="#cb16-139"></a></span>
<span id="cb16-140"><a href="#cb16-140"></a><span class="co"># get order of the clustered observations. The order is for the groups</span></span>
<span id="cb16-141"><a href="#cb16-141"></a>ord <span class="ot">&lt;-</span> hc<span class="sc">$</span>order</span>
<span id="cb16-142"><a href="#cb16-142"></a></span>
<span id="cb16-143"><a href="#cb16-143"></a><span class="co"># get cluster membership. Again memebership is for the groups</span></span>
<span id="cb16-144"><a href="#cb16-144"></a>myclust <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc, <span class="at">k =</span> <span class="dv">6</span>)</span>
<span id="cb16-145"><a href="#cb16-145"></a></span>
<span id="cb16-146"><a href="#cb16-146"></a>df2plot <span class="ot">&lt;-</span> df_grouped <span class="sc">%&gt;%</span> </span>
<span id="cb16-147"><a href="#cb16-147"></a>  <span class="fu">mutate</span>(<span class="at">cluster =</span> myclust[group_id],</span>
<span id="cb16-148"><a href="#cb16-148"></a>         <span class="at">order =</span> ord[group_id]) <span class="sc">%&gt;%</span> </span>
<span id="cb16-149"><a href="#cb16-149"></a>  <span class="co"># convert group_id into a factor that is ordered by the hierarchical clustering</span></span>
<span id="cb16-150"><a href="#cb16-150"></a>  <span class="co"># done above</span></span>
<span id="cb16-151"><a href="#cb16-151"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">factor</span>(group_id, <span class="at">levels =</span> ord),</span>
<span id="cb16-152"><a href="#cb16-152"></a>         <span class="at">cluster =</span> <span class="fu">factor</span>(cluster)) <span class="sc">%&gt;%</span></span>
<span id="cb16-153"><a href="#cb16-153"></a>  <span class="fu">mutate</span>(<span class="at">day =</span> <span class="fu">factor</span>(time_days), </span>
<span id="cb16-154"><a href="#cb16-154"></a>         <span class="at">pos =</span> <span class="fu">factor</span>(pos),</span>
<span id="cb16-155"><a href="#cb16-155"></a>         <span class="at">mylab =</span> <span class="fu">interaction</span>(replicate, prey_history, predator_history, <span class="fu">paste0</span>(locus_tag, <span class="st">"_"</span>, pos)),</span>
<span id="cb16-156"><a href="#cb16-156"></a>         <span class="at">strainID2 =</span> <span class="fu">case_when</span>(strainID <span class="sc">==</span> <span class="st">"HAMBI_0403"</span> <span class="sc">~</span> <span class="dv">5</span>,</span>
<span id="cb16-157"><a href="#cb16-157"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span> <span class="sc">~</span> <span class="dv">4</span>,</span>
<span id="cb16-158"><a href="#cb16-158"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span> <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb16-159"><a href="#cb16-159"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span> <span class="sc">~</span> <span class="dv">3</span>,</span>
<span id="cb16-160"><a href="#cb16-160"></a>                               strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span> <span class="sc">~</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-161"><a href="#cb16-161"></a>  <span class="fu">mutate</span>(<span class="at">mylab2 =</span> <span class="fu">fct_reorder</span>(group_id, strainID2)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-162"><a href="#cb16-162"></a>  <span class="fu">relocate</span>(mylab2, order, strainID2) </span>
<span id="cb16-163"><a href="#cb16-163"></a><span class="in">```</span></span>
<span id="cb16-164"><a href="#cb16-164"></a></span>
<span id="cb16-165"><a href="#cb16-165"></a><span class="fu"># Main text figure</span></span>
<span id="cb16-166"><a href="#cb16-166"></a></span>
<span id="cb16-167"><a href="#cb16-167"></a>Idea is to have a one heatmap for the alt allele frequencies for every detected mutation that passes our thresholds and then have some horiztonal bars which designate the different treatment combinations, the longitudinal clusters, and the species identieis stacked on top. Then below this we will have some simple line plots show the highly parallel mutations identified in the prior analysis.</span>
<span id="cb16-168"><a href="#cb16-168"></a></span>
<span id="cb16-169"><a href="#cb16-169"></a><span class="fu">## Heatmap all species together</span></span>
<span id="cb16-170"><a href="#cb16-170"></a></span>
<span id="cb16-173"><a href="#cb16-173"></a><span class="in">```{r}</span></span>
<span id="cb16-174"><a href="#cb16-174"></a>blank_theme01 <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb16-175"><a href="#cb16-175"></a>  <span class="fu">theme</span>(</span>
<span id="cb16-176"><a href="#cb16-176"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-177"><a href="#cb16-177"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-178"><a href="#cb16-178"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-179"><a href="#cb16-179"></a>    <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-180"><a href="#cb16-180"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-181"><a href="#cb16-181"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-182"><a href="#cb16-182"></a>    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">"cm"</span>),</span>
<span id="cb16-183"><a href="#cb16-183"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-184"><a href="#cb16-184"></a>    <span class="at">strip.text =</span> <span class="fu">element_blank</span>()</span>
<span id="cb16-185"><a href="#cb16-185"></a>  )</span>
<span id="cb16-186"><a href="#cb16-186"></a>}</span>
<span id="cb16-187"><a href="#cb16-187"></a></span>
<span id="cb16-188"><a href="#cb16-188"></a>blank_theme02 <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb16-189"><a href="#cb16-189"></a>  <span class="fu">theme</span>(</span>
<span id="cb16-190"><a href="#cb16-190"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-191"><a href="#cb16-191"></a>    <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-192"><a href="#cb16-192"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-193"><a href="#cb16-193"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-194"><a href="#cb16-194"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-195"><a href="#cb16-195"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-196"><a href="#cb16-196"></a>    <span class="at">strip.text =</span> <span class="fu">element_blank</span>()</span>
<span id="cb16-197"><a href="#cb16-197"></a>  )</span>
<span id="cb16-198"><a href="#cb16-198"></a>}</span>
<span id="cb16-199"><a href="#cb16-199"></a></span>
<span id="cb16-200"><a href="#cb16-200"></a>phoriz_bar <span class="ot">&lt;-</span> <span class="cf">function</span>(df, fillvar, mypal){</span>
<span id="cb16-201"><a href="#cb16-201"></a>  <span class="co"># only take the first time point so to make one bar</span></span>
<span id="cb16-202"><a href="#cb16-202"></a>  <span class="fu">filter</span>(df, day <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-203"><a href="#cb16-203"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> day,  <span class="at">x=</span>mylab2, <span class="at">fill =</span> {{ fillvar }})) <span class="sc">+</span></span>
<span id="cb16-204"><a href="#cb16-204"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb16-205"><a href="#cb16-205"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb16-206"><a href="#cb16-206"></a>    <span class="fu">facet_wrap</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-207"><a href="#cb16-207"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> {{ mypal }}) <span class="sc">+</span> </span>
<span id="cb16-208"><a href="#cb16-208"></a>    <span class="fu">blank_theme01</span>()</span>
<span id="cb16-209"><a href="#cb16-209"></a>}</span>
<span id="cb16-210"><a href="#cb16-210"></a></span>
<span id="cb16-211"><a href="#cb16-211"></a>pheat_nolb <span class="ot">&lt;-</span> <span class="cf">function</span>(df, fillvar){</span>
<span id="cb16-212"><a href="#cb16-212"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> day,  <span class="at">x=</span>mylab2, <span class="at">fill =</span> {{ fillvar }} )) <span class="sc">+</span></span>
<span id="cb16-213"><a href="#cb16-213"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb16-214"><a href="#cb16-214"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb16-215"><a href="#cb16-215"></a>    <span class="fu">facet_wrap</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-216"><a href="#cb16-216"></a>    <span class="fu">scale_fill_viridis_c</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">trans =</span> <span class="st">"sqrt"</span>) <span class="sc">+</span></span>
<span id="cb16-217"><a href="#cb16-217"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb16-218"><a href="#cb16-218"></a>    <span class="fu">blank_theme01</span>()</span>
<span id="cb16-219"><a href="#cb16-219"></a>}</span>
<span id="cb16-220"><a href="#cb16-220"></a></span>
<span id="cb16-221"><a href="#cb16-221"></a>pheat_labs <span class="ot">&lt;-</span> <span class="cf">function</span>(df, fillvar){</span>
<span id="cb16-222"><a href="#cb16-222"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">y =</span> day,  <span class="at">x=</span>mylab2, <span class="at">fill =</span> {{ fillvar }})) <span class="sc">+</span></span>
<span id="cb16-223"><a href="#cb16-223"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb16-224"><a href="#cb16-224"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb16-225"><a href="#cb16-225"></a>    <span class="fu">facet_wrap</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">nrow =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-226"><a href="#cb16-226"></a>    <span class="fu">scale_fill_viridis_c</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">trans =</span> <span class="st">"sqrt"</span>) <span class="sc">+</span></span>
<span id="cb16-227"><a href="#cb16-227"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb16-228"><a href="#cb16-228"></a>    <span class="fu">blank_theme02</span>()</span>
<span id="cb16-229"><a href="#cb16-229"></a>}</span>
<span id="cb16-230"><a href="#cb16-230"></a><span class="in">```</span></span>
<span id="cb16-231"><a href="#cb16-231"></a></span>
<span id="cb16-232"><a href="#cb16-232"></a></span>
<span id="cb16-233"><a href="#cb16-233"></a><span class="fu">### All alleles</span></span>
<span id="cb16-236"><a href="#cb16-236"></a><span class="in">```{r}</span></span>
<span id="cb16-237"><a href="#cb16-237"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-238"><a href="#cb16-238"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb16-239"><a href="#cb16-239"></a><span class="co">#clupal &lt;- unname(createPalette(length(unique(myclust)), c("#F3874AFF", "#FCD125FF"), M=5000))</span></span>
<span id="cb16-240"><a href="#cb16-240"></a>clupal <span class="ot">&lt;-</span> <span class="fu">gray.colors</span>(<span class="fu">length</span>(<span class="fu">unique</span>(myclust)), <span class="at">start =</span> <span class="fl">0.1</span>, <span class="at">end =</span> <span class="fl">0.9</span>, <span class="at">gamma =</span> <span class="fl">2.2</span>, <span class="dv">1</span>, <span class="at">rev =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-241"><a href="#cb16-241"></a><span class="fu">names</span>(clupal) <span class="ot">&lt;-</span> <span class="fu">unique</span>(myclust)</span>
<span id="cb16-242"><a href="#cb16-242"></a></span>
<span id="cb16-243"><a href="#cb16-243"></a>pheat_all <span class="ot">&lt;-</span> <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), cluster, clupal) <span class="sc">+</span></span>
<span id="cb16-244"><a href="#cb16-244"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), freq_alt_complete) <span class="sc">+</span></span>
<span id="cb16-245"><a href="#cb16-245"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb16-246"><a href="#cb16-246"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-247"><a href="#cb16-247"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb16-248"><a href="#cb16-248"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-249"><a href="#cb16-249"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb16-250"><a href="#cb16-250"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-251"><a href="#cb16-251"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_0403"</span>), cluster, clupal) <span class="sc">+</span> </span>
<span id="cb16-252"><a href="#cb16-252"></a>  <span class="fu">pheat_nolb</span>(<span class="fu">filter</span>(df2plot, strainID <span class="sc">==</span> <span class="st">"HAMBI_0403"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-253"><a href="#cb16-253"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">10</span>,</span>
<span id="cb16-254"><a href="#cb16-254"></a>              <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>),</span>
<span id="cb16-255"><a href="#cb16-255"></a>              <span class="at">guides =</span> <span class="st">"collect"</span>)</span>
<span id="cb16-256"><a href="#cb16-256"></a></span>
<span id="cb16-257"><a href="#cb16-257"></a>pheat_all</span>
<span id="cb16-258"><a href="#cb16-258"></a><span class="in">```</span></span>
<span id="cb16-259"><a href="#cb16-259"></a></span>
<span id="cb16-260"><a href="#cb16-260"></a><span class="fu">#### Save</span></span>
<span id="cb16-263"><a href="#cb16-263"></a><span class="in">```{r}</span></span>
<span id="cb16-264"><a href="#cb16-264"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"heatmap_all_alleles.svg"</span>), pheat_all, <span class="at">width=</span><span class="dv">11</span>, <span class="at">height=</span><span class="dv">4</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb16-265"><a href="#cb16-265"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb16-266"><a href="#cb16-266"></a><span class="in">```</span></span>
<span id="cb16-267"><a href="#cb16-267"></a></span>
<span id="cb16-268"><a href="#cb16-268"></a><span class="fu">### Parallel genes</span></span>
<span id="cb16-269"><a href="#cb16-269"></a></span>
<span id="cb16-272"><a href="#cb16-272"></a><span class="in">```{r}</span></span>
<span id="cb16-273"><a href="#cb16-273"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-274"><a href="#cb16-274"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb16-275"><a href="#cb16-275"></a><span class="co">#| warning: false</span></span>
<span id="cb16-276"><a href="#cb16-276"></a></span>
<span id="cb16-277"><a href="#cb16-277"></a>pdfr_evo <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, df2plot, </span>
<span id="cb16-278"><a href="#cb16-278"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-279"><a href="#cb16-279"></a>  <span class="fu">filter</span>(prey_history <span class="sc">==</span> <span class="st">"evo"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-280"><a href="#cb16-280"></a>  <span class="co"># some manual renaming</span></span>
<span id="cb16-281"><a href="#cb16-281"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">case_when</span>(locus_tag <span class="sc">==</span> <span class="st">"H1287_02172"</span> <span class="sc">~</span> <span class="st">"yddV_2"</span>,</span>
<span id="cb16-282"><a href="#cb16-282"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1977_02612"</span> <span class="sc">~</span> <span class="st">"dppA/oppA"</span>,</span>
<span id="cb16-283"><a href="#cb16-283"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00299"</span> <span class="sc">~</span> <span class="st">"PAP2_like"</span>,</span>
<span id="cb16-284"><a href="#cb16-284"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02826"</span> <span class="sc">~</span> <span class="st">"ynjC"</span>,</span>
<span id="cb16-285"><a href="#cb16-285"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00671"</span> <span class="sc">~</span> <span class="st">"yqaA"</span>,</span>
<span id="cb16-286"><a href="#cb16-286"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00421"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb16-287"><a href="#cb16-287"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02723"</span> <span class="sc">~</span> <span class="st">"yifE"</span>,</span>
<span id="cb16-288"><a href="#cb16-288"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H2659_01789"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb16-289"><a href="#cb16-289"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-290"><a href="#cb16-290"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-291"><a href="#cb16-291"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, locus_tag, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-292"><a href="#cb16-292"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">paste0</span>(gene_lab, <span class="st">" | "</span>, replicate, <span class="st">" | "</span>,  cluster, <span class="st">"|"</span>, group_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-293"><a href="#cb16-293"></a>  <span class="fu">arrange</span>(COG_category_long, gene_lab) <span class="sc">%&gt;%</span> </span>
<span id="cb16-294"><a href="#cb16-294"></a>  <span class="fu">mutate</span>(<span class="at">mylab2 =</span> <span class="fu">factor</span>(gene_lab, <span class="fu">unique</span>(gene_lab)))</span>
<span id="cb16-295"><a href="#cb16-295"></a></span>
<span id="cb16-296"><a href="#cb16-296"></a>pdfr_anc <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, df2plot, </span>
<span id="cb16-297"><a href="#cb16-297"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-298"><a href="#cb16-298"></a>  <span class="fu">filter</span>(prey_history <span class="sc">==</span> <span class="st">"anc"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-299"><a href="#cb16-299"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">case_when</span>(locus_tag <span class="sc">==</span> <span class="st">"H1972_02826"</span> <span class="sc">~</span> <span class="st">"ynjC"</span>,</span>
<span id="cb16-300"><a href="#cb16-300"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00421"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb16-301"><a href="#cb16-301"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-302"><a href="#cb16-302"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-303"><a href="#cb16-303"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, locus_tag, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-304"><a href="#cb16-304"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">paste0</span>(gene_lab, <span class="st">" | "</span>, replicate, <span class="st">" | "</span>,  cluster, <span class="st">"|"</span>, group_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-305"><a href="#cb16-305"></a>  <span class="fu">arrange</span>(COG_category_long, gene_lab) <span class="sc">%&gt;%</span> </span>
<span id="cb16-306"><a href="#cb16-306"></a>  <span class="fu">mutate</span>(<span class="at">mylab2 =</span> <span class="fu">factor</span>(gene_lab, <span class="fu">unique</span>(gene_lab)))</span>
<span id="cb16-307"><a href="#cb16-307"></a>  </span>
<span id="cb16-308"><a href="#cb16-308"></a><span class="co"># evolved</span></span>
<span id="cb16-309"><a href="#cb16-309"></a>pheat_par <span class="ot">&lt;-</span> <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb16-310"><a href="#cb16-310"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_2659"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-311"><a href="#cb16-311"></a>  <span class="co"># evolved 1972</span></span>
<span id="cb16-312"><a href="#cb16-312"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb16-313"><a href="#cb16-313"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-314"><a href="#cb16-314"></a>  <span class="co"># ancestral 1972 - the only species that has significantly parallel genes in the ancestral treatment</span></span>
<span id="cb16-315"><a href="#cb16-315"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_anc, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb16-316"><a href="#cb16-316"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_anc, strainID <span class="sc">==</span> <span class="st">"HAMBI_1972"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-317"><a href="#cb16-317"></a>  <span class="co"># evolved 1977</span></span>
<span id="cb16-318"><a href="#cb16-318"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb16-319"><a href="#cb16-319"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-320"><a href="#cb16-320"></a>  <span class="co"># evolved 1287</span></span>
<span id="cb16-321"><a href="#cb16-321"></a>  <span class="fu">phoriz_bar</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), COG_category_long, cogpal) <span class="sc">+</span></span>
<span id="cb16-322"><a href="#cb16-322"></a>  <span class="fu">pheat_labs</span>(<span class="fu">filter</span>(pdfr_evo, strainID <span class="sc">==</span> <span class="st">"HAMBI_1287"</span>), freq_alt_complete) <span class="sc">+</span> </span>
<span id="cb16-323"><a href="#cb16-323"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">10</span>,</span>
<span id="cb16-324"><a href="#cb16-324"></a>              <span class="at">heights =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>, <span class="fl">0.25</span>, <span class="dv">1</span>),</span>
<span id="cb16-325"><a href="#cb16-325"></a>              <span class="at">guides =</span> <span class="st">"collect"</span>)</span>
<span id="cb16-326"><a href="#cb16-326"></a></span>
<span id="cb16-327"><a href="#cb16-327"></a>pheat_par</span>
<span id="cb16-328"><a href="#cb16-328"></a><span class="in">```</span></span>
<span id="cb16-329"><a href="#cb16-329"></a></span>
<span id="cb16-330"><a href="#cb16-330"></a><span class="fu">#### Save</span></span>
<span id="cb16-333"><a href="#cb16-333"></a><span class="in">```{r}</span></span>
<span id="cb16-334"><a href="#cb16-334"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"heatmap_par_alleles.svg"</span>), pheat_par, <span class="at">width=</span><span class="dv">12</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb16-335"><a href="#cb16-335"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb16-336"><a href="#cb16-336"></a><span class="in">```</span></span>
<span id="cb16-337"><a href="#cb16-337"></a></span>
<span id="cb16-338"><a href="#cb16-338"></a><span class="fu">## Individual allelle trajectories</span></span>
<span id="cb16-339"><a href="#cb16-339"></a></span>
<span id="cb16-340"><a href="#cb16-340"></a>Read data from parallelism analysis</span>
<span id="cb16-341"><a href="#cb16-341"></a></span>
<span id="cb16-344"><a href="#cb16-344"></a><span class="in">```{r}</span></span>
<span id="cb16-345"><a href="#cb16-345"></a><span class="co">#| output: false</span></span>
<span id="cb16-346"><a href="#cb16-346"></a><span class="co">#| warning: false</span></span>
<span id="cb16-347"><a href="#cb16-347"></a><span class="co">#| error: false</span></span>
<span id="cb16-348"><a href="#cb16-348"></a><span class="in">```</span></span>
<span id="cb16-349"><a href="#cb16-349"></a></span>
<span id="cb16-350"><a href="#cb16-350"></a>Function for plotting trajectories for indivdidual genes</span>
<span id="cb16-351"><a href="#cb16-351"></a></span>
<span id="cb16-354"><a href="#cb16-354"></a><span class="in">```{r}</span></span>
<span id="cb16-355"><a href="#cb16-355"></a>plotpargenes <span class="ot">&lt;-</span> <span class="cf">function</span>(strainID){</span>
<span id="cb16-356"><a href="#cb16-356"></a>  pdfr <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, df2plot, </span>
<span id="cb16-357"><a href="#cb16-357"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-358"><a href="#cb16-358"></a>  <span class="co"># to reduce size only include genes that are mutated in at least two of the replicates</span></span>
<span id="cb16-359"><a href="#cb16-359"></a>  <span class="fu">filter</span>(n_replicate <span class="sc">&gt;=</span><span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-360"><a href="#cb16-360"></a>  <span class="co"># name the genes for plotting</span></span>
<span id="cb16-361"><a href="#cb16-361"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene),</span>
<span id="cb16-362"><a href="#cb16-362"></a>         <span class="at">treat =</span> <span class="fu">paste0</span>(<span class="st">"prey:"</span>, prey_history, <span class="st">" | pred:"</span>, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-363"><a href="#cb16-363"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, Description, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-364"><a href="#cb16-364"></a>  <span class="fu">filter</span>(strainID <span class="sc">==</span> {{ strainID }})</span>
<span id="cb16-365"><a href="#cb16-365"></a>  </span>
<span id="cb16-366"><a href="#cb16-366"></a>  <span class="co"># make random color set to help differentiate alleles </span></span>
<span id="cb16-367"><a href="#cb16-367"></a>  mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(pdfr<span class="sc">$</span>mylab2)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb16-368"><a href="#cb16-368"></a>  </span>
<span id="cb16-369"><a href="#cb16-369"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb16-370"><a href="#cb16-370"></a>  <span class="fu">geom_line</span>(<span class="at">data=</span>pdfr[<span class="sc">!</span><span class="fu">is.na</span>(pdfr<span class="sc">$</span>freq_alt_complete), ],</span>
<span id="cb16-371"><a href="#cb16-371"></a>            <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete,</span>
<span id="cb16-372"><a href="#cb16-372"></a>                <span class="at">group =</span> mylab2,</span>
<span id="cb16-373"><a href="#cb16-373"></a>                <span class="at">color =</span> hgvs_p)) <span class="sc">+</span></span>
<span id="cb16-374"><a href="#cb16-374"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> pdfr, </span>
<span id="cb16-375"><a href="#cb16-375"></a>             <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete, <span class="at">shape =</span> replicate), </span>
<span id="cb16-376"><a href="#cb16-376"></a>             <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-377"><a href="#cb16-377"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb16-378"><a href="#cb16-378"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal) <span class="sc">+</span> </span>
<span id="cb16-379"><a href="#cb16-379"></a>  <span class="fu">facet_wrap</span>(gene_lab <span class="sc">~</span> treat) <span class="sc">+</span></span>
<span id="cb16-380"><a href="#cb16-380"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Days"</span>, <span class="at">y =</span> <span class="st">"Allele frequency"</span>) <span class="sc">+</span></span>
<span id="cb16-381"><a href="#cb16-381"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb16-382"><a href="#cb16-382"></a>  <span class="fu">theme</span>(</span>
<span id="cb16-383"><a href="#cb16-383"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb16-384"><a href="#cb16-384"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb16-385"><a href="#cb16-385"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-386"><a href="#cb16-386"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb16-387"><a href="#cb16-387"></a>}</span>
<span id="cb16-388"><a href="#cb16-388"></a><span class="in">```</span></span>
<span id="cb16-389"><a href="#cb16-389"></a></span>
<span id="cb16-390"><a href="#cb16-390"></a><span class="fu">### HAMBI_1287</span></span>
<span id="cb16-391"><a href="#cb16-391"></a></span>
<span id="cb16-392"><a href="#cb16-392"></a>::: {#fig-01}</span>
<span id="cb16-395"><a href="#cb16-395"></a><span class="in">```{r}</span></span>
<span id="cb16-396"><a href="#cb16-396"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-397"><a href="#cb16-397"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb16-398"><a href="#cb16-398"></a><span class="co">#| echo: false</span></span>
<span id="cb16-399"><a href="#cb16-399"></a><span class="fu">plotpargenes</span>(<span class="st">"HAMBI_1287"</span>)</span>
<span id="cb16-400"><a href="#cb16-400"></a><span class="in">```</span></span>
<span id="cb16-401"><a href="#cb16-401"></a></span>
<span id="cb16-402"><a href="#cb16-402"></a>Nonsynonymous variant frequencies (vertical axis) over time (horizontal axis) for combinations of individual genes (title bar top row) and treatment combinations (title bar bottom row). Point shape denotes the biological replicate from the treatment and line colors indicate the amino acid changing variant (some genes have multiple non-synonymous variants at different positions). Focal genes are from *Citrobacter koserii* HAMBI_1287 and are those exhibiting significant parallelism for the duration of the experiment.</span>
<span id="cb16-403"><a href="#cb16-403"></a>:::</span>
<span id="cb16-404"><a href="#cb16-404"></a></span>
<span id="cb16-405"><a href="#cb16-405"></a><span class="fu">### HAMBI_1972</span></span>
<span id="cb16-406"><a href="#cb16-406"></a></span>
<span id="cb16-407"><a href="#cb16-407"></a>::: {#fig-02}</span>
<span id="cb16-410"><a href="#cb16-410"></a><span class="in">```{r}</span></span>
<span id="cb16-411"><a href="#cb16-411"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-412"><a href="#cb16-412"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb16-413"><a href="#cb16-413"></a><span class="co">#| echo: false</span></span>
<span id="cb16-414"><a href="#cb16-414"></a><span class="fu">plotpargenes</span>(<span class="st">"HAMBI_1972"</span>)</span>
<span id="cb16-415"><a href="#cb16-415"></a><span class="in">```</span></span>
<span id="cb16-416"><a href="#cb16-416"></a></span>
<span id="cb16-417"><a href="#cb16-417"></a>As in @fig-01 but for *Aeromonas caviae* HAMBI_1972.</span>
<span id="cb16-418"><a href="#cb16-418"></a>:::</span>
<span id="cb16-419"><a href="#cb16-419"></a></span>
<span id="cb16-420"><a href="#cb16-420"></a><span class="fu">### HAMBI_1977</span></span>
<span id="cb16-421"><a href="#cb16-421"></a></span>
<span id="cb16-422"><a href="#cb16-422"></a>::: {#fig-03}</span>
<span id="cb16-425"><a href="#cb16-425"></a><span class="in">```{r}</span></span>
<span id="cb16-426"><a href="#cb16-426"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-427"><a href="#cb16-427"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb16-428"><a href="#cb16-428"></a><span class="co">#| echo: false</span></span>
<span id="cb16-429"><a href="#cb16-429"></a><span class="fu">plotpargenes</span>(<span class="st">"HAMBI_1977"</span>)</span>
<span id="cb16-430"><a href="#cb16-430"></a><span class="in">```</span></span>
<span id="cb16-431"><a href="#cb16-431"></a></span>
<span id="cb16-432"><a href="#cb16-432"></a>As in @fig-01 but for *Pseudomonas chlororaphis* HAMBI_1977.</span>
<span id="cb16-433"><a href="#cb16-433"></a>:::</span>
<span id="cb16-434"><a href="#cb16-434"></a></span>
<span id="cb16-435"><a href="#cb16-435"></a><span class="fu">### HAMBI_2659</span></span>
<span id="cb16-436"><a href="#cb16-436"></a></span>
<span id="cb16-437"><a href="#cb16-437"></a>::: {#fig-04}</span>
<span id="cb16-438"><a href="#cb16-438"></a></span>
<span id="cb16-441"><a href="#cb16-441"></a><span class="in">```{r}</span></span>
<span id="cb16-442"><a href="#cb16-442"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-443"><a href="#cb16-443"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb16-444"><a href="#cb16-444"></a><span class="co">#| echo: false</span></span>
<span id="cb16-445"><a href="#cb16-445"></a><span class="fu">plotpargenes</span>(<span class="st">"HAMBI_2659"</span>)</span>
<span id="cb16-446"><a href="#cb16-446"></a><span class="in">```</span></span>
<span id="cb16-447"><a href="#cb16-447"></a></span>
<span id="cb16-448"><a href="#cb16-448"></a>As in @fig-01 but for *Stenotrophomonas maltophilia* HAMBI_2659.</span>
<span id="cb16-449"><a href="#cb16-449"></a>:::</span>
<span id="cb16-450"><a href="#cb16-450"></a></span>
<span id="cb16-451"><a href="#cb16-451"></a><span class="fu">### Selection of parallel genes to plot</span></span>
<span id="cb16-452"><a href="#cb16-452"></a></span>
<span id="cb16-453"><a href="#cb16-453"></a>::: {#fig-05}</span>
<span id="cb16-454"><a href="#cb16-454"></a></span>
<span id="cb16-457"><a href="#cb16-457"></a><span class="in">```{r}</span></span>
<span id="cb16-458"><a href="#cb16-458"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb16-459"><a href="#cb16-459"></a><span class="co">#| fig.height: 8</span></span>
<span id="cb16-460"><a href="#cb16-460"></a></span>
<span id="cb16-461"><a href="#cb16-461"></a>pdfr <span class="ot">&lt;-</span> par_genes <span class="sc">%&gt;%</span> </span>
<span id="cb16-462"><a href="#cb16-462"></a>  <span class="fu">filter</span>(n_replicate <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-463"><a href="#cb16-463"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(gene_multiplicity_m_i)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-464"><a href="#cb16-464"></a>  <span class="co"># take top 20 highest multiplicitiies</span></span>
<span id="cb16-465"><a href="#cb16-465"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-466"><a href="#cb16-466"></a>  <span class="fu">left_join</span>(df2plot, <span class="at">by =</span> <span class="fu">join_by</span>(locus_tag, strainID, prey_history, predator_history, chrom)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-467"><a href="#cb16-467"></a>  <span class="co"># name the genes for plotting</span></span>
<span id="cb16-468"><a href="#cb16-468"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene),</span>
<span id="cb16-469"><a href="#cb16-469"></a>         <span class="at">treat =</span> <span class="fu">paste0</span>(<span class="st">"prey:"</span>, prey_history, <span class="st">" | pred:"</span>, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-470"><a href="#cb16-470"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, Description, gene_lab))</span>
<span id="cb16-471"><a href="#cb16-471"></a></span>
<span id="cb16-472"><a href="#cb16-472"></a></span>
<span id="cb16-473"><a href="#cb16-473"></a>mypal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(pdfr<span class="sc">$</span>mylab2)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb16-474"><a href="#cb16-474"></a>  </span>
<span id="cb16-475"><a href="#cb16-475"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb16-476"><a href="#cb16-476"></a>  <span class="fu">geom_line</span>(<span class="at">data=</span>pdfr[<span class="sc">!</span><span class="fu">is.na</span>(pdfr<span class="sc">$</span>freq_alt_complete), ],</span>
<span id="cb16-477"><a href="#cb16-477"></a>            <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete,</span>
<span id="cb16-478"><a href="#cb16-478"></a>                <span class="at">group =</span> mylab2,</span>
<span id="cb16-479"><a href="#cb16-479"></a>                <span class="at">color =</span> hgvs_p)) <span class="sc">+</span></span>
<span id="cb16-480"><a href="#cb16-480"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> pdfr, </span>
<span id="cb16-481"><a href="#cb16-481"></a>             <span class="fu">aes</span>(<span class="at">x=</span>time_days, <span class="at">y=</span>freq_alt_complete, <span class="at">shape =</span> replicate), </span>
<span id="cb16-482"><a href="#cb16-482"></a>             <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-483"><a href="#cb16-483"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb16-484"><a href="#cb16-484"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> mypal) <span class="sc">+</span> </span>
<span id="cb16-485"><a href="#cb16-485"></a>  <span class="fu">facet_wrap</span>(gene_lab <span class="sc">~</span> treat) <span class="sc">+</span></span>
<span id="cb16-486"><a href="#cb16-486"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Days"</span>, <span class="at">y =</span> <span class="st">"Allele frequency"</span>) <span class="sc">+</span></span>
<span id="cb16-487"><a href="#cb16-487"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb16-488"><a href="#cb16-488"></a>  <span class="fu">theme</span>(</span>
<span id="cb16-489"><a href="#cb16-489"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb16-490"><a href="#cb16-490"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb16-491"><a href="#cb16-491"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-492"><a href="#cb16-492"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb16-493"><a href="#cb16-493"></a><span class="in">```</span></span>
<span id="cb16-494"><a href="#cb16-494"></a>As in @fig-01 but for the genes with the top 20 higest gene multiplicities across HAMBI_1287, HAMBI_1972, HAMBI_1977, and HAMBI_2659.</span>
<span id="cb16-495"><a href="#cb16-495"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>