<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-10-30">

<title>hambiDoubleEvo - Mutational dynamics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">Metagenome sequencing</a></li><li class="breadcrumb-item"><a href="../../R/metagenome/04_mutation_dynamics.html">4. Mutational dynamics</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">hambiDoubleEvo</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/01_rpkm2tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/illumina_v3v4/02_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Analyze and plot</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Population WGS</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/wgs/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Metagenome sequencing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/01_format_variants.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Read and format</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/02_parallelism.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Parallelism analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/03_analyze_metagenome_variant_timeseries.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Variant plotting</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/metagenome/04_mutation_dynamics.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">4. Mutational dynamics</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">2</span> Read data</a></li>
  <li><a href="#metagenomic-coverage" id="toc-metagenomic-coverage" class="nav-link" data-scroll-target="#metagenomic-coverage"><span class="header-section-number">3</span> Metagenomic coverage</a></li>
  <li><a href="#mutational-spectra" id="toc-mutational-spectra" class="nav-link" data-scroll-target="#mutational-spectra"><span class="header-section-number">4</span> Mutational spectra</a></li>
  <li><a href="#mutational-dynamics" id="toc-mutational-dynamics" class="nav-link" data-scroll-target="#mutational-dynamics"><span class="header-section-number">5</span> Mutational dynamics</a>
  <ul class="collapse">
  <li><a href="#some-data-formatting" id="toc-some-data-formatting" class="nav-link" data-scroll-target="#some-data-formatting"><span class="header-section-number">5.1</span> Some data formatting</a></li>
  <li><a href="#sec-allvars" id="toc-sec-allvars" class="nav-link" data-scroll-target="#sec-allvars"><span class="header-section-number">5.2</span> All variants</a>
  <ul class="collapse">
  <li><a href="#cumulative-mutation-trajectories-mt" id="toc-cumulative-mutation-trajectories-mt" class="nav-link" data-scroll-target="#cumulative-mutation-trajectories-mt"><span class="header-section-number">5.2.1</span> Cumulative mutation trajectories (M(t))</a></li>
  <li><a href="#total-mutation-trajectories" id="toc-total-mutation-trajectories" class="nav-link" data-scroll-target="#total-mutation-trajectories"><span class="header-section-number">5.2.2</span> Total mutation trajectories</a></li>
  <li><a href="#fixed-mutation-trajectories" id="toc-fixed-mutation-trajectories" class="nav-link" data-scroll-target="#fixed-mutation-trajectories"><span class="header-section-number">5.2.3</span> Fixed mutation trajectories</a></li>
  </ul></li>
  <li><a href="#de-novo-variants" id="toc-de-novo-variants" class="nav-link" data-scroll-target="#de-novo-variants"><span class="header-section-number">5.3</span> De novo variants</a>
  <ul class="collapse">
  <li><a href="#cumulative-mutation-trajectories-mt-1" id="toc-cumulative-mutation-trajectories-mt-1" class="nav-link" data-scroll-target="#cumulative-mutation-trajectories-mt-1"><span class="header-section-number">5.3.1</span> Cumulative mutation trajectories (M(t))</a></li>
  <li><a href="#total-mutation-trajectories-1" id="toc-total-mutation-trajectories-1" class="nav-link" data-scroll-target="#total-mutation-trajectories-1"><span class="header-section-number">5.3.2</span> Total mutation trajectories</a></li>
  <li><a href="#fixed-mutation-trajectories-1" id="toc-fixed-mutation-trajectories-1" class="nav-link" data-scroll-target="#fixed-mutation-trajectories-1"><span class="header-section-number">5.3.3</span> Fixed mutation trajectories</a></li>
  </ul></li>
  <li><a href="#sec-nsvars" id="toc-sec-nsvars" class="nav-link" data-scroll-target="#sec-nsvars"><span class="header-section-number">5.4</span> Non-synonymous variants</a>
  <ul class="collapse">
  <li><a href="#cumulative-mutation-trajectories-mt-2" id="toc-cumulative-mutation-trajectories-mt-2" class="nav-link" data-scroll-target="#cumulative-mutation-trajectories-mt-2"><span class="header-section-number">5.4.1</span> Cumulative mutation trajectories (M(t))</a></li>
  <li><a href="#total-mutation-trajectories-2" id="toc-total-mutation-trajectories-2" class="nav-link" data-scroll-target="#total-mutation-trajectories-2"><span class="header-section-number">5.4.2</span> Total mutation trajectories</a></li>
  <li><a href="#fixed-mutation-trajectories-2" id="toc-fixed-mutation-trajectories-2" class="nav-link" data-scroll-target="#fixed-mutation-trajectories-2"><span class="header-section-number">5.4.3</span> Fixed mutation trajectories</a></li>
  </ul></li>
  <li><a href="#de-novo-non-synonymous-variants" id="toc-de-novo-non-synonymous-variants" class="nav-link" data-scroll-target="#de-novo-non-synonymous-variants"><span class="header-section-number">5.5</span> De novo, non-synonymous variants</a>
  <ul class="collapse">
  <li><a href="#cumulative-mutation-trajectories-mt-3" id="toc-cumulative-mutation-trajectories-mt-3" class="nav-link" data-scroll-target="#cumulative-mutation-trajectories-mt-3"><span class="header-section-number">5.5.1</span> Cumulative mutation trajectories (M(t))</a></li>
  <li><a href="#total-mutation-trajectories-3" id="toc-total-mutation-trajectories-3" class="nav-link" data-scroll-target="#total-mutation-trajectories-3"><span class="header-section-number">5.5.2</span> Total mutation trajectories</a></li>
  <li><a href="#fixed-mutation-trajectories-3" id="toc-fixed-mutation-trajectories-3" class="nav-link" data-scroll-target="#fixed-mutation-trajectories-3"><span class="header-section-number">5.5.3</span> Fixed mutation trajectories</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#variant-time-to-majority" id="toc-variant-time-to-majority" class="nav-link" data-scroll-target="#variant-time-to-majority"><span class="header-section-number">6</span> Variant time to majority</a>
  <ul class="collapse">
  <li><a href="#de-novo-variants-both-amino-acid-chaning-and-non-changing" id="toc-de-novo-variants-both-amino-acid-chaning-and-non-changing" class="nav-link" data-scroll-target="#de-novo-variants-both-amino-acid-chaning-and-non-changing"><span class="header-section-number">6.1</span> De novo variants (both amino acid chaning and non-changing)</a></li>
  <li><a href="#de-novo-variants-non-synonymous" id="toc-de-novo-variants-non-synonymous" class="nav-link" data-scroll-target="#de-novo-variants-non-synonymous"><span class="header-section-number">6.2</span> De novo variants, non-synonymous</a></li>
  </ul></li>
  <li><a href="#time-to-mutation-emergence-in-parallel-genes" id="toc-time-to-mutation-emergence-in-parallel-genes" class="nav-link" data-scroll-target="#time-to-mutation-emergence-in-parallel-genes"><span class="header-section-number">7</span> Time to mutation emergence in parallel genes</a></li>
  <li><a href="#relationship-between-gene-parallelism-and-f_max" id="toc-relationship-between-gene-parallelism-and-f_max" class="nav-link" data-scroll-target="#relationship-between-gene-parallelism-and-f_max"><span class="header-section-number">8</span> Relationship between gene parallelism and <span class="math inline">\(f_{max}\)</span></a>
  <ul class="collapse">
  <li><a href="#delta-l" id="toc-delta-l" class="nav-link" data-scroll-target="#delta-l"><span class="header-section-number">8.1</span> <span class="math inline">\(\Delta l\)</span></a></li>
  <li><a href="#g-score" id="toc-g-score" class="nav-link" data-scroll-target="#g-score"><span class="header-section-number">8.2</span> G score</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/metagenome/01_format_variants.html">Metagenome sequencing</a></li><li class="breadcrumb-item"><a href="../../R/metagenome/04_mutation_dynamics.html">4. Mutational dynamics</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Mutational dynamics</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 30, 2024</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    This notebook reads in the formatted metagenome variant frequencies and annotations. It makes some plots and does some basic analysis looking at the temporal dynamics and ultimate fates (e.g., extinction or fixation) of total and nonsynomous mutations from the metagenomes.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<p>Libraries and global variables</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb1-10"><a href="#cb1-10"></a></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"metagenome"</span>, <span class="st">"utils_parallelism.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Set up some directories</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-7"><a href="#cb2-7"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-8"><a href="#cb2-8"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read data</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># these were already filtered in the last step</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metagenome_variant_timeseries.tsv"</span>))</span>
<span id="cb3-3"><a href="#cb3-3"></a>degentab <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb3-4"><a href="#cb3-4"></a>genome_len <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_genome_len.tsv"</span>))</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co"># annotations</span></span>
<span id="cb3-6"><a href="#cb3-6"></a>annotations <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb3-7"><a href="#cb3-7"></a></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co"># coverage</span></span>
<span id="cb3-9"><a href="#cb3-9"></a>covslurpedfmtflt <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"coverage.tsv"</span>))</span>
<span id="cb3-10"><a href="#cb3-10"></a></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co"># parallel genes</span></span>
<span id="cb3-12"><a href="#cb3-12"></a>par_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"enriched_parallel_genes.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>cog_description <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="sc">~</span>COG_category_single,                                                  <span class="sc">~</span>COG_category_long,</span>
<span id="cb4-3"><a href="#cb4-3"></a>            <span class="st">"J"</span>,               <span class="st">"J - Translation, ribosomal structure and biogenesis"</span>,</span>
<span id="cb4-4"><a href="#cb4-4"></a>            <span class="st">"A"</span>,                               <span class="st">"A - RNA processing and modification"</span>,</span>
<span id="cb4-5"><a href="#cb4-5"></a>            <span class="st">"K"</span>,                                                 <span class="st">"K – Transcription"</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>            <span class="st">"L"</span>,                         <span class="st">"L - Replication, recombination and repair"</span>,</span>
<span id="cb4-7"><a href="#cb4-7"></a>            <span class="st">"B"</span>,                              <span class="st">"B - Chromatin structure and dynamics"</span>,</span>
<span id="cb4-8"><a href="#cb4-8"></a>            <span class="st">"D"</span>,    <span class="st">"D - Cell cycle control, cell division, chromosome partitioning"</span>,</span>
<span id="cb4-9"><a href="#cb4-9"></a>            <span class="st">"Y"</span>,                                             <span class="st">"Y - Nuclear structure"</span>,</span>
<span id="cb4-10"><a href="#cb4-10"></a>            <span class="st">"V"</span>,                                            <span class="st">"V - Defense mechanisms"</span>,</span>
<span id="cb4-11"><a href="#cb4-11"></a>            <span class="st">"T"</span>,                                <span class="st">"T - Signal transduction mechanisms"</span>,</span>
<span id="cb4-12"><a href="#cb4-12"></a>            <span class="st">"M"</span>,                        <span class="st">"M - Cell wall/membrane/envelope biogenesis"</span>,</span>
<span id="cb4-13"><a href="#cb4-13"></a>            <span class="st">"N"</span>,                                                 <span class="st">"N - Cell motility"</span>,</span>
<span id="cb4-14"><a href="#cb4-14"></a>            <span class="st">"Z"</span>,                                                  <span class="st">"Z – Cytoskeleton"</span>,</span>
<span id="cb4-15"><a href="#cb4-15"></a>            <span class="st">"W"</span>,                                      <span class="st">"W - Extracellular structures"</span>,</span>
<span id="cb4-16"><a href="#cb4-16"></a>            <span class="st">"U"</span>, <span class="st">"U - Intracellular trafficking, secretion, and vesicular transport"</span>,</span>
<span id="cb4-17"><a href="#cb4-17"></a>            <span class="st">"O"</span>,  <span class="st">"O - Posttranslational modification, protein turnover, chaperones"</span>,</span>
<span id="cb4-18"><a href="#cb4-18"></a>            <span class="st">"X"</span>,                              <span class="st">"X - Mobilome: prophages, transposons"</span>,</span>
<span id="cb4-19"><a href="#cb4-19"></a>            <span class="st">"C"</span>,                              <span class="st">"C - Energy production and conversion"</span>,</span>
<span id="cb4-20"><a href="#cb4-20"></a>            <span class="st">"G"</span>,                         <span class="st">"G - Carbohydrate transport and metabolism"</span>,</span>
<span id="cb4-21"><a href="#cb4-21"></a>            <span class="st">"E"</span>,                           <span class="st">"E - Amino acid transport and metabolism"</span>,</span>
<span id="cb4-22"><a href="#cb4-22"></a>            <span class="st">"F"</span>,                           <span class="st">"F - Nucleotide transport and metabolism"</span>,</span>
<span id="cb4-23"><a href="#cb4-23"></a>            <span class="st">"H"</span>,                             <span class="st">"H - Coenzyme transport and metabolism"</span>,</span>
<span id="cb4-24"><a href="#cb4-24"></a>            <span class="st">"I"</span>,                                <span class="st">"I - Lipid transport and metabolism"</span>,</span>
<span id="cb4-25"><a href="#cb4-25"></a>            <span class="st">"P"</span>,                        <span class="st">"P - Inorganic ion transport and metabolism"</span>,</span>
<span id="cb4-26"><a href="#cb4-26"></a>            <span class="st">"Q"</span>,  <span class="st">"Q - Secondary metabolites biosynthesis, transport and catabolism"</span>,</span>
<span id="cb4-27"><a href="#cb4-27"></a>            <span class="st">"R"</span>,                              <span class="st">"R - General function prediction only"</span>,</span>
<span id="cb4-28"><a href="#cb4-28"></a>            <span class="st">"S"</span>,                                              <span class="st">"S - Function unknown"</span></span>
<span id="cb4-29"><a href="#cb4-29"></a>  )</span>
<span id="cb4-30"><a href="#cb4-30"></a></span>
<span id="cb4-31"><a href="#cb4-31"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">12367</span>,</span>
<span id="cb4-32"><a href="#cb4-32"></a>                 cogpal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(cog_description<span class="sc">$</span>COG_category_single)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb4-33"><a href="#cb4-33"></a>)</span>
<span id="cb4-34"><a href="#cb4-34"></a><span class="fu">names</span>(cogpal) <span class="ot">&lt;-</span> cog_description<span class="sc">$</span>COG_category_long</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="metagenomic-coverage" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Metagenomic coverage</h1>
<p>Only looking at the 5 species for which we called variants. Everything else was below a mean coverage of 5 and even calling variants at a coverage of 5 is a bit sketchy. Generally, the higher the coverage your genome is the confidence you can have calling polymorphisms where the alternate allele is rare. Basically this shows that we can really only same something of confidence for HAMBI_1287, HAMBI_1972, HAMBI_1977, and HAMBI_2659.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>covslurpedfmtflt <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_2659"</span>, <span class="st">"HAMBI_0403"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(scaffold, <span class="st">"plas"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="fu">left_join</span>(<span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(mgvars, sample, time_days, replicate, prey_history, predator_history)),</span>
<span id="cb5-5"><a href="#cb5-5"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb5-7"><a href="#cb5-7"></a>  <span class="fu">mutate</span>(<span class="at">replicate_mean =</span> <span class="fu">mean</span>(trimmed_mean)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-8"><a href="#cb5-8"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> time_days, <span class="at">y =</span> trimmed_mean)) <span class="sc">+</span></span>
<span id="cb5-9"><a href="#cb5-9"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">5</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb5-10"><a href="#cb5-10"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, <span class="at">group =</span> <span class="fu">interaction</span>(strainID, prey_history, predator_history, replicate)),</span>
<span id="cb5-11"><a href="#cb5-11"></a>            <span class="at">alpha =</span> <span class="fl">0.35</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb5-12"><a href="#cb5-12"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, <span class="at">shape =</span> replicate),</span>
<span id="cb5-13"><a href="#cb5-13"></a>             <span class="at">alpha =</span> <span class="fl">0.35</span>) <span class="sc">+</span></span>
<span id="cb5-14"><a href="#cb5-14"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> replicate_mean, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb5-15"><a href="#cb5-15"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> replicate_mean, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb5-16"><a href="#cb5-16"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb5-17"><a href="#cb5-17"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb5-18"><a href="#cb5-18"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Experiment time (days)"</span>, <span class="at">y =</span> <span class="st">"Mean genome coverage"</span>, <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb5-19"><a href="#cb5-19"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>), <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb5-20"><a href="#cb5-20"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">300</span>)) <span class="sc">+</span></span>
<span id="cb5-21"><a href="#cb5-21"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb5-22"><a href="#cb5-22"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">28</span>, <span class="dv">60</span>)) <span class="sc">+</span></span>
<span id="cb5-23"><a href="#cb5-23"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb5-24"><a href="#cb5-24"></a>  <span class="fu">theme</span>(</span>
<span id="cb5-25"><a href="#cb5-25"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb5-26"><a href="#cb5-26"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb5-27"><a href="#cb5-27"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb5-28"><a href="#cb5-28"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Genomic coverage in the metagenomes over time. The genomes of these five species exceeded a coverage of at least 5X (black dashed horizontal line) at at least one sampling time from the experiment. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.
</figcaption>
</figure>
</div>
</section>
<section id="mutational-spectra" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Mutational spectra</h1>
<p>Some broad scale patterns in how mutations trajectories vary between species and treatments can be observed from the mutational spectra. Below we examine how the fraction of mutations exceeding the maximum observed allele frequency <span class="math inline">\(f_{max}\)</span> relates to different levels of <span class="math inline">\(f_{max}\)</span> and compare this between different evolution treatments and across species.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>allelespectra <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, replicate, prey_history, predator_history, freq_alt_complete) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="fu">group_by</span>(strainID, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5"></a>  <span class="co"># the rbeta is to just add some noise at the lower and upper ends where there are ties</span></span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt_complete =</span> <span class="fu">case_when</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> freq_alt_complete <span class="sc">-</span> <span class="fu">rbeta</span>(<span class="fu">n</span>(), <span class="fl">0.1</span>, <span class="dv">10</span>, <span class="at">ncp =</span> <span class="dv">0</span>),</span>
<span id="cb6-7"><a href="#cb6-7"></a>                                       freq_alt_complete <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> freq_alt_complete <span class="sc">+</span> <span class="fu">rbeta</span>(<span class="fu">n</span>(), <span class="fl">0.1</span>, <span class="dv">10</span>, <span class="at">ncp =</span> <span class="dv">0</span>),</span>
<span id="cb6-8"><a href="#cb6-8"></a>                                       <span class="cn">TRUE</span> <span class="sc">~</span> freq_alt_complete)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-9"><a href="#cb6-9"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-10"><a href="#cb6-10"></a>  <span class="fu">arrange</span>(strainID, replicate, prey_history, predator_history, <span class="fu">desc</span>(freq_alt_complete)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-11"><a href="#cb6-11"></a>  <span class="fu">group_by</span>(strainID, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb6-12"><a href="#cb6-12"></a>  <span class="co"># get 1 - emiprical cumulative distribution</span></span>
<span id="cb6-13"><a href="#cb6-13"></a>  <span class="fu">mutate</span>(<span class="at">f_gt_fmax =</span> <span class="fl">1.001</span> <span class="sc">-</span> <span class="fu">ecdf</span>(freq_alt_complete)(freq_alt_complete))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">ggplot</span>(allelespectra, <span class="fu">aes</span>(<span class="at">x =</span> freq_alt_complete, <span class="at">y =</span> f_gt_fmax)) <span class="sc">+</span> </span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="fu">geom_step</span>(<span class="fu">aes</span>(<span class="at">color =</span> predator_history, <span class="at">linetype =</span> replicate, <span class="at">group =</span> <span class="fu">interaction</span>(predator_history, replicate))) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Maximum observed allele frequency $</span><span class="sc">\\</span><span class="st">textit{f_{max}}$"</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">"Fraction of mutations $</span><span class="sc">\\</span><span class="st">geq </span><span class="sc">\\</span><span class="st">textit{f_{max}}$"</span>)) <span class="sc">+</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> strainID, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"bl"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb7-6"><a href="#cb7-6"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.01</span>, <span class="fl">0.001</span>)) <span class="sc">+</span></span>
<span id="cb7-7"><a href="#cb7-7"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb7-9"><a href="#cb7-9"></a>    <span class="fu">theme</span>(</span>
<span id="cb7-10"><a href="#cb7-10"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb7-11"><a href="#cb7-11"></a>      <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb7-12"><a href="#cb7-12"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb7-13"><a href="#cb7-13"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb7-14"><a href="#cb7-14"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-02" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: The empirical cumulative distribution of the maximum frequency that mutations reached <span class="math inline">\(f_{max}\)</span> for each taxon-treatment combination.
</figcaption>
</figure>
</div>
<p>Mutational spectra from the ancestral bacteria are of lower resolution because there are fewer mutations in those populations during the course of the experiment. Spectra from evolved bacteria are of higher resolution but include both standing genetic variation from the start of the experiment and de novo mutions. Thus, comparing ancestral to evolved bacteria spectra is not particularly meaningful. However, some insights may be gained from comparing different predator evolutionary histories nested with bacterial evolutionary histories. For evolved bacteria it appears that for 1972, 1977, and 2659 that the trajectory of <span class="math inline">\(f_{max}\)</span> in the ancestral predator treatment drops relative to the evolved predator treatment. This would then imply that, on the whole, bacterial alleles from the evolved predator treatments increased to higher frequencies than alleles in the ancestral predator treatments. However, I am not sure there is strong support for this assertion - there are only three replicates and usually at least one replicate from the evolved predator treatment overlaps with the ancestral predator treatment. A comparison of the spectra between species indicates that, generally, some species (HAMBI_1972 and HAMBI_0403) have more high frequency mutations than others (HAMBI_1287 and HAMBI_2659).</p>
</section>
<section id="mutational-dynamics" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Mutational dynamics</h1>
<p>We calculated the accumulation of mutations by time t as the sum of derived allele frequencies</p>
<p><span class="math display">\[ M\left(t\right)\equiv\sum_m {\widehat f}_{pmt} \]</span> where <span class="math inline">\({\widehat f}_{pmt}=A_{pmt}/D_{pmt}\)</span> and A is the coverage of the alternative allele and D is the total depth of coverage. To partially mitigate noise associated with calling low frequency variants in medium-low coverage samples (~20X) we restricted our analysis to variants where <span class="math inline">\({\widehat f}_{pmt} \geq 0.15\)</span>.</p>
<section id="some-data-formatting" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="some-data-formatting"><span class="header-section-number">5.1</span> Some data formatting</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># only denovo (i.e. not observed at time 0)</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>mgvars <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb8-6"><a href="#cb8-6"></a>  </span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="co"># only denovo (i.e. not observed at time 0)</span></span>
<span id="cb8-8"><a href="#cb8-8"></a>mgvars_denovo <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb8-9"><a href="#cb8-9"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb8-11"><a href="#cb8-11"></a>  <span class="fu">mutate</span>(<span class="at">denovo =</span> <span class="fu">if_else</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> time_days <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="cn">NA_real_</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-12"><a href="#cb8-12"></a>  <span class="fu">fill</span>(denovo, <span class="at">.direction =</span> <span class="st">"down"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-13"><a href="#cb8-13"></a>  <span class="fu">relocate</span>(group_id, freq_alt_complete, denovo) <span class="sc">%&gt;%</span> </span>
<span id="cb8-14"><a href="#cb8-14"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-15"><a href="#cb8-15"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(denovo))</span>
<span id="cb8-16"><a href="#cb8-16"></a></span>
<span id="cb8-17"><a href="#cb8-17"></a><span class="co"># only nonsynonymous variants</span></span>
<span id="cb8-18"><a href="#cb8-18"></a>mgvars_ns <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb8-19"><a href="#cb8-19"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(effect, <span class="st">"intergenic|intragenic|synonymous|fusion"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-20"><a href="#cb8-20"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(impact, <span class="st">"MODIFIER"</span>))</span>
<span id="cb8-21"><a href="#cb8-21"></a></span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="co"># only denovo (i.e. not observed at time 0) and nonsynonymous variants</span></span>
<span id="cb8-23"><a href="#cb8-23"></a>mgvars_ns_denovo <span class="ot">&lt;-</span> mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb8-24"><a href="#cb8-24"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb8-25"><a href="#cb8-25"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb8-26"><a href="#cb8-26"></a>  <span class="fu">mutate</span>(<span class="at">denovo =</span> <span class="fu">if_else</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> time_days <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="cn">NA_real_</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-27"><a href="#cb8-27"></a>  <span class="fu">fill</span>(denovo, <span class="at">.direction =</span> <span class="st">"down"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-28"><a href="#cb8-28"></a>  <span class="fu">relocate</span>(group_id, freq_alt_complete, denovo) <span class="sc">%&gt;%</span> </span>
<span id="cb8-29"><a href="#cb8-29"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-30"><a href="#cb8-30"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(denovo)) </span>
<span id="cb8-31"><a href="#cb8-31"></a></span>
<span id="cb8-32"><a href="#cb8-32"></a>cummt <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb8-33"><a href="#cb8-33"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-34"><a href="#cb8-34"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb8-35"><a href="#cb8-35"></a></span>
<span id="cb8-36"><a href="#cb8-36"></a>cummt_denovo <span class="ot">&lt;-</span> mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb8-37"><a href="#cb8-37"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-38"><a href="#cb8-38"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb8-39"><a href="#cb8-39"></a></span>
<span id="cb8-40"><a href="#cb8-40"></a>cummt_ns <span class="ot">&lt;-</span> mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb8-41"><a href="#cb8-41"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-42"><a href="#cb8-42"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb8-43"><a href="#cb8-43"></a></span>
<span id="cb8-44"><a href="#cb8-44"></a>cummt_ns_denovo <span class="ot">&lt;-</span> mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb8-45"><a href="#cb8-45"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-46"><a href="#cb8-46"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>plotmuttraj <span class="ot">&lt;-</span> <span class="cf">function</span>(df, x, alpha, ylab){</span>
<span id="cb9-2"><a href="#cb9-2"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> time_days, <span class="at">y =</span> {{ x }})) <span class="sc">+</span></span>
<span id="cb9-3"><a href="#cb9-3"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, <span class="at">shape =</span> replicate), <span class="at">alpha =</span> alpha) <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, </span>
<span id="cb9-5"><a href="#cb9-5"></a>                  <span class="at">group =</span> <span class="fu">interaction</span>(strainID, prey_history, predator_history, replicate)),</span>
<span id="cb9-6"><a href="#cb9-6"></a>              <span class="at">alpha =</span> alpha, </span>
<span id="cb9-7"><a href="#cb9-7"></a>              <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb9-8"><a href="#cb9-8"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb9-9"><a href="#cb9-9"></a>    <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb9-10"><a href="#cb9-10"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Experiment time (days)"</span>, <span class="at">y =</span> ylab, <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb9-11"><a href="#cb9-11"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">28</span>, <span class="dv">60</span>)) <span class="sc">+</span></span>
<span id="cb9-12"><a href="#cb9-12"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb9-13"><a href="#cb9-13"></a>    <span class="fu">theme</span>(</span>
<span id="cb9-14"><a href="#cb9-14"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb9-15"><a href="#cb9-15"></a>      <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb9-16"><a href="#cb9-16"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb9-17"><a href="#cb9-17"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb9-18"><a href="#cb9-18"></a>    )</span>
<span id="cb9-19"><a href="#cb9-19"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="sec-allvars" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="sec-allvars"><span class="header-section-number">5.2</span> All variants</h2>
<p>This section considers all variants from the experiment including standing genetic variation present in the evolved populations at T0 and de novo mutations that arise later. It also includes both protein changing and protein non-changing variants.</p>
<section id="cumulative-mutation-trajectories-mt" class="level3" data-number="5.2.1">
<h3 data-number="5.2.1" class="anchored" data-anchor-id="cumulative-mutation-trajectories-mt"><span class="header-section-number">5.2.1</span> Cumulative mutation trajectories (M(t))</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>cummt <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t)"</span>) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-03" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Cumulative mutation (M(t)) trajectories of both amino acid altering and amino acid non-altering variants in HAMBI species genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.
</figcaption>
</figure>
</div>
</section>
<section id="total-mutation-trajectories" class="level3" data-number="5.2.2">
<h3 data-number="5.2.2" class="anchored" data-anchor-id="total-mutation-trajectories"><span class="header-section-number">5.2.2</span> Total mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb11-3"><a href="#cb11-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of mutations"</span>) <span class="sc">+</span></span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb11-8"><a href="#cb11-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb11-9"><a href="#cb11-9"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb11-10"><a href="#cb11-10"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb11-11"><a href="#cb11-11"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Trajectories of the <strong>total number of non-zero mutations</strong> detected at each time point for both amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.
</figcaption>
</figure>
</div>
</section>
<section id="fixed-mutation-trajectories" class="level3" data-number="5.2.3">
<h3 data-number="5.2.3" class="anchored" data-anchor-id="fixed-mutation-trajectories"><span class="header-section-number">5.2.3</span> Fixed mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb12-3"><a href="#cb12-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-05" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Trajectories of the total number of <strong>fixed</strong> mutations detected at each time point for both amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="de-novo-variants" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="de-novo-variants"><span class="header-section-number">5.3</span> De novo variants</h2>
<p>This section considers only de novo mutations that did not exist within the evolved populations at T0. It includes both protein changing and protein non-changing variants. Thus, the results here for the ancestral bacteria populations here will be the same as in <a href="#sec-allvars" class="quarto-xref">Section&nbsp;5.2</a>.</p>
<section id="cumulative-mutation-trajectories-mt-1" class="level3" data-number="5.3.1">
<h3 data-number="5.3.1" class="anchored" data-anchor-id="cumulative-mutation-trajectories-mt-1"><span class="header-section-number">5.3.1</span> Cumulative mutation trajectories (M(t))</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>fig06 <span class="ot">&lt;-</span> cummt_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t)"</span>) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb13-7"><a href="#cb13-7"></a></span>
<span id="cb13-8"><a href="#cb13-8"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"Mt_denovo.svg"</span>), fig06, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">5</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb13-9"><a href="#cb13-9"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb13-10"><a href="#cb13-10"></a></span>
<span id="cb13-11"><a href="#cb13-11"></a>fig06</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-06" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Cumulative mutation (M(t)) trajectories of de novo amino acid altering and amino acid non-altering variants in HAMBI species genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-03" class="quarto-xref">Figure&nbsp;3</a> but only includes de novo variants (i.e.&nbsp;excludes standing genetic variation in the evolved populations).
</figcaption>
</figure>
</div>
</section>
<section id="total-mutation-trajectories-1" class="level3" data-number="5.3.2">
<h3 data-number="5.3.2" class="anchored" data-anchor-id="total-mutation-trajectories-1"><span class="header-section-number">5.3.2</span> Total mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb14-3"><a href="#cb14-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-6"><a href="#cb14-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of mutations"</span>) <span class="sc">+</span></span>
<span id="cb14-7"><a href="#cb14-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb14-8"><a href="#cb14-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb14-9"><a href="#cb14-9"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb14-10"><a href="#cb14-10"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb14-11"><a href="#cb14-11"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-07" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-07-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-07-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Trajectories of the <strong>total number of non-zero mutations</strong> detected at each time point for de novo amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-04" class="quarto-xref">Figure&nbsp;4</a> but only includes de novo variants (i.e.&nbsp;excludes standing genetic variation in the evolved populations).
</figcaption>
</figure>
</div>
</section>
<section id="fixed-mutation-trajectories-1" class="level3" data-number="5.3.3">
<h3 data-number="5.3.3" class="anchored" data-anchor-id="fixed-mutation-trajectories-1"><span class="header-section-number">5.3.3</span> Fixed mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb15-3"><a href="#cb15-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb15-7"><a href="#cb15-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb15-8"><a href="#cb15-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-08" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-08-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-08-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Trajectories of the total number of <strong>fixed</strong> mutations detected at each time point for de novo amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-05" class="quarto-xref">Figure&nbsp;5</a> but only includes de novo variants (i.e.&nbsp;excludes standing genetic variation in the evolved populations).
</figcaption>
</figure>
</div>
</section>
</section>
<section id="sec-nsvars" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="sec-nsvars"><span class="header-section-number">5.4</span> Non-synonymous variants</h2>
<p>This section considers all nonsynonymous (i.e., amino acid altering) variants from the experiment including standing genetic variation present in the evolved populations at T0 and de novo mutations that arise later.</p>
<section id="cumulative-mutation-trajectories-mt-2" class="level3" data-number="5.4.1">
<h3 data-number="5.4.1" class="anchored" data-anchor-id="cumulative-mutation-trajectories-mt-2"><span class="header-section-number">5.4.1</span> Cumulative mutation trajectories (M(t))</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>cummt_ns <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-4"><a href="#cb16-4"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t) (nonsynonymous)"</span>) <span class="sc">+</span></span>
<span id="cb16-5"><a href="#cb16-5"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-09" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-09-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-09-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Cumulative mutation (M(t)) trajectories of amino acid altering variants in HAMBI species genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-03" class="quarto-xref">Figure&nbsp;3</a> but only includes nonsynonymous (amino acid changing) variants.
</figcaption>
</figure>
</div>
</section>
<section id="total-mutation-trajectories-2" class="level3" data-number="5.4.2">
<h3 data-number="5.4.2" class="anchored" data-anchor-id="total-mutation-trajectories-2"><span class="header-section-number">5.4.2</span> Total mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb17-3"><a href="#cb17-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of nonsynonymous mutations"</span>) <span class="sc">+</span></span>
<span id="cb17-7"><a href="#cb17-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb17-8"><a href="#cb17-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb17-9"><a href="#cb17-9"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb17-10"><a href="#cb17-10"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb17-11"><a href="#cb17-11"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-10" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-10-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-10-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Trajectories of the <strong>total number of non-zero mutations</strong> detected at each time point for amino acid altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-04" class="quarto-xref">Figure&nbsp;4</a> but only includes nonsynonymous (amino acid changing) variants.
</figcaption>
</figure>
</div>
</section>
<section id="fixed-mutation-trajectories-2" class="level3" data-number="5.4.3">
<h3 data-number="5.4.3" class="anchored" data-anchor-id="fixed-mutation-trajectories-2"><span class="header-section-number">5.4.3</span> Fixed mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb18-3"><a href="#cb18-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-6"><a href="#cb18-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of nonsynonymous fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb18-8"><a href="#cb18-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-11" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-11-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-11-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Trajectories of the total number of <strong>fixed</strong> mutations detected at each time point for amino acid altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-05" class="quarto-xref">Figure&nbsp;5</a> but only includes non-synonymous (amino acid changing) variants.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="de-novo-non-synonymous-variants" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="de-novo-non-synonymous-variants"><span class="header-section-number">5.5</span> De novo, non-synonymous variants</h2>
<p>This section considers only de novo mutations that did not exist within the evolved populations at T0 for only non-synonymous (i.e., amino acid altering) variants. Thus, the results here for the ancestral bacteria populations will be the same as in <a href="#sec-nsvars" class="quarto-xref">Section&nbsp;5.4</a></p>
<section id="cumulative-mutation-trajectories-mt-3" class="level3" data-number="5.5.1">
<h3 data-number="5.5.1" class="anchored" data-anchor-id="cumulative-mutation-trajectories-mt-3"><span class="header-section-number">5.5.1</span> Cumulative mutation trajectories (M(t))</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>cummt_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t) (de novo, nonsynonymous)"</span>) <span class="sc">+</span></span>
<span id="cb19-5"><a href="#cb19-5"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-12" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-12-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-12-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Cumulative mutation (M(t)) trajectories of de novo amino acid altering variants in HAMBI species genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-03" class="quarto-xref">Figure&nbsp;3</a> but only includes de novo (i.e., not present at T0) and non-synonymous (amino acid changing) variants.
</figcaption>
</figure>
</div>
</section>
<section id="total-mutation-trajectories-3" class="level3" data-number="5.5.2">
<h3 data-number="5.5.2" class="anchored" data-anchor-id="total-mutation-trajectories-3"><span class="header-section-number">5.5.2</span> Total mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>fig13 <span class="ot">&lt;-</span> mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb20-3"><a href="#cb20-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of de novo nonsynonymous mutations"</span>) <span class="sc">+</span></span>
<span id="cb20-7"><a href="#cb20-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb20-8"><a href="#cb20-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb20-9"><a href="#cb20-9"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb20-10"><a href="#cb20-10"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb20-11"><a href="#cb20-11"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>))</span>
<span id="cb20-12"><a href="#cb20-12"></a></span>
<span id="cb20-13"><a href="#cb20-13"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"cummuts_denovo_nonsyn.svg"</span>), fig13, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">5</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb20-14"><a href="#cb20-14"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb20-15"><a href="#cb20-15"></a></span>
<span id="cb20-16"><a href="#cb20-16"></a>fig13</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-13" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-13-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-13-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Trajectories of the <strong>total number of non-zero mutations</strong> detected at each time point for de novo amino acid altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-04" class="quarto-xref">Figure&nbsp;4</a> but only includes de novo (i.e., not present at T0) and non-synonymous (amino acid changing) variants.
</figcaption>
</figure>
</div>
</section>
<section id="fixed-mutation-trajectories-3" class="level3" data-number="5.5.3">
<h3 data-number="5.5.3" class="anchored" data-anchor-id="fixed-mutation-trajectories-3"><span class="header-section-number">5.5.3</span> Fixed mutation trajectories</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb21-3"><a href="#cb21-3"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of de novo, nonsynonymous fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb21-7"><a href="#cb21-7"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb21-8"><a href="#cb21-8"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-14" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-14-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-14-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14: Trajectories of the total number of <strong>fixed</strong> mutations detected at each time point for de novo amino acid altering variants in HAMBI genomes with metagenomic coverage <span class="math inline">\(\geq\)</span> 5X in at least one time point (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to <a href="#fig-05" class="quarto-xref">Figure&nbsp;5</a> but only includes de novo (i.e., not present at T0) and non-synonymous (amino acid changing) variants.
</figcaption>
</figure>
</div>
</section>
</section>
</section>
<section id="variant-time-to-majority" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Variant time to majority</h1>
<p>Does the evolutionary history of prey/predator impact how long it takes alleles to reach majority status (<span class="math inline">\(f_{max} \geq 0.5\)</span>). For every alternative allele that emerged in the experiment de novo, we plot the time that it took for that alt allele to reach majority status (i.e., the alt allele dominates the reference allele).</p>
<section id="de-novo-variants-both-amino-acid-chaning-and-non-changing" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="de-novo-variants-both-amino-acid-chaning-and-non-changing"><span class="header-section-number">6.1</span> De novo variants (both amino acid chaning and non-changing)</h2>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2"></a>  <span class="fu">group_by</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete) <span class="sc">&amp;</span> freq_alt_complete <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-4"><a href="#cb22-4"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-5"><a href="#cb22-5"></a>  <span class="fu">slice_min</span>(time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb22-6"><a href="#cb22-6"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-7"><a href="#cb22-7"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(time_days))) <span class="sc">+</span></span>
<span id="cb22-8"><a href="#cb22-8"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID)) <span class="sc">+</span></span>
<span id="cb22-9"><a href="#cb22-9"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb22-10"><a href="#cb22-10"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb22-11"><a href="#cb22-11"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Time to major allele ( $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ )"</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">" $n_{allele}$ with $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ "</span>), <span class="at">fill =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb22-13"><a href="#cb22-13"></a>  <span class="fu">theme</span>(</span>
<span id="cb22-14"><a href="#cb22-14"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb22-15"><a href="#cb22-15"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb22-16"><a href="#cb22-16"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-17"><a href="#cb22-17"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-15" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-15-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-15-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;15: Count of alternative alleles with <span class="math inline">\(f_{max} \geq 0.5\)</span> plotted against the time (days) when the alternative allele first reaches major allele status <span class="math inline">\((f_{max} \geq 0.5)\)</span>. Colors represent the proportion of alleles from different HAMBI species. Plot includes both amino acid changing and non-changing de novo mutations (i.e.&nbsp;excludes standing genetic variation from T0 in the evolved bacterial populations).
</figcaption>
</figure>
</div>
</section>
<section id="de-novo-variants-non-synonymous" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="de-novo-variants-non-synonymous"><span class="header-section-number">6.2</span> De novo variants, non-synonymous</h2>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">group_by</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete) <span class="sc">&amp;</span> freq_alt_complete <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5"></a>  <span class="fu">slice_min</span>(time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb23-6"><a href="#cb23-6"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-7"><a href="#cb23-7"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(time_days))) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID)) <span class="sc">+</span></span>
<span id="cb23-9"><a href="#cb23-9"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb23-10"><a href="#cb23-10"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb23-11"><a href="#cb23-11"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Time to major allele ( $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ )"</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">" $n_{allele}$ with $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ "</span>), <span class="at">fill =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb23-12"><a href="#cb23-12"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb23-13"><a href="#cb23-13"></a>  <span class="fu">theme</span>(</span>
<span id="cb23-14"><a href="#cb23-14"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb23-15"><a href="#cb23-15"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb23-16"><a href="#cb23-16"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb23-17"><a href="#cb23-17"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-16" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-16-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-16-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;16: Count of alternative alleles with <span class="math inline">\(f_{max} \geq 0.5\)</span> plotted against the time (days) when the alternative allele first reaches major allele status <span class="math inline">\((f_{max} \geq 0.5)\)</span>. Colors represent the proportion of alleles from different HAMBI species. Plot includes only amino acid changing de novo variants (i.e.&nbsp;excludes standing genetic variation from T0 in the evolved bacterial populations).
</figcaption>
</figure>
</div>
<p>This plot suggests that most de novo alternative alleles (either amino acid changing or non chaning) become the major allele by either day 28 or 60 for the ancestral bacteria populations. For the evolved populations this evolution seems to happen faster where some alt alleles become the major allele already by day 28 but this is only for HAMBI_2659.</p>
</section>
</section>
<section id="time-to-mutation-emergence-in-parallel-genes" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Time to mutation emergence in parallel genes</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>df_ttm <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, mgvars_ns_denovo) <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2"></a>  <span class="fu">group_by</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete) <span class="sc">&amp;</span> freq_alt_complete <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="co"># uncomment to only look at emergence of minor-to-major allele transition</span></span>
<span id="cb24-5"><a href="#cb24-5"></a>  <span class="co">#filter(freq_alt_complete &gt;= 0.50) %&gt;% </span></span>
<span id="cb24-6"><a href="#cb24-6"></a>  <span class="fu">slice_min</span>(time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb24-7"><a href="#cb24-7"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-8"><a href="#cb24-8"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb24-9"><a href="#cb24-9"></a>  <span class="co"># some manual renaming</span></span>
<span id="cb24-10"><a href="#cb24-10"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">case_when</span>(locus_tag <span class="sc">==</span> <span class="st">"H1287_02172"</span> <span class="sc">~</span> <span class="st">"yddV_2"</span>,</span>
<span id="cb24-11"><a href="#cb24-11"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1977_02612"</span> <span class="sc">~</span> <span class="st">"dppA/oppA"</span>,</span>
<span id="cb24-12"><a href="#cb24-12"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00299"</span> <span class="sc">~</span> <span class="st">"PAP2_like"</span>,</span>
<span id="cb24-13"><a href="#cb24-13"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02826"</span> <span class="sc">~</span> <span class="st">"ynjC"</span>,</span>
<span id="cb24-14"><a href="#cb24-14"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00671"</span> <span class="sc">~</span> <span class="st">"yqaA"</span>,</span>
<span id="cb24-15"><a href="#cb24-15"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00421"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb24-16"><a href="#cb24-16"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02723"</span> <span class="sc">~</span> <span class="st">"yifE"</span>,</span>
<span id="cb24-17"><a href="#cb24-17"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H2659_01789"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb24-18"><a href="#cb24-18"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-19"><a href="#cb24-19"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene),</span>
<span id="cb24-20"><a href="#cb24-20"></a>         <span class="at">treat =</span> <span class="fu">paste0</span>(<span class="st">"prey:"</span>, prey_history, <span class="st">" | pred:"</span>, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-21"><a href="#cb24-21"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, Description, gene_lab)) <span class="sc">%&gt;%</span></span>
<span id="cb24-22"><a href="#cb24-22"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">paste0</span>(treat,<span class="st">"-"</span>, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-23"><a href="#cb24-23"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab2 =</span> <span class="fu">fct_reorder</span>(<span class="fu">factor</span>(gene_lab), time_days))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(strainID, prey_history, predator_history, locus_tag,
chrom)`</code></pre>
</div>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>pbar <span class="ot">&lt;-</span> df_ttm <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="fu">distinct</span>(gene_lab2, COG_category_long) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> gene_lab2, <span class="at">fill =</span> COG_category_long)) <span class="sc">+</span> </span>
<span id="cb26-4"><a href="#cb26-4"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> cogpal) <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6"></a>  <span class="fu">theme_void</span>()</span>
<span id="cb26-7"><a href="#cb26-7"></a></span>
<span id="cb26-8"><a href="#cb26-8"></a>ptext <span class="ot">&lt;-</span> df_ttm <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9"></a>  <span class="fu">distinct</span>(gene_lab2, COG_category_long, COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="#cb26-10"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">COG_category_long =</span> <span class="st">"S - Function unknown"</span>, <span class="at">COG_category_single =</span> <span class="st">"S"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-11"><a href="#cb26-11"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> gene_lab2)) <span class="sc">+</span> </span>
<span id="cb26-12"><a href="#cb26-12"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> COG_category_single, <span class="at">color =</span> COG_category_long)) <span class="sc">+</span></span>
<span id="cb26-13"><a href="#cb26-13"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cogpal) <span class="sc">+</span></span>
<span id="cb26-14"><a href="#cb26-14"></a>  <span class="fu">theme_void</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>ppoint <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_ttm, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(time_days), <span class="at">y =</span> gene_lab2)) <span class="sc">+</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> treat, <span class="at">shape =</span> strainID), <span class="at">size =</span> <span class="dv">2</span>, <span class="at">width =</span> <span class="fl">0.15</span>, <span class="at">height =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb27-3"><a href="#cb27-3"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Time to mutation emergence"</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb27-4"><a href="#cb27-4"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span> </span>
<span id="cb27-5"><a href="#cb27-5"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb27-6"><a href="#cb27-6"></a>  <span class="fu">theme</span>(</span>
<span id="cb27-7"><a href="#cb27-7"></a>    <span class="at">legend.position =</span> <span class="st">"left"</span>,</span>
<span id="cb27-8"><a href="#cb27-8"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb27-9"><a href="#cb27-9"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb27-10"><a href="#cb27-10"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>ptogether <span class="ot">&lt;-</span> ppoint <span class="sc">+</span> ptext <span class="sc">+</span></span>
<span id="cb28-2"><a href="#cb28-2"></a>    <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.15</span>), <span class="at">guides =</span> <span class="st">"collect"</span>)</span>
<span id="cb28-3"><a href="#cb28-3"></a></span>
<span id="cb28-4"><a href="#cb28-4"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"par_gene_time2emergence.svg"</span>), ptogether, <span class="at">width=</span><span class="dv">9</span>, <span class="at">height=</span><span class="dv">7</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb28-5"><a href="#cb28-5"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb28-6"><a href="#cb28-6"></a></span>
<span id="cb28-7"><a href="#cb28-7"></a>ptogether</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-17" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-17-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-17-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;17
</figcaption>
</figure>
</div>
<p>dhaR has two associated COG categories 1. K – Transcription 2. Q - Secondary metabolites biosynthesis transport and catabolism</p>
<p>smf-1 (Major fimbrial subunit SMF-1) has two associated COG categoires 1. N - Cell motility 2. U - intracellular trafficiking, secretion, and vesicular transport</p>
</section>
<section id="relationship-between-gene-parallelism-and-f_max" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Relationship between gene parallelism and <span class="math inline">\(f_{max}\)</span></h1>
<p>Look if there is a relationship between extent of gene parallelism (<span class="math inline">\(\Delta L\)</span>) and the max frequency that variants reach during the experiment.</p>
<p>As seen from the results below we observe that the degree of parallelism <span class="math inline">\(\Delta l\)</span> or <span class="math inline">\(G_{i}\)</span> increases with <span class="math inline">\(f_{max}\)</span> across experimental treatment combinations and species, which is consistent with the idea that mutations with higher <span class="math inline">\(f_{max}\)</span> are primarily being driven by positive selection across these treatments.</p>
<section id="delta-l" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="delta-l"><span class="header-section-number">8.1</span> <span class="math inline">\(\Delta l\)</span></h2>
<p>This is the same formatting that is done in <a href="../../R/metagenome/02_parallelism.html">the prior parallelism analysis.</a></p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>tofilter <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="fu">filter</span>(time_days <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="#cb29-3"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-4"><a href="#cb29-4"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb29-5"><a href="#cb29-5"></a>  <span class="fu">distinct</span>()</span>
<span id="cb29-6"><a href="#cb29-6"></a></span>
<span id="cb29-7"><a href="#cb29-7"></a>mgvars_filt_mb_ns <span class="ot">&lt;-</span> <span class="fu">anti_join</span>(mgvars, tofilter, <span class="at">by =</span> <span class="fu">join_by</span>(chrom, pos, ref, alt)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-8"><a href="#cb29-8"></a>  <span class="co"># exclude variants that were added to make complete time course</span></span>
<span id="cb29-9"><a href="#cb29-9"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-10"><a href="#cb29-10"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-11"><a href="#cb29-11"></a>  <span class="fu">filter</span>(time_days <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-12"><a href="#cb29-12"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(effect, <span class="st">"intergenic|intragenic|synonymous|fusion"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-13"><a href="#cb29-13"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(impact, <span class="st">"MODIFIER"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-14"><a href="#cb29-14"></a>  <span class="co"># double check to remove any noncoding mutations</span></span>
<span id="cb29-15"><a href="#cb29-15"></a>  <span class="fu">filter</span>(locus_tag <span class="sc">%in%</span> <span class="fu">pull</span>(<span class="fu">filter</span>(degentab, <span class="sc">!</span><span class="fu">is.na</span>(ns_length)), locus_tag))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We now want to look at the extent of multiplicity at different binned <span class="math inline">\(f_{max}\)</span></p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>multiplicity_binned <span class="ot">&lt;-</span> mgvars_filt_mb_ns <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2"></a>  <span class="co"># automatically bin f_max into 5 categories</span></span>
<span id="cb30-3"><a href="#cb30-3"></a>  <span class="fu">mutate</span>(<span class="at">fbin =</span> <span class="fu">cut_interval</span>(freq_alt_complete, <span class="dv">5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="#cb30-4"></a>  <span class="fu">summarize</span>(<span class="at">n_i =</span> <span class="fu">n</span>(), </span>
<span id="cb30-5"><a href="#cb30-5"></a>            <span class="at">n_replicate =</span> <span class="fu">n_distinct</span>(replicate),</span>
<span id="cb30-6"><a href="#cb30-6"></a>    <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"locus_tag"</span>, <span class="st">"strainID"</span>, <span class="st">"fbin"</span>, <span class="st">"prey_history"</span>, <span class="st">"predator_history"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-7"><a href="#cb30-7"></a>  <span class="fu">group_by</span>(locus_tag, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb30-8"><a href="#cb30-8"></a>  <span class="fu">complete</span>(fbin, prey_history, predator_history,</span>
<span id="cb30-9"><a href="#cb30-9"></a>           <span class="at">fill =</span> <span class="fu">list</span>(<span class="at">n_i =</span> <span class="dv">0</span>, <span class="at">n_replicate =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-10"><a href="#cb30-10"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-11"><a href="#cb30-11"></a>  <span class="fu">group_by</span>(strainID, fbin, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb30-12"><a href="#cb30-12"></a>  <span class="fu">mutate</span>(<span class="at">Ntot =</span> <span class="fu">sum</span>(n_i)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-13"><a href="#cb30-13"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Make df with information about gene lengths</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>gene_tab <span class="ot">&lt;-</span> degentab <span class="sc">%&gt;%</span> </span>
<span id="cb31-2"><a href="#cb31-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, locus_tag, <span class="at">l_i =</span> ns_length) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb31-5"><a href="#cb31-5"></a>  <span class="fu">mutate</span>(</span>
<span id="cb31-6"><a href="#cb31-6"></a>    <span class="co"># compute the number of genes in the genome</span></span>
<span id="cb31-7"><a href="#cb31-7"></a>    <span class="at">n_genes =</span> <span class="fu">n</span>(),</span>
<span id="cb31-8"><a href="#cb31-8"></a>    <span class="co"># compute the sum of non-syn sites in genome</span></span>
<span id="cb31-9"><a href="#cb31-9"></a>    <span class="at">Ltot =</span> <span class="fu">sum</span>(l_i),</span>
<span id="cb31-10"><a href="#cb31-10"></a>    <span class="co"># compute the average non-syn sites per gene</span></span>
<span id="cb31-11"><a href="#cb31-11"></a>    <span class="at">Lavg =</span> <span class="fu">mean</span>(l_i)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-12"><a href="#cb31-12"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-13"><a href="#cb31-13"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_2659"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-14"><a href="#cb31-14"></a>  <span class="co"># expanding this now to make later joining faster</span></span>
<span id="cb31-15"><a href="#cb31-15"></a>  <span class="fu">expand_grid</span>(<span class="at">prey_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>), </span>
<span id="cb31-16"><a href="#cb31-16"></a>              <span class="at">predator_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>),</span>
<span id="cb31-17"><a href="#cb31-17"></a>              <span class="at">fbin =</span> <span class="fu">unique</span>(multiplicity_binned<span class="sc">$</span>fbin))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>This prepares the tibble that we can use in the next step to calculate the G scores and <span class="math inline">\(\Delta L\)</span></p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>g_prepped <span class="ot">&lt;-</span> <span class="fu">left_join</span>(gene_tab, multiplicity_binned) <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2"></a>  <span class="fu">arrange</span>(strainID, prey_history, predator_history, fbin, locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb32-3"><a href="#cb32-3"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">n_i =</span> <span class="dv">0</span>, <span class="at">n_replicate =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history, fbin) <span class="sc">%&gt;%</span> </span>
<span id="cb32-5"><a href="#cb32-5"></a>  <span class="fu">fill</span>(Ntot, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-6"><a href="#cb32-6"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-7"><a href="#cb32-7"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb32-8"><a href="#cb32-8"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(Ntot, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-9"><a href="#cb32-9"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(strainID, locus_tag, prey_history, predator_history,
fbin)`</code></pre>
</div>
</div>
<p>This function calculate <span class="math inline">\(\Delta L\)</span> but from a bootstrapping of the source dataframes to get sample means plus confidence intervals. We use <code>rsample</code> package hence the use of <code>analysis(df).</code> See the next section for implementation</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>G_score_boot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">boot=</span><span class="cn">TRUE</span>){</span>
<span id="cb34-2"><a href="#cb34-2"></a>  </span>
<span id="cb34-3"><a href="#cb34-3"></a>  <span class="cf">if</span> (boot) {</span>
<span id="cb34-4"><a href="#cb34-4"></a>    <span class="co"># convert the rsample splits to tibble</span></span>
<span id="cb34-5"><a href="#cb34-5"></a>    df <span class="ot">&lt;-</span> rsample<span class="sc">::</span><span class="fu">analysis</span>(df)</span>
<span id="cb34-6"><a href="#cb34-6"></a>  }</span>
<span id="cb34-7"><a href="#cb34-7"></a>  <span class="co"># compute the expected multiplicity value</span></span>
<span id="cb34-8"><a href="#cb34-8"></a>  m_exp <span class="ot">&lt;-</span> <span class="fu">unique</span>(df<span class="sc">$</span>Ntot) <span class="sc">/</span> <span class="fu">unique</span>(df<span class="sc">$</span>n_genes)</span>
<span id="cb34-9"><a href="#cb34-9"></a>  <span class="co"># compute gene multiplicity for gene i</span></span>
<span id="cb34-10"><a href="#cb34-10"></a>  m_i <span class="ot">&lt;-</span>  df<span class="sc">$</span>n_i <span class="sc">*</span> <span class="fu">unique</span>(df<span class="sc">$</span>Lavg) <span class="sc">/</span> df<span class="sc">$</span>l_i</span>
<span id="cb34-11"><a href="#cb34-11"></a>  <span class="co"># compute the G score for each gene</span></span>
<span id="cb34-12"><a href="#cb34-12"></a>  g_score <span class="ot">&lt;-</span> df<span class="sc">$</span>n_i <span class="sc">*</span> <span class="fu">log</span>(m_i <span class="sc">/</span> m_exp)</span>
<span id="cb34-13"><a href="#cb34-13"></a>  <span class="co"># net increase in log-likelihood compared to the null model of m_i / m_exp =1</span></span>
<span id="cb34-14"><a href="#cb34-14"></a>  <span class="co"># normalized to the total mutations</span></span>
<span id="cb34-15"><a href="#cb34-15"></a>  <span class="fu">sum</span>(g_score, <span class="at">na.rm=</span>T) <span class="sc">/</span> <span class="fu">unique</span>(df<span class="sc">$</span>Ntot)</span>
<span id="cb34-16"><a href="#cb34-16"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We need the package <code>rsample</code> to do the boostrap analysis and we need <code>Hmisc</code> to get the bootstrapped confidence intervals</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="fu">library</span>(rsample)</span>
<span id="cb35-2"><a href="#cb35-2"></a><span class="fu">library</span>(Hmisc)</span>
<span id="cb35-3"><a href="#cb35-3"></a></span>
<span id="cb35-4"><a href="#cb35-4"></a><span class="co"># this is a bit slow so saving the results</span></span>
<span id="cb35-5"><a href="#cb35-5"></a>g_boot <span class="ot">&lt;-</span> g_prepped <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="#cb35-6"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">.by =</span> <span class="fu">c</span>(strainID, prey_history, predator_history, fbin)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-7"><a href="#cb35-7"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">boots =</span> purrr<span class="sc">::</span><span class="fu">map</span>(data, <span class="cf">function</span>(df) rsample<span class="sc">::</span><span class="fu">bootstraps</span>(df, <span class="at">times =</span> <span class="dv">1000</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-8"><a href="#cb35-8"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">g_scores =</span> purrr<span class="sc">::</span><span class="fu">map</span>(boots, \(boot) purrr<span class="sc">::</span><span class="fu">map_dbl</span>(boot<span class="sc">$</span>splits, G_score_boot))) <span class="sc">%&gt;%</span> </span>
<span id="cb35-9"><a href="#cb35-9"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">meanboot =</span> purrr<span class="sc">::</span><span class="fu">map</span>(g_scores, ggplot2<span class="sc">::</span>mean_cl_boot),</span>
<span id="cb35-10"><a href="#cb35-10"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, prey_history, predator_history, fbin)) <span class="sc">%&gt;%</span> </span>
<span id="cb35-11"><a href="#cb35-11"></a>  tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(meanboot))</span>
<span id="cb35-12"><a href="#cb35-12"></a></span>
<span id="cb35-13"><a href="#cb35-13"></a><span class="co"># Save this for later</span></span>
<span id="cb35-14"><a href="#cb35-14"></a><span class="fu">write_rds</span>(g_boot, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"g_score_bootstap.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Read in saved bootstrapped data</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="co"># Read in previous results</span></span>
<span id="cb36-2"><a href="#cb36-2"></a>g_boot <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"g_score_bootstap.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Plot bootstrapped <span class="math inline">\(\Delta l\)</span> range with <span class="math inline">\(\Delta l\)</span> from full dataset at different values of <span class="math inline">\(f_{max}\)</span></p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>pg_boot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(g_boot, <span class="fu">aes</span>(<span class="at">x =</span> fbin, <span class="at">y =</span> y, <span class="at">ymin =</span> ymin, <span class="at">ymax =</span> ymax)) <span class="sc">+</span></span>
<span id="cb37-2"><a href="#cb37-2"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID), </span>
<span id="cb37-3"><a href="#cb37-3"></a>                  <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb37-4"><a href="#cb37-4"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Binned maximum observed allele frequency ($f_{max}$) "</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">"Extent of parallel evolution ($</span><span class="sc">\\</span><span class="st">Delta l$)"</span>), <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb37-7"><a href="#cb37-7"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb37-8"><a href="#cb37-8"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9"></a>  <span class="fu">theme</span>(</span>
<span id="cb37-10"><a href="#cb37-10"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb37-11"><a href="#cb37-11"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb37-12"><a href="#cb37-12"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb37-13"><a href="#cb37-13"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb37-14"><a href="#cb37-14"></a></span>
<span id="cb37-15"><a href="#cb37-15"></a>pg_boot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-18" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-18-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-18-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;18
</figcaption>
</figure>
</div>
</section>
<section id="g-score" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="g-score"><span class="header-section-number">8.2</span> G score</h2>
<p>Also calculate the G score from <span class="citation" data-cites="tenaillon2016">(<a href="#ref-tenaillon2016" role="doc-biblioref">Tenaillon et al. 2016</a>)</span> and look at the relationship with <span class="math inline">\(f_{max}\)</span>. The difference between the analysis above is that G-score is the non-summed and non-normalized form of <span class="math inline">\(\Delta l\)</span> and it is calculated for every gene uniquely instead of at different <span class="math inline">\(f_{max}\)</span> bins. Briefly,</p>
<p><span class="math display">\[
G_{i} = n_{i} \log \left ( \frac{m_{i}}{\overline{m}} \right )
\]</span> where the expected multiplicity <span class="math inline">\(\overline{m}\)</span> is</p>
<p><span class="math display">\[
\overline{m} = n_{tot}/n_{genes}
\]</span></p>
<p>and the observed multiplicity is</p>
<p><span class="math display">\[
m_{i} = n_{i} \cdot \frac{\overline{L}}{L_{i}}
\]</span></p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>multiplicity <span class="ot">&lt;-</span> mgvars_filt_mb_ns <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="#cb38-2"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">n_i =</span> <span class="fu">n</span>(),</span>
<span id="cb38-3"><a href="#cb38-3"></a>            <span class="at">n_replicate =</span> <span class="fu">n_distinct</span>(replicate), </span>
<span id="cb38-4"><a href="#cb38-4"></a>            <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"locus_tag"</span>, <span class="st">"strainID"</span>, <span class="st">"prey_history"</span>, <span class="st">"predator_history"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="#cb38-5"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span></span>
<span id="cb38-6"><a href="#cb38-6"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Ntot =</span> <span class="fu">sum</span>(n_i)) <span class="sc">%&gt;%</span></span>
<span id="cb38-7"><a href="#cb38-7"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb38-8"><a href="#cb38-8"></a></span>
<span id="cb38-9"><a href="#cb38-9"></a>g_prepped_nobin <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(gene_tab, <span class="sc">-</span>fbin)), multiplicity) <span class="sc">%&gt;%</span> </span>
<span id="cb38-10"><a href="#cb38-10"></a>  <span class="fu">arrange</span>(strainID, prey_history, predator_history, locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb38-11"><a href="#cb38-11"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">n_i =</span> <span class="dv">0</span>, <span class="at">n_replicate =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-12"><a href="#cb38-12"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb38-13"><a href="#cb38-13"></a>  <span class="fu">fill</span>(Ntot, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-14"><a href="#cb38-14"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-15"><a href="#cb38-15"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb38-16"><a href="#cb38-16"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(Ntot, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-17"><a href="#cb38-17"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(strainID, locus_tag, prey_history,
predator_history)`</code></pre>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>G_score_tidy <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb40-2"><a href="#cb40-2"></a>  df <span class="sc">%&gt;%</span> </span>
<span id="cb40-3"><a href="#cb40-3"></a>  <span class="co"># compute the expected multiplicity value</span></span>
<span id="cb40-4"><a href="#cb40-4"></a>  <span class="fu">mutate</span>(<span class="at">m_exp =</span> Ntot <span class="sc">/</span> n_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb40-5"><a href="#cb40-5"></a>  <span class="co"># compute gene multiplicity for gene i</span></span>
<span id="cb40-6"><a href="#cb40-6"></a>  <span class="fu">mutate</span>(<span class="at">m_i =</span>  n_i <span class="sc">*</span> Lavg <span class="sc">/</span> l_i) <span class="sc">%&gt;%</span> </span>
<span id="cb40-7"><a href="#cb40-7"></a>  <span class="co"># compute the G score for each gene</span></span>
<span id="cb40-8"><a href="#cb40-8"></a>  <span class="fu">mutate</span>(<span class="at">g_score =</span> n_i <span class="sc">*</span> <span class="fu">log</span>(m_i <span class="sc">/</span> m_exp)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-9"><a href="#cb40-9"></a>  <span class="co"># net increase in log-likelihood compared to the null model of m_i / m_exp =1</span></span>
<span id="cb40-10"><a href="#cb40-10"></a>  <span class="co"># normalized to the total mutations</span></span>
<span id="cb40-11"><a href="#cb40-11"></a>  <span class="fu">mutate</span>(<span class="at">delta_l =</span> <span class="fu">sum</span>(g_score, <span class="at">na.rm=</span>T) <span class="sc">/</span> Ntot)</span>
<span id="cb40-12"><a href="#cb40-12"></a>}</span>
<span id="cb40-13"><a href="#cb40-13"></a></span>
<span id="cb40-14"><a href="#cb40-14"></a>g_scores <span class="ot">&lt;-</span> g_prepped_nobin <span class="sc">%&gt;%</span> </span>
<span id="cb40-15"><a href="#cb40-15"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb40-16"><a href="#cb40-16"></a>  <span class="fu">G_score_tidy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb40-17"><a href="#cb40-17"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb40-18"><a href="#cb40-18"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb40-19"><a href="#cb40-19"></a></span>
<span id="cb40-20"><a href="#cb40-20"></a>g_scores_fmax <span class="ot">&lt;-</span> <span class="fu">left_join</span>(g_scores, mgvars,</span>
<span id="cb40-21"><a href="#cb40-21"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(strainID, locus_tag, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-22"><a href="#cb40-22"></a>  <span class="fu">group_by</span>(locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb40-23"><a href="#cb40-23"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">==</span> <span class="fu">max</span>(freq_alt_complete, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-24"><a href="#cb40-24"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>fig19 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(g_scores_fmax, <span class="fu">aes</span>(<span class="at">x =</span> freq_alt_complete, <span class="at">y =</span> g_score)) <span class="sc">+</span></span>
<span id="cb41-2"><a href="#cb41-2"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">'y ~ x'</span>) <span class="sc">+</span></span>
<span id="cb41-3"><a href="#cb41-3"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb41-4"><a href="#cb41-4"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb41-5"><a href="#cb41-5"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb41-6"><a href="#cb41-6"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Maximum observed allele frequency ($f_{max}$) "</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">"Extent of parallel evolution (Gene-wise $G_{i}$)"</span>), <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"bl"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb41-8"><a href="#cb41-8"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb41-9"><a href="#cb41-9"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb41-10"><a href="#cb41-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb41-11"><a href="#cb41-11"></a>  <span class="fu">theme</span>(</span>
<span id="cb41-12"><a href="#cb41-12"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb41-13"><a href="#cb41-13"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb41-14"><a href="#cb41-14"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-15"><a href="#cb41-15"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb41-16"><a href="#cb41-16"></a></span>
<span id="cb41-17"><a href="#cb41-17"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"G_score_fmax.svg"</span>), fig19, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb41-18"><a href="#cb41-18"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb41-19"><a href="#cb41-19"></a></span>
<span id="cb41-20"><a href="#cb41-20"></a>fig19</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-19" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-19-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="04_mutation_dynamics_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-19-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;19
</figcaption>
</figure>
</div>
<p>Linear regression for different treatment combinations (over species). Overall, we find a clear positive relationship between the gene-wise extent of parallel evolution (G score) and the maximum observed allele frequency for that gene in 3/4 treatment combinations. We also find a positive relationship for the prey: evo | predator: anc treatment combination but the significance of this relationship is unclear (P value = 0.07).</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a>g_scores_fmax <span class="sc">%&gt;%</span> </span>
<span id="cb42-2"><a href="#cb42-2"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span><span class="fu">c</span>(prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-3"><a href="#cb42-3"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, \(df) <span class="fu">lm</span>(g_score <span class="sc">~</span> freq_alt_complete, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">data =</span> df))) <span class="sc">%&gt;%</span> </span>
<span id="cb42-4"><a href="#cb42-4"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">tidy =</span> <span class="fu">map</span>(model, broom<span class="sc">::</span>tidy)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-5"><a href="#cb42-5"></a>  <span class="fu">unnest</span>(tidy) <span class="sc">%&gt;%</span> </span>
<span id="cb42-6"><a href="#cb42-6"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>data, <span class="sc">-</span>model) <span class="sc">%&gt;%</span> </span>
<span id="cb42-7"><a href="#cb42-7"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-8"><a href="#cb42-8"></a>  <span class="fu">arrange</span>(p.value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 4 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `model = map(...)`.
Caused by warning:
! In lm.fit(x, y, offset = offset, singular.ok = singular.ok, ...) :
 extra argument 'na.rm' will be disregarded
ℹ Run `dplyr::last_dplyr_warnings()` to see the 3 remaining warnings.</code></pre>
</div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["prey_history"],"name":[1],"type":["chr"],"align":["left"]},{"label":["predator_history"],"name":[2],"type":["chr"],"align":["left"]},{"label":["term"],"name":[3],"type":["chr"],"align":["left"]},{"label":["estimate"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["std.error"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["statistic"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["p.value"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"evo","2":"anc","3":"freq_alt_complete","4":"35.909096","5":"3.677987","6":"9.763248","7":"5.664531e-20"},{"1":"anc","2":"evo","3":"freq_alt_complete","4":"13.015740","5":"3.962569","6":"3.284672","7":"1.324835e-03"},{"1":"evo","2":"evo","3":"freq_alt_complete","4":"10.140530","5":"3.201669","6":"3.167264","7":"2.622928e-03"},{"1":"anc","2":"anc","3":"freq_alt_complete","4":"9.737615","5":"4.855621","6":"2.005432","7":"7.588524e-02"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>Linear regression for all different species in different treatment combinations. By including every species we lose statistical power because some species have relatively few observations. However, for all 4 species in the prey: evo | predator: anc treatment combination we find a clear positive relationship between the gene-wise extent of parallel evolution (G score) and the maximum observed allele frequency for that gene. We also find this positive relationmship for HAMBI_2659 in the prey: evo | predator: evo treatment combination.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a>g_scores_fmax <span class="sc">%&gt;%</span> </span>
<span id="cb44-2"><a href="#cb44-2"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span><span class="fu">c</span>(strainID, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-3"><a href="#cb44-3"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, \(df) <span class="fu">lm</span>(g_score <span class="sc">~</span> freq_alt_complete, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">data =</span> df))) <span class="sc">%&gt;%</span> </span>
<span id="cb44-4"><a href="#cb44-4"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">tidy =</span> <span class="fu">map</span>(model, broom<span class="sc">::</span>tidy)) <span class="sc">%&gt;%</span> </span>
<span id="cb44-5"><a href="#cb44-5"></a>  <span class="fu">unnest</span>(tidy) <span class="sc">%&gt;%</span> </span>
<span id="cb44-6"><a href="#cb44-6"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>data, <span class="sc">-</span>model) <span class="sc">%&gt;%</span> </span>
<span id="cb44-7"><a href="#cb44-7"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb44-8"><a href="#cb44-8"></a>  <span class="fu">arrange</span>(p.value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: There were 15 warnings in `dplyr::mutate()`.
The first warning was:
ℹ In argument: `model = map(...)`.
Caused by warning:
! In lm.fit(x, y, offset = offset, singular.ok = singular.ok, ...) :
 extra argument 'na.rm' will be disregarded
ℹ Run `dplyr::last_dplyr_warnings()` to see the 14 remaining warnings.</code></pre>
</div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["strainID"],"name":[1],"type":["chr"],"align":["left"]},{"label":["prey_history"],"name":[2],"type":["chr"],"align":["left"]},{"label":["predator_history"],"name":[3],"type":["chr"],"align":["left"]},{"label":["term"],"name":[4],"type":["chr"],"align":["left"]},{"label":["estimate"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["std.error"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["statistic"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["p.value"],"name":[8],"type":["dbl"],"align":["right"]}],"data":[{"1":"HAMBI_1977","2":"evo","3":"anc","4":"freq_alt_complete","5":"71.910210","6":"8.814624","7":"8.1580573","8":"2.964323e-13"},{"1":"HAMBI_1972","2":"evo","3":"anc","4":"freq_alt_complete","5":"18.429894","6":"2.791730","7":"6.6016034","8":"1.754208e-09"},{"1":"HAMBI_2659","2":"evo","3":"anc","4":"freq_alt_complete","5":"23.635339","6":"4.881690","7":"4.8416306","8":"8.109997e-06"},{"1":"HAMBI_1287","2":"evo","3":"anc","4":"freq_alt_complete","5":"81.985997","6":"17.175856","7":"4.7733282","8":"3.835964e-05"},{"1":"HAMBI_2659","2":"evo","3":"evo","4":"freq_alt_complete","5":"30.329421","6":"6.570987","7":"4.6156565","8":"9.569190e-04"},{"1":"HAMBI_1977","2":"anc","3":"anc","4":"freq_alt_complete","5":"124.854069","6":"32.678466","7":"3.8206833","8":"1.629689e-01"},{"1":"HAMBI_1287","2":"evo","3":"evo","4":"freq_alt_complete","5":"22.650958","6":"14.826941","7":"1.5276892","8":"1.651098e-01"},{"1":"HAMBI_1972","2":"anc","3":"evo","4":"freq_alt_complete","5":"10.449933","6":"7.426150","7":"1.4071804","8":"1.681928e-01"},{"1":"HAMBI_2659","2":"anc","3":"anc","4":"freq_alt_complete","5":"8.926562","6":"6.152820","7":"1.4508081","8":"2.204522e-01"},{"1":"HAMBI_1977","2":"evo","3":"evo","4":"freq_alt_complete","5":"10.135013","6":"8.492395","7":"1.1934223","8":"2.777475e-01"},{"1":"HAMBI_0403","2":"evo","3":"evo","4":"freq_alt_complete","5":"-1.332596","6":"1.325657","7":"-1.0052344","8":"4.206403e-01"},{"1":"HAMBI_1972","2":"evo","3":"evo","4":"freq_alt_complete","5":"2.496275","6":"4.200643","7":"0.5942601","8":"5.606477e-01"},{"1":"HAMBI_1287","2":"anc","3":"evo","4":"freq_alt_complete","5":"9.116231","6":"39.183607","7":"0.2326542","8":"8.199507e-01"},{"1":"HAMBI_2659","2":"anc","3":"evo","4":"freq_alt_complete","5":"1.339296","6":"9.750416","7":"0.1373578","8":"8.911210e-01"},{"1":"HAMBI_1972","2":"anc","3":"anc","4":"freq_alt_complete","5":"2.556070","6":"NaN","7":"NaN","8":"NaN"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>


<!-- -->


</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-tenaillon2016" class="csl-entry" role="listitem">
Tenaillon, Olivier, Jeffrey E. Barrick, Noah Ribeck, Daniel E. Deatherage, Jeffrey L. Blanchard, Aurko Dasgupta, Gabriel C. Wu, et al. 2016. <span>“Tempo and Mode of Genome Evolution in a 50,000-Generation Experiment.”</span> <em>Nature</em> 536 (7615): 165–70. <a href="https://doi.org/10.1038/nature18959">https://doi.org/10.1038/nature18959</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb46" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb46-1"><a href="#cb46-1"></a><span class="co">---</span></span>
<span id="cb46-2"><a href="#cb46-2"></a><span class="an">title:</span><span class="co"> "Mutational dynamics"</span></span>
<span id="cb46-3"><a href="#cb46-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb46-4"><a href="#cb46-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb46-5"><a href="#cb46-5"></a><span class="an">link-citations:</span><span class="co"> true</span></span>
<span id="cb46-6"><a href="#cb46-6"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb46-7"><a href="#cb46-7"></a><span class="an">abstract:</span><span class="co"> "This notebook reads in the formatted metagenome variant frequencies and annotations. It makes some plots and does some basic analysis looking at the temporal dynamics and ultimate fates (e.g., extinction or fixation) of total and nonsynomous mutations from the metagenomes."</span></span>
<span id="cb46-8"><a href="#cb46-8"></a><span class="co">---</span></span>
<span id="cb46-9"><a href="#cb46-9"></a></span>
<span id="cb46-10"><a href="#cb46-10"></a><span class="fu"># Setup</span></span>
<span id="cb46-11"><a href="#cb46-11"></a></span>
<span id="cb46-12"><a href="#cb46-12"></a>Libraries and global variables</span>
<span id="cb46-13"><a href="#cb46-13"></a></span>
<span id="cb46-16"><a href="#cb46-16"></a><span class="in">```{r}</span></span>
<span id="cb46-17"><a href="#cb46-17"></a><span class="co">#| output: false</span></span>
<span id="cb46-18"><a href="#cb46-18"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb46-19"><a href="#cb46-19"></a><span class="fu">library</span>(here)</span>
<span id="cb46-20"><a href="#cb46-20"></a><span class="fu">library</span>(fs)</span>
<span id="cb46-21"><a href="#cb46-21"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb46-22"><a href="#cb46-22"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb46-23"><a href="#cb46-23"></a><span class="fu">library</span>(latex2exp)</span>
<span id="cb46-24"><a href="#cb46-24"></a><span class="fu">library</span>(vctrs)</span>
<span id="cb46-25"><a href="#cb46-25"></a><span class="fu">library</span>(scales)</span>
<span id="cb46-26"><a href="#cb46-26"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb46-27"><a href="#cb46-27"></a></span>
<span id="cb46-28"><a href="#cb46-28"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb46-29"><a href="#cb46-29"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"metagenome"</span>, <span class="st">"utils_parallelism.R"</span>))</span>
<span id="cb46-30"><a href="#cb46-30"></a><span class="in">```</span></span>
<span id="cb46-31"><a href="#cb46-31"></a></span>
<span id="cb46-32"><a href="#cb46-32"></a>Set up some directories</span>
<span id="cb46-33"><a href="#cb46-33"></a></span>
<span id="cb46-36"><a href="#cb46-36"></a><span class="in">```{r}</span></span>
<span id="cb46-37"><a href="#cb46-37"></a><span class="co">#| output: false</span></span>
<span id="cb46-38"><a href="#cb46-38"></a><span class="co">#| warning: false</span></span>
<span id="cb46-39"><a href="#cb46-39"></a><span class="co">#| error: false</span></span>
<span id="cb46-40"><a href="#cb46-40"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb46-41"><a href="#cb46-41"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb46-42"><a href="#cb46-42"></a>shared <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"shared"</span>)</span>
<span id="cb46-43"><a href="#cb46-43"></a>figs <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"metagenome"</span>)</span>
<span id="cb46-44"><a href="#cb46-44"></a></span>
<span id="cb46-45"><a href="#cb46-45"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb46-46"><a href="#cb46-46"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb46-47"><a href="#cb46-47"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(figs)</span>
<span id="cb46-48"><a href="#cb46-48"></a><span class="in">```</span></span>
<span id="cb46-49"><a href="#cb46-49"></a></span>
<span id="cb46-50"><a href="#cb46-50"></a><span class="fu"># Read data</span></span>
<span id="cb46-51"><a href="#cb46-51"></a></span>
<span id="cb46-54"><a href="#cb46-54"></a><span class="in">```{r}</span></span>
<span id="cb46-55"><a href="#cb46-55"></a><span class="co">#| output: false</span></span>
<span id="cb46-56"><a href="#cb46-56"></a><span class="co">#| warning: false</span></span>
<span id="cb46-57"><a href="#cb46-57"></a><span class="co">#| error: false</span></span>
<span id="cb46-58"><a href="#cb46-58"></a></span>
<span id="cb46-59"><a href="#cb46-59"></a><span class="co"># these were already filtered in the last step</span></span>
<span id="cb46-60"><a href="#cb46-60"></a>mgvars <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"metagenome_variant_timeseries.tsv"</span>))</span>
<span id="cb46-61"><a href="#cb46-61"></a>degentab <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb46-62"><a href="#cb46-62"></a>genome_len <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"HAMBI_genome_len.tsv"</span>))</span>
<span id="cb46-63"><a href="#cb46-63"></a><span class="co"># annotations</span></span>
<span id="cb46-64"><a href="#cb46-64"></a>annotations <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(shared, <span class="st">"annotations_codon_degeneracy.rds"</span>))</span>
<span id="cb46-65"><a href="#cb46-65"></a></span>
<span id="cb46-66"><a href="#cb46-66"></a><span class="co"># coverage</span></span>
<span id="cb46-67"><a href="#cb46-67"></a>covslurpedfmtflt <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"coverage.tsv"</span>))</span>
<span id="cb46-68"><a href="#cb46-68"></a></span>
<span id="cb46-69"><a href="#cb46-69"></a><span class="co"># parallel genes</span></span>
<span id="cb46-70"><a href="#cb46-70"></a>par_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"enriched_parallel_genes.tsv"</span>))</span>
<span id="cb46-71"><a href="#cb46-71"></a><span class="in">```</span></span>
<span id="cb46-72"><a href="#cb46-72"></a></span>
<span id="cb46-75"><a href="#cb46-75"></a><span class="in">```{r}</span></span>
<span id="cb46-76"><a href="#cb46-76"></a>cog_description <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb46-77"><a href="#cb46-77"></a>  <span class="sc">~</span>COG_category_single,                                                  <span class="sc">~</span>COG_category_long,</span>
<span id="cb46-78"><a href="#cb46-78"></a>            <span class="st">"J"</span>,               <span class="st">"J - Translation, ribosomal structure and biogenesis"</span>,</span>
<span id="cb46-79"><a href="#cb46-79"></a>            <span class="st">"A"</span>,                               <span class="st">"A - RNA processing and modification"</span>,</span>
<span id="cb46-80"><a href="#cb46-80"></a>            <span class="st">"K"</span>,                                                 <span class="st">"K – Transcription"</span>,</span>
<span id="cb46-81"><a href="#cb46-81"></a>            <span class="st">"L"</span>,                         <span class="st">"L - Replication, recombination and repair"</span>,</span>
<span id="cb46-82"><a href="#cb46-82"></a>            <span class="st">"B"</span>,                              <span class="st">"B - Chromatin structure and dynamics"</span>,</span>
<span id="cb46-83"><a href="#cb46-83"></a>            <span class="st">"D"</span>,    <span class="st">"D - Cell cycle control, cell division, chromosome partitioning"</span>,</span>
<span id="cb46-84"><a href="#cb46-84"></a>            <span class="st">"Y"</span>,                                             <span class="st">"Y - Nuclear structure"</span>,</span>
<span id="cb46-85"><a href="#cb46-85"></a>            <span class="st">"V"</span>,                                            <span class="st">"V - Defense mechanisms"</span>,</span>
<span id="cb46-86"><a href="#cb46-86"></a>            <span class="st">"T"</span>,                                <span class="st">"T - Signal transduction mechanisms"</span>,</span>
<span id="cb46-87"><a href="#cb46-87"></a>            <span class="st">"M"</span>,                        <span class="st">"M - Cell wall/membrane/envelope biogenesis"</span>,</span>
<span id="cb46-88"><a href="#cb46-88"></a>            <span class="st">"N"</span>,                                                 <span class="st">"N - Cell motility"</span>,</span>
<span id="cb46-89"><a href="#cb46-89"></a>            <span class="st">"Z"</span>,                                                  <span class="st">"Z – Cytoskeleton"</span>,</span>
<span id="cb46-90"><a href="#cb46-90"></a>            <span class="st">"W"</span>,                                      <span class="st">"W - Extracellular structures"</span>,</span>
<span id="cb46-91"><a href="#cb46-91"></a>            <span class="st">"U"</span>, <span class="st">"U - Intracellular trafficking, secretion, and vesicular transport"</span>,</span>
<span id="cb46-92"><a href="#cb46-92"></a>            <span class="st">"O"</span>,  <span class="st">"O - Posttranslational modification, protein turnover, chaperones"</span>,</span>
<span id="cb46-93"><a href="#cb46-93"></a>            <span class="st">"X"</span>,                              <span class="st">"X - Mobilome: prophages, transposons"</span>,</span>
<span id="cb46-94"><a href="#cb46-94"></a>            <span class="st">"C"</span>,                              <span class="st">"C - Energy production and conversion"</span>,</span>
<span id="cb46-95"><a href="#cb46-95"></a>            <span class="st">"G"</span>,                         <span class="st">"G - Carbohydrate transport and metabolism"</span>,</span>
<span id="cb46-96"><a href="#cb46-96"></a>            <span class="st">"E"</span>,                           <span class="st">"E - Amino acid transport and metabolism"</span>,</span>
<span id="cb46-97"><a href="#cb46-97"></a>            <span class="st">"F"</span>,                           <span class="st">"F - Nucleotide transport and metabolism"</span>,</span>
<span id="cb46-98"><a href="#cb46-98"></a>            <span class="st">"H"</span>,                             <span class="st">"H - Coenzyme transport and metabolism"</span>,</span>
<span id="cb46-99"><a href="#cb46-99"></a>            <span class="st">"I"</span>,                                <span class="st">"I - Lipid transport and metabolism"</span>,</span>
<span id="cb46-100"><a href="#cb46-100"></a>            <span class="st">"P"</span>,                        <span class="st">"P - Inorganic ion transport and metabolism"</span>,</span>
<span id="cb46-101"><a href="#cb46-101"></a>            <span class="st">"Q"</span>,  <span class="st">"Q - Secondary metabolites biosynthesis, transport and catabolism"</span>,</span>
<span id="cb46-102"><a href="#cb46-102"></a>            <span class="st">"R"</span>,                              <span class="st">"R - General function prediction only"</span>,</span>
<span id="cb46-103"><a href="#cb46-103"></a>            <span class="st">"S"</span>,                                              <span class="st">"S - Function unknown"</span></span>
<span id="cb46-104"><a href="#cb46-104"></a>  )</span>
<span id="cb46-105"><a href="#cb46-105"></a></span>
<span id="cb46-106"><a href="#cb46-106"></a>withr<span class="sc">::</span><span class="fu">with_seed</span>(<span class="dv">12367</span>,</span>
<span id="cb46-107"><a href="#cb46-107"></a>                 cogpal <span class="ot">&lt;-</span> <span class="fu">unname</span>(<span class="fu">createPalette</span>(<span class="fu">length</span>(<span class="fu">unique</span>(cog_description<span class="sc">$</span>COG_category_single)), <span class="fu">c</span>(<span class="st">"#F3874AFF"</span>, <span class="st">"#FCD125FF"</span>), <span class="at">M=</span><span class="dv">5000</span>))</span>
<span id="cb46-108"><a href="#cb46-108"></a>)</span>
<span id="cb46-109"><a href="#cb46-109"></a><span class="fu">names</span>(cogpal) <span class="ot">&lt;-</span> cog_description<span class="sc">$</span>COG_category_long</span>
<span id="cb46-110"><a href="#cb46-110"></a><span class="in">```</span></span>
<span id="cb46-111"><a href="#cb46-111"></a></span>
<span id="cb46-112"><a href="#cb46-112"></a><span class="fu"># Metagenomic coverage</span></span>
<span id="cb46-113"><a href="#cb46-113"></a></span>
<span id="cb46-114"><a href="#cb46-114"></a>Only looking at the 5 species for which we called variants. Everything else was below a mean coverage of 5 and even calling variants at a coverage of 5 is a bit sketchy. Generally, the higher the coverage your genome is the confidence you can have calling polymorphisms where the alternate allele is rare. Basically this shows that we can really only same something of confidence for HAMBI_1287, HAMBI_1972, HAMBI_1977, and HAMBI_2659.</span>
<span id="cb46-115"><a href="#cb46-115"></a></span>
<span id="cb46-116"><a href="#cb46-116"></a>::: {#fig-01}</span>
<span id="cb46-119"><a href="#cb46-119"></a><span class="in">```{r}</span></span>
<span id="cb46-120"><a href="#cb46-120"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-121"><a href="#cb46-121"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb46-122"><a href="#cb46-122"></a><span class="co">#| warning: false</span></span>
<span id="cb46-123"><a href="#cb46-123"></a><span class="co">#| error: false</span></span>
<span id="cb46-124"><a href="#cb46-124"></a>covslurpedfmtflt <span class="sc">%&gt;%</span> </span>
<span id="cb46-125"><a href="#cb46-125"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_2659"</span>, <span class="st">"HAMBI_0403"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-126"><a href="#cb46-126"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(scaffold, <span class="st">"plas"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-127"><a href="#cb46-127"></a>  <span class="fu">left_join</span>(<span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(mgvars, sample, time_days, replicate, prey_history, predator_history)),</span>
<span id="cb46-128"><a href="#cb46-128"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-129"><a href="#cb46-129"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-130"><a href="#cb46-130"></a>  <span class="fu">mutate</span>(<span class="at">replicate_mean =</span> <span class="fu">mean</span>(trimmed_mean)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-131"><a href="#cb46-131"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> time_days, <span class="at">y =</span> trimmed_mean)) <span class="sc">+</span></span>
<span id="cb46-132"><a href="#cb46-132"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">5</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb46-133"><a href="#cb46-133"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, <span class="at">group =</span> <span class="fu">interaction</span>(strainID, prey_history, predator_history, replicate)),</span>
<span id="cb46-134"><a href="#cb46-134"></a>            <span class="at">alpha =</span> <span class="fl">0.35</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb46-135"><a href="#cb46-135"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, <span class="at">shape =</span> replicate),</span>
<span id="cb46-136"><a href="#cb46-136"></a>             <span class="at">alpha =</span> <span class="fl">0.35</span>) <span class="sc">+</span></span>
<span id="cb46-137"><a href="#cb46-137"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> replicate_mean, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-138"><a href="#cb46-138"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> replicate_mean, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb46-139"><a href="#cb46-139"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb46-140"><a href="#cb46-140"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-141"><a href="#cb46-141"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Experiment time (days)"</span>, <span class="at">y =</span> <span class="st">"Mean genome coverage"</span>, <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb46-142"><a href="#cb46-142"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>), <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb46-143"><a href="#cb46-143"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">300</span>)) <span class="sc">+</span></span>
<span id="cb46-144"><a href="#cb46-144"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-145"><a href="#cb46-145"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">28</span>, <span class="dv">60</span>)) <span class="sc">+</span></span>
<span id="cb46-146"><a href="#cb46-146"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-147"><a href="#cb46-147"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-148"><a href="#cb46-148"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-149"><a href="#cb46-149"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-150"><a href="#cb46-150"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-151"><a href="#cb46-151"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb46-152"><a href="#cb46-152"></a><span class="in">```</span></span>
<span id="cb46-153"><a href="#cb46-153"></a></span>
<span id="cb46-154"><a href="#cb46-154"></a>Genomic coverage in the metagenomes over time. The genomes of these five species exceeded a coverage of at least 5X (black dashed horizontal line) at at least one sampling time from the experiment. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.</span>
<span id="cb46-155"><a href="#cb46-155"></a>:::</span>
<span id="cb46-156"><a href="#cb46-156"></a></span>
<span id="cb46-157"><a href="#cb46-157"></a><span class="fu"># Mutational spectra</span></span>
<span id="cb46-158"><a href="#cb46-158"></a></span>
<span id="cb46-159"><a href="#cb46-159"></a>Some broad scale patterns in how mutations trajectories vary between species and treatments can be observed from the mutational spectra. Below we examine how the fraction of mutations exceeding the maximum observed allele frequency $f_{max}$ relates to different levels of $f_{max}$ and compare this between different evolution treatments and across species.</span>
<span id="cb46-160"><a href="#cb46-160"></a></span>
<span id="cb46-163"><a href="#cb46-163"></a><span class="in">```{r}</span></span>
<span id="cb46-164"><a href="#cb46-164"></a>allelespectra <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-165"><a href="#cb46-165"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, replicate, prey_history, predator_history, freq_alt_complete) <span class="sc">%&gt;%</span> </span>
<span id="cb46-166"><a href="#cb46-166"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-167"><a href="#cb46-167"></a>  <span class="fu">group_by</span>(strainID, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-168"><a href="#cb46-168"></a>  <span class="co"># the rbeta is to just add some noise at the lower and upper ends where there are ties</span></span>
<span id="cb46-169"><a href="#cb46-169"></a>  <span class="fu">mutate</span>(<span class="at">freq_alt_complete =</span> <span class="fu">case_when</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> freq_alt_complete <span class="sc">-</span> <span class="fu">rbeta</span>(<span class="fu">n</span>(), <span class="fl">0.1</span>, <span class="dv">10</span>, <span class="at">ncp =</span> <span class="dv">0</span>),</span>
<span id="cb46-170"><a href="#cb46-170"></a>                                       freq_alt_complete <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> freq_alt_complete <span class="sc">+</span> <span class="fu">rbeta</span>(<span class="fu">n</span>(), <span class="fl">0.1</span>, <span class="dv">10</span>, <span class="at">ncp =</span> <span class="dv">0</span>),</span>
<span id="cb46-171"><a href="#cb46-171"></a>                                       <span class="cn">TRUE</span> <span class="sc">~</span> freq_alt_complete)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-172"><a href="#cb46-172"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-173"><a href="#cb46-173"></a>  <span class="fu">arrange</span>(strainID, replicate, prey_history, predator_history, <span class="fu">desc</span>(freq_alt_complete)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-174"><a href="#cb46-174"></a>  <span class="fu">group_by</span>(strainID, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-175"><a href="#cb46-175"></a>  <span class="co"># get 1 - emiprical cumulative distribution</span></span>
<span id="cb46-176"><a href="#cb46-176"></a>  <span class="fu">mutate</span>(<span class="at">f_gt_fmax =</span> <span class="fl">1.001</span> <span class="sc">-</span> <span class="fu">ecdf</span>(freq_alt_complete)(freq_alt_complete))</span>
<span id="cb46-177"><a href="#cb46-177"></a><span class="in">```</span></span>
<span id="cb46-178"><a href="#cb46-178"></a></span>
<span id="cb46-179"><a href="#cb46-179"></a>::: {#fig-02}</span>
<span id="cb46-182"><a href="#cb46-182"></a><span class="in">```{r}</span></span>
<span id="cb46-183"><a href="#cb46-183"></a><span class="fu">ggplot</span>(allelespectra, <span class="fu">aes</span>(<span class="at">x =</span> freq_alt_complete, <span class="at">y =</span> f_gt_fmax)) <span class="sc">+</span> </span>
<span id="cb46-184"><a href="#cb46-184"></a>  <span class="fu">geom_step</span>(<span class="fu">aes</span>(<span class="at">color =</span> predator_history, <span class="at">linetype =</span> replicate, <span class="at">group =</span> <span class="fu">interaction</span>(predator_history, replicate))) <span class="sc">+</span></span>
<span id="cb46-185"><a href="#cb46-185"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Maximum observed allele frequency $</span><span class="sc">\\</span><span class="st">textit{f_{max}}$"</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">"Fraction of mutations $</span><span class="sc">\\</span><span class="st">geq </span><span class="sc">\\</span><span class="st">textit{f_{max}}$"</span>)) <span class="sc">+</span></span>
<span id="cb46-186"><a href="#cb46-186"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> strainID, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-187"><a href="#cb46-187"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"bl"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-188"><a href="#cb46-188"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>, <span class="fl">0.01</span>, <span class="fl">0.001</span>)) <span class="sc">+</span></span>
<span id="cb46-189"><a href="#cb46-189"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.1</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">1e-1</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb46-190"><a href="#cb46-190"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-191"><a href="#cb46-191"></a>    <span class="fu">theme</span>(</span>
<span id="cb46-192"><a href="#cb46-192"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-193"><a href="#cb46-193"></a>      <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-194"><a href="#cb46-194"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-195"><a href="#cb46-195"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb46-196"><a href="#cb46-196"></a>    )</span>
<span id="cb46-197"><a href="#cb46-197"></a><span class="in">```</span></span>
<span id="cb46-198"><a href="#cb46-198"></a></span>
<span id="cb46-199"><a href="#cb46-199"></a>The empirical cumulative distribution of the maximum frequency that mutations reached $f_{max}$ for each taxon-treatment combination.</span>
<span id="cb46-200"><a href="#cb46-200"></a>:::</span>
<span id="cb46-201"><a href="#cb46-201"></a></span>
<span id="cb46-202"><a href="#cb46-202"></a>Mutational spectra from the ancestral bacteria are of lower resolution because there are fewer mutations in those populations during the course of the experiment. Spectra from evolved bacteria are of higher resolution but include both standing genetic variation from the start of the experiment and de novo mutions. Thus, comparing ancestral to evolved bacteria spectra is not particularly meaningful. However, some insights may be gained from comparing different predator evolutionary histories nested with bacterial evolutionary histories. For evolved bacteria it appears that for 1972, 1977, and 2659 that the trajectory of $f_{max}$ in the ancestral predator treatment drops relative to the evolved predator treatment. This would then imply that, on the whole, bacterial alleles from the evolved predator treatments increased to higher frequencies than alleles in the ancestral predator treatments. However, I am not sure there is strong support for this assertion - there are only three replicates and usually at least one replicate from the evolved predator treatment overlaps with the ancestral predator treatment. A comparison of the spectra between species indicates that, generally, some species (HAMBI_1972 and HAMBI_0403) have more high frequency mutations than others (HAMBI_1287 and HAMBI_2659).</span>
<span id="cb46-203"><a href="#cb46-203"></a></span>
<span id="cb46-204"><a href="#cb46-204"></a><span class="fu"># Mutational dynamics</span></span>
<span id="cb46-205"><a href="#cb46-205"></a></span>
<span id="cb46-206"><a href="#cb46-206"></a>We calculated the accumulation of mutations by time t as the sum of derived allele frequencies</span>
<span id="cb46-207"><a href="#cb46-207"></a></span>
<span id="cb46-208"><a href="#cb46-208"></a>$$ M\left(t\right)\equiv\sum_m {\widehat f}_{pmt} $$ where ${\widehat f}_{pmt}=A_{pmt}/D_{pmt}$ and A is the coverage of the alternative allele and D is the total depth of coverage. To partially mitigate noise associated with calling low frequency variants in medium-low coverage samples (\~20X) we restricted our analysis to variants where ${\widehat f}_{pmt} \geq 0.15$.</span>
<span id="cb46-209"><a href="#cb46-209"></a></span>
<span id="cb46-210"><a href="#cb46-210"></a><span class="fu">## Some data formatting</span></span>
<span id="cb46-211"><a href="#cb46-211"></a></span>
<span id="cb46-214"><a href="#cb46-214"></a><span class="in">```{r}</span></span>
<span id="cb46-215"><a href="#cb46-215"></a><span class="co"># only denovo (i.e. not observed at time 0)</span></span>
<span id="cb46-216"><a href="#cb46-216"></a>mgvars <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-217"><a href="#cb46-217"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-218"><a href="#cb46-218"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb46-219"><a href="#cb46-219"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb46-220"><a href="#cb46-220"></a>  </span>
<span id="cb46-221"><a href="#cb46-221"></a><span class="co"># only denovo (i.e. not observed at time 0)</span></span>
<span id="cb46-222"><a href="#cb46-222"></a>mgvars_denovo <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-223"><a href="#cb46-223"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-224"><a href="#cb46-224"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb46-225"><a href="#cb46-225"></a>  <span class="fu">mutate</span>(<span class="at">denovo =</span> <span class="fu">if_else</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> time_days <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="cn">NA_real_</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-226"><a href="#cb46-226"></a>  <span class="fu">fill</span>(denovo, <span class="at">.direction =</span> <span class="st">"down"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-227"><a href="#cb46-227"></a>  <span class="fu">relocate</span>(group_id, freq_alt_complete, denovo) <span class="sc">%&gt;%</span> </span>
<span id="cb46-228"><a href="#cb46-228"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-229"><a href="#cb46-229"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(denovo))</span>
<span id="cb46-230"><a href="#cb46-230"></a></span>
<span id="cb46-231"><a href="#cb46-231"></a><span class="co"># only nonsynonymous variants</span></span>
<span id="cb46-232"><a href="#cb46-232"></a>mgvars_ns <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-233"><a href="#cb46-233"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(effect, <span class="st">"intergenic|intragenic|synonymous|fusion"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-234"><a href="#cb46-234"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(impact, <span class="st">"MODIFIER"</span>))</span>
<span id="cb46-235"><a href="#cb46-235"></a></span>
<span id="cb46-236"><a href="#cb46-236"></a><span class="co"># only denovo (i.e. not observed at time 0) and nonsynonymous variants</span></span>
<span id="cb46-237"><a href="#cb46-237"></a>mgvars_ns_denovo <span class="ot">&lt;-</span> mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb46-238"><a href="#cb46-238"></a>  <span class="fu">group_by</span>(strainID, chrom, pos, ref, alt, replicate, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-239"><a href="#cb46-239"></a>  <span class="fu">mutate</span>(<span class="at">group_id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb46-240"><a href="#cb46-240"></a>  <span class="fu">mutate</span>(<span class="at">denovo =</span> <span class="fu">if_else</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> time_days <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">1</span>, <span class="cn">NA_real_</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-241"><a href="#cb46-241"></a>  <span class="fu">fill</span>(denovo, <span class="at">.direction =</span> <span class="st">"down"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-242"><a href="#cb46-242"></a>  <span class="fu">relocate</span>(group_id, freq_alt_complete, denovo) <span class="sc">%&gt;%</span> </span>
<span id="cb46-243"><a href="#cb46-243"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-244"><a href="#cb46-244"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(denovo)) </span>
<span id="cb46-245"><a href="#cb46-245"></a></span>
<span id="cb46-246"><a href="#cb46-246"></a>cummt <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-247"><a href="#cb46-247"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-248"><a href="#cb46-248"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb46-249"><a href="#cb46-249"></a></span>
<span id="cb46-250"><a href="#cb46-250"></a>cummt_denovo <span class="ot">&lt;-</span> mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-251"><a href="#cb46-251"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-252"><a href="#cb46-252"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb46-253"><a href="#cb46-253"></a></span>
<span id="cb46-254"><a href="#cb46-254"></a>cummt_ns <span class="ot">&lt;-</span> mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb46-255"><a href="#cb46-255"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-256"><a href="#cb46-256"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb46-257"><a href="#cb46-257"></a></span>
<span id="cb46-258"><a href="#cb46-258"></a>cummt_ns_denovo <span class="ot">&lt;-</span> mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-259"><a href="#cb46-259"></a>  <span class="fu">summarize</span>(<span class="at">Mt =</span> <span class="fu">sum</span>(freq_alt_complete[freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.15</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-260"><a href="#cb46-260"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history))</span>
<span id="cb46-261"><a href="#cb46-261"></a><span class="in">```</span></span>
<span id="cb46-262"><a href="#cb46-262"></a></span>
<span id="cb46-265"><a href="#cb46-265"></a><span class="in">```{r}</span></span>
<span id="cb46-266"><a href="#cb46-266"></a>plotmuttraj <span class="ot">&lt;-</span> <span class="cf">function</span>(df, x, alpha, ylab){</span>
<span id="cb46-267"><a href="#cb46-267"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> time_days, <span class="at">y =</span> {{ x }})) <span class="sc">+</span></span>
<span id="cb46-268"><a href="#cb46-268"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, <span class="at">shape =</span> replicate), <span class="at">alpha =</span> alpha) <span class="sc">+</span></span>
<span id="cb46-269"><a href="#cb46-269"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID, </span>
<span id="cb46-270"><a href="#cb46-270"></a>                  <span class="at">group =</span> <span class="fu">interaction</span>(strainID, prey_history, predator_history, replicate)),</span>
<span id="cb46-271"><a href="#cb46-271"></a>              <span class="at">alpha =</span> alpha, </span>
<span id="cb46-272"><a href="#cb46-272"></a>              <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb46-273"><a href="#cb46-273"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb46-274"><a href="#cb46-274"></a>    <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-275"><a href="#cb46-275"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Experiment time (days)"</span>, <span class="at">y =</span> ylab, <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb46-276"><a href="#cb46-276"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">28</span>, <span class="dv">60</span>)) <span class="sc">+</span></span>
<span id="cb46-277"><a href="#cb46-277"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-278"><a href="#cb46-278"></a>    <span class="fu">theme</span>(</span>
<span id="cb46-279"><a href="#cb46-279"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-280"><a href="#cb46-280"></a>      <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-281"><a href="#cb46-281"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-282"><a href="#cb46-282"></a>      <span class="at">panel.grid =</span> <span class="fu">element_blank</span>()</span>
<span id="cb46-283"><a href="#cb46-283"></a>    )</span>
<span id="cb46-284"><a href="#cb46-284"></a>}</span>
<span id="cb46-285"><a href="#cb46-285"></a><span class="in">```</span></span>
<span id="cb46-286"><a href="#cb46-286"></a></span>
<span id="cb46-287"><a href="#cb46-287"></a><span class="fu">## All variants {#sec-allvars}</span></span>
<span id="cb46-288"><a href="#cb46-288"></a></span>
<span id="cb46-289"><a href="#cb46-289"></a>This section considers all variants from the experiment including standing genetic variation present in the evolved populations at T0 and de novo mutations that arise later. It also includes both protein changing and protein non-changing variants.</span>
<span id="cb46-290"><a href="#cb46-290"></a></span>
<span id="cb46-291"><a href="#cb46-291"></a><span class="fu">### Cumulative mutation trajectories (M(t))</span></span>
<span id="cb46-292"><a href="#cb46-292"></a></span>
<span id="cb46-293"><a href="#cb46-293"></a>::: {#fig-03}</span>
<span id="cb46-296"><a href="#cb46-296"></a><span class="in">```{r}</span></span>
<span id="cb46-297"><a href="#cb46-297"></a><span class="co">#| warning: false</span></span>
<span id="cb46-298"><a href="#cb46-298"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-299"><a href="#cb46-299"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-300"><a href="#cb46-300"></a>cummt <span class="sc">%&gt;%</span> </span>
<span id="cb46-301"><a href="#cb46-301"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-302"><a href="#cb46-302"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-303"><a href="#cb46-303"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t)"</span>) <span class="sc">+</span></span>
<span id="cb46-304"><a href="#cb46-304"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-305"><a href="#cb46-305"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-306"><a href="#cb46-306"></a><span class="in">```</span></span>
<span id="cb46-307"><a href="#cb46-307"></a></span>
<span id="cb46-308"><a href="#cb46-308"></a>Cumulative mutation (M(t)) trajectories of both amino acid altering and amino acid non-altering variants in HAMBI species genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.</span>
<span id="cb46-309"><a href="#cb46-309"></a>:::</span>
<span id="cb46-310"><a href="#cb46-310"></a></span>
<span id="cb46-311"><a href="#cb46-311"></a><span class="fu">### Total mutation trajectories</span></span>
<span id="cb46-312"><a href="#cb46-312"></a></span>
<span id="cb46-313"><a href="#cb46-313"></a>::: {#fig-04}</span>
<span id="cb46-316"><a href="#cb46-316"></a><span class="in">```{r}</span></span>
<span id="cb46-317"><a href="#cb46-317"></a><span class="co">#| warning: false</span></span>
<span id="cb46-318"><a href="#cb46-318"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-319"><a href="#cb46-319"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-320"><a href="#cb46-320"></a>mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-321"><a href="#cb46-321"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-322"><a href="#cb46-322"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-323"><a href="#cb46-323"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-324"><a href="#cb46-324"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-325"><a href="#cb46-325"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-326"><a href="#cb46-326"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-327"><a href="#cb46-327"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb46-328"><a href="#cb46-328"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-329"><a href="#cb46-329"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb46-330"><a href="#cb46-330"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb46-331"><a href="#cb46-331"></a><span class="in">```</span></span>
<span id="cb46-332"><a href="#cb46-332"></a></span>
<span id="cb46-333"><a href="#cb46-333"></a>Trajectories of the **total number of non-zero mutations** detected at each time point for both amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.</span>
<span id="cb46-334"><a href="#cb46-334"></a>:::</span>
<span id="cb46-335"><a href="#cb46-335"></a></span>
<span id="cb46-336"><a href="#cb46-336"></a><span class="fu">### Fixed mutation trajectories</span></span>
<span id="cb46-337"><a href="#cb46-337"></a></span>
<span id="cb46-338"><a href="#cb46-338"></a>::: {#fig-05}</span>
<span id="cb46-341"><a href="#cb46-341"></a><span class="in">```{r}</span></span>
<span id="cb46-342"><a href="#cb46-342"></a><span class="co">#| warning: false</span></span>
<span id="cb46-343"><a href="#cb46-343"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-344"><a href="#cb46-344"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-345"><a href="#cb46-345"></a>mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-346"><a href="#cb46-346"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-347"><a href="#cb46-347"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-348"><a href="#cb46-348"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-349"><a href="#cb46-349"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-350"><a href="#cb46-350"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-351"><a href="#cb46-351"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-352"><a href="#cb46-352"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-353"><a href="#cb46-353"></a><span class="in">```</span></span>
<span id="cb46-354"><a href="#cb46-354"></a></span>
<span id="cb46-355"><a href="#cb46-355"></a>Trajectories of the total number of **fixed** mutations detected at each time point for both amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates.</span>
<span id="cb46-356"><a href="#cb46-356"></a>:::</span>
<span id="cb46-357"><a href="#cb46-357"></a></span>
<span id="cb46-358"><a href="#cb46-358"></a><span class="fu">## De novo variants</span></span>
<span id="cb46-359"><a href="#cb46-359"></a></span>
<span id="cb46-360"><a href="#cb46-360"></a>This section considers only de novo mutations that did not exist within the evolved populations at T0. It includes both protein changing and protein non-changing variants. Thus, the results here for the ancestral bacteria populations here will be the same as in @sec-allvars.</span>
<span id="cb46-361"><a href="#cb46-361"></a></span>
<span id="cb46-362"><a href="#cb46-362"></a><span class="fu">### Cumulative mutation trajectories (M(t))</span></span>
<span id="cb46-363"><a href="#cb46-363"></a></span>
<span id="cb46-364"><a href="#cb46-364"></a>::: {#fig-06}</span>
<span id="cb46-367"><a href="#cb46-367"></a><span class="in">```{r}</span></span>
<span id="cb46-368"><a href="#cb46-368"></a><span class="co">#| warning: false</span></span>
<span id="cb46-369"><a href="#cb46-369"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-370"><a href="#cb46-370"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-371"><a href="#cb46-371"></a>fig06 <span class="ot">&lt;-</span> cummt_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-372"><a href="#cb46-372"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-373"><a href="#cb46-373"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-374"><a href="#cb46-374"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t)"</span>) <span class="sc">+</span></span>
<span id="cb46-375"><a href="#cb46-375"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-376"><a href="#cb46-376"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-377"><a href="#cb46-377"></a></span>
<span id="cb46-378"><a href="#cb46-378"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"Mt_denovo.svg"</span>), fig06, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">5</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb46-379"><a href="#cb46-379"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb46-380"><a href="#cb46-380"></a></span>
<span id="cb46-381"><a href="#cb46-381"></a>fig06</span>
<span id="cb46-382"><a href="#cb46-382"></a><span class="in">```</span></span>
<span id="cb46-383"><a href="#cb46-383"></a></span>
<span id="cb46-384"><a href="#cb46-384"></a>Cumulative mutation (M(t)) trajectories of de novo amino acid altering and amino acid non-altering variants in HAMBI species genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-03 but only includes de novo variants (i.e. excludes standing genetic variation in the evolved populations).</span>
<span id="cb46-385"><a href="#cb46-385"></a>:::</span>
<span id="cb46-386"><a href="#cb46-386"></a></span>
<span id="cb46-387"><a href="#cb46-387"></a><span class="fu">### Total mutation trajectories</span></span>
<span id="cb46-388"><a href="#cb46-388"></a></span>
<span id="cb46-389"><a href="#cb46-389"></a>::: {#fig-07}</span>
<span id="cb46-392"><a href="#cb46-392"></a><span class="in">```{r}</span></span>
<span id="cb46-393"><a href="#cb46-393"></a><span class="co">#| warning: false</span></span>
<span id="cb46-394"><a href="#cb46-394"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-395"><a href="#cb46-395"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-396"><a href="#cb46-396"></a>mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-397"><a href="#cb46-397"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-398"><a href="#cb46-398"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-399"><a href="#cb46-399"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-400"><a href="#cb46-400"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-401"><a href="#cb46-401"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-402"><a href="#cb46-402"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-403"><a href="#cb46-403"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb46-404"><a href="#cb46-404"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-405"><a href="#cb46-405"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb46-406"><a href="#cb46-406"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb46-407"><a href="#cb46-407"></a><span class="in">```</span></span>
<span id="cb46-408"><a href="#cb46-408"></a></span>
<span id="cb46-409"><a href="#cb46-409"></a>Trajectories of the **total number of non-zero mutations** detected at each time point for de novo amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-04 but only includes de novo variants (i.e. excludes standing genetic variation in the evolved populations).</span>
<span id="cb46-410"><a href="#cb46-410"></a>:::</span>
<span id="cb46-411"><a href="#cb46-411"></a></span>
<span id="cb46-412"><a href="#cb46-412"></a><span class="fu">### Fixed mutation trajectories</span></span>
<span id="cb46-413"><a href="#cb46-413"></a></span>
<span id="cb46-414"><a href="#cb46-414"></a>::: {#fig-08}</span>
<span id="cb46-417"><a href="#cb46-417"></a><span class="in">```{r}</span></span>
<span id="cb46-418"><a href="#cb46-418"></a><span class="co">#| warning: false</span></span>
<span id="cb46-419"><a href="#cb46-419"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-420"><a href="#cb46-420"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-421"><a href="#cb46-421"></a>mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-422"><a href="#cb46-422"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-423"><a href="#cb46-423"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-424"><a href="#cb46-424"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-425"><a href="#cb46-425"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-426"><a href="#cb46-426"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-427"><a href="#cb46-427"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-428"><a href="#cb46-428"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-429"><a href="#cb46-429"></a><span class="in">```</span></span>
<span id="cb46-430"><a href="#cb46-430"></a></span>
<span id="cb46-431"><a href="#cb46-431"></a>Trajectories of the total number of **fixed** mutations detected at each time point for de novo amino acid altering and amino acid non-altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-05 but only includes de novo variants (i.e. excludes standing genetic variation in the evolved populations).</span>
<span id="cb46-432"><a href="#cb46-432"></a>:::</span>
<span id="cb46-433"><a href="#cb46-433"></a></span>
<span id="cb46-434"><a href="#cb46-434"></a><span class="fu">## Non-synonymous variants {#sec-nsvars}</span></span>
<span id="cb46-435"><a href="#cb46-435"></a></span>
<span id="cb46-436"><a href="#cb46-436"></a>This section considers all nonsynonymous (i.e., amino acid altering) variants from the experiment including standing genetic variation present in the evolved populations at T0 and de novo mutations that arise later.</span>
<span id="cb46-437"><a href="#cb46-437"></a></span>
<span id="cb46-438"><a href="#cb46-438"></a><span class="fu">### Cumulative mutation trajectories (M(t))</span></span>
<span id="cb46-439"><a href="#cb46-439"></a></span>
<span id="cb46-440"><a href="#cb46-440"></a>::: {#fig-09}</span>
<span id="cb46-443"><a href="#cb46-443"></a><span class="in">```{r}</span></span>
<span id="cb46-444"><a href="#cb46-444"></a><span class="co">#| warning: false</span></span>
<span id="cb46-445"><a href="#cb46-445"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-446"><a href="#cb46-446"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-447"><a href="#cb46-447"></a>cummt_ns <span class="sc">%&gt;%</span> </span>
<span id="cb46-448"><a href="#cb46-448"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-449"><a href="#cb46-449"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-450"><a href="#cb46-450"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t) (nonsynonymous)"</span>) <span class="sc">+</span></span>
<span id="cb46-451"><a href="#cb46-451"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-452"><a href="#cb46-452"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-453"><a href="#cb46-453"></a><span class="in">```</span></span>
<span id="cb46-454"><a href="#cb46-454"></a></span>
<span id="cb46-455"><a href="#cb46-455"></a>Cumulative mutation (M(t)) trajectories of amino acid altering variants in HAMBI species genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-03 but only includes nonsynonymous (amino acid changing) variants.</span>
<span id="cb46-456"><a href="#cb46-456"></a>:::</span>
<span id="cb46-457"><a href="#cb46-457"></a></span>
<span id="cb46-458"><a href="#cb46-458"></a><span class="fu">### Total mutation trajectories</span></span>
<span id="cb46-459"><a href="#cb46-459"></a></span>
<span id="cb46-460"><a href="#cb46-460"></a>::: {#fig-10}</span>
<span id="cb46-463"><a href="#cb46-463"></a><span class="in">```{r}</span></span>
<span id="cb46-464"><a href="#cb46-464"></a><span class="co">#| warning: false</span></span>
<span id="cb46-465"><a href="#cb46-465"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-466"><a href="#cb46-466"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-467"><a href="#cb46-467"></a>mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb46-468"><a href="#cb46-468"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-469"><a href="#cb46-469"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-470"><a href="#cb46-470"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-471"><a href="#cb46-471"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-472"><a href="#cb46-472"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of nonsynonymous mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-473"><a href="#cb46-473"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-474"><a href="#cb46-474"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb46-475"><a href="#cb46-475"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-476"><a href="#cb46-476"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb46-477"><a href="#cb46-477"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>)) </span>
<span id="cb46-478"><a href="#cb46-478"></a><span class="in">```</span></span>
<span id="cb46-479"><a href="#cb46-479"></a></span>
<span id="cb46-480"><a href="#cb46-480"></a>Trajectories of the **total number of non-zero mutations** detected at each time point for amino acid altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-04 but only includes nonsynonymous (amino acid changing) variants.</span>
<span id="cb46-481"><a href="#cb46-481"></a>:::</span>
<span id="cb46-482"><a href="#cb46-482"></a></span>
<span id="cb46-483"><a href="#cb46-483"></a><span class="fu">### Fixed mutation trajectories</span></span>
<span id="cb46-484"><a href="#cb46-484"></a></span>
<span id="cb46-485"><a href="#cb46-485"></a>::: {#fig-11}</span>
<span id="cb46-488"><a href="#cb46-488"></a><span class="in">```{r}</span></span>
<span id="cb46-489"><a href="#cb46-489"></a><span class="co">#| warning: false</span></span>
<span id="cb46-490"><a href="#cb46-490"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-491"><a href="#cb46-491"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-492"><a href="#cb46-492"></a>mgvars_ns <span class="sc">%&gt;%</span> </span>
<span id="cb46-493"><a href="#cb46-493"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-494"><a href="#cb46-494"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-495"><a href="#cb46-495"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-496"><a href="#cb46-496"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-497"><a href="#cb46-497"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of nonsynonymous fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-498"><a href="#cb46-498"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-499"><a href="#cb46-499"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-500"><a href="#cb46-500"></a><span class="in">```</span></span>
<span id="cb46-501"><a href="#cb46-501"></a></span>
<span id="cb46-502"><a href="#cb46-502"></a>Trajectories of the total number of **fixed** mutations detected at each time point for amino acid altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). For evolved prey treatments this includes both standing mutations at T0 and de novo mutations. Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-05 but only includes non-synonymous (amino acid changing) variants.</span>
<span id="cb46-503"><a href="#cb46-503"></a>:::</span>
<span id="cb46-504"><a href="#cb46-504"></a></span>
<span id="cb46-505"><a href="#cb46-505"></a><span class="fu">## De novo, non-synonymous variants</span></span>
<span id="cb46-506"><a href="#cb46-506"></a></span>
<span id="cb46-507"><a href="#cb46-507"></a>This section considers only de novo mutations that did not exist within the evolved populations at T0 for only non-synonymous (i.e., amino acid altering) variants. Thus, the results here for the ancestral bacteria populations will be the same as in @sec-nsvars</span>
<span id="cb46-508"><a href="#cb46-508"></a></span>
<span id="cb46-509"><a href="#cb46-509"></a><span class="fu">### Cumulative mutation trajectories (M(t))</span></span>
<span id="cb46-510"><a href="#cb46-510"></a></span>
<span id="cb46-511"><a href="#cb46-511"></a>::: {#fig-12}</span>
<span id="cb46-514"><a href="#cb46-514"></a><span class="in">```{r}</span></span>
<span id="cb46-515"><a href="#cb46-515"></a><span class="co">#| warning: false</span></span>
<span id="cb46-516"><a href="#cb46-516"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-517"><a href="#cb46-517"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-518"><a href="#cb46-518"></a>cummt_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-519"><a href="#cb46-519"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-520"><a href="#cb46-520"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(Mt)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-521"><a href="#cb46-521"></a>  <span class="fu">plotmuttraj</span>(Mt, <span class="fl">0.35</span>, <span class="st">"M(t) (de novo, nonsynonymous)"</span>) <span class="sc">+</span></span>
<span id="cb46-522"><a href="#cb46-522"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-523"><a href="#cb46-523"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-524"><a href="#cb46-524"></a><span class="in">```</span></span>
<span id="cb46-525"><a href="#cb46-525"></a></span>
<span id="cb46-526"><a href="#cb46-526"></a>Cumulative mutation (M(t)) trajectories of de novo amino acid altering variants in HAMBI species genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-03 but only includes de novo (i.e., not present at T0) and non-synonymous (amino acid changing) variants.</span>
<span id="cb46-527"><a href="#cb46-527"></a>:::</span>
<span id="cb46-528"><a href="#cb46-528"></a></span>
<span id="cb46-529"><a href="#cb46-529"></a><span class="fu">### Total mutation trajectories</span></span>
<span id="cb46-530"><a href="#cb46-530"></a></span>
<span id="cb46-531"><a href="#cb46-531"></a>::: {#fig-13}</span>
<span id="cb46-534"><a href="#cb46-534"></a><span class="in">```{r}</span></span>
<span id="cb46-535"><a href="#cb46-535"></a><span class="co">#| warning: false</span></span>
<span id="cb46-536"><a href="#cb46-536"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-537"><a href="#cb46-537"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-538"><a href="#cb46-538"></a>fig13 <span class="ot">&lt;-</span> mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-539"><a href="#cb46-539"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-540"><a href="#cb46-540"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-541"><a href="#cb46-541"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-542"><a href="#cb46-542"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-543"><a href="#cb46-543"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of de novo nonsynonymous mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-544"><a href="#cb46-544"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-545"><a href="#cb46-545"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb46-546"><a href="#cb46-546"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"l"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-547"><a href="#cb46-547"></a>  <span class="fu">scale_y_log10</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">10</span>, <span class="dv">100</span>),</span>
<span id="cb46-548"><a href="#cb46-548"></a>                <span class="at">labels =</span> <span class="fu">label_log</span>(<span class="at">base =</span> <span class="dv">10</span>, <span class="at">digits =</span> <span class="dv">3</span>))</span>
<span id="cb46-549"><a href="#cb46-549"></a></span>
<span id="cb46-550"><a href="#cb46-550"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"cummuts_denovo_nonsyn.svg"</span>), fig13, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">5</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb46-551"><a href="#cb46-551"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb46-552"><a href="#cb46-552"></a></span>
<span id="cb46-553"><a href="#cb46-553"></a>fig13</span>
<span id="cb46-554"><a href="#cb46-554"></a><span class="in">```</span></span>
<span id="cb46-555"><a href="#cb46-555"></a></span>
<span id="cb46-556"><a href="#cb46-556"></a>Trajectories of the **total number of non-zero mutations** detected at each time point for de novo amino acid altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-04 but only includes de novo (i.e., not present at T0) and non-synonymous (amino acid changing) variants.</span>
<span id="cb46-557"><a href="#cb46-557"></a>:::</span>
<span id="cb46-558"><a href="#cb46-558"></a></span>
<span id="cb46-559"><a href="#cb46-559"></a><span class="fu">### Fixed mutation trajectories</span></span>
<span id="cb46-560"><a href="#cb46-560"></a></span>
<span id="cb46-561"><a href="#cb46-561"></a>::: {#fig-14}</span>
<span id="cb46-564"><a href="#cb46-564"></a><span class="in">```{r}</span></span>
<span id="cb46-565"><a href="#cb46-565"></a><span class="co">#| warning: false</span></span>
<span id="cb46-566"><a href="#cb46-566"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-567"><a href="#cb46-567"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb46-568"><a href="#cb46-568"></a>mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-569"><a href="#cb46-569"></a>  <span class="fu">summarize</span>(<span class="at">n =</span> <span class="fu">sum</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb46-570"><a href="#cb46-570"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, time_days, replicate, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-571"><a href="#cb46-571"></a>  <span class="fu">group_by</span>(strainID, time_days, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-572"><a href="#cb46-572"></a>  <span class="fu">mutate</span>(<span class="at">mn =</span> <span class="fu">mean</span>(n)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-573"><a href="#cb46-573"></a>  <span class="fu">plotmuttraj</span>(n, <span class="fl">0.35</span>, <span class="st">"Number of de novo, nonsynonymous fixed mutations"</span>) <span class="sc">+</span></span>
<span id="cb46-574"><a href="#cb46-574"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-575"><a href="#cb46-575"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> mn, <span class="at">x=</span> time_days, <span class="at">color =</span> strainID), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb46-576"><a href="#cb46-576"></a><span class="in">```</span></span>
<span id="cb46-577"><a href="#cb46-577"></a></span>
<span id="cb46-578"><a href="#cb46-578"></a>Trajectories of the total number of **fixed** mutations detected at each time point for de novo amino acid altering variants in HAMBI genomes with metagenomic coverage $\geq$ 5X in at least one time point (@fig-01). Transparent points/lines represent individual replicates, while opaque points/lines represent the mean over the three replicates. Plot is analogous to @fig-05 but only includes de novo (i.e., not present at T0) and non-synonymous (amino acid changing) variants.</span>
<span id="cb46-579"><a href="#cb46-579"></a>:::</span>
<span id="cb46-580"><a href="#cb46-580"></a></span>
<span id="cb46-581"><a href="#cb46-581"></a><span class="fu"># Variant time to majority</span></span>
<span id="cb46-582"><a href="#cb46-582"></a></span>
<span id="cb46-583"><a href="#cb46-583"></a>Does the evolutionary history of prey/predator impact how long it takes alleles to reach majority status ($f_{max} \geq 0.5$). For every alternative allele that emerged in the experiment de novo, we plot the time that it took for that alt allele to reach majority status (i.e., the alt allele dominates the reference allele).</span>
<span id="cb46-584"><a href="#cb46-584"></a></span>
<span id="cb46-585"><a href="#cb46-585"></a><span class="fu">## De novo variants (both amino acid chaning and non-changing)</span></span>
<span id="cb46-586"><a href="#cb46-586"></a></span>
<span id="cb46-587"><a href="#cb46-587"></a>::: {#fig-15}</span>
<span id="cb46-590"><a href="#cb46-590"></a><span class="in">```{r}</span></span>
<span id="cb46-591"><a href="#cb46-591"></a><span class="co">#| warning: false</span></span>
<span id="cb46-592"><a href="#cb46-592"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-593"><a href="#cb46-593"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb46-594"><a href="#cb46-594"></a>mgvars_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-595"><a href="#cb46-595"></a>  <span class="fu">group_by</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb46-596"><a href="#cb46-596"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete) <span class="sc">&amp;</span> freq_alt_complete <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-597"><a href="#cb46-597"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-598"><a href="#cb46-598"></a>  <span class="fu">slice_min</span>(time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb46-599"><a href="#cb46-599"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-600"><a href="#cb46-600"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(time_days))) <span class="sc">+</span></span>
<span id="cb46-601"><a href="#cb46-601"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-602"><a href="#cb46-602"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb46-603"><a href="#cb46-603"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-604"><a href="#cb46-604"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Time to major allele ( $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ )"</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">" $n_{allele}$ with $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ "</span>), <span class="at">fill =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb46-605"><a href="#cb46-605"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-606"><a href="#cb46-606"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-607"><a href="#cb46-607"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-608"><a href="#cb46-608"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-609"><a href="#cb46-609"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-610"><a href="#cb46-610"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb46-611"><a href="#cb46-611"></a><span class="in">```</span></span>
<span id="cb46-612"><a href="#cb46-612"></a></span>
<span id="cb46-613"><a href="#cb46-613"></a>Count of alternative alleles with $f_{max} \geq 0.5$ plotted against the time (days) when the alternative allele first reaches major allele status $(f_{max} \geq 0.5)$. Colors represent the proportion of alleles from different HAMBI species. Plot includes both amino acid changing and non-changing de novo mutations (i.e. excludes standing genetic variation from T0 in the evolved bacterial populations).</span>
<span id="cb46-614"><a href="#cb46-614"></a>:::</span>
<span id="cb46-615"><a href="#cb46-615"></a></span>
<span id="cb46-616"><a href="#cb46-616"></a><span class="fu">## De novo variants, non-synonymous</span></span>
<span id="cb46-617"><a href="#cb46-617"></a></span>
<span id="cb46-618"><a href="#cb46-618"></a>::: {#fig-16}</span>
<span id="cb46-621"><a href="#cb46-621"></a><span class="in">```{r}</span></span>
<span id="cb46-622"><a href="#cb46-622"></a><span class="co">#| warning: false</span></span>
<span id="cb46-623"><a href="#cb46-623"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-624"><a href="#cb46-624"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb46-625"><a href="#cb46-625"></a>mgvars_ns_denovo <span class="sc">%&gt;%</span> </span>
<span id="cb46-626"><a href="#cb46-626"></a>  <span class="fu">group_by</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb46-627"><a href="#cb46-627"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete) <span class="sc">&amp;</span> freq_alt_complete <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-628"><a href="#cb46-628"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;=</span> <span class="fl">0.50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-629"><a href="#cb46-629"></a>  <span class="fu">slice_min</span>(time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb46-630"><a href="#cb46-630"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-631"><a href="#cb46-631"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(time_days))) <span class="sc">+</span></span>
<span id="cb46-632"><a href="#cb46-632"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-633"><a href="#cb46-633"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb46-634"><a href="#cb46-634"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-635"><a href="#cb46-635"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Time to major allele ( $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ )"</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">" $n_{allele}$ with $f_{max} </span><span class="sc">\\</span><span class="st">geq 0.5$ "</span>), <span class="at">fill =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb46-636"><a href="#cb46-636"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-637"><a href="#cb46-637"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-638"><a href="#cb46-638"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-639"><a href="#cb46-639"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-640"><a href="#cb46-640"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-641"><a href="#cb46-641"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb46-642"><a href="#cb46-642"></a><span class="in">```</span></span>
<span id="cb46-643"><a href="#cb46-643"></a></span>
<span id="cb46-644"><a href="#cb46-644"></a>Count of alternative alleles with $f_{max} \geq 0.5$ plotted against the time (days) when the alternative allele first reaches major allele status $(f_{max} \geq 0.5)$. Colors represent the proportion of alleles from different HAMBI species. Plot includes only amino acid changing de novo variants (i.e. excludes standing genetic variation from T0 in the evolved bacterial populations).</span>
<span id="cb46-645"><a href="#cb46-645"></a>:::</span>
<span id="cb46-646"><a href="#cb46-646"></a></span>
<span id="cb46-647"><a href="#cb46-647"></a>This plot suggests that most de novo alternative alleles (either amino acid changing or non chaning) become the major allele by either day 28 or 60 for the ancestral bacteria populations. For the evolved populations this evolution seems to happen faster where some alt alleles become the major allele already by day 28 but this is only for HAMBI_2659.</span>
<span id="cb46-648"><a href="#cb46-648"></a></span>
<span id="cb46-649"><a href="#cb46-649"></a><span class="fu"># Time to mutation emergence in parallel genes</span></span>
<span id="cb46-650"><a href="#cb46-650"></a></span>
<span id="cb46-653"><a href="#cb46-653"></a><span class="in">```{r}</span></span>
<span id="cb46-654"><a href="#cb46-654"></a>df_ttm <span class="ot">&lt;-</span> <span class="fu">left_join</span>(par_genes, mgvars_ns_denovo) <span class="sc">%&gt;%</span> </span>
<span id="cb46-655"><a href="#cb46-655"></a>  <span class="fu">group_by</span>(group_id) <span class="sc">%&gt;%</span> </span>
<span id="cb46-656"><a href="#cb46-656"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete) <span class="sc">&amp;</span> freq_alt_complete <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-657"><a href="#cb46-657"></a>  <span class="co"># uncomment to only look at emergence of minor-to-major allele transition</span></span>
<span id="cb46-658"><a href="#cb46-658"></a>  <span class="co">#filter(freq_alt_complete &gt;= 0.50) %&gt;% </span></span>
<span id="cb46-659"><a href="#cb46-659"></a>  <span class="fu">slice_min</span>(time_days) <span class="sc">%&gt;%</span> </span>
<span id="cb46-660"><a href="#cb46-660"></a>  <span class="fu">slice</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-661"><a href="#cb46-661"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-662"><a href="#cb46-662"></a>  <span class="co"># some manual renaming</span></span>
<span id="cb46-663"><a href="#cb46-663"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">case_when</span>(locus_tag <span class="sc">==</span> <span class="st">"H1287_02172"</span> <span class="sc">~</span> <span class="st">"yddV_2"</span>,</span>
<span id="cb46-664"><a href="#cb46-664"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1977_02612"</span> <span class="sc">~</span> <span class="st">"dppA/oppA"</span>,</span>
<span id="cb46-665"><a href="#cb46-665"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00299"</span> <span class="sc">~</span> <span class="st">"PAP2_like"</span>,</span>
<span id="cb46-666"><a href="#cb46-666"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02826"</span> <span class="sc">~</span> <span class="st">"ynjC"</span>,</span>
<span id="cb46-667"><a href="#cb46-667"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00671"</span> <span class="sc">~</span> <span class="st">"yqaA"</span>,</span>
<span id="cb46-668"><a href="#cb46-668"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_00421"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb46-669"><a href="#cb46-669"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H1972_02723"</span> <span class="sc">~</span> <span class="st">"yifE"</span>,</span>
<span id="cb46-670"><a href="#cb46-670"></a>                          locus_tag <span class="sc">==</span> <span class="st">"H2659_01789"</span> <span class="sc">~</span> <span class="st">"hyp"</span>,</span>
<span id="cb46-671"><a href="#cb46-671"></a>                          <span class="cn">TRUE</span> <span class="sc">~</span> gene)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-672"><a href="#cb46-672"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(gene), Preferred_name, gene),</span>
<span id="cb46-673"><a href="#cb46-673"></a>         <span class="at">treat =</span> <span class="fu">paste0</span>(<span class="st">"prey:"</span>, prey_history, <span class="st">" | pred:"</span>, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-674"><a href="#cb46-674"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">if_else</span>(gene_lab <span class="sc">==</span> <span class="st">"-"</span>, Description, gene_lab)) <span class="sc">%&gt;%</span></span>
<span id="cb46-675"><a href="#cb46-675"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab =</span> <span class="fu">paste0</span>(treat,<span class="st">"-"</span>, gene_lab)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-676"><a href="#cb46-676"></a>  <span class="fu">mutate</span>(<span class="at">gene_lab2 =</span> <span class="fu">fct_reorder</span>(<span class="fu">factor</span>(gene_lab), time_days))</span>
<span id="cb46-677"><a href="#cb46-677"></a><span class="in">```</span></span>
<span id="cb46-678"><a href="#cb46-678"></a></span>
<span id="cb46-679"><a href="#cb46-679"></a></span>
<span id="cb46-682"><a href="#cb46-682"></a><span class="in">```{r}</span></span>
<span id="cb46-683"><a href="#cb46-683"></a>pbar <span class="ot">&lt;-</span> df_ttm <span class="sc">%&gt;%</span> </span>
<span id="cb46-684"><a href="#cb46-684"></a>  <span class="fu">distinct</span>(gene_lab2, COG_category_long) <span class="sc">%&gt;%</span> </span>
<span id="cb46-685"><a href="#cb46-685"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> gene_lab2, <span class="at">fill =</span> COG_category_long)) <span class="sc">+</span> </span>
<span id="cb46-686"><a href="#cb46-686"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb46-687"><a href="#cb46-687"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> cogpal) <span class="sc">+</span></span>
<span id="cb46-688"><a href="#cb46-688"></a>  <span class="fu">theme_void</span>()</span>
<span id="cb46-689"><a href="#cb46-689"></a></span>
<span id="cb46-690"><a href="#cb46-690"></a>ptext <span class="ot">&lt;-</span> df_ttm <span class="sc">%&gt;%</span> </span>
<span id="cb46-691"><a href="#cb46-691"></a>  <span class="fu">distinct</span>(gene_lab2, COG_category_long, COG_category_single) <span class="sc">%&gt;%</span> </span>
<span id="cb46-692"><a href="#cb46-692"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">COG_category_long =</span> <span class="st">"S - Function unknown"</span>, <span class="at">COG_category_single =</span> <span class="st">"S"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-693"><a href="#cb46-693"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> gene_lab2)) <span class="sc">+</span> </span>
<span id="cb46-694"><a href="#cb46-694"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> COG_category_single, <span class="at">color =</span> COG_category_long)) <span class="sc">+</span></span>
<span id="cb46-695"><a href="#cb46-695"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> cogpal) <span class="sc">+</span></span>
<span id="cb46-696"><a href="#cb46-696"></a>  <span class="fu">theme_void</span>()</span>
<span id="cb46-697"><a href="#cb46-697"></a><span class="in">```</span></span>
<span id="cb46-698"><a href="#cb46-698"></a></span>
<span id="cb46-699"><a href="#cb46-699"></a></span>
<span id="cb46-702"><a href="#cb46-702"></a><span class="in">```{r}</span></span>
<span id="cb46-703"><a href="#cb46-703"></a>ppoint <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_ttm, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(time_days), <span class="at">y =</span> gene_lab2)) <span class="sc">+</span></span>
<span id="cb46-704"><a href="#cb46-704"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">color =</span> treat, <span class="at">shape =</span> strainID), <span class="at">size =</span> <span class="dv">2</span>, <span class="at">width =</span> <span class="fl">0.15</span>, <span class="at">height =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb46-705"><a href="#cb46-705"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Time to mutation emergence"</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb46-706"><a href="#cb46-706"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span> </span>
<span id="cb46-707"><a href="#cb46-707"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-708"><a href="#cb46-708"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-709"><a href="#cb46-709"></a>    <span class="at">legend.position =</span> <span class="st">"left"</span>,</span>
<span id="cb46-710"><a href="#cb46-710"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-711"><a href="#cb46-711"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-712"><a href="#cb46-712"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb46-713"><a href="#cb46-713"></a><span class="in">```</span></span>
<span id="cb46-714"><a href="#cb46-714"></a></span>
<span id="cb46-715"><a href="#cb46-715"></a>::: {#fig-17}</span>
<span id="cb46-718"><a href="#cb46-718"></a><span class="in">```{r}</span></span>
<span id="cb46-719"><a href="#cb46-719"></a><span class="co">#| warning: false</span></span>
<span id="cb46-720"><a href="#cb46-720"></a><span class="co">#| fig.width: 9</span></span>
<span id="cb46-721"><a href="#cb46-721"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb46-722"><a href="#cb46-722"></a>ptogether <span class="ot">&lt;-</span> ppoint <span class="sc">+</span> ptext <span class="sc">+</span></span>
<span id="cb46-723"><a href="#cb46-723"></a>    <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.15</span>), <span class="at">guides =</span> <span class="st">"collect"</span>)</span>
<span id="cb46-724"><a href="#cb46-724"></a></span>
<span id="cb46-725"><a href="#cb46-725"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"par_gene_time2emergence.svg"</span>), ptogether, <span class="at">width=</span><span class="dv">9</span>, <span class="at">height=</span><span class="dv">7</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb46-726"><a href="#cb46-726"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb46-727"><a href="#cb46-727"></a></span>
<span id="cb46-728"><a href="#cb46-728"></a>ptogether</span>
<span id="cb46-729"><a href="#cb46-729"></a><span class="in">```</span></span>
<span id="cb46-730"><a href="#cb46-730"></a></span>
<span id="cb46-731"><a href="#cb46-731"></a>:::</span>
<span id="cb46-732"><a href="#cb46-732"></a></span>
<span id="cb46-733"><a href="#cb46-733"></a>dhaR has two associated COG categories</span>
<span id="cb46-734"><a href="#cb46-734"></a><span class="ss">1. </span>K – Transcription</span>
<span id="cb46-735"><a href="#cb46-735"></a><span class="ss">2. </span>Q - Secondary metabolites biosynthesis transport and catabolism</span>
<span id="cb46-736"><a href="#cb46-736"></a></span>
<span id="cb46-737"><a href="#cb46-737"></a>smf-1 (Major fimbrial subunit SMF-1) has two associated COG categoires</span>
<span id="cb46-738"><a href="#cb46-738"></a><span class="ss">1. </span>N - Cell motility</span>
<span id="cb46-739"><a href="#cb46-739"></a><span class="ss">2. </span>U - intracellular trafficiking, secretion, and vesicular transport</span>
<span id="cb46-740"><a href="#cb46-740"></a></span>
<span id="cb46-741"><a href="#cb46-741"></a></span>
<span id="cb46-742"><a href="#cb46-742"></a><span class="fu"># Relationship between gene parallelism and $f_{max}$</span></span>
<span id="cb46-743"><a href="#cb46-743"></a>Look if there is a relationship between extent of gene parallelism ($\Delta L$) and the max frequency that variants reach during the experiment.</span>
<span id="cb46-744"><a href="#cb46-744"></a></span>
<span id="cb46-745"><a href="#cb46-745"></a>As seen from the results below we observe that the degree of parallelism $\Delta l$ or $G_{i}$ increases with $f_{max}$ across experimental treatment combinations and species, which is consistent with the idea that mutations with higher $f_{max}$ are primarily being driven by positive selection across these treatments. </span>
<span id="cb46-746"><a href="#cb46-746"></a></span>
<span id="cb46-747"><a href="#cb46-747"></a><span class="fu">## $\Delta l$</span></span>
<span id="cb46-748"><a href="#cb46-748"></a></span>
<span id="cb46-749"><a href="#cb46-749"></a>This is the same formatting that is done in <span class="co">[</span><span class="ot">the prior parallelism analysis.</span><span class="co">](02_parallelism.qmd)</span></span>
<span id="cb46-752"><a href="#cb46-752"></a><span class="in">```{r}</span></span>
<span id="cb46-753"><a href="#cb46-753"></a>tofilter <span class="ot">&lt;-</span> mgvars <span class="sc">%&gt;%</span> </span>
<span id="cb46-754"><a href="#cb46-754"></a>  <span class="fu">filter</span>(time_days <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-755"><a href="#cb46-755"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-756"><a href="#cb46-756"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(chrom, pos, ref, alt) <span class="sc">%&gt;%</span> </span>
<span id="cb46-757"><a href="#cb46-757"></a>  <span class="fu">distinct</span>()</span>
<span id="cb46-758"><a href="#cb46-758"></a></span>
<span id="cb46-759"><a href="#cb46-759"></a>mgvars_filt_mb_ns <span class="ot">&lt;-</span> <span class="fu">anti_join</span>(mgvars, tofilter, <span class="at">by =</span> <span class="fu">join_by</span>(chrom, pos, ref, alt)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-760"><a href="#cb46-760"></a>  <span class="co"># exclude variants that were added to make complete time course</span></span>
<span id="cb46-761"><a href="#cb46-761"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(freq_alt_complete)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-762"><a href="#cb46-762"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-763"><a href="#cb46-763"></a>  <span class="fu">filter</span>(time_days <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-764"><a href="#cb46-764"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(effect, <span class="st">"intergenic|intragenic|synonymous|fusion"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-765"><a href="#cb46-765"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(impact, <span class="st">"MODIFIER"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-766"><a href="#cb46-766"></a>  <span class="co"># double check to remove any noncoding mutations</span></span>
<span id="cb46-767"><a href="#cb46-767"></a>  <span class="fu">filter</span>(locus_tag <span class="sc">%in%</span> <span class="fu">pull</span>(<span class="fu">filter</span>(degentab, <span class="sc">!</span><span class="fu">is.na</span>(ns_length)), locus_tag))</span>
<span id="cb46-768"><a href="#cb46-768"></a><span class="in">```</span></span>
<span id="cb46-769"><a href="#cb46-769"></a></span>
<span id="cb46-770"><a href="#cb46-770"></a>We now want to look at the extent of multiplicity at different binned $f_{max}$</span>
<span id="cb46-773"><a href="#cb46-773"></a><span class="in">```{r}</span></span>
<span id="cb46-774"><a href="#cb46-774"></a>multiplicity_binned <span class="ot">&lt;-</span> mgvars_filt_mb_ns <span class="sc">%&gt;%</span></span>
<span id="cb46-775"><a href="#cb46-775"></a>  <span class="co"># automatically bin f_max into 5 categories</span></span>
<span id="cb46-776"><a href="#cb46-776"></a>  <span class="fu">mutate</span>(<span class="at">fbin =</span> <span class="fu">cut_interval</span>(freq_alt_complete, <span class="dv">5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-777"><a href="#cb46-777"></a>  <span class="fu">summarize</span>(<span class="at">n_i =</span> <span class="fu">n</span>(), </span>
<span id="cb46-778"><a href="#cb46-778"></a>            <span class="at">n_replicate =</span> <span class="fu">n_distinct</span>(replicate),</span>
<span id="cb46-779"><a href="#cb46-779"></a>    <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"locus_tag"</span>, <span class="st">"strainID"</span>, <span class="st">"fbin"</span>, <span class="st">"prey_history"</span>, <span class="st">"predator_history"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-780"><a href="#cb46-780"></a>  <span class="fu">group_by</span>(locus_tag, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb46-781"><a href="#cb46-781"></a>  <span class="fu">complete</span>(fbin, prey_history, predator_history,</span>
<span id="cb46-782"><a href="#cb46-782"></a>           <span class="at">fill =</span> <span class="fu">list</span>(<span class="at">n_i =</span> <span class="dv">0</span>, <span class="at">n_replicate =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-783"><a href="#cb46-783"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-784"><a href="#cb46-784"></a>  <span class="fu">group_by</span>(strainID, fbin, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-785"><a href="#cb46-785"></a>  <span class="fu">mutate</span>(<span class="at">Ntot =</span> <span class="fu">sum</span>(n_i)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-786"><a href="#cb46-786"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb46-787"><a href="#cb46-787"></a><span class="in">```</span></span>
<span id="cb46-788"><a href="#cb46-788"></a></span>
<span id="cb46-789"><a href="#cb46-789"></a>Make df with information about gene lengths</span>
<span id="cb46-792"><a href="#cb46-792"></a><span class="in">```{r}</span></span>
<span id="cb46-793"><a href="#cb46-793"></a>gene_tab <span class="ot">&lt;-</span> degentab <span class="sc">%&gt;%</span> </span>
<span id="cb46-794"><a href="#cb46-794"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(strainID, locus_tag, <span class="at">l_i =</span> ns_length) <span class="sc">%&gt;%</span> </span>
<span id="cb46-795"><a href="#cb46-795"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-796"><a href="#cb46-796"></a>  <span class="fu">group_by</span>(strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb46-797"><a href="#cb46-797"></a>  <span class="fu">mutate</span>(</span>
<span id="cb46-798"><a href="#cb46-798"></a>    <span class="co"># compute the number of genes in the genome</span></span>
<span id="cb46-799"><a href="#cb46-799"></a>    <span class="at">n_genes =</span> <span class="fu">n</span>(),</span>
<span id="cb46-800"><a href="#cb46-800"></a>    <span class="co"># compute the sum of non-syn sites in genome</span></span>
<span id="cb46-801"><a href="#cb46-801"></a>    <span class="at">Ltot =</span> <span class="fu">sum</span>(l_i),</span>
<span id="cb46-802"><a href="#cb46-802"></a>    <span class="co"># compute the average non-syn sites per gene</span></span>
<span id="cb46-803"><a href="#cb46-803"></a>    <span class="at">Lavg =</span> <span class="fu">mean</span>(l_i)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-804"><a href="#cb46-804"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-805"><a href="#cb46-805"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1972"</span>, <span class="st">"HAMBI_1977"</span>, <span class="st">"HAMBI_2659"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-806"><a href="#cb46-806"></a>  <span class="co"># expanding this now to make later joining faster</span></span>
<span id="cb46-807"><a href="#cb46-807"></a>  <span class="fu">expand_grid</span>(<span class="at">prey_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>), </span>
<span id="cb46-808"><a href="#cb46-808"></a>              <span class="at">predator_history =</span> <span class="fu">c</span>(<span class="st">"anc"</span>, <span class="st">"evo"</span>),</span>
<span id="cb46-809"><a href="#cb46-809"></a>              <span class="at">fbin =</span> <span class="fu">unique</span>(multiplicity_binned<span class="sc">$</span>fbin))</span>
<span id="cb46-810"><a href="#cb46-810"></a><span class="in">```</span></span>
<span id="cb46-811"><a href="#cb46-811"></a></span>
<span id="cb46-812"><a href="#cb46-812"></a>This prepares the tibble that we can use in the next step to calculate the G scores and $\Delta L$</span>
<span id="cb46-815"><a href="#cb46-815"></a><span class="in">```{r}</span></span>
<span id="cb46-816"><a href="#cb46-816"></a>g_prepped <span class="ot">&lt;-</span> <span class="fu">left_join</span>(gene_tab, multiplicity_binned) <span class="sc">%&gt;%</span> </span>
<span id="cb46-817"><a href="#cb46-817"></a>  <span class="fu">arrange</span>(strainID, prey_history, predator_history, fbin, locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb46-818"><a href="#cb46-818"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">n_i =</span> <span class="dv">0</span>, <span class="at">n_replicate =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-819"><a href="#cb46-819"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history, fbin) <span class="sc">%&gt;%</span> </span>
<span id="cb46-820"><a href="#cb46-820"></a>  <span class="fu">fill</span>(Ntot, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-821"><a href="#cb46-821"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-822"><a href="#cb46-822"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-823"><a href="#cb46-823"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(Ntot, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-824"><a href="#cb46-824"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb46-825"><a href="#cb46-825"></a><span class="in">```</span></span>
<span id="cb46-826"><a href="#cb46-826"></a></span>
<span id="cb46-827"><a href="#cb46-827"></a>This function calculate $\Delta L$ but from a bootstrapping of the source dataframes to get sample means plus confidence intervals. We use <span class="in">`rsample`</span> package hence the use of <span class="in">`analysis(df).`</span> See the next section for implementation</span>
<span id="cb46-828"><a href="#cb46-828"></a></span>
<span id="cb46-831"><a href="#cb46-831"></a><span class="in">```{r}</span></span>
<span id="cb46-832"><a href="#cb46-832"></a>G_score_boot <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">boot=</span><span class="cn">TRUE</span>){</span>
<span id="cb46-833"><a href="#cb46-833"></a>  </span>
<span id="cb46-834"><a href="#cb46-834"></a>  <span class="cf">if</span> (boot) {</span>
<span id="cb46-835"><a href="#cb46-835"></a>    <span class="co"># convert the rsample splits to tibble</span></span>
<span id="cb46-836"><a href="#cb46-836"></a>    df <span class="ot">&lt;-</span> rsample<span class="sc">::</span><span class="fu">analysis</span>(df)</span>
<span id="cb46-837"><a href="#cb46-837"></a>  }</span>
<span id="cb46-838"><a href="#cb46-838"></a>  <span class="co"># compute the expected multiplicity value</span></span>
<span id="cb46-839"><a href="#cb46-839"></a>  m_exp <span class="ot">&lt;-</span> <span class="fu">unique</span>(df<span class="sc">$</span>Ntot) <span class="sc">/</span> <span class="fu">unique</span>(df<span class="sc">$</span>n_genes)</span>
<span id="cb46-840"><a href="#cb46-840"></a>  <span class="co"># compute gene multiplicity for gene i</span></span>
<span id="cb46-841"><a href="#cb46-841"></a>  m_i <span class="ot">&lt;-</span>  df<span class="sc">$</span>n_i <span class="sc">*</span> <span class="fu">unique</span>(df<span class="sc">$</span>Lavg) <span class="sc">/</span> df<span class="sc">$</span>l_i</span>
<span id="cb46-842"><a href="#cb46-842"></a>  <span class="co"># compute the G score for each gene</span></span>
<span id="cb46-843"><a href="#cb46-843"></a>  g_score <span class="ot">&lt;-</span> df<span class="sc">$</span>n_i <span class="sc">*</span> <span class="fu">log</span>(m_i <span class="sc">/</span> m_exp)</span>
<span id="cb46-844"><a href="#cb46-844"></a>  <span class="co"># net increase in log-likelihood compared to the null model of m_i / m_exp =1</span></span>
<span id="cb46-845"><a href="#cb46-845"></a>  <span class="co"># normalized to the total mutations</span></span>
<span id="cb46-846"><a href="#cb46-846"></a>  <span class="fu">sum</span>(g_score, <span class="at">na.rm=</span>T) <span class="sc">/</span> <span class="fu">unique</span>(df<span class="sc">$</span>Ntot)</span>
<span id="cb46-847"><a href="#cb46-847"></a>}</span>
<span id="cb46-848"><a href="#cb46-848"></a><span class="in">```</span></span>
<span id="cb46-849"><a href="#cb46-849"></a></span>
<span id="cb46-850"><a href="#cb46-850"></a>We need the package <span class="in">`rsample`</span> to do the boostrap analysis and we need <span class="in">`Hmisc`</span> to get the bootstrapped confidence intervals</span>
<span id="cb46-851"><a href="#cb46-851"></a></span>
<span id="cb46-854"><a href="#cb46-854"></a><span class="in">```{r}</span></span>
<span id="cb46-855"><a href="#cb46-855"></a><span class="co">#| eval: false</span></span>
<span id="cb46-856"><a href="#cb46-856"></a><span class="co">#| echo: true</span></span>
<span id="cb46-857"><a href="#cb46-857"></a><span class="fu">library</span>(rsample)</span>
<span id="cb46-858"><a href="#cb46-858"></a><span class="fu">library</span>(Hmisc)</span>
<span id="cb46-859"><a href="#cb46-859"></a></span>
<span id="cb46-860"><a href="#cb46-860"></a><span class="co"># this is a bit slow so saving the results</span></span>
<span id="cb46-861"><a href="#cb46-861"></a>g_boot <span class="ot">&lt;-</span> g_prepped <span class="sc">%&gt;%</span> </span>
<span id="cb46-862"><a href="#cb46-862"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">.by =</span> <span class="fu">c</span>(strainID, prey_history, predator_history, fbin)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-863"><a href="#cb46-863"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">boots =</span> purrr<span class="sc">::</span><span class="fu">map</span>(data, <span class="cf">function</span>(df) rsample<span class="sc">::</span><span class="fu">bootstraps</span>(df, <span class="at">times =</span> <span class="dv">1000</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb46-864"><a href="#cb46-864"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">g_scores =</span> purrr<span class="sc">::</span><span class="fu">map</span>(boots, \(boot) purrr<span class="sc">::</span><span class="fu">map_dbl</span>(boot<span class="sc">$</span>splits, G_score_boot))) <span class="sc">%&gt;%</span> </span>
<span id="cb46-865"><a href="#cb46-865"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">meanboot =</span> purrr<span class="sc">::</span><span class="fu">map</span>(g_scores, ggplot2<span class="sc">::</span>mean_cl_boot),</span>
<span id="cb46-866"><a href="#cb46-866"></a>            <span class="at">.by =</span> <span class="fu">c</span>(strainID, prey_history, predator_history, fbin)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-867"><a href="#cb46-867"></a>  tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(meanboot))</span>
<span id="cb46-868"><a href="#cb46-868"></a></span>
<span id="cb46-869"><a href="#cb46-869"></a><span class="co"># Save this for later</span></span>
<span id="cb46-870"><a href="#cb46-870"></a><span class="fu">write_rds</span>(g_boot, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"g_score_bootstap.rds"</span>))</span>
<span id="cb46-871"><a href="#cb46-871"></a><span class="in">```</span></span>
<span id="cb46-872"><a href="#cb46-872"></a></span>
<span id="cb46-873"><a href="#cb46-873"></a>Read in saved bootstrapped data</span>
<span id="cb46-876"><a href="#cb46-876"></a><span class="in">```{r}</span></span>
<span id="cb46-877"><a href="#cb46-877"></a><span class="co"># Read in previous results</span></span>
<span id="cb46-878"><a href="#cb46-878"></a>g_boot <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"g_score_bootstap.rds"</span>))</span>
<span id="cb46-879"><a href="#cb46-879"></a><span class="in">```</span></span>
<span id="cb46-880"><a href="#cb46-880"></a></span>
<span id="cb46-881"><a href="#cb46-881"></a>Plot bootstrapped $\Delta l$ range with $\Delta l$ from full dataset at different values of $f_{max}$</span>
<span id="cb46-882"><a href="#cb46-882"></a></span>
<span id="cb46-883"><a href="#cb46-883"></a>::: {#fig-18}</span>
<span id="cb46-886"><a href="#cb46-886"></a><span class="in">```{r}</span></span>
<span id="cb46-887"><a href="#cb46-887"></a><span class="co">#| warning: false</span></span>
<span id="cb46-888"><a href="#cb46-888"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-889"><a href="#cb46-889"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb46-890"><a href="#cb46-890"></a>pg_boot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(g_boot, <span class="fu">aes</span>(<span class="at">x =</span> fbin, <span class="at">y =</span> y, <span class="at">ymin =</span> ymin, <span class="at">ymax =</span> ymax)) <span class="sc">+</span></span>
<span id="cb46-891"><a href="#cb46-891"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID), </span>
<span id="cb46-892"><a href="#cb46-892"></a>                  <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb46-893"><a href="#cb46-893"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-894"><a href="#cb46-894"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Binned maximum observed allele frequency ($f_{max}$) "</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">"Extent of parallel evolution ($</span><span class="sc">\\</span><span class="st">Delta l$)"</span>), <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb46-895"><a href="#cb46-895"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb46-896"><a href="#cb46-896"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">45</span>)) <span class="sc">+</span></span>
<span id="cb46-897"><a href="#cb46-897"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-898"><a href="#cb46-898"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-899"><a href="#cb46-899"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-900"><a href="#cb46-900"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-901"><a href="#cb46-901"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-902"><a href="#cb46-902"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb46-903"><a href="#cb46-903"></a></span>
<span id="cb46-904"><a href="#cb46-904"></a>pg_boot</span>
<span id="cb46-905"><a href="#cb46-905"></a><span class="in">```</span></span>
<span id="cb46-906"><a href="#cb46-906"></a></span>
<span id="cb46-907"><a href="#cb46-907"></a>:::</span>
<span id="cb46-908"><a href="#cb46-908"></a></span>
<span id="cb46-909"><a href="#cb46-909"></a><span class="fu">## G score</span></span>
<span id="cb46-910"><a href="#cb46-910"></a></span>
<span id="cb46-911"><a href="#cb46-911"></a>Also calculate the G score from <span class="co">[</span><span class="ot">@tenaillon2016</span><span class="co">]</span> and look at the relationship with $f_{max}$. The difference between the analysis above is that G-score is the non-summed and non-normalized form of $\Delta l$ and it is calculated for every gene uniquely instead of at different $f_{max}$ bins. Briefly,</span>
<span id="cb46-912"><a href="#cb46-912"></a></span>
<span id="cb46-913"><a href="#cb46-913"></a>$$</span>
<span id="cb46-914"><a href="#cb46-914"></a>G_{i} = n_{i} \log \left ( \frac{m_{i}}{\overline{m}} \right )</span>
<span id="cb46-915"><a href="#cb46-915"></a>$$</span>
<span id="cb46-916"><a href="#cb46-916"></a>where the expected multiplicity $\overline{m}$ is</span>
<span id="cb46-917"><a href="#cb46-917"></a></span>
<span id="cb46-918"><a href="#cb46-918"></a>$$</span>
<span id="cb46-919"><a href="#cb46-919"></a>\overline{m} = n_{tot}/n_{genes}</span>
<span id="cb46-920"><a href="#cb46-920"></a>$$</span>
<span id="cb46-921"><a href="#cb46-921"></a></span>
<span id="cb46-922"><a href="#cb46-922"></a>and the observed multiplicity is</span>
<span id="cb46-923"><a href="#cb46-923"></a></span>
<span id="cb46-924"><a href="#cb46-924"></a>$$</span>
<span id="cb46-925"><a href="#cb46-925"></a>m_{i} = n_{i} \cdot \frac{\overline{L}}{L_{i}}</span>
<span id="cb46-926"><a href="#cb46-926"></a>$$</span>
<span id="cb46-927"><a href="#cb46-927"></a></span>
<span id="cb46-930"><a href="#cb46-930"></a><span class="in">```{r}</span></span>
<span id="cb46-931"><a href="#cb46-931"></a>multiplicity <span class="ot">&lt;-</span> mgvars_filt_mb_ns <span class="sc">%&gt;%</span></span>
<span id="cb46-932"><a href="#cb46-932"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">n_i =</span> <span class="fu">n</span>(),</span>
<span id="cb46-933"><a href="#cb46-933"></a>            <span class="at">n_replicate =</span> <span class="fu">n_distinct</span>(replicate), </span>
<span id="cb46-934"><a href="#cb46-934"></a>            <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"locus_tag"</span>, <span class="st">"strainID"</span>, <span class="st">"prey_history"</span>, <span class="st">"predator_history"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-935"><a href="#cb46-935"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span></span>
<span id="cb46-936"><a href="#cb46-936"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Ntot =</span> <span class="fu">sum</span>(n_i)) <span class="sc">%&gt;%</span></span>
<span id="cb46-937"><a href="#cb46-937"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb46-938"><a href="#cb46-938"></a></span>
<span id="cb46-939"><a href="#cb46-939"></a>g_prepped_nobin <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(gene_tab, <span class="sc">-</span>fbin)), multiplicity) <span class="sc">%&gt;%</span> </span>
<span id="cb46-940"><a href="#cb46-940"></a>  <span class="fu">arrange</span>(strainID, prey_history, predator_history, locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb46-941"><a href="#cb46-941"></a>  <span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">n_i =</span> <span class="dv">0</span>, <span class="at">n_replicate =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-942"><a href="#cb46-942"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-943"><a href="#cb46-943"></a>  <span class="fu">fill</span>(Ntot, <span class="at">.direction =</span> <span class="st">"updown"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-944"><a href="#cb46-944"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-945"><a href="#cb46-945"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-946"><a href="#cb46-946"></a>  <span class="fu">filter</span>(<span class="fu">sum</span>(Ntot, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-947"><a href="#cb46-947"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb46-948"><a href="#cb46-948"></a></span>
<span id="cb46-949"><a href="#cb46-949"></a>G_score_tidy <span class="ot">&lt;-</span> <span class="cf">function</span>(df){</span>
<span id="cb46-950"><a href="#cb46-950"></a>  df <span class="sc">%&gt;%</span> </span>
<span id="cb46-951"><a href="#cb46-951"></a>  <span class="co"># compute the expected multiplicity value</span></span>
<span id="cb46-952"><a href="#cb46-952"></a>  <span class="fu">mutate</span>(<span class="at">m_exp =</span> Ntot <span class="sc">/</span> n_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb46-953"><a href="#cb46-953"></a>  <span class="co"># compute gene multiplicity for gene i</span></span>
<span id="cb46-954"><a href="#cb46-954"></a>  <span class="fu">mutate</span>(<span class="at">m_i =</span>  n_i <span class="sc">*</span> Lavg <span class="sc">/</span> l_i) <span class="sc">%&gt;%</span> </span>
<span id="cb46-955"><a href="#cb46-955"></a>  <span class="co"># compute the G score for each gene</span></span>
<span id="cb46-956"><a href="#cb46-956"></a>  <span class="fu">mutate</span>(<span class="at">g_score =</span> n_i <span class="sc">*</span> <span class="fu">log</span>(m_i <span class="sc">/</span> m_exp)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-957"><a href="#cb46-957"></a>  <span class="co"># net increase in log-likelihood compared to the null model of m_i / m_exp =1</span></span>
<span id="cb46-958"><a href="#cb46-958"></a>  <span class="co"># normalized to the total mutations</span></span>
<span id="cb46-959"><a href="#cb46-959"></a>  <span class="fu">mutate</span>(<span class="at">delta_l =</span> <span class="fu">sum</span>(g_score, <span class="at">na.rm=</span>T) <span class="sc">/</span> Ntot)</span>
<span id="cb46-960"><a href="#cb46-960"></a>}</span>
<span id="cb46-961"><a href="#cb46-961"></a></span>
<span id="cb46-962"><a href="#cb46-962"></a>g_scores <span class="ot">&lt;-</span> g_prepped_nobin <span class="sc">%&gt;%</span> </span>
<span id="cb46-963"><a href="#cb46-963"></a>  <span class="fu">group_by</span>(strainID, prey_history, predator_history) <span class="sc">%&gt;%</span> </span>
<span id="cb46-964"><a href="#cb46-964"></a>  <span class="fu">G_score_tidy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-965"><a href="#cb46-965"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb46-966"><a href="#cb46-966"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb46-967"><a href="#cb46-967"></a></span>
<span id="cb46-968"><a href="#cb46-968"></a>g_scores_fmax <span class="ot">&lt;-</span> <span class="fu">left_join</span>(g_scores, mgvars,</span>
<span id="cb46-969"><a href="#cb46-969"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(strainID, locus_tag, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-970"><a href="#cb46-970"></a>  <span class="fu">group_by</span>(locus_tag) <span class="sc">%&gt;%</span> </span>
<span id="cb46-971"><a href="#cb46-971"></a>  <span class="fu">filter</span>(freq_alt_complete <span class="sc">==</span> <span class="fu">max</span>(freq_alt_complete, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-972"><a href="#cb46-972"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb46-973"><a href="#cb46-973"></a><span class="in">```</span></span>
<span id="cb46-974"><a href="#cb46-974"></a></span>
<span id="cb46-975"><a href="#cb46-975"></a>::: {#fig-19}</span>
<span id="cb46-978"><a href="#cb46-978"></a><span class="in">```{r}</span></span>
<span id="cb46-979"><a href="#cb46-979"></a><span class="co">#| warning: false</span></span>
<span id="cb46-980"><a href="#cb46-980"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb46-981"><a href="#cb46-981"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb46-982"><a href="#cb46-982"></a>fig19 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(g_scores_fmax, <span class="fu">aes</span>(<span class="at">x =</span> freq_alt_complete, <span class="at">y =</span> g_score)) <span class="sc">+</span></span>
<span id="cb46-983"><a href="#cb46-983"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">'y ~ x'</span>) <span class="sc">+</span></span>
<span id="cb46-984"><a href="#cb46-984"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> strainID)) <span class="sc">+</span></span>
<span id="cb46-985"><a href="#cb46-985"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span> </span>
<span id="cb46-986"><a href="#cb46-986"></a>  <span class="fu">facet_grid</span>(prey_history <span class="sc">~</span> predator_history, <span class="at">labeller =</span> label_both) <span class="sc">+</span></span>
<span id="cb46-987"><a href="#cb46-987"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">TeX</span>(<span class="st">"Maximum observed allele frequency ($f_{max}$) "</span>), <span class="at">y =</span> <span class="fu">TeX</span>(<span class="st">"Extent of parallel evolution (Gene-wise $G_{i}$)"</span>), <span class="at">color =</span> <span class="cn">NULL</span>, <span class="at">shape =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb46-988"><a href="#cb46-988"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"bl"</span>, <span class="at">color =</span> <span class="st">"grey40"</span>) <span class="sc">+</span></span>
<span id="cb46-989"><a href="#cb46-989"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb46-990"><a href="#cb46-990"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb46-991"><a href="#cb46-991"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb46-992"><a href="#cb46-992"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-993"><a href="#cb46-993"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb46-994"><a href="#cb46-994"></a>    <span class="at">strip.placement =</span> <span class="st">'outside'</span>,</span>
<span id="cb46-995"><a href="#cb46-995"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-996"><a href="#cb46-996"></a>    <span class="at">panel.grid =</span> <span class="fu">element_blank</span>())</span>
<span id="cb46-997"><a href="#cb46-997"></a></span>
<span id="cb46-998"><a href="#cb46-998"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(figs, <span class="st">"G_score_fmax.svg"</span>), fig19, <span class="at">width=</span><span class="dv">7</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">units=</span><span class="st">"in"</span>,</span>
<span id="cb46-999"><a href="#cb46-999"></a>       <span class="at">device=</span><span class="st">"svg"</span>)</span>
<span id="cb46-1000"><a href="#cb46-1000"></a></span>
<span id="cb46-1001"><a href="#cb46-1001"></a>fig19</span>
<span id="cb46-1002"><a href="#cb46-1002"></a><span class="in">```</span></span>
<span id="cb46-1003"><a href="#cb46-1003"></a></span>
<span id="cb46-1004"><a href="#cb46-1004"></a>:::</span>
<span id="cb46-1005"><a href="#cb46-1005"></a></span>
<span id="cb46-1006"><a href="#cb46-1006"></a>Linear regression for different treatment combinations (over species). Overall, we find a clear positive relationship between the gene-wise extent of parallel evolution (G score) and the maximum observed allele frequency for that gene in 3/4 treatment combinations. We also find a positive relationship for the prey: evo | predator: anc treatment combination but the significance of this relationship is unclear (P value = 0.07).</span>
<span id="cb46-1007"><a href="#cb46-1007"></a></span>
<span id="cb46-1010"><a href="#cb46-1010"></a><span class="in">```{r}</span></span>
<span id="cb46-1011"><a href="#cb46-1011"></a>g_scores_fmax <span class="sc">%&gt;%</span> </span>
<span id="cb46-1012"><a href="#cb46-1012"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span><span class="fu">c</span>(prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1013"><a href="#cb46-1013"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, \(df) <span class="fu">lm</span>(g_score <span class="sc">~</span> freq_alt_complete, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">data =</span> df))) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1014"><a href="#cb46-1014"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">tidy =</span> <span class="fu">map</span>(model, broom<span class="sc">::</span>tidy)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1015"><a href="#cb46-1015"></a>  <span class="fu">unnest</span>(tidy) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1016"><a href="#cb46-1016"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>data, <span class="sc">-</span>model) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1017"><a href="#cb46-1017"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1018"><a href="#cb46-1018"></a>  <span class="fu">arrange</span>(p.value)</span>
<span id="cb46-1019"><a href="#cb46-1019"></a><span class="in">```</span></span>
<span id="cb46-1020"><a href="#cb46-1020"></a></span>
<span id="cb46-1021"><a href="#cb46-1021"></a>Linear regression for all different species in different treatment combinations. By including every species we lose statistical power because some species have relatively few observations. However, for all 4 species in the prey: evo | predator: anc treatment combination we find a clear positive relationship between the gene-wise extent of parallel evolution (G score) and the maximum observed allele frequency for that gene. We also find this positive relationmship for HAMBI_2659 in the prey: evo | predator: evo treatment combination.</span>
<span id="cb46-1022"><a href="#cb46-1022"></a></span>
<span id="cb46-1025"><a href="#cb46-1025"></a><span class="in">```{r}</span></span>
<span id="cb46-1026"><a href="#cb46-1026"></a>g_scores_fmax <span class="sc">%&gt;%</span> </span>
<span id="cb46-1027"><a href="#cb46-1027"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="sc">-</span><span class="fu">c</span>(strainID, prey_history, predator_history)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1028"><a href="#cb46-1028"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">model =</span> <span class="fu">map</span>(data, \(df) <span class="fu">lm</span>(g_score <span class="sc">~</span> freq_alt_complete, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">data =</span> df))) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1029"><a href="#cb46-1029"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">tidy =</span> <span class="fu">map</span>(model, broom<span class="sc">::</span>tidy)) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1030"><a href="#cb46-1030"></a>  <span class="fu">unnest</span>(tidy) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1031"><a href="#cb46-1031"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>data, <span class="sc">-</span>model) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1032"><a href="#cb46-1032"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(term <span class="sc">!=</span> <span class="st">"(Intercept)"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb46-1033"><a href="#cb46-1033"></a>  <span class="fu">arrange</span>(p.value)</span>
<span id="cb46-1034"><a href="#cb46-1034"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>